{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "b11ba192-32a0-4ec3-9e30-072f2f43f8ee",
   "metadata": {
    "kernel": "SoS",
    "tags": []
   },
   "source": [
    "# ARHI x sex interaction analysis\n",
    "\n",
    "Use sex as the interactive covariate. Exclude sex in the covariates."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "17bece4e-f845-4088-9588-6930be3cc299",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Common variables Columbia's cluster\n",
    "UKBB_PATH=$HOME/UKBiobank\n",
    "UKBB_yale=$HOME/UKBiobank_Yale_transfer\n",
    "USER_PATH=$HOME/working\n",
    "container_lmm=$HOME/containers/lmm.sif\n",
    "container_marp=$HOME/containers/marp.sif\n",
    "container_annovar=$HOME/containers/gatk4-annovar.sif\n",
    "hearing_pheno_path=$UKBB_PATH/phenotype_files/hearing_impairment\n",
    "tpl_file=$USER_PATH/bioworkflows/admin/csg.yml\n",
    "formatFile_fastgwa=$USER_PATH/UKBB_GWAS_dev/data/fastGWA_template.yml\n",
    "formatFile_bolt=$USER_PATH/UKBB_GWAS_dev/data/boltlmm_template.yml\n",
    "formatFile_saige=$USER_PATH/UKBB_GWAS_dev/data/saige_template.yml\n",
    "formatFile_regenie=$USER_PATH/UKBB_GWAS_dev/data/regenie_template.yml\n",
    "# Workflows\n",
    "lmm_sos=$USER_PATH/bioworkflows/GWAS/LMM.ipynb\n",
    "anno_sos=$USER_PATH/bioworkflows/variant-annotation/annovar.ipynb\n",
    "clumping_sos=$USER_PATH/bioworkflows/GWAS/LD_Clumping.ipynb\n",
    "extract_sos=$USER_PATH/bioworkflows/GWAS/Region_Extraction.ipynb\n",
    "snptogene_sos=$USER_PATH/UKBB_GWAS_dev/workflow/snptogene.ipynb\n",
    "\n",
    "# LMM directories for array data data\n",
    "lmm_imp_dir_fastgwa=$UKBB_PATH/results/FastGWA_results/results_array_data\n",
    "lmm_imp_dir_bolt=$UKBB_PATH/results/BOLTLMM_results/results_array_data\n",
    "lmm_imp_dir_saige=$UKBB_PATH/results/SAIGE_results/results_array_data\n",
    "lmm_imp_dir_regenie=$UKBB_PATH/results/REGENIE_results/results_array_data\n",
    "\n",
    "# LMM directories for exome data\n",
    "lmm_exome_dir_fastgwa=$UKBB_PATH/results/FastGWA_results/results_array_data\n",
    "lmm_exome_dir_bolt=$UKBB_PATH/results/BOLTLMM_results/results_array_data\n",
    "lmm_exome_dir_saige=$UKBB_PATH/results/SAIGE_results/results_array_data\n",
    "lmm_exome_dir_regenie=$UKBB_PATH/results/REGENIE_results/results_array_data\n",
    "### Specific to REGENIE\n",
    "bsize=1000\n",
    "trait=bt\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "minMAC=4\n",
    "maf_filter=0.01\n",
    "geno_filter=0.01\n",
    "hwe_filter=0\n",
    "mind_filter=0.1\n",
    "reverse_log_p=True\n",
    "numThreads=2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "46190c12-c966-45aa-a7fb-7759789ec320",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f3393/f3393_sex_interaction-regenie_2022-05-17.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wd7f9119b68345bce) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "# 200k exome\n",
    "# f3393\n",
    "lmm_dir_regenie=/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f3393\n",
    "lmm_sbatch_regenie=$lmm_dir_regenie/f3393_sex_interaction-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "phenoFile=/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/090321_UKBB_Hearing_aid_f3393_expandedwhite_6436cases_96601ctrl_PC1_2.tsv\n",
    "phenoCol=f3393\n",
    "covarCol=age\n",
    "qCovarCol=\"PC1 PC2\"\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=/home/gl2776/UKBiobank/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the gene bfile here for the association\n",
    "genoFile=`echo /home/gl2776/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "lmm_job_size=20\n",
    "regenie_sos=~/working/interaction/newv/Regenie_interaction_GxE.ipynb\n",
    "regenieFile=/mnt/mfs/statgen/guangyou/containers/regenie.v.2.3\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "covariates=sex\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --regenieFile $regenieFile\n",
    "    --sampleFile $sampleFile\n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --covariates $covariates\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $regenie_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "5f74c6c6-01ae-4cba-81c0-26040cde1d49",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f2247/f2247_sex_interaction-regenie_2022-05-17.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w9fcc09c866ca20d0) is executed successfully with 1 completed step.\n",
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f2257/f2257_sex_interaction-regenie_2022-05-17.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wea5aaf6103093b94) is executed successfully with 1 completed step.\n",
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f2247_f2257/f2247_f2257_sex_interaction-regenie_2022-05-17.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=we0a6c07501e22a7e) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "# f2247\n",
    "lmm_dir_regenie=/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f2247\n",
    "lmm_sbatch_regenie=$lmm_dir_regenie/f2247_sex_interaction-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "phenoFile=/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/090321_UKBB_Hearing_difficulty_f2247_expandedwhite_45502cases_96601ctrl_PC1_2.tsv\n",
    "phenoCol=f2247\n",
    "covarCol=age\n",
    "qCovarCol=\"PC1 PC2\"\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=/home/gl2776/UKBiobank/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the gene bfile here for the association\n",
    "genoFile=`echo /home/gl2776/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "lmm_job_size=20\n",
    "regenie_sos=~/working/interaction/newv/Regenie_interaction_GxE.ipynb\n",
    "regenieFile=/mnt/mfs/statgen/guangyou/containers/regenie.v.2.3\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "covariates=sex\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --regenieFile $regenieFile\n",
    "    --sampleFile $sampleFile\n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --covariates $covariates\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $regenie_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" \n",
    "\n",
    "# f2257\n",
    "lmm_dir_regenie=/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f2257\n",
    "lmm_sbatch_regenie=$lmm_dir_regenie/f2257_sex_interaction-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "phenoFile=/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/090321_UKBB_Hearing_noise_f2257_expandedwhite_65660cases_96601ctrl_PC1_2.tsv\n",
    "phenoCol=f2257\n",
    "covarCol=age\n",
    "qCovarCol=\"PC1 PC2\"\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=/home/gl2776/UKBiobank/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the gene bfile here for the association\n",
    "genoFile=`echo /home/gl2776/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "lmm_job_size=20\n",
    "regenie_sos=~/working/interaction/newv/Regenie_interaction_GxE.ipynb\n",
    "regenieFile=/mnt/mfs/statgen/guangyou/containers/regenie.v.2.3\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "covariates=sex\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --regenieFile $regenieFile\n",
    "    --sampleFile $sampleFile\n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --covariates $covariates\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $regenie_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" \n",
    "\n",
    "\n",
    "# f2247_f2257\n",
    "lmm_dir_regenie=/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f2247_f2257\n",
    "lmm_sbatch_regenie=$lmm_dir_regenie/f2247_f2257_sex_interaction-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "phenoFile=/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/090321_UKBB_Combined_f2247_f2257_expandedwhite_38410cases_96601ctrl_PC1_2.tsv\n",
    "phenoCol=f2247_f2257\n",
    "covarCol=age\n",
    "qCovarCol=\"PC1 PC2\"\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=/home/gl2776/UKBiobank/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the gene bfile here for the association\n",
    "genoFile=`echo /home/gl2776/UKBiobank/data/exome_files/project_VCF/072721_run/plink/ukb23156_c{1..22}.merged.filtered.bed`\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "lmm_job_size=20\n",
    "regenie_sos=~/working/interaction/newv/Regenie_interaction_GxE.ipynb\n",
    "regenieFile=/mnt/mfs/statgen/guangyou/containers/regenie.v.2.3\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "covariates=sex\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --regenieFile $regenieFile\n",
    "    --sampleFile $sampleFile\n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --covariates $covariates\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $regenie_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "markdown",
   "id": "556b183b-3050-4bdd-a971-074764a1ab64",
   "metadata": {
    "kernel": "Bash"
   },
   "source": [
    "## Results analysis with discovered variants\n",
    "\n",
    "The following variants are of interest\n",
    "```\n",
    "chr1:46001049:T:C\n",
    "chr1:46021880:G:A\n",
    "chr1:46027788:T:G\n",
    "chr1:46132597:C:A\n",
    "chr5:272741:A:G\n",
    "chr5:272748:G:C\n",
    "chr5:272755:A:G\n",
    "chr5:73776529:T:C\n",
    "chr5:73780632:G:A\n",
    "chr5:73780649:GT:G\n",
    "chr5:73780686:C:A\n",
    "chr5:73794436:T:C\n",
    "chr5:73795301:T:A\n",
    "chr5:73795403:C:T\n",
    "chr6:158071628:C:T\n",
    "chr6:43301291:A:C\n",
    "chr6:43305866:A:G\n",
    "chr6:43308652:G:A\n",
    "chr6:75362956:T:C\n",
    "chr6:75841299:A:G\n",
    "chr7:73573024:C:T\n",
    "chr7:73605645:A:G\n",
    "chr20:62065259:A:G\n",
    "chr22:38086345:GTGCGGGAGCGGGACTGGCCATCCCAGTACTCCGAGGGTGCTA:G\n",
    "chr22:50549676:G:A\n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "b5ba592d-03dd-4a4a-b49f-8076ef1cdaff",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>ID</th>\n",
       "      <th>ALLELE0</th>\n",
       "      <th>ALLELE1</th>\n",
       "      <th>A1FREQ</th>\n",
       "      <th>N</th>\n",
       "      <th>TEST</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>LOG10P</th>\n",
       "      <th>EXTRA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>925862</td>\n",
       "      <td>chr1:925862:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>93212</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>-1.448210</td>\n",
       "      <td>1.333390</td>\n",
       "      <td>1.179640</td>\n",
       "      <td>0.556848</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>925862</td>\n",
       "      <td>chr1:925862:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>93212</td>\n",
       "      <td>ADD-INT_SNP</td>\n",
       "      <td>-7.746010</td>\n",
       "      <td>42.909800</td>\n",
       "      <td>0.032587</td>\n",
       "      <td>0.067148</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>925862</td>\n",
       "      <td>chr1:925862:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>93212</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>2.476520</td>\n",
       "      <td>69.468200</td>\n",
       "      <td>0.001271</td>\n",
       "      <td>0.012530</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>925862</td>\n",
       "      <td>chr1:925862:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000027</td>\n",
       "      <td>93212</td>\n",
       "      <td>ADD-INT_2DF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.042282</td>\n",
       "      <td>0.009181</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>925881</td>\n",
       "      <td>chr1:925881:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.004690</td>\n",
       "      <td>100210</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>-0.209793</td>\n",
       "      <td>0.143623</td>\n",
       "      <td>2.133700</td>\n",
       "      <td>0.841357</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1164288</th>\n",
       "      <td>1</td>\n",
       "      <td>248918380</td>\n",
       "      <td>chr1:248918380:T:G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000146</td>\n",
       "      <td>103030</td>\n",
       "      <td>ADD-INT_2DF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.731550</td>\n",
       "      <td>0.376002</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1164289</th>\n",
       "      <td>1</td>\n",
       "      <td>248918390</td>\n",
       "      <td>chr1:248918390:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>102992</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>-1.052730</td>\n",
       "      <td>2.325280</td>\n",
       "      <td>0.204966</td>\n",
       "      <td>0.186591</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1164290</th>\n",
       "      <td>1</td>\n",
       "      <td>248918390</td>\n",
       "      <td>chr1:248918390:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>102992</td>\n",
       "      <td>ADD-INT_SNP</td>\n",
       "      <td>-6.260560</td>\n",
       "      <td>53.837500</td>\n",
       "      <td>0.013522</td>\n",
       "      <td>0.042189</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1164291</th>\n",
       "      <td>1</td>\n",
       "      <td>248918390</td>\n",
       "      <td>chr1:248918390:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>102992</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>-1.112130</td>\n",
       "      <td>199.283000</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>0.001938</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1164292</th>\n",
       "      <td>1</td>\n",
       "      <td>248918390</td>\n",
       "      <td>chr1:248918390:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000029</td>\n",
       "      <td>102992</td>\n",
       "      <td>ADD-INT_2DF</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.014701</td>\n",
       "      <td>0.003192</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1164293 rows Ã— 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         CHROM     GENPOS                  ID ALLELE0 ALLELE1    A1FREQ  \\\n",
       "0            1     925862     chr1:925862:G:A       G       A  0.000027   \n",
       "1            1     925862     chr1:925862:G:A       G       A  0.000027   \n",
       "2            1     925862     chr1:925862:G:A       G       A  0.000027   \n",
       "3            1     925862     chr1:925862:G:A       G       A  0.000027   \n",
       "4            1     925881     chr1:925881:G:A       G       A  0.004690   \n",
       "...        ...        ...                 ...     ...     ...       ...   \n",
       "1164288      1  248918380  chr1:248918380:T:G       T       G  0.000146   \n",
       "1164289      1  248918390  chr1:248918390:A:G       A       G  0.000029   \n",
       "1164290      1  248918390  chr1:248918390:A:G       A       G  0.000029   \n",
       "1164291      1  248918390  chr1:248918390:A:G       A       G  0.000029   \n",
       "1164292      1  248918390  chr1:248918390:A:G       A       G  0.000029   \n",
       "\n",
       "              N             TEST      BETA          SE     CHISQ    LOG10P  \\\n",
       "0         93212       ADD-CONDTL -1.448210    1.333390  1.179640  0.556848   \n",
       "1         93212      ADD-INT_SNP -7.746010   42.909800  0.032587  0.067148   \n",
       "2         93212  ADD-INT_SNPxsex  2.476520   69.468200  0.001271  0.012530   \n",
       "3         93212      ADD-INT_2DF       NaN         NaN  0.042282  0.009181   \n",
       "4        100210       ADD-CONDTL -0.209793    0.143623  2.133700  0.841357   \n",
       "...         ...              ...       ...         ...       ...       ...   \n",
       "1164288  103030      ADD-INT_2DF       NaN         NaN  1.731550  0.376002   \n",
       "1164289  102992       ADD-CONDTL -1.052730    2.325280  0.204966  0.186591   \n",
       "1164290  102992      ADD-INT_SNP -6.260560   53.837500  0.013522  0.042189   \n",
       "1164291  102992  ADD-INT_SNPxsex -1.112130  199.283000  0.000031  0.001938   \n",
       "1164292  102992      ADD-INT_2DF       NaN         NaN  0.014701  0.003192   \n",
       "\n",
       "         EXTRA  \n",
       "0          NaN  \n",
       "1          NaN  \n",
       "2          NaN  \n",
       "3          NaN  \n",
       "4          NaN  \n",
       "...        ...  \n",
       "1164288    NaN  \n",
       "1164289    NaN  \n",
       "1164290    NaN  \n",
       "1164291    NaN  \n",
       "1164292    NaN  \n",
       "\n",
       "[1164293 rows x 13 columns]"
      ]
     },
     "execution_count": 1,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "import pandas as pd\n",
    "file = \"~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f3393/cache/ukb23156_c1.merged.filtered.bed.sex_f3393.regenie.gz\"\n",
    "stat = pd.read_csv(file,header=0,sep=\" \")\n",
    "stat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "e0185752-3a78-43e0-83c5-9dd1bacd5816",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>ID</th>\n",
       "      <th>ALLELE0</th>\n",
       "      <th>ALLELE1</th>\n",
       "      <th>A1FREQ</th>\n",
       "      <th>N</th>\n",
       "      <th>TEST</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>LOG10P</th>\n",
       "      <th>EXTRA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>421246</th>\n",
       "      <td>1</td>\n",
       "      <td>46001049</td>\n",
       "      <td>chr1:46001049:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.427120</td>\n",
       "      <td>99183</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.036216</td>\n",
       "      <td>0.040272</td>\n",
       "      <td>0.808703</td>\n",
       "      <td>0.433558</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421410</th>\n",
       "      <td>1</td>\n",
       "      <td>46021880</td>\n",
       "      <td>chr1:46021880:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.434137</td>\n",
       "      <td>101574</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.039930</td>\n",
       "      <td>0.039117</td>\n",
       "      <td>1.041960</td>\n",
       "      <td>0.512345</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>421659</th>\n",
       "      <td>1</td>\n",
       "      <td>46027788</td>\n",
       "      <td>chr1:46027788:T:G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.437853</td>\n",
       "      <td>102966</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.031215</td>\n",
       "      <td>0.038690</td>\n",
       "      <td>0.650910</td>\n",
       "      <td>0.376970</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>423001</th>\n",
       "      <td>1</td>\n",
       "      <td>46132597</td>\n",
       "      <td>chr1:46132597:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.438899</td>\n",
       "      <td>103034</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.033714</td>\n",
       "      <td>0.038751</td>\n",
       "      <td>0.756930</td>\n",
       "      <td>0.415340</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "        CHROM    GENPOS                 ID ALLELE0 ALLELE1    A1FREQ       N  \\\n",
       "421246      1  46001049  chr1:46001049:T:C       T       C  0.427120   99183   \n",
       "421410      1  46021880  chr1:46021880:G:A       G       A  0.434137  101574   \n",
       "421659      1  46027788  chr1:46027788:T:G       T       G  0.437853  102966   \n",
       "423001      1  46132597  chr1:46132597:C:A       C       A  0.438899  103034   \n",
       "\n",
       "                   TEST      BETA        SE     CHISQ    LOG10P  EXTRA  \n",
       "421246  ADD-INT_SNPxsex  0.036216  0.040272  0.808703  0.433558    NaN  \n",
       "421410  ADD-INT_SNPxsex  0.039930  0.039117  1.041960  0.512345    NaN  \n",
       "421659  ADD-INT_SNPxsex  0.031215  0.038690  0.650910  0.376970    NaN  \n",
       "423001  ADD-INT_SNPxsex  0.033714  0.038751  0.756930  0.415340    NaN  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snplist = pd.read_csv(\"~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/vairants_of_interest.txt\",header=None)[0].to_list()\n",
    "tmp = stat[stat[\"ID\"].isin(snplist)]\n",
    "tmp[tmp[\"TEST\"]==\"ADD-INT_SNPxsex\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7b6310e4-3a41-47d6-a394-3b9892856f37",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>ID</th>\n",
       "      <th>ALLELE0</th>\n",
       "      <th>ALLELE1</th>\n",
       "      <th>A1FREQ</th>\n",
       "      <th>N</th>\n",
       "      <th>TEST</th>\n",
       "      <th>BETA</th>\n",
       "      <th>SE</th>\n",
       "      <th>CHISQ</th>\n",
       "      <th>LOG10P</th>\n",
       "      <th>EXTRA</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>46001049</td>\n",
       "      <td>chr1:46001049:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.42712</td>\n",
       "      <td>99183</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.036216</td>\n",
       "      <td>0.040272</td>\n",
       "      <td>0.808703</td>\n",
       "      <td>0.433558</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>46021880</td>\n",
       "      <td>chr1:46021880:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.434137</td>\n",
       "      <td>101574</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.03993</td>\n",
       "      <td>0.039117</td>\n",
       "      <td>1.04196</td>\n",
       "      <td>0.512345</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>46027788</td>\n",
       "      <td>chr1:46027788:T:G</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.437853</td>\n",
       "      <td>102966</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.031215</td>\n",
       "      <td>0.03869</td>\n",
       "      <td>0.65091</td>\n",
       "      <td>0.37697</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>46132597</td>\n",
       "      <td>chr1:46132597:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.438899</td>\n",
       "      <td>103034</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.033714</td>\n",
       "      <td>0.038751</td>\n",
       "      <td>0.75693</td>\n",
       "      <td>0.41534</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>272741</td>\n",
       "      <td>chr5:272741:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000539</td>\n",
       "      <td>103025</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.216107</td>\n",
       "      <td>0.491409</td>\n",
       "      <td>0.193397</td>\n",
       "      <td>0.180387</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>272748</td>\n",
       "      <td>chr5:272748:G:C</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000447</td>\n",
       "      <td>103008</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>-0.38988</td>\n",
       "      <td>0.564171</td>\n",
       "      <td>0.477574</td>\n",
       "      <td>0.310227</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5</td>\n",
       "      <td>272755</td>\n",
       "      <td>chr5:272755:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000578</td>\n",
       "      <td>103030</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.153732</td>\n",
       "      <td>0.474756</td>\n",
       "      <td>0.104855</td>\n",
       "      <td>0.127215</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>5</td>\n",
       "      <td>73776529</td>\n",
       "      <td>chr5:73776529:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.46584</td>\n",
       "      <td>103031</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.085258</td>\n",
       "      <td>0.0385</td>\n",
       "      <td>4.90034</td>\n",
       "      <td>1.57103</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>5</td>\n",
       "      <td>73780632</td>\n",
       "      <td>chr5:73780632:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.461324</td>\n",
       "      <td>102208</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.079997</td>\n",
       "      <td>0.038733</td>\n",
       "      <td>4.26267</td>\n",
       "      <td>1.4094</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>5</td>\n",
       "      <td>73780649</td>\n",
       "      <td>chr5:73780649:GT:G</td>\n",
       "      <td>GT</td>\n",
       "      <td>G</td>\n",
       "      <td>0.469417</td>\n",
       "      <td>101396</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.08182</td>\n",
       "      <td>0.038903</td>\n",
       "      <td>4.42026</td>\n",
       "      <td>1.44959</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>5</td>\n",
       "      <td>73780686</td>\n",
       "      <td>chr5:73780686:C:A</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.500039</td>\n",
       "      <td>103006</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.057869</td>\n",
       "      <td>0.038817</td>\n",
       "      <td>2.22246</td>\n",
       "      <td>0.866408</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>5</td>\n",
       "      <td>73794436</td>\n",
       "      <td>chr5:73794436:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.522717</td>\n",
       "      <td>102699</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.080494</td>\n",
       "      <td>0.038596</td>\n",
       "      <td>4.34773</td>\n",
       "      <td>1.43111</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5</td>\n",
       "      <td>73795301</td>\n",
       "      <td>chr5:73795301:T:A</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>0.477072</td>\n",
       "      <td>102733</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.05526</td>\n",
       "      <td>0.038674</td>\n",
       "      <td>2.04167</td>\n",
       "      <td>0.815191</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>5</td>\n",
       "      <td>73795403</td>\n",
       "      <td>chr5:73795403:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.477184</td>\n",
       "      <td>102867</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.049815</td>\n",
       "      <td>0.038677</td>\n",
       "      <td>1.65888</td>\n",
       "      <td>0.703873</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>6</td>\n",
       "      <td>43301291</td>\n",
       "      <td>chr6:43301291:A:C</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.605344</td>\n",
       "      <td>103015</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.059571</td>\n",
       "      <td>0.040074</td>\n",
       "      <td>2.20971</td>\n",
       "      <td>0.862821</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>6</td>\n",
       "      <td>43305866</td>\n",
       "      <td>chr6:43305866:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.605361</td>\n",
       "      <td>103036</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.060616</td>\n",
       "      <td>0.040076</td>\n",
       "      <td>2.28769</td>\n",
       "      <td>0.884709</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>6</td>\n",
       "      <td>43308652</td>\n",
       "      <td>chr6:43308652:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.603251</td>\n",
       "      <td>100977</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.059866</td>\n",
       "      <td>0.040503</td>\n",
       "      <td>2.18466</td>\n",
       "      <td>0.855761</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>6</td>\n",
       "      <td>158071628</td>\n",
       "      <td>chr6:158071628:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.005042</td>\n",
       "      <td>103037</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.117421</td>\n",
       "      <td>0.231253</td>\n",
       "      <td>0.25782</td>\n",
       "      <td>0.213517</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>7</td>\n",
       "      <td>73573024</td>\n",
       "      <td>chr7:73573024:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.281331</td>\n",
       "      <td>103037</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.056335</td>\n",
       "      <td>0.042068</td>\n",
       "      <td>1.79329</td>\n",
       "      <td>0.743461</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>7</td>\n",
       "      <td>73605645</td>\n",
       "      <td>chr7:73605645:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.279967</td>\n",
       "      <td>102846</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.056899</td>\n",
       "      <td>0.042361</td>\n",
       "      <td>1.80418</td>\n",
       "      <td>0.746643</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>20</td>\n",
       "      <td>62065259</td>\n",
       "      <td>chr20:62065259:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.240964</td>\n",
       "      <td>92767</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.066438</td>\n",
       "      <td>0.036573</td>\n",
       "      <td>3.30003</td>\n",
       "      <td>1.1594</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22</td>\n",
       "      <td>38086345</td>\n",
       "      <td>chr22:38086345:GTGCGGGAGCGGGACTGGCCATCCCAGTACT...</td>\n",
       "      <td>GTGCGGGAGCGGGACTGGCCATCCCAGTACTCCGAGGGTGCTA</td>\n",
       "      <td>G</td>\n",
       "      <td>0.290592</td>\n",
       "      <td>98351</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.02216</td>\n",
       "      <td>0.043424</td>\n",
       "      <td>0.260436</td>\n",
       "      <td>0.214798</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>22</td>\n",
       "      <td>50549676</td>\n",
       "      <td>chr22:50549676:G:A</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.042208</td>\n",
       "      <td>103037</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>-0.015832</td>\n",
       "      <td>0.095091</td>\n",
       "      <td>0.02772</td>\n",
       "      <td>0.061596</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM     GENPOS                                                 ID  \\\n",
       "0      1   46001049                                  chr1:46001049:T:C   \n",
       "1      1   46021880                                  chr1:46021880:G:A   \n",
       "2      1   46027788                                  chr1:46027788:T:G   \n",
       "3      1   46132597                                  chr1:46132597:C:A   \n",
       "4      5     272741                                    chr5:272741:A:G   \n",
       "5      5     272748                                    chr5:272748:G:C   \n",
       "6      5     272755                                    chr5:272755:A:G   \n",
       "7      5   73776529                                  chr5:73776529:T:C   \n",
       "8      5   73780632                                  chr5:73780632:G:A   \n",
       "9      5   73780649                                 chr5:73780649:GT:G   \n",
       "10     5   73780686                                  chr5:73780686:C:A   \n",
       "11     5   73794436                                  chr5:73794436:T:C   \n",
       "12     5   73795301                                  chr5:73795301:T:A   \n",
       "13     5   73795403                                  chr5:73795403:C:T   \n",
       "14     6   43301291                                  chr6:43301291:A:C   \n",
       "15     6   43305866                                  chr6:43305866:A:G   \n",
       "16     6   43308652                                  chr6:43308652:G:A   \n",
       "17     6  158071628                                 chr6:158071628:C:T   \n",
       "18     7   73573024                                  chr7:73573024:C:T   \n",
       "19     7   73605645                                  chr7:73605645:A:G   \n",
       "20    20   62065259                                 chr20:62065259:A:G   \n",
       "21    22   38086345  chr22:38086345:GTGCGGGAGCGGGACTGGCCATCCCAGTACT...   \n",
       "22    22   50549676                                 chr22:50549676:G:A   \n",
       "\n",
       "                                        ALLELE0 ALLELE1    A1FREQ       N  \\\n",
       "0                                             T       C   0.42712   99183   \n",
       "1                                             G       A  0.434137  101574   \n",
       "2                                             T       G  0.437853  102966   \n",
       "3                                             C       A  0.438899  103034   \n",
       "4                                             A       G  0.000539  103025   \n",
       "5                                             G       C  0.000447  103008   \n",
       "6                                             A       G  0.000578  103030   \n",
       "7                                             T       C   0.46584  103031   \n",
       "8                                             G       A  0.461324  102208   \n",
       "9                                            GT       G  0.469417  101396   \n",
       "10                                            C       A  0.500039  103006   \n",
       "11                                            T       C  0.522717  102699   \n",
       "12                                            T       A  0.477072  102733   \n",
       "13                                            C       T  0.477184  102867   \n",
       "14                                            A       C  0.605344  103015   \n",
       "15                                            A       G  0.605361  103036   \n",
       "16                                            G       A  0.603251  100977   \n",
       "17                                            C       T  0.005042  103037   \n",
       "18                                            C       T  0.281331  103037   \n",
       "19                                            A       G  0.279967  102846   \n",
       "20                                            A       G  0.240964   92767   \n",
       "21  GTGCGGGAGCGGGACTGGCCATCCCAGTACTCCGAGGGTGCTA       G  0.290592   98351   \n",
       "22                                            G       A  0.042208  103037   \n",
       "\n",
       "               TEST      BETA        SE     CHISQ    LOG10P EXTRA  \n",
       "0   ADD-INT_SNPxsex  0.036216  0.040272  0.808703  0.433558   NaN  \n",
       "1   ADD-INT_SNPxsex   0.03993  0.039117   1.04196  0.512345   NaN  \n",
       "2   ADD-INT_SNPxsex  0.031215   0.03869   0.65091   0.37697   NaN  \n",
       "3   ADD-INT_SNPxsex  0.033714  0.038751   0.75693   0.41534   NaN  \n",
       "4   ADD-INT_SNPxsex  0.216107  0.491409  0.193397  0.180387   NaN  \n",
       "5   ADD-INT_SNPxsex  -0.38988  0.564171  0.477574  0.310227   NaN  \n",
       "6   ADD-INT_SNPxsex  0.153732  0.474756  0.104855  0.127215   NaN  \n",
       "7   ADD-INT_SNPxsex  0.085258    0.0385   4.90034   1.57103   NaN  \n",
       "8   ADD-INT_SNPxsex  0.079997  0.038733   4.26267    1.4094   NaN  \n",
       "9   ADD-INT_SNPxsex   0.08182  0.038903   4.42026   1.44959   NaN  \n",
       "10  ADD-INT_SNPxsex  0.057869  0.038817   2.22246  0.866408   NaN  \n",
       "11  ADD-INT_SNPxsex  0.080494  0.038596   4.34773   1.43111   NaN  \n",
       "12  ADD-INT_SNPxsex   0.05526  0.038674   2.04167  0.815191   NaN  \n",
       "13  ADD-INT_SNPxsex  0.049815  0.038677   1.65888  0.703873   NaN  \n",
       "14  ADD-INT_SNPxsex  0.059571  0.040074   2.20971  0.862821   NaN  \n",
       "15  ADD-INT_SNPxsex  0.060616  0.040076   2.28769  0.884709   NaN  \n",
       "16  ADD-INT_SNPxsex  0.059866  0.040503   2.18466  0.855761   NaN  \n",
       "17  ADD-INT_SNPxsex  0.117421  0.231253   0.25782  0.213517   NaN  \n",
       "18  ADD-INT_SNPxsex  0.056335  0.042068   1.79329  0.743461   NaN  \n",
       "19  ADD-INT_SNPxsex  0.056899  0.042361   1.80418  0.746643   NaN  \n",
       "20  ADD-INT_SNPxsex  0.066438  0.036573   3.30003    1.1594   NaN  \n",
       "21  ADD-INT_SNPxsex   0.02216  0.043424  0.260436  0.214798   NaN  \n",
       "22  ADD-INT_SNPxsex -0.015832  0.095091   0.02772  0.061596   NaN  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snplist = pd.read_csv(\"~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/vairants_of_interest.txt\",header=None)[0].to_list()\n",
    "result = pd.DataFrame(columns=stat.columns)\n",
    "for i in [1,5,6,7,20,22]:\n",
    "    file = \"~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f3393/cache/ukb23156_c\"+str(i)+\".merged.filtered.bed.sex_f3393.regenie.gz\"\n",
    "    stat = pd.read_csv(file,header=0,sep=\" \")\n",
    "    tmp = stat[stat[\"ID\"].isin(snplist)]\n",
    "    tmp = tmp[tmp[\"TEST\"]==\"ADD-INT_SNPxsex\"]\n",
    "    result = pd.concat([result,tmp], axis=0,ignore_index=True)\n",
    "result"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "7d594c96-0431-456c-9d97-939b12f5da0c",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>ID</th>\n",
       "      <th>TEST</th>\n",
       "      <th>LOG10P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>46001049</td>\n",
       "      <td>chr1:46001049:T:C</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.433558</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>46021880</td>\n",
       "      <td>chr1:46021880:G:A</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.512345</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>46027788</td>\n",
       "      <td>chr1:46027788:T:G</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.37697</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>46132597</td>\n",
       "      <td>chr1:46132597:C:A</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.41534</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>272741</td>\n",
       "      <td>chr5:272741:A:G</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.180387</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>272748</td>\n",
       "      <td>chr5:272748:G:C</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.310227</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5</td>\n",
       "      <td>272755</td>\n",
       "      <td>chr5:272755:A:G</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.127215</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>5</td>\n",
       "      <td>73776529</td>\n",
       "      <td>chr5:73776529:T:C</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>1.57103</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>5</td>\n",
       "      <td>73780632</td>\n",
       "      <td>chr5:73780632:G:A</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>1.4094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>5</td>\n",
       "      <td>73780649</td>\n",
       "      <td>chr5:73780649:GT:G</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>1.44959</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>5</td>\n",
       "      <td>73780686</td>\n",
       "      <td>chr5:73780686:C:A</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.866408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>5</td>\n",
       "      <td>73794436</td>\n",
       "      <td>chr5:73794436:T:C</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>1.43111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5</td>\n",
       "      <td>73795301</td>\n",
       "      <td>chr5:73795301:T:A</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.815191</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>5</td>\n",
       "      <td>73795403</td>\n",
       "      <td>chr5:73795403:C:T</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.703873</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>6</td>\n",
       "      <td>43301291</td>\n",
       "      <td>chr6:43301291:A:C</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.862821</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>6</td>\n",
       "      <td>43305866</td>\n",
       "      <td>chr6:43305866:A:G</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.884709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>6</td>\n",
       "      <td>43308652</td>\n",
       "      <td>chr6:43308652:G:A</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.855761</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>6</td>\n",
       "      <td>158071628</td>\n",
       "      <td>chr6:158071628:C:T</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.213517</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>7</td>\n",
       "      <td>73573024</td>\n",
       "      <td>chr7:73573024:C:T</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.743461</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>7</td>\n",
       "      <td>73605645</td>\n",
       "      <td>chr7:73605645:A:G</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.746643</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>20</td>\n",
       "      <td>62065259</td>\n",
       "      <td>chr20:62065259:A:G</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>1.1594</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>22</td>\n",
       "      <td>38086345</td>\n",
       "      <td>chr22:38086345:GTGCGGGAGCGGGACTGGCCATCCCAGTACT...</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.214798</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>22</td>\n",
       "      <td>50549676</td>\n",
       "      <td>chr22:50549676:G:A</td>\n",
       "      <td>ADD-INT_SNPxsex</td>\n",
       "      <td>0.061596</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM     GENPOS                                                 ID  \\\n",
       "0      1   46001049                                  chr1:46001049:T:C   \n",
       "1      1   46021880                                  chr1:46021880:G:A   \n",
       "2      1   46027788                                  chr1:46027788:T:G   \n",
       "3      1   46132597                                  chr1:46132597:C:A   \n",
       "4      5     272741                                    chr5:272741:A:G   \n",
       "5      5     272748                                    chr5:272748:G:C   \n",
       "6      5     272755                                    chr5:272755:A:G   \n",
       "7      5   73776529                                  chr5:73776529:T:C   \n",
       "8      5   73780632                                  chr5:73780632:G:A   \n",
       "9      5   73780649                                 chr5:73780649:GT:G   \n",
       "10     5   73780686                                  chr5:73780686:C:A   \n",
       "11     5   73794436                                  chr5:73794436:T:C   \n",
       "12     5   73795301                                  chr5:73795301:T:A   \n",
       "13     5   73795403                                  chr5:73795403:C:T   \n",
       "14     6   43301291                                  chr6:43301291:A:C   \n",
       "15     6   43305866                                  chr6:43305866:A:G   \n",
       "16     6   43308652                                  chr6:43308652:G:A   \n",
       "17     6  158071628                                 chr6:158071628:C:T   \n",
       "18     7   73573024                                  chr7:73573024:C:T   \n",
       "19     7   73605645                                  chr7:73605645:A:G   \n",
       "20    20   62065259                                 chr20:62065259:A:G   \n",
       "21    22   38086345  chr22:38086345:GTGCGGGAGCGGGACTGGCCATCCCAGTACT...   \n",
       "22    22   50549676                                 chr22:50549676:G:A   \n",
       "\n",
       "               TEST    LOG10P  \n",
       "0   ADD-INT_SNPxsex  0.433558  \n",
       "1   ADD-INT_SNPxsex  0.512345  \n",
       "2   ADD-INT_SNPxsex   0.37697  \n",
       "3   ADD-INT_SNPxsex   0.41534  \n",
       "4   ADD-INT_SNPxsex  0.180387  \n",
       "5   ADD-INT_SNPxsex  0.310227  \n",
       "6   ADD-INT_SNPxsex  0.127215  \n",
       "7   ADD-INT_SNPxsex   1.57103  \n",
       "8   ADD-INT_SNPxsex    1.4094  \n",
       "9   ADD-INT_SNPxsex   1.44959  \n",
       "10  ADD-INT_SNPxsex  0.866408  \n",
       "11  ADD-INT_SNPxsex   1.43111  \n",
       "12  ADD-INT_SNPxsex  0.815191  \n",
       "13  ADD-INT_SNPxsex  0.703873  \n",
       "14  ADD-INT_SNPxsex  0.862821  \n",
       "15  ADD-INT_SNPxsex  0.884709  \n",
       "16  ADD-INT_SNPxsex  0.855761  \n",
       "17  ADD-INT_SNPxsex  0.213517  \n",
       "18  ADD-INT_SNPxsex  0.743461  \n",
       "19  ADD-INT_SNPxsex  0.746643  \n",
       "20  ADD-INT_SNPxsex    1.1594  \n",
       "21  ADD-INT_SNPxsex  0.214798  \n",
       "22  ADD-INT_SNPxsex  0.061596  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snplist = pd.read_csv(\"~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/vairants_of_interest.txt\",header=None)[0].to_list()\n",
    "result = pd.DataFrame(columns=stat.columns)\n",
    "for i in [1,5,6,7,20,22]:\n",
    "    file = \"~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f3393/cache/ukb23156_c\"+str(i)+\".merged.filtered.bed.sex_f3393.regenie.gz\"\n",
    "    stat = pd.read_csv(file,header=0,sep=\" \")\n",
    "    tmp = stat[stat[\"ID\"].isin(snplist)]\n",
    "    tmp = tmp[tmp[\"TEST\"]==\"ADD-INT_SNPxsex\"]\n",
    "    result = pd.concat([result,tmp], axis=0,ignore_index=True)\n",
    "result[[\"CHROM\",\"GENPOS\",\"ID\",\"TEST\",\"LOG10P\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "id": "9231790f-c1a0-4e9b-9000-672159ee9f2b",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>ID</th>\n",
       "      <th>TEST</th>\n",
       "      <th>LOG10P</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>46001049</td>\n",
       "      <td>chr1:46001049:T:C</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>1.48726</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>46021880</td>\n",
       "      <td>chr1:46021880:G:A</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>1.56105</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>1</td>\n",
       "      <td>46027788</td>\n",
       "      <td>chr1:46027788:T:G</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>1.36076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>1</td>\n",
       "      <td>46132597</td>\n",
       "      <td>chr1:46132597:C:A</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>1.36003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>5</td>\n",
       "      <td>272741</td>\n",
       "      <td>chr5:272741:A:G</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>10.1892</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>5</td>\n",
       "      <td>272748</td>\n",
       "      <td>chr5:272748:G:C</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>7.68709</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>5</td>\n",
       "      <td>272755</td>\n",
       "      <td>chr5:272755:A:G</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>9.95175</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>5</td>\n",
       "      <td>73776529</td>\n",
       "      <td>chr5:73776529:T:C</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>7.7842</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>5</td>\n",
       "      <td>73780632</td>\n",
       "      <td>chr5:73780632:G:A</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>7.26544</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>5</td>\n",
       "      <td>73780649</td>\n",
       "      <td>chr5:73780649:GT:G</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>7.83956</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>5</td>\n",
       "      <td>73780686</td>\n",
       "      <td>chr5:73780686:C:A</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>8.06261</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>5</td>\n",
       "      <td>73794436</td>\n",
       "      <td>chr5:73794436:T:C</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>7.86406</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>5</td>\n",
       "      <td>73795301</td>\n",
       "      <td>chr5:73795301:T:A</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>7.58669</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>5</td>\n",
       "      <td>73795403</td>\n",
       "      <td>chr5:73795403:C:T</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>7.3362</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>6</td>\n",
       "      <td>43301291</td>\n",
       "      <td>chr6:43301291:A:C</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>6.12865</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>60</th>\n",
       "      <td>6</td>\n",
       "      <td>43305866</td>\n",
       "      <td>chr6:43305866:A:G</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>6.07358</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>64</th>\n",
       "      <td>6</td>\n",
       "      <td>43308652</td>\n",
       "      <td>chr6:43308652:G:A</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>6.17006</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>68</th>\n",
       "      <td>6</td>\n",
       "      <td>75362956</td>\n",
       "      <td>chr6:75362956:T:C</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>8.40233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>69</th>\n",
       "      <td>6</td>\n",
       "      <td>75841299</td>\n",
       "      <td>chr6:75841299:A:G</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>13.7617</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>70</th>\n",
       "      <td>6</td>\n",
       "      <td>158071628</td>\n",
       "      <td>chr6:158071628:C:T</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>4.10978</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>74</th>\n",
       "      <td>7</td>\n",
       "      <td>73573024</td>\n",
       "      <td>chr7:73573024:C:T</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>1.76537</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>78</th>\n",
       "      <td>7</td>\n",
       "      <td>73605645</td>\n",
       "      <td>chr7:73605645:A:G</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>1.67518</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>82</th>\n",
       "      <td>20</td>\n",
       "      <td>62065259</td>\n",
       "      <td>chr20:62065259:A:G</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>0.266011</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>86</th>\n",
       "      <td>22</td>\n",
       "      <td>38086345</td>\n",
       "      <td>chr22:38086345:GTGCGGGAGCGGGACTGGCCATCCCAGTACT...</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>4.14478</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>90</th>\n",
       "      <td>22</td>\n",
       "      <td>50549676</td>\n",
       "      <td>chr22:50549676:G:A</td>\n",
       "      <td>ADD-CONDTL</td>\n",
       "      <td>0.862699</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM     GENPOS                                                 ID  \\\n",
       "0      1   46001049                                  chr1:46001049:T:C   \n",
       "4      1   46021880                                  chr1:46021880:G:A   \n",
       "8      1   46027788                                  chr1:46027788:T:G   \n",
       "12     1   46132597                                  chr1:46132597:C:A   \n",
       "16     5     272741                                    chr5:272741:A:G   \n",
       "20     5     272748                                    chr5:272748:G:C   \n",
       "24     5     272755                                    chr5:272755:A:G   \n",
       "28     5   73776529                                  chr5:73776529:T:C   \n",
       "32     5   73780632                                  chr5:73780632:G:A   \n",
       "36     5   73780649                                 chr5:73780649:GT:G   \n",
       "40     5   73780686                                  chr5:73780686:C:A   \n",
       "44     5   73794436                                  chr5:73794436:T:C   \n",
       "48     5   73795301                                  chr5:73795301:T:A   \n",
       "52     5   73795403                                  chr5:73795403:C:T   \n",
       "56     6   43301291                                  chr6:43301291:A:C   \n",
       "60     6   43305866                                  chr6:43305866:A:G   \n",
       "64     6   43308652                                  chr6:43308652:G:A   \n",
       "68     6   75362956                                  chr6:75362956:T:C   \n",
       "69     6   75841299                                  chr6:75841299:A:G   \n",
       "70     6  158071628                                 chr6:158071628:C:T   \n",
       "74     7   73573024                                  chr7:73573024:C:T   \n",
       "78     7   73605645                                  chr7:73605645:A:G   \n",
       "82    20   62065259                                 chr20:62065259:A:G   \n",
       "86    22   38086345  chr22:38086345:GTGCGGGAGCGGGACTGGCCATCCCAGTACT...   \n",
       "90    22   50549676                                 chr22:50549676:G:A   \n",
       "\n",
       "          TEST    LOG10P  \n",
       "0   ADD-CONDTL   1.48726  \n",
       "4   ADD-CONDTL   1.56105  \n",
       "8   ADD-CONDTL   1.36076  \n",
       "12  ADD-CONDTL   1.36003  \n",
       "16  ADD-CONDTL   10.1892  \n",
       "20  ADD-CONDTL   7.68709  \n",
       "24  ADD-CONDTL   9.95175  \n",
       "28  ADD-CONDTL    7.7842  \n",
       "32  ADD-CONDTL   7.26544  \n",
       "36  ADD-CONDTL   7.83956  \n",
       "40  ADD-CONDTL   8.06261  \n",
       "44  ADD-CONDTL   7.86406  \n",
       "48  ADD-CONDTL   7.58669  \n",
       "52  ADD-CONDTL    7.3362  \n",
       "56  ADD-CONDTL   6.12865  \n",
       "60  ADD-CONDTL   6.07358  \n",
       "64  ADD-CONDTL   6.17006  \n",
       "68  ADD-CONDTL   8.40233  \n",
       "69  ADD-CONDTL   13.7617  \n",
       "70  ADD-CONDTL   4.10978  \n",
       "74  ADD-CONDTL   1.76537  \n",
       "78  ADD-CONDTL   1.67518  \n",
       "82  ADD-CONDTL  0.266011  \n",
       "86  ADD-CONDTL   4.14478  \n",
       "90  ADD-CONDTL  0.862699  "
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snplist = pd.read_csv(\"~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/vairants_of_interest.txt\",header=None)[0].to_list()\n",
    "result = pd.DataFrame(columns=stat.columns)\n",
    "for i in [1,5,6,7,20,22]:\n",
    "    file = \"~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f3393/cache/ukb23156_c\"+str(i)+\".merged.filtered.bed.sex_f3393.regenie.gz\"\n",
    "    stat = pd.read_csv(file,header=0,sep=\" \")\n",
    "    tmp = stat[stat[\"ID\"].isin(snplist)]\n",
    "    #tmp = tmp[tmp[\"TEST\"]==\"ADD-INT_SNPxsex\"]\n",
    "    result = pd.concat([result,tmp], axis=0,ignore_index=True)\n",
    "result[result[\"TEST\"]==\"ADD-CONDTL\"][[\"CHROM\",\"GENPOS\",\"ID\",\"TEST\",\"LOG10P\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "a64ca222-2e43-4046-8802-f0ca14b613b6",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(25, 5)"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "result[result[\"TEST\"]==\"ADD-CONDTL\"][[\"CHROM\",\"GENPOS\",\"ID\",\"TEST\",\"LOG10P\"]].shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 41,
   "id": "5143491a-30d3-4ca8-a608-87cbcda4d08c",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "col = result[result[\"TEST\"]==\"ADD-CONDTL\"][[\"CHROM\",\"GENPOS\",\"ID\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "1a741b09-bfa6-4894-a1e4-1fc89a3336bf",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>GENPOS</th>\n",
       "      <th>ID</th>\n",
       "      <th>f3393</th>\n",
       "      <th>f2247</th>\n",
       "      <th>f2257</th>\n",
       "      <th>f2247_f2257</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>46001049</td>\n",
       "      <td>chr1:46001049:T:C</td>\n",
       "      <td>0.368504</td>\n",
       "      <td>0.977637</td>\n",
       "      <td>0.425022</td>\n",
       "      <td>0.97753</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>46021880</td>\n",
       "      <td>chr1:46021880:G:A</td>\n",
       "      <td>0.307365</td>\n",
       "      <td>0.727522</td>\n",
       "      <td>0.238529</td>\n",
       "      <td>0.730976</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>46027788</td>\n",
       "      <td>chr1:46027788:T:G</td>\n",
       "      <td>0.419788</td>\n",
       "      <td>0.937475</td>\n",
       "      <td>0.387529</td>\n",
       "      <td>0.925932</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>46132597</td>\n",
       "      <td>chr1:46132597:C:A</td>\n",
       "      <td>0.384291</td>\n",
       "      <td>0.82291</td>\n",
       "      <td>0.380452</td>\n",
       "      <td>0.886132</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>5</td>\n",
       "      <td>272741</td>\n",
       "      <td>chr5:272741:A:G</td>\n",
       "      <td>0.660105</td>\n",
       "      <td>0.153036</td>\n",
       "      <td>0.249944</td>\n",
       "      <td>0.118207</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>5</td>\n",
       "      <td>272748</td>\n",
       "      <td>chr5:272748:G:C</td>\n",
       "      <td>0.489523</td>\n",
       "      <td>0.142472</td>\n",
       "      <td>0.213779</td>\n",
       "      <td>0.137861</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>5</td>\n",
       "      <td>272755</td>\n",
       "      <td>chr5:272755:A:G</td>\n",
       "      <td>0.746079</td>\n",
       "      <td>0.09104</td>\n",
       "      <td>0.151607</td>\n",
       "      <td>0.06307</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>5</td>\n",
       "      <td>73776529</td>\n",
       "      <td>chr5:73776529:T:C</td>\n",
       "      <td>0.026852</td>\n",
       "      <td>0.14838</td>\n",
       "      <td>0.359496</td>\n",
       "      <td>0.056573</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>5</td>\n",
       "      <td>73780632</td>\n",
       "      <td>chr5:73780632:G:A</td>\n",
       "      <td>0.038958</td>\n",
       "      <td>0.273677</td>\n",
       "      <td>0.516853</td>\n",
       "      <td>0.108467</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>5</td>\n",
       "      <td>73780649</td>\n",
       "      <td>chr5:73780649:GT:G</td>\n",
       "      <td>0.035515</td>\n",
       "      <td>0.191505</td>\n",
       "      <td>0.489894</td>\n",
       "      <td>0.052426</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>5</td>\n",
       "      <td>73780686</td>\n",
       "      <td>chr5:73780686:C:A</td>\n",
       "      <td>0.136017</td>\n",
       "      <td>0.269732</td>\n",
       "      <td>0.672063</td>\n",
       "      <td>0.08785</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>5</td>\n",
       "      <td>73794436</td>\n",
       "      <td>chr5:73794436:T:C</td>\n",
       "      <td>0.037059</td>\n",
       "      <td>0.386213</td>\n",
       "      <td>0.852975</td>\n",
       "      <td>0.240332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>5</td>\n",
       "      <td>73795301</td>\n",
       "      <td>chr5:73795301:T:A</td>\n",
       "      <td>0.153041</td>\n",
       "      <td>0.67078</td>\n",
       "      <td>0.875821</td>\n",
       "      <td>0.358223</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>5</td>\n",
       "      <td>73795403</td>\n",
       "      <td>chr5:73795403:C:T</td>\n",
       "      <td>0.197755</td>\n",
       "      <td>0.686218</td>\n",
       "      <td>0.874326</td>\n",
       "      <td>0.381449</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>6</td>\n",
       "      <td>43301291</td>\n",
       "      <td>chr6:43301291:A:C</td>\n",
       "      <td>0.137145</td>\n",
       "      <td>0.477044</td>\n",
       "      <td>0.8604</td>\n",
       "      <td>0.429153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>6</td>\n",
       "      <td>43305866</td>\n",
       "      <td>chr6:43305866:A:G</td>\n",
       "      <td>0.130404</td>\n",
       "      <td>0.469152</td>\n",
       "      <td>0.857057</td>\n",
       "      <td>0.417968</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>6</td>\n",
       "      <td>43308652</td>\n",
       "      <td>chr6:43308652:G:A</td>\n",
       "      <td>0.139392</td>\n",
       "      <td>0.442898</td>\n",
       "      <td>0.776614</td>\n",
       "      <td>0.361744</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>6</td>\n",
       "      <td>75362956</td>\n",
       "      <td>chr6:75362956:T:C</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.57773</td>\n",
       "      <td>0.527134</td>\n",
       "      <td>0.661503</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>6</td>\n",
       "      <td>75841299</td>\n",
       "      <td>chr6:75841299:A:G</td>\n",
       "      <td>NaN</td>\n",
       "      <td>0.675966</td>\n",
       "      <td>0.634579</td>\n",
       "      <td>0.677996</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>6</td>\n",
       "      <td>158071628</td>\n",
       "      <td>chr6:158071628:C:T</td>\n",
       "      <td>0.611622</td>\n",
       "      <td>0.380353</td>\n",
       "      <td>0.853001</td>\n",
       "      <td>0.257583</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>7</td>\n",
       "      <td>73573024</td>\n",
       "      <td>chr7:73573024:C:T</td>\n",
       "      <td>0.180526</td>\n",
       "      <td>0.00138</td>\n",
       "      <td>0.038382</td>\n",
       "      <td>0.007663</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>7</td>\n",
       "      <td>73605645</td>\n",
       "      <td>chr7:73605645:A:G</td>\n",
       "      <td>0.179208</td>\n",
       "      <td>0.001001</td>\n",
       "      <td>0.043188</td>\n",
       "      <td>0.0071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>20</td>\n",
       "      <td>62065259</td>\n",
       "      <td>chr20:62065259:A:G</td>\n",
       "      <td>0.069279</td>\n",
       "      <td>0.167429</td>\n",
       "      <td>0.544786</td>\n",
       "      <td>0.313764</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>22</td>\n",
       "      <td>38086345</td>\n",
       "      <td>chr22:38086345:GTGCGGGAGCGGGACTGGCCATCCCAGTACT...</td>\n",
       "      <td>0.60982</td>\n",
       "      <td>0.480117</td>\n",
       "      <td>0.182525</td>\n",
       "      <td>0.682736</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>22</td>\n",
       "      <td>50549676</td>\n",
       "      <td>chr22:50549676:G:A</td>\n",
       "      <td>0.867769</td>\n",
       "      <td>0.950848</td>\n",
       "      <td>0.815275</td>\n",
       "      <td>0.841334</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   CHROM     GENPOS                                                 ID  \\\n",
       "0      1   46001049                                  chr1:46001049:T:C   \n",
       "1      1   46021880                                  chr1:46021880:G:A   \n",
       "2      1   46027788                                  chr1:46027788:T:G   \n",
       "3      1   46132597                                  chr1:46132597:C:A   \n",
       "4      5     272741                                    chr5:272741:A:G   \n",
       "5      5     272748                                    chr5:272748:G:C   \n",
       "6      5     272755                                    chr5:272755:A:G   \n",
       "7      5   73776529                                  chr5:73776529:T:C   \n",
       "8      5   73780632                                  chr5:73780632:G:A   \n",
       "9      5   73780649                                 chr5:73780649:GT:G   \n",
       "10     5   73780686                                  chr5:73780686:C:A   \n",
       "11     5   73794436                                  chr5:73794436:T:C   \n",
       "12     5   73795301                                  chr5:73795301:T:A   \n",
       "13     5   73795403                                  chr5:73795403:C:T   \n",
       "14     6   43301291                                  chr6:43301291:A:C   \n",
       "15     6   43305866                                  chr6:43305866:A:G   \n",
       "16     6   43308652                                  chr6:43308652:G:A   \n",
       "17     6   75362956                                  chr6:75362956:T:C   \n",
       "18     6   75841299                                  chr6:75841299:A:G   \n",
       "19     6  158071628                                 chr6:158071628:C:T   \n",
       "20     7   73573024                                  chr7:73573024:C:T   \n",
       "21     7   73605645                                  chr7:73605645:A:G   \n",
       "22    20   62065259                                 chr20:62065259:A:G   \n",
       "23    22   38086345  chr22:38086345:GTGCGGGAGCGGGACTGGCCATCCCAGTACT...   \n",
       "24    22   50549676                                 chr22:50549676:G:A   \n",
       "\n",
       "       f3393     f2247     f2257 f2247_f2257  \n",
       "0   0.368504  0.977637  0.425022     0.97753  \n",
       "1   0.307365  0.727522  0.238529    0.730976  \n",
       "2   0.419788  0.937475  0.387529    0.925932  \n",
       "3   0.384291   0.82291  0.380452    0.886132  \n",
       "4   0.660105  0.153036  0.249944    0.118207  \n",
       "5   0.489523  0.142472  0.213779    0.137861  \n",
       "6   0.746079   0.09104  0.151607     0.06307  \n",
       "7   0.026852   0.14838  0.359496    0.056573  \n",
       "8   0.038958  0.273677  0.516853    0.108467  \n",
       "9   0.035515  0.191505  0.489894    0.052426  \n",
       "10  0.136017  0.269732  0.672063     0.08785  \n",
       "11  0.037059  0.386213  0.852975    0.240332  \n",
       "12  0.153041   0.67078  0.875821    0.358223  \n",
       "13  0.197755  0.686218  0.874326    0.381449  \n",
       "14  0.137145  0.477044    0.8604    0.429153  \n",
       "15  0.130404  0.469152  0.857057    0.417968  \n",
       "16  0.139392  0.442898  0.776614    0.361744  \n",
       "17       NaN   0.57773  0.527134    0.661503  \n",
       "18       NaN  0.675966  0.634579    0.677996  \n",
       "19  0.611622  0.380353  0.853001    0.257583  \n",
       "20  0.180526   0.00138  0.038382    0.007663  \n",
       "21  0.179208  0.001001  0.043188      0.0071  \n",
       "22  0.069279  0.167429  0.544786    0.313764  \n",
       "23   0.60982  0.480117  0.182525    0.682736  \n",
       "24  0.867769  0.950848  0.815275    0.841334  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snplist = pd.read_csv(\"~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/vairants_of_interest.txt\",header=None)[0].to_list()\n",
    "# f3393\n",
    "result = pd.DataFrame(columns=stat.columns)\n",
    "for i in [1,5,6,7,20,22]:\n",
    "    file = \"~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f3393/cache/ukb23156_c\"+str(i)+\".merged.filtered.bed.sex_f3393.regenie.gz\"\n",
    "    stat = pd.read_csv(file,header=0,sep=\" \")\n",
    "    tmp = stat[stat[\"ID\"].isin(snplist)]\n",
    "    tmp = tmp[tmp[\"TEST\"]==\"ADD-INT_SNPxsex\"]\n",
    "    result = pd.concat([result,tmp], axis=0,ignore_index=True)\n",
    "col = col.merge(result[[\"ID\",\"LOG10P\"]],how=\"left\",left_on=\"ID\",right_on=\"ID\")\n",
    "col = col.rename(columns={\"LOG10P\":\"f3393\"})\n",
    "# f2247\n",
    "result = pd.DataFrame(columns=stat.columns)\n",
    "for i in [1,5,6,7,20,22]:\n",
    "    file = \"~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f2247/cache/ukb23156_c\"+str(i)+\".merged.filtered.bed.sex_f2247.regenie.gz\"\n",
    "    stat = pd.read_csv(file,header=0,sep=\" \")\n",
    "    tmp = stat[stat[\"ID\"].isin(snplist)]\n",
    "    tmp = tmp[tmp[\"TEST\"]==\"ADD-INT_SNPxsex\"]\n",
    "    result = pd.concat([result,tmp], axis=0,ignore_index=True)\n",
    "col = col.merge(result[[\"ID\",\"LOG10P\"]],how=\"left\",left_on=\"ID\",right_on=\"ID\")\n",
    "col= col.rename(columns={\"LOG10P\":\"f2247\"})\n",
    "# f2257\n",
    "result = pd.DataFrame(columns=stat.columns)\n",
    "for i in [1,5,6,7,20,22]:\n",
    "    file = \"~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f2257/cache/ukb23156_c\"+str(i)+\".merged.filtered.bed.sex_f2257.regenie.gz\"\n",
    "    stat = pd.read_csv(file,header=0,sep=\" \")\n",
    "    tmp = stat[stat[\"ID\"].isin(snplist)]\n",
    "    tmp = tmp[tmp[\"TEST\"]==\"ADD-INT_SNPxsex\"]\n",
    "    result = pd.concat([result,tmp], axis=0,ignore_index=True)\n",
    "col = col.merge(result[[\"ID\",\"LOG10P\"]],how=\"left\",left_on=\"ID\",right_on=\"ID\")\n",
    "col= col.rename(columns={\"LOG10P\":\"f2257\"})\n",
    "# f2247_f2257\n",
    "result = pd.DataFrame(columns=stat.columns)\n",
    "for i in [1,5,6,7,20,22]:\n",
    "    file = \"~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f2247_f2257/cache/ukb23156_c\"+str(i)+\".merged.filtered.bed.sex_f2247_f2257.regenie.gz\"\n",
    "    stat = pd.read_csv(file,header=0,sep=\" \")\n",
    "    tmp = stat[stat[\"ID\"].isin(snplist)]\n",
    "    tmp = tmp[tmp[\"TEST\"]==\"ADD-INT_SNPxsex\"]\n",
    "    result = pd.concat([result,tmp], axis=0,ignore_index=True)\n",
    "col = col.merge(result[[\"ID\",\"LOG10P\"]],how=\"left\",left_on=\"ID\",right_on=\"ID\")\n",
    "col= col.rename(columns={\"LOG10P\":\"f2247_f2257\"})\n",
    "col.iloc[:,3:] = 10**(-col.iloc[:,3:])\n",
    "col"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "id": "4465d9a6-f93d-453c-af46-87477fe6ed59",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.002"
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "0.05/25"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "id": "8fb15934-978a-443c-80d1-6c0ccd664360",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "col.to_csv(\"~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/vof.inter.sex.p.csv\",index=False)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0c6f1a75-313a-428f-8454-80cfbfab5b86",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "source": [
    "## Burden test (discovered genes * sex)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "36d3b9b3-b55b-4498-9808-390b7aa9baa2",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "# Common variables Columbia's cluster\n",
    "UKBB_PATH=$HOME/UKBiobank\n",
    "UKBB_yale=$HOME/UKBiobank_Yale_transfer\n",
    "USER_PATH=$HOME/working\n",
    "container_lmm=$HOME/containers/lmm.sif\n",
    "container_marp=$HOME/containers/marp.sif\n",
    "container_annovar=$HOME/containers/gatk4-annovar.sif\n",
    "hearing_pheno_path=$UKBB_PATH/phenotype_files/hearing_impairment\n",
    "tpl_file=$USER_PATH/bioworkflows/admin/csg.yml\n",
    "formatFile_fastgwa=$USER_PATH/UKBB_GWAS_dev/data/fastGWA_template.yml\n",
    "formatFile_bolt=$USER_PATH/UKBB_GWAS_dev/data/boltlmm_template.yml\n",
    "formatFile_saige=$USER_PATH/UKBB_GWAS_dev/data/saige_template.yml\n",
    "formatFile_regenie=$USER_PATH/UKBB_GWAS_dev/data/regenie_template.yml\n",
    "# Workflows\n",
    "lmm_sos=$USER_PATH/bioworkflows/GWAS/LMM.ipynb\n",
    "anno_sos=$USER_PATH/bioworkflows/variant-annotation/annovar.ipynb\n",
    "clumping_sos=$USER_PATH/bioworkflows/GWAS/LD_Clumping.ipynb\n",
    "extract_sos=$USER_PATH/bioworkflows/GWAS/Region_Extraction.ipynb\n",
    "snptogene_sos=$USER_PATH/UKBB_GWAS_dev/workflow/snptogene.ipynb\n",
    "### Specific to REGENIE\n",
    "bsize=1000\n",
    "trait=bt\n",
    "bgenMinMAF=0.001\n",
    "bgenMinINFO=0.8\n",
    "minMAC=4\n",
    "maf_filter=0.01\n",
    "geno_filter=0.01\n",
    "hwe_filter=0\n",
    "mind_filter=0.1\n",
    "reverse_log_p=True\n",
    "numThreads=2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "b4420892-96cd-4535-af09-0dbccdc19fed",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/burden/f3393/f3393_sex_interaction-regenie_2022-05-18.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w4a342b370c30b54b) is executed successfully with 1 completed step.\n",
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/burden/f2247/f2247_sex_interaction-regenie_2022-05-18.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w44cc62da23f9a14a) is executed successfully with 1 completed step.\n",
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/burden/f2257/f2257_sex_interaction-regenie_2022-05-18.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=w6cb3369d3204eca0) is executed successfully with 1 completed step.\n",
      "INFO: Running \u001b[32mcsg\u001b[0m: Configuration for Columbia csg partition cluster\n",
      "INFO: \u001b[32mcsg\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mcsg\u001b[0m output:   \u001b[32m/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/burden/f2247_f2257/f2247_f2257_sex_interaction-regenie_2022-05-18.sbatch\u001b[0m\n",
      "INFO: Workflow csg (ID=wb8fdf25b4f9aaaa2) is executed successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "# 200k exome\n",
    "# f3393\n",
    "lmm_dir_regenie=/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/burden/f3393\n",
    "lmm_sbatch_regenie=$lmm_dir_regenie/f3393_sex_interaction-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "phenoFile=/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/090321_UKBB_Hearing_aid_f3393_expandedwhite_6436cases_96601ctrl_PC1_2.tsv\n",
    "phenoCol=f3393\n",
    "covarCol=age\n",
    "qCovarCol=\"PC1 PC2\"\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=/home/gl2776/UKBiobank/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the gene bfile here for the association\n",
    "genoFile=\"~/project/guangyou/epistasis/chr1/ukb23155_c1_b0_v1.filtered.gene.bed\"\n",
    "for i in {2..22}\n",
    "do\n",
    "genoFile=`echo \"${genoFile} ~/project/guangyou/epistasis/chr${i}/ukb23155_c${i}_b0_v1.filtered.gene.bed\"`\n",
    "done\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "lmm_job_size=20\n",
    "regenie_sos=~/working/interaction/newv/Regenie_interaction_GxE.ipynb\n",
    "regenieFile=/mnt/mfs/statgen/guangyou/containers/regenie.v.2.3\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "covariates=sex\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --regenieFile $regenieFile\n",
    "    --sampleFile $sampleFile\n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --covariates $covariates\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $regenie_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" \n",
    "\n",
    "# f2247\n",
    "lmm_dir_regenie=/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/burden/f2247\n",
    "lmm_sbatch_regenie=$lmm_dir_regenie/f2247_sex_interaction-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "phenoFile=/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/090321_UKBB_Hearing_difficulty_f2247_expandedwhite_45502cases_96601ctrl_PC1_2.tsv\n",
    "phenoCol=f2247\n",
    "covarCol=age\n",
    "qCovarCol=\"PC1 PC2\"\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=/home/gl2776/UKBiobank/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the gene bfile here for the association\n",
    "genoFile=\"~/project/guangyou/epistasis/chr1/ukb23155_c1_b0_v1.filtered.gene.bed\"\n",
    "for i in {2..22}\n",
    "do\n",
    "genoFile=`echo \"${genoFile} ~/project/guangyou/epistasis/chr${i}/ukb23155_c${i}_b0_v1.filtered.gene.bed\"`\n",
    "done\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "lmm_job_size=20\n",
    "regenie_sos=~/working/interaction/newv/Regenie_interaction_GxE.ipynb\n",
    "regenieFile=/mnt/mfs/statgen/guangyou/containers/regenie.v.2.3\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "covariates=sex\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --regenieFile $regenieFile\n",
    "    --sampleFile $sampleFile\n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --covariates $covariates\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $regenie_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" \n",
    "\n",
    "# f2257\n",
    "lmm_dir_regenie=/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/burden/f2257\n",
    "lmm_sbatch_regenie=$lmm_dir_regenie/f2257_sex_interaction-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "phenoFile=/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/090321_UKBB_Hearing_noise_f2257_expandedwhite_65660cases_96601ctrl_PC1_2.tsv\n",
    "phenoCol=f2257\n",
    "covarCol=age\n",
    "qCovarCol=\"PC1 PC2\"\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=/home/gl2776/UKBiobank/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the gene bfile here for the association\n",
    "genoFile=\"~/project/guangyou/epistasis/chr1/ukb23155_c1_b0_v1.filtered.gene.bed\"\n",
    "for i in {2..22}\n",
    "do\n",
    "genoFile=`echo \"${genoFile} ~/project/guangyou/epistasis/chr${i}/ukb23155_c${i}_b0_v1.filtered.gene.bed\"`\n",
    "done\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "lmm_job_size=20\n",
    "regenie_sos=~/working/interaction/newv/Regenie_interaction_GxE.ipynb\n",
    "regenieFile=/mnt/mfs/statgen/guangyou/containers/regenie.v.2.3\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "covariates=sex\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --regenieFile $regenieFile\n",
    "    --sampleFile $sampleFile\n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --covariates $covariates\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $regenie_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" \n",
    "\n",
    "\n",
    "# f2247_f2257\n",
    "lmm_dir_regenie=/home/gl2776/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/burden/f2247_f2257\n",
    "lmm_sbatch_regenie=$lmm_dir_regenie/f2247_f2257_sex_interaction-regenie_$(date +\"%Y-%m-%d\").sbatch\n",
    "phenoFile=/home/gl2776/UKBiobank/phenotype_files/hearing_impairment/090321_UKBB_Combined_f2247_f2257_expandedwhite_38410cases_96601ctrl_PC1_2.tsv\n",
    "phenoCol=f2247_f2257\n",
    "covarCol=age\n",
    "qCovarCol=\"PC1 PC2\"\n",
    "#Use the original bed files that passed QC using Megan's parameters geno=0.01, mind=0.1, maf=0.01, hwe=5e-08\n",
    "bfile=/home/gl2776/UKBiobank/genotype_files_processed/090221_sample_variant_qc_final_callrate90/cache/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.bed\n",
    "# Use the gene bfile here for the association\n",
    "genoFile=\"~/project/guangyou/epistasis/chr1/ukb23155_c1_b0_v1.filtered.gene.bed\"\n",
    "for i in {2..22}\n",
    "do\n",
    "genoFile=`echo \"${genoFile} ~/project/guangyou/epistasis/chr${i}/ukb23155_c${i}_b0_v1.filtered.gene.bed\"`\n",
    "done\n",
    "## After doing the annotation you can create the annotatted manhatan plot\n",
    "lmm_job_size=20\n",
    "regenie_sos=~/working/interaction/newv/Regenie_interaction_GxE.ipynb\n",
    "regenieFile=/mnt/mfs/statgen/guangyou/containers/regenie.v.2.3\n",
    "lowmem_dir=$lmm_dir_regenie/scratch60/\n",
    "covariates=sex\n",
    "# If --annotate then it will add the label to the plot otherwise --no-annotate\n",
    "\n",
    "lmm_args=\"\"\"regenie\n",
    "    --cwd $lmm_dir_regenie \n",
    "    --bfile $bfile \n",
    "    --genoFile $genoFile\n",
    "    --phenoFile $phenoFile \n",
    "    --formatFile $formatFile_regenie \n",
    "    --regenieFile $regenieFile\n",
    "    --sampleFile $sampleFile\n",
    "    --phenoCol $phenoCol\n",
    "    --covarCol $covarCol  \n",
    "    --qCovarCol $qCovarCol\n",
    "    --bsize $bsize\n",
    "    --lowmem_dir $lowmem_dir\n",
    "    --trait $trait\n",
    "    --covariates $covariates\n",
    "    --bgenMinMAF $bgenMinMAF \n",
    "    --bgenMinINFO $bgenMinINFO\n",
    "    --maf_filter $maf_filter\n",
    "    --geno_filter $geno_filter\n",
    "    --hwe_filter $hwe_filter\n",
    "    --mind_filter $mind_filter\n",
    "    --minMAC $minMAC\n",
    "    --job_size $lmm_job_size\n",
    "    --container_lmm $container_lmm\n",
    "    --container_marp $container_marp\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/working/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg\\\n",
    "    --template-file $tpl_file \\\n",
    "    --workflow-file $regenie_sos \\\n",
    "    --to-script $lmm_sbatch_regenie \\\n",
    "    --args \"$lmm_args\" "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "id": "a489768f-78e2-4e20-bd57-88021ef09258",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "from=~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f3393/\n",
    "to=~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/burden/f3393/\n",
    "cp -r $from/090321* $to/\n",
    "cp -r $from/cache/UKB_geno* $to/cache/\n",
    "from=~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f2247/\n",
    "to=~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/burden/f2247/\n",
    "cp -r $from/090321* $to/\n",
    "cp -r $from/cache/UKB_geno* $to/cache/\n",
    "from=~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f2257/\n",
    "to=~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/burden/f2257/\n",
    "cp -r $from/090321* $to/\n",
    "cp -r $from/cache/UKB_geno* $to/cache/\n",
    "from=~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/f2247_f2257/\n",
    "to=~/UKBiobank/results/REGENIE_results/results_exome_data/051722_interaction_sex/burden/f2247_f2257/\n",
    "cp -r $from/090321* $to/\n",
    "cp -r $from/cache/UKB_geno* $to/cache/"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e2185dcf-a1c0-4b8e-ab8c-4727a432768d",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SoS",
   "language": "sos",
   "name": "sos"
  },
  "language_info": {
   "codemirror_mode": "sos",
   "file_extension": ".sos",
   "mimetype": "text/x-sos",
   "name": "sos",
   "nbconvert_exporter": "sos_notebook.converter.SoS_Exporter",
   "pygments_lexer": "sos"
  },
  "sos": {
   "kernels": [
    [
     "Bash",
     "bash",
     "Bash",
     "#E6EEFF",
     ""
    ],
    [
     "Python3",
     "python3",
     "Python3",
     "#FFD91A",
     ""
    ]
   ],
   "version": "0.23.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
