{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "41c3b683-12a0-4eeb-b641-350529df6ccd",
   "metadata": {
    "kernel": "SoS"
   },
   "source": [
    "## Redo the *z file so that it does not include rsid in the variant column "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "2bee6f3b-7fec-46bd-8ab1-74bc4b17730d",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Creation of the z file: it has to be space delimited and not tab delimited\n",
    "import glob, os, sys\n",
    "import pandas as pd\n",
    "log_file= open('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/regions1_22.log', \"w\")\n",
    "regions=glob.glob('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/*.variants', recursive=True)\n",
    "mfi_files=glob.glob('/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr*[0-9]_v3.txt', recursive=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "652a96b2-5e27-47df-8359-92d8a1eef87a",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_16050408_17674295.variants\n",
      "22_16050408_17674295.variants\n",
      "The number of variants in the region is: 38544\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_17674295_18296088.variants\n",
      "22_17674295_18296088.variants\n",
      "The number of variants in the region is: 24377\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_18296088_19912358.variants\n",
      "22_18296088_19912358.variants\n",
      "The number of variants in the region is: 54633\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_19912358_22357325.variants\n",
      "22_19912358_22357325.variants\n",
      "The number of variants in the region is: 65393\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_22357325_23712647.variants\n",
      "22_22357325_23712647.variants\n",
      "The number of variants in the region is: 55506\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_23712647_24984204.variants\n",
      "22_23712647_24984204.variants\n",
      "The number of variants in the region is: 46195\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_24984204_26791628.variants\n",
      "22_24984204_26791628.variants\n",
      "The number of variants in the region is: 66302\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_26791628_27834752.variants\n",
      "22_26791628_27834752.variants\n",
      "The number of variants in the region is: 39498\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_27834752_29651799.variants\n",
      "22_27834752_29651799.variants\n",
      "The number of variants in the region is: 56374\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_29651799_31439918.variants\n",
      "22_29651799_31439918.variants\n",
      "The number of variants in the region is: 58345\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_31439918_32664986.variants\n",
      "22_31439918_32664986.variants\n",
      "The number of variants in the region is: 39618\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_32664986_34494914.variants\n",
      "22_32664986_34494914.variants\n",
      "The number of variants in the region is: 65743\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_34494914_35530985.variants\n",
      "22_34494914_35530985.variants\n",
      "The number of variants in the region is: 40322\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_35530985_37570269.variants\n",
      "22_35530985_37570269.variants\n",
      "The number of variants in the region is: 74372\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_37570269_39307894.variants\n",
      "22_37570269_39307894.variants\n",
      "The number of variants in the region is: 60893\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_39307894_40545797.variants\n",
      "22_39307894_40545797.variants\n",
      "The number of variants in the region is: 43309\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_40545797_42690818.variants\n",
      "22_40545797_42690818.variants\n",
      "The number of variants in the region is: 69192\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_42690818_43714200.variants\n",
      "22_42690818_43714200.variants\n",
      "The number of variants in the region is: 39944\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_43714200_44995308.variants\n",
      "22_43714200_44995308.variants\n",
      "The number of variants in the region is: 49027\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_44995308_46470495.variants\n",
      "22_44995308_46470495.variants\n",
      "The number of variants in the region is: 60203\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_46470495_47596318.variants\n",
      "22_46470495_47596318.variants\n",
      "The number of variants in the region is: 49047\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_47596318_48903703.variants\n",
      "22_47596318_48903703.variants\n",
      "The number of variants in the region is: 56612\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_48903703_49824534.variants\n",
      "22_48903703_49824534.variants\n",
      "The number of variants in the region is: 45195\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/22_49824534_51243298.variants\n",
      "22_49824534_51243298.variants\n",
      "The number of variants in the region is: 57054\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr22_v3.txt is: 1255683\n"
     ]
    }
   ],
   "source": [
    "for filename in regions:\n",
    "    print(filename)\n",
    "    basename=os.path.basename(filename)\n",
    "    sp = basename.split(\"_\")[0]\n",
    "    chrom= sp.lstrip(\"0\")\n",
    "    print(basename)\n",
    "    fullregion = pd.read_csv(filename,header=0,sep=\"\\t\", skiprows=1, skipfooter=1, names=['alternate_ids', 'rsid', 'chromosome', 'position', 'number_of_alleles', 'allele1', 'allele2'])\n",
    "    print('The number of variants in the region is:',fullregion.shape[0])\n",
    "    mfi_file = [file for file in mfi_files if \"chr\" + chrom + \"_\" in file][0]\n",
    "    chrom_file = pd.read_csv(mfi_file,header=None,sep=\"\\t\", names=[\"alternate_id\", \"rsid\", \"position\", \"allele1\", \"allele2\", \"maf\", \"minor_allele\", \"info_score\"])\n",
    "    print('The number of variants in',mfi_file,'is:',chrom_file.shape[0])\n",
    "    subset_region=chrom_file.join(fullregion.set_index(['alternate_ids', 'rsid', 'position', 'allele1', 'allele2']), on=['alternate_id', 'rsid', 'position', 'allele1', 'allele2'], how='right')\n",
    "    maf=0.001\n",
    "    subset_maf=subset_region[subset_region['maf'] > maf]\n",
    "    #print(subset_maf)\n",
    "    #df=subset_maf.copy()\n",
    "    #df= df.drop('rsid', axis=1)\n",
    "    #df[\"rsid\"]=df[\"chromosome\"].astype(str) + \"_\" + df[\"position\"].astype(str) + \"_\" + df[\"allele1\"] + \"_\" + df[\"allele2\"]\n",
    "    #print(df)\n",
    "    subset_maf[[\"rsid\",\"chromosome\", \"position\", \"allele1\", \"allele2\", \"alternate_id\"]].to_csv(filename + '_' + str(maf) + '.z', sep=' ', index=False, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "c9d7a8bd-2354-4e17-a4b6-3dfae44f0b8c",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3441: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    }
   ],
   "source": [
    "master_list = pd.read_csv(\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/22/masterfile22_v0.001_list_unrelated_whiteEur.txt\",\";\")\n",
    "#input_list = master_list.iloc[:,[0,4]].values.tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "5c5bbe2b-4d7e-4614-a229-c4bcd3295b07",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>z</th>\n",
       "      <th>bgen</th>\n",
       "      <th>bgi</th>\n",
       "      <th>bcor</th>\n",
       "      <th>ld</th>\n",
       "      <th>n_samples</th>\n",
       "      <th>sample</th>\n",
       "      <th>incl</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "      <td>351430</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...</td>\n",
       "      <td>/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                    z  \\\n",
       "0   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "1   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "2   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "3   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "4   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "5   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "6   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "7   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "8   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "9   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "10  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "11  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "12  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "13  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "14  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "15  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "16  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "17  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "18  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "19  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "20  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "21  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "22  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "23  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "\n",
       "                                                 bgen  \\\n",
       "0   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "1   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "2   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "3   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "4   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "5   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "6   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "7   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "8   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "9   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "10  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "11  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "12  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "13  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "14  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "15  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "16  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "17  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "18  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "19  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "20  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "21  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "22  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "23  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "\n",
       "                                                  bgi  \\\n",
       "0   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "1   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "2   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "3   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "4   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "5   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "6   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "7   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "8   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "9   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "10  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "11  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "12  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "13  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "14  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "15  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "16  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "17  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "18  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "19  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "20  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "21  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "22  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "23  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "\n",
       "                                                 bcor  \\\n",
       "0   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "1   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "2   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "3   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "4   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "5   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "6   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "7   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "8   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "9   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "10  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "11  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "12  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "13  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "14  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "15  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "16  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "17  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "18  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "19  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "20  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "21  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "22  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "23  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...   \n",
       "\n",
       "                                                   ld  n_samples  \\\n",
       "0   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "1   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "2   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "3   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "4   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "5   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "6   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "7   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "8   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "9   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "10  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "11  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "12  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "13  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "14  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "15  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "16  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "17  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "18  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "19  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "20  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "21  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "22  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "23  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...     351430   \n",
       "\n",
       "                                               sample  \\\n",
       "0   /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "1   /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "2   /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "3   /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "4   /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "5   /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "6   /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "7   /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "8   /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "9   /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "10  /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "11  /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "12  /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "13  /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "14  /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "15  /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "16  /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "17  /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "18  /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "19  /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "20  /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "21  /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "22  /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "23  /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb3...   \n",
       "\n",
       "                                                 incl  \n",
       "0   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "1   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "2   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "3   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "4   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "5   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "6   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "7   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "8   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "9   /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "10  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "11  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "12  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "13  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "14  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "15  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "16  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "17  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "18  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "19  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "20  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "21  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "22  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  \n",
       "23  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy...  "
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "master_list"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4367494c-3273-4ecc-90d0-4f24448d24af",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "## Read bgen files to figure out problem with variants name"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "4cb1ec03-f546-4e64-bbfe-aca807e3c25f",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/runpy.py\", line 194, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/site-packages/ipykernel_launcher.py\", line 15, in <module>\n",
      "    from ipykernel import kernelapp as app\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/site-packages/ipykernel/kernelapp.py\", line 21, in <module>\n",
      "    from IPython.core.application import (\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/site-packages/IPython/__init__.py\", line 61, in <module>\n",
      "    from .utils.frame import extract_module_locals\n",
      "  File \"<frozen importlib._bootstrap>\", line 991, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 975, in _find_and_load_unlocked\n",
      "  File \"<frozen importlib._bootstrap>\", line 671, in _load_unlocked\n",
      "  File \"<frozen importlib._bootstrap_external>\", line 779, in exec_module\n",
      "  File \"<frozen importlib._bootstrap_external>\", line 874, in get_code\n",
      "  File \"<frozen importlib._bootstrap_external>\", line 973, in get_data\n",
      "KeyboardInterrupt\n",
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/sos_notebook/kernel.py:1334: DeprecationWarning: Kernel._parent_header is deprecated in ipykernel 6. Use .get_parent()\n",
      "  msg['msg_id'] = self._parent_header['header']['msg_id']\n"
     ]
    }
   ],
   "source": [
    "from bgen_reader import example_filepath"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 141,
   "id": "aea745ca-3a0c-4d2e-b3c2-3e070d68e35c",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "from bgen_reader import open_bgen \n",
    "from bgen_reader import read_bgen"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "db434c8c-d8de-4f89-8bd3-9afb9112dad1",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "bgen_file = '/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_31571218_32682664.bgen'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 142,
   "id": "332a392c-3779-4b48-a161-79edac941933",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "bgen_file = '/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.bgen'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 155,
   "id": "c73f1049-1101-444a-9d71-9e5cf1c169d3",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "bgen = open_bgen(bgen_file, verbose=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 157,
   "id": "3306c2bd-fb5f-4c8d-975a-b15730520cd8",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "bgen = read_bgen(bgen_file, verbose=False)\n",
    "\n",
    "variants = bgen[\"variants\"]\n",
    "\n",
    "samples = bgen[\"samples\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 149,
   "id": "a01e33bc-1a0d-4a2d-b5c3-277ca5506784",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "v = variants.compute()\n",
    "\n",
    "g = bgen[\"genotype\"][0].compute()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "id": "2eaa03a7-4bbe-48f2-8196-3e0f0263e7c8",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "print(v)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 151,
   "id": "2c705bfe-eca0-4a18-99c5-cd4bef4a9fe5",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>nalleles</th>\n",
       "      <th>allele_ids</th>\n",
       "      <th>vaddr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [id, rsid, chrom, pos, nalleles, allele_ids, vaddr]\n",
       "Index: []"
      ]
     },
     "execution_count": 151,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "v.loc[(v['id'] == 'rs76250508')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 152,
   "id": "8665d4ca-6892-4426-b0b1-d935d368cc9b",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "print(samples)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 153,
   "id": "3cbec0a2-5fa1-42c5-86b0-a2edf8a30a3d",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'dict' object has no attribute 'samples'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m                            Traceback (most recent call last)",
      "\u001b[0;32m/tmp/5053897.1.csg2.q/ipykernel_10694/3175417391.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbgen\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msamples\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;36m5\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mAttributeError\u001b[0m: 'dict' object has no attribute 'samples'"
     ]
    }
   ],
   "source": [
    "print(bgen.samples[:5])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 159,
   "id": "a174155a-8c50-4479-8325-958d21ff0476",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chrom</th>\n",
       "      <th>pos</th>\n",
       "      <th>nalleles</th>\n",
       "      <th>allele_ids</th>\n",
       "      <th>vaddr</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td></td>\n",
       "      <td>chr6:31571218:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>31571218</td>\n",
       "      <td>2</td>\n",
       "      <td>C,T</td>\n",
       "      <td>8285137</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td></td>\n",
       "      <td>chr6:31571228:A:C</td>\n",
       "      <td>6</td>\n",
       "      <td>31571228</td>\n",
       "      <td>2</td>\n",
       "      <td>A,C</td>\n",
       "      <td>8296855</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td></td>\n",
       "      <td>chr6:31571296:A:G</td>\n",
       "      <td>6</td>\n",
       "      <td>31571296</td>\n",
       "      <td>2</td>\n",
       "      <td>A,G</td>\n",
       "      <td>8299119</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td></td>\n",
       "      <td>chr6:31571308:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>31571308</td>\n",
       "      <td>2</td>\n",
       "      <td>C,T</td>\n",
       "      <td>8301500</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td></td>\n",
       "      <td>chr6:31571330:C:A</td>\n",
       "      <td>6</td>\n",
       "      <td>31571330</td>\n",
       "      <td>2</td>\n",
       "      <td>C,A</td>\n",
       "      <td>8303601</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200</th>\n",
       "      <td></td>\n",
       "      <td>chr6:32682566:G:C</td>\n",
       "      <td>6</td>\n",
       "      <td>32682566</td>\n",
       "      <td>2</td>\n",
       "      <td>G,C</td>\n",
       "      <td>2853863591</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>201</th>\n",
       "      <td></td>\n",
       "      <td>chr6:32682590:G:A</td>\n",
       "      <td>6</td>\n",
       "      <td>32682590</td>\n",
       "      <td>2</td>\n",
       "      <td>G,A</td>\n",
       "      <td>2853866372</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>202</th>\n",
       "      <td></td>\n",
       "      <td>chr6:32682623:G:T</td>\n",
       "      <td>6</td>\n",
       "      <td>32682623</td>\n",
       "      <td>2</td>\n",
       "      <td>G,T</td>\n",
       "      <td>2853888780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>203</th>\n",
       "      <td></td>\n",
       "      <td>chr6:32682656:G:A</td>\n",
       "      <td>6</td>\n",
       "      <td>32682656</td>\n",
       "      <td>2</td>\n",
       "      <td>G,A</td>\n",
       "      <td>2853890415</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>204</th>\n",
       "      <td></td>\n",
       "      <td>chr6:32682664:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>32682664</td>\n",
       "      <td>2</td>\n",
       "      <td>C,T</td>\n",
       "      <td>2853892752</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>43053 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    id               rsid chrom       pos  nalleles allele_ids       vaddr\n",
       "0       chr6:31571218:C:T     6  31571218         2        C,T     8285137\n",
       "1       chr6:31571228:A:C     6  31571228         2        A,C     8296855\n",
       "2       chr6:31571296:A:G     6  31571296         2        A,G     8299119\n",
       "3       chr6:31571308:C:T     6  31571308         2        C,T     8301500\n",
       "4       chr6:31571330:C:A     6  31571330         2        C,A     8303601\n",
       "..  ..                ...   ...       ...       ...        ...         ...\n",
       "200     chr6:32682566:G:C     6  32682566         2        G,C  2853863591\n",
       "201     chr6:32682590:G:A     6  32682590         2        G,A  2853866372\n",
       "202     chr6:32682623:G:T     6  32682623         2        G,T  2853888780\n",
       "203     chr6:32682656:G:A     6  32682656         2        G,A  2853890415\n",
       "204     chr6:32682664:C:T     6  32682664         2        C,T  2853892752\n",
       "\n",
       "[43053 rows x 7 columns]"
      ]
     },
     "execution_count": 159,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants.compute()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 156,
   "id": "540bfc8e-b353-4f66-b5c8-74b75acbecaa",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "print(bgen.ids) #first 5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "id": "2dad3860-d6cc-4585-a3ff-ef435ae9d7b9",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "'open_bgen' object is not subscriptable",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/tmp/5053897.1.csg2.q/ipykernel_10694/2151850230.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mvariants\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mbgen\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"variants\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m: 'open_bgen' object is not subscriptable"
     ]
    }
   ],
   "source": [
    "variants = bgen[\"variants\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "98c2a563-96c3-48c8-8d29-205d59dde047",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "import sys\n",
    "sys.stdout = open('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_31571218_32682664.variants_in_bgen','wt')\n",
    "print(bgen.ids)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "bc27130d-a929-420a-8419-bbf35a4bea8e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "43053\n"
     ]
    }
   ],
   "source": [
    "print(bgen.nvariants)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "8299ebb5-6c31-4c74-9be5-c6e63ff1f3bf",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "region = pd.read_csv(\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_31571218_32682664.variants_0.001.z\", header=0, sep=\" \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 65,
   "id": "5a2c4d54-0f15-4e7f-951b-72e4758b8001",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsid</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "      <th>rsid_old</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6:31571330_C_A</td>\n",
       "      <td>6</td>\n",
       "      <td>31571330</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs115490198</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6:31571337_C_T</td>\n",
       "      <td>6</td>\n",
       "      <td>31571337</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs140472229</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6:31571570_T_C</td>\n",
       "      <td>6</td>\n",
       "      <td>31571570</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs115755747</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6:31571571_C_T</td>\n",
       "      <td>6</td>\n",
       "      <td>31571571</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs558100393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6:31571665_T_C</td>\n",
       "      <td>6</td>\n",
       "      <td>31571665</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs148929231</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24047</th>\n",
       "      <td>6:32682452_G_A</td>\n",
       "      <td>6</td>\n",
       "      <td>32682452</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs3104407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24048</th>\n",
       "      <td>6:32682509_GGTT_G</td>\n",
       "      <td>6</td>\n",
       "      <td>32682509</td>\n",
       "      <td>GGTT</td>\n",
       "      <td>G</td>\n",
       "      <td>6:32682509_GGTT_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24049</th>\n",
       "      <td>6:32682540_T_C</td>\n",
       "      <td>6</td>\n",
       "      <td>32682540</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs193258417</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24050</th>\n",
       "      <td>rs113721192</td>\n",
       "      <td>6</td>\n",
       "      <td>32682590</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs113721192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24051</th>\n",
       "      <td>rs9275601</td>\n",
       "      <td>6</td>\n",
       "      <td>32682664</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs9275601</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24052 rows  6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    rsid  chromosome  position allele1 allele2  \\\n",
       "0         6:31571330_C_A           6  31571330       C       A   \n",
       "1         6:31571337_C_T           6  31571337       C       T   \n",
       "2         6:31571570_T_C           6  31571570       T       C   \n",
       "3         6:31571571_C_T           6  31571571       C       T   \n",
       "4         6:31571665_T_C           6  31571665       T       C   \n",
       "...                  ...         ...       ...     ...     ...   \n",
       "24047     6:32682452_G_A           6  32682452       G       A   \n",
       "24048  6:32682509_GGTT_G           6  32682509    GGTT       G   \n",
       "24049     6:32682540_T_C           6  32682540       T       C   \n",
       "24050        rs113721192           6  32682590       G       A   \n",
       "24051          rs9275601           6  32682664       C       T   \n",
       "\n",
       "                rsid_old  \n",
       "0            rs115490198  \n",
       "1            rs140472229  \n",
       "2            rs115755747  \n",
       "3            rs558100393  \n",
       "4            rs148929231  \n",
       "...                  ...  \n",
       "24047          rs3104407  \n",
       "24048  6:32682509_GGTT_G  \n",
       "24049        rs193258417  \n",
       "24050        rs113721192  \n",
       "24051          rs9275601  \n",
       "\n",
       "[24052 rows x 6 columns]"
      ]
     },
     "execution_count": 65,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "id": "e76e6721-d7c5-4bcc-831d-7440d603dd66",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(24052, 6)"
      ]
     },
     "execution_count": 67,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "region.shape"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "b806db16-4a4f-4f69-9a00-4e8ea97db436",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "144312"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "region.size"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "id": "acb20da6-27dd-4d0c-a942-7f0dd1cf85f5",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "region['chromosome'] = '0' + region['chromosome'].astype(str)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 79,
   "id": "75ff6cc0-3023-43ef-aa15-fa4238ea7d97",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "region['chromosome'] = region.chromosome.astype(str).str.zfill(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 80,
   "id": "6d3bf115-7c16-4945-9ced-7c54b5fbe27a",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alternate_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chromosome_new</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6:31571330_C_A</td>\n",
       "      <td>06</td>\n",
       "      <td>31571330</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>rs115490198</td>\n",
       "      <td>06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6:31571337_C_T</td>\n",
       "      <td>06</td>\n",
       "      <td>31571337</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs140472229</td>\n",
       "      <td>06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6:31571570_T_C</td>\n",
       "      <td>06</td>\n",
       "      <td>31571570</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs115755747</td>\n",
       "      <td>06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6:31571571_C_T</td>\n",
       "      <td>06</td>\n",
       "      <td>31571571</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs558100393</td>\n",
       "      <td>06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6:31571665_T_C</td>\n",
       "      <td>06</td>\n",
       "      <td>31571665</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs148929231</td>\n",
       "      <td>06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24047</th>\n",
       "      <td>6:32682452_G_A</td>\n",
       "      <td>06</td>\n",
       "      <td>32682452</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs3104407</td>\n",
       "      <td>06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24048</th>\n",
       "      <td>6:32682509_GGTT_G</td>\n",
       "      <td>06</td>\n",
       "      <td>32682509</td>\n",
       "      <td>GGTT</td>\n",
       "      <td>G</td>\n",
       "      <td>6:32682509_GGTT_G</td>\n",
       "      <td>06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24049</th>\n",
       "      <td>6:32682540_T_C</td>\n",
       "      <td>06</td>\n",
       "      <td>32682540</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>rs193258417</td>\n",
       "      <td>06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24050</th>\n",
       "      <td>rs113721192</td>\n",
       "      <td>06</td>\n",
       "      <td>32682590</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs113721192</td>\n",
       "      <td>06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24051</th>\n",
       "      <td>rs9275601</td>\n",
       "      <td>06</td>\n",
       "      <td>32682664</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs9275601</td>\n",
       "      <td>06</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24052 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            alternate_id chromosome  position allele1 allele2  \\\n",
       "0         6:31571330_C_A         06  31571330       C       A   \n",
       "1         6:31571337_C_T         06  31571337       C       T   \n",
       "2         6:31571570_T_C         06  31571570       T       C   \n",
       "3         6:31571571_C_T         06  31571571       C       T   \n",
       "4         6:31571665_T_C         06  31571665       T       C   \n",
       "...                  ...        ...       ...     ...     ...   \n",
       "24047     6:32682452_G_A         06  32682452       G       A   \n",
       "24048  6:32682509_GGTT_G         06  32682509    GGTT       G   \n",
       "24049     6:32682540_T_C         06  32682540       T       C   \n",
       "24050        rs113721192         06  32682590       G       A   \n",
       "24051          rs9275601         06  32682664       C       T   \n",
       "\n",
       "                    rsid chromosome_new  \n",
       "0            rs115490198             06  \n",
       "1            rs140472229             06  \n",
       "2            rs115755747             06  \n",
       "3            rs558100393             06  \n",
       "4            rs148929231             06  \n",
       "...                  ...            ...  \n",
       "24047          rs3104407             06  \n",
       "24048  6:32682509_GGTT_G             06  \n",
       "24049        rs193258417             06  \n",
       "24050        rs113721192             06  \n",
       "24051          rs9275601             06  \n",
       "\n",
       "[24052 rows x 7 columns]"
      ]
     },
     "execution_count": 80,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 75,
   "id": "f9034fdb-2e1f-4ace-8e0f-03f160aebb68",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "region.rename(columns={'rsid_old': 'rsid',  'rsid': 'alternate_id'}, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 77,
   "id": "47b31b0e-7683-4695-b8f3-099dda6f794e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "region[['rsid', 'chromosome', 'position', 'allele1', 'allele2', 'alternate_id']].to_csv(\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_31571218_32682664.variants_0.001.v2.z\", index=False, sep=\" \")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f9c66f4b-3d2b-4db1-9803-e1c823d19fd8",
   "metadata": {
    "kernel": "Python3"
   },
   "source": [
    "## Try creation of the z file for a particular block"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "a3044452-4078-4596-981b-677a9e420a68",
   "metadata": {
    "kernel": "Python3",
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_100630146_102636772.variants\n",
      "06_100630146_102636772.variants\n",
      "The number of variants in the region is: 64958\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 14038\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_102636772_103983395.variants\n",
      "06_102636772_103983395.variants\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/pandas/core/frame.py:5034: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  return super().rename(\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "The number of variants in the region is: 46383\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 10890\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_103983395_106056733.variants\n",
      "06_103983395_106056733.variants\n",
      "The number of variants in the region is: 68107\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 14269\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_10540317_11790816.variants\n",
      "06_10540317_11790816.variants\n",
      "The number of variants in the region is: 43884\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 9809\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_106056733_107450306.variants\n",
      "06_106056733_107450306.variants\n",
      "The number of variants in the region is: 47219\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 10015\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_107450306_108464380.variants\n",
      "06_107450306_108464380.variants\n",
      "The number of variants in the region is: 31858\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 6756\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_108464380_110304247.variants\n",
      "06_108464380_110304247.variants\n",
      "The number of variants in the region is: 57682\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 11188\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_110304247_112345014.variants\n",
      "06_110304247_112345014.variants\n",
      "The number of variants in the region is: 64350\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 13200\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_112345014_114220833.variants\n",
      "06_112345014_114220833.variants\n",
      "The number of variants in the region is: 60911\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 11880\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_114220833_116131396.variants\n",
      "06_114220833_116131396.variants\n",
      "The number of variants in the region is: 63042\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 13030\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_116131396_117672972.variants\n",
      "06_116131396_117672972.variants\n",
      "The number of variants in the region is: 47243\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 10025\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_117672972_118963115.variants\n",
      "06_117672972_118963115.variants\n",
      "The number of variants in the region is: 41034\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 8316\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_11790816_13209388.variants\n",
      "06_11790816_13209388.variants\n",
      "The number of variants in the region is: 47738\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 9967\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_118963115_119963278.variants\n",
      "06_118963115_119963278.variants\n",
      "The number of variants in the region is: 32086\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 6770\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_119963278_120512128.variants\n",
      "06_119963278_120512128.variants\n",
      "The number of variants in the region is: 17713\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 4008\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_120512128_121905676.variants\n",
      "06_120512128_121905676.variants\n",
      "The number of variants in the region is: 46137\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 10824\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_121905676_123856181.variants\n",
      "06_121905676_123856181.variants\n",
      "The number of variants in the region is: 63045\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 12740\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_123856181_125424383.variants\n",
      "06_123856181_125424383.variants\n",
      "The number of variants in the region is: 52537\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 11067\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_125424383_127540461.variants\n",
      "06_125424383_127540461.variants\n",
      "The number of variants in the region is: 63716\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 10985\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_127540461_129861031.variants\n",
      "06_127540461_129861031.variants\n",
      "The number of variants in the region is: 74924\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 14826\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_129861031_130935454.variants\n",
      "06_129861031_130935454.variants\n",
      "The number of variants in the region is: 36844\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 8171\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_130935454_132765669.variants\n",
      "06_130935454_132765669.variants\n",
      "The number of variants in the region is: 60875\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 12922\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_13209388_14802924.variants\n",
      "06_13209388_14802924.variants\n",
      "The number of variants in the region is: 56501\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 11460\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_132765669_134244243.variants\n",
      "06_132765669_134244243.variants\n",
      "The number of variants in the region is: 50050\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 11029\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_134244243_136224177.variants\n",
      "06_134244243_136224177.variants\n",
      "The number of variants in the region is: 64573\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 13096\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_136224177_137614218.variants\n",
      "06_136224177_137614218.variants\n",
      "The number of variants in the region is: 44689\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 8648\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_137614218_138822629.variants\n",
      "06_137614218_138822629.variants\n",
      "The number of variants in the region is: 42715\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 8818\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_138822629_139845436.variants\n",
      "06_138822629_139845436.variants\n",
      "The number of variants in the region is: 34427\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 7673\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_139845436_142288479.variants\n",
      "06_139845436_142288479.variants\n",
      "The number of variants in the region is: 77910\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 15306\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_142288479_145319810.variants\n",
      "06_142288479_145319810.variants\n",
      "The number of variants in the region is: 98753\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 19910\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_1452362_2458936.variants\n",
      "06_1452362_2458936.variants\n",
      "The number of variants in the region is: 35638\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 7391\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_145319810_146665424.variants\n",
      "06_145319810_146665424.variants\n",
      "The number of variants in the region is: 43042\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 7949\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_146665424_148920352.variants\n",
      "06_146665424_148920352.variants\n",
      "The number of variants in the region is: 75214\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 15544\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_14802924_17386405.variants\n",
      "06_14802924_17386405.variants\n",
      "The number of variants in the region is: 89761\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 18241\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_148920352_150253404.variants\n",
      "06_148920352_150253404.variants\n",
      "The number of variants in the region is: 44692\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 9262\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_150253404_151912703.variants\n",
      "06_150253404_151912703.variants\n",
      "The number of variants in the region is: 60515\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 13985\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_151912703_153094496.variants\n",
      "06_151912703_153094496.variants\n",
      "The number of variants in the region is: 39895\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 8532\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_153094496_154974120.variants\n",
      "06_153094496_154974120.variants\n",
      "The number of variants in the region is: 65076\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 14667\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_154974120_156763326.variants\n",
      "06_154974120_156763326.variants\n",
      "The number of variants in the region is: 62264\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 13007\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_156763326_158218719.variants\n",
      "06_156763326_158218719.variants\n",
      "The number of variants in the region is: 46785\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 9575\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_158218719_160580497.variants\n",
      "06_158218719_160580497.variants\n",
      "The number of variants in the region is: 85734\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 18350\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_160580497_162169564.variants\n",
      "06_160580497_162169564.variants\n",
      "The number of variants in the region is: 56633\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 12778\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_162169564_162931203.variants\n",
      "06_162169564_162931203.variants\n",
      "The number of variants in the region is: 29226\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 6373\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_162931203_164383521.variants\n",
      "06_162931203_164383521.variants\n",
      "The number of variants in the region is: 51134\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 11095\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_164383521_165586864.variants\n",
      "06_164383521_165586864.variants\n",
      "The number of variants in the region is: 44191\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 9822\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_165586864_167024733.variants\n",
      "06_165586864_167024733.variants\n",
      "The number of variants in the region is: 53617\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 11795\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_167024733_168042835.variants\n",
      "06_167024733_168042835.variants\n",
      "The number of variants in the region is: 35907\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 8906\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_168042835_169382050.variants\n",
      "06_168042835_169382050.variants\n",
      "The number of variants in the region is: 54485\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 13145\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_169382050_170330173.variants\n",
      "06_169382050_170330173.variants\n",
      "The number of variants in the region is: 33651\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 7274\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_170330173_171051270.variants\n",
      "06_170330173_171051270.variants\n",
      "The number of variants in the region is: 23913\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 5827\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_17386405_19207487.variants\n",
      "06_17386405_19207487.variants\n",
      "The number of variants in the region is: 62566\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 14114\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_19207487_21684065.variants\n",
      "06_19207487_21684065.variants\n",
      "The number of variants in the region is: 84811\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 18230\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_21684065_22748307.variants\n",
      "06_21684065_22748307.variants\n",
      "The number of variants in the region is: 36592\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 8400\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_22748307_23936619.variants\n",
      "06_22748307_23936619.variants\n",
      "The number of variants in the region is: 46084\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 10063\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_23936619_24852275.variants\n",
      "06_23936619_24852275.variants\n",
      "The number of variants in the region is: 32935\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 7793\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_2458936_3573593.variants\n",
      "06_2458936_3573593.variants\n",
      "The number of variants in the region is: 43142\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 9646\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_24852275_25684587.variants\n",
      "06_24852275_25684587.variants\n",
      "The number of variants in the region is: 27468\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 6777\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_25684587_26791233.variants\n",
      "06_25684587_26791233.variants\n",
      "The number of variants in the region is: 34333\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 7745\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_26791233_28017819.variants\n",
      "06_26791233_28017819.variants\n",
      "The number of variants in the region is: 35810\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 8133\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_28017819_28917608.variants\n",
      "06_28017819_28917608.variants\n",
      "The number of variants in the region is: 27172\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 6573\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_28917608_29737971.variants\n",
      "06_28917608_29737971.variants\n",
      "The number of variants in the region is: 25395\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 7785\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_29737971_30798168.variants\n",
      "06_29737971_30798168.variants\n",
      "The number of variants in the region is: 39527\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 16707\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_30798168_31571218.variants\n",
      "06_30798168_31571218.variants\n",
      "The number of variants in the region is: 32459\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 16652\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_31571218_32682664.variants\n",
      "06_31571218_32682664.variants\n",
      "The number of variants in the region is: 43053\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 24052\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_32682664_33236497.variants\n",
      "06_32682664_33236497.variants\n",
      "The number of variants in the region is: 21532\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 8897\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_33236497_35455756.variants\n",
      "06_33236497_35455756.variants\n",
      "The number of variants in the region is: 75085\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 17032\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_35455756_37572596.variants\n",
      "06_35455756_37572596.variants\n",
      "The number of variants in the region is: 71828\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 14942\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_3573593_4617817.variants\n",
      "06_3573593_4617817.variants\n",
      "The number of variants in the region is: 39080\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 8928\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_37572596_39295677.variants\n",
      "06_37572596_39295677.variants\n",
      "The number of variants in the region is: 59084\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 12522\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_39295677_40345115.variants\n",
      "06_39295677_40345115.variants\n",
      "The number of variants in the region is: 36810\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 7717\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_40345115_42038721.variants\n",
      "06_40345115_42038721.variants\n",
      "The number of variants in the region is: 59490\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 12645\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_42038721_43756169.variants\n",
      "06_42038721_43756169.variants\n",
      "The number of variants in the region is: 55859\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 11047\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_43756169_45406563.variants\n",
      "06_43756169_45406563.variants\n",
      "The number of variants in the region is: 56161\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 11705\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_45406563_47311898.variants\n",
      "06_45406563_47311898.variants\n",
      "The number of variants in the region is: 66826\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 12210\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_4617817_5791472.variants\n",
      "06_4617817_5791472.variants\n",
      "The number of variants in the region is: 43139\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 9521\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_47311898_48391125.variants\n",
      "06_47311898_48391125.variants\n",
      "The number of variants in the region is: 38197\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 6861\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_48391125_50386145.variants\n",
      "06_48391125_50386145.variants\n",
      "The number of variants in the region is: 72191\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 13965\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_50386145_52210476.variants\n",
      "06_50386145_52210476.variants\n",
      "The number of variants in the region is: 60297\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 12317\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_52210476_53279102.variants\n",
      "06_52210476_53279102.variants\n",
      "The number of variants in the region is: 37004\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 7910\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_53279102_55468270.variants\n",
      "06_53279102_55468270.variants\n",
      "The number of variants in the region is: 76736\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 17147\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_55468270_56105313.variants\n",
      "06_55468270_56105313.variants\n",
      "The number of variants in the region is: 21582\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 4932\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_56105313_57599465.variants\n",
      "06_56105313_57599465.variants\n",
      "The number of variants in the region is: 46968\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 9796\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_57599465_61880512.variants\n",
      "06_57599465_61880512.variants\n",
      "The number of variants in the region is: 44019\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 9202\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_5791472_6785207.variants\n",
      "06_5791472_6785207.variants\n",
      "The number of variants in the region is: 36830\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 8297\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_61880512_63552888.variants\n",
      "06_61880512_63552888.variants\n",
      "The number of variants in the region is: 68300\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 13829\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_63552888_65765742.variants\n",
      "06_63552888_65765742.variants\n",
      "The number of variants in the region is: 80153\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 16275\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_65765742_67329215.variants\n",
      "06_65765742_67329215.variants\n",
      "The number of variants in the region is: 57865\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 13335\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_67329215_68849483.variants\n",
      "06_67329215_68849483.variants\n",
      "The number of variants in the region is: 58548\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 13813\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_6785207_7808936.variants\n",
      "06_6785207_7808936.variants\n",
      "The number of variants in the region is: 38278\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 8357\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_68849483_69523448.variants\n",
      "06_68849483_69523448.variants\n",
      "The number of variants in the region is: 23418\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 5549\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_69523448_71609510.variants\n",
      "06_69523448_71609510.variants\n",
      "The number of variants in the region is: 71711\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 15231\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_71609510_73450097.variants\n",
      "06_71609510_73450097.variants\n",
      "The number of variants in the region is: 62269\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 13707\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_73450097_75462003.variants\n",
      "06_73450097_75462003.variants\n",
      "The number of variants in the region is: 68243\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 14861\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_73924_1452362.variants\n",
      "06_73924_1452362.variants\n",
      "The number of variants in the region is: 49322\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 11214\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_75462003_77414881.variants\n",
      "06_75462003_77414881.variants\n",
      "The number of variants in the region is: 67053\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 13199\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_77414881_78957728.variants\n",
      "06_77414881_78957728.variants\n",
      "The number of variants in the region is: 53909\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 12323\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_7808936_9020026.variants\n",
      "06_7808936_9020026.variants\n",
      "The number of variants in the region is: 42099\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 9445\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_78957728_80691170.variants\n",
      "06_78957728_80691170.variants\n",
      "The number of variants in the region is: 60727\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 12518\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_80691170_81929222.variants\n",
      "06_80691170_81929222.variants\n",
      "The number of variants in the region is: 46002\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 10846\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_81929222_83127149.variants\n",
      "06_81929222_83127149.variants\n",
      "The number of variants in the region is: 40626\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 8330\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_83127149_85209989.variants\n",
      "06_83127149_85209989.variants\n",
      "The number of variants in the region is: 66885\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 12777\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_85209989_87069500.variants\n",
      "06_85209989_87069500.variants\n",
      "The number of variants in the region is: 69195\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 13704\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_87069500_88822140.variants\n",
      "06_87069500_88822140.variants\n",
      "The number of variants in the region is: 63649\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 12743\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_88822140_89973052.variants\n",
      "06_88822140_89973052.variants\n",
      "The number of variants in the region is: 38643\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 8237\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_89973052_91843196.variants\n",
      "06_89973052_91843196.variants\n",
      "The number of variants in the region is: 60324\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 11824\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_9020026_10540317.variants\n",
      "06_9020026_10540317.variants\n",
      "The number of variants in the region is: 55503\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 12033\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_91843196_93428644.variants\n",
      "06_91843196_93428644.variants\n",
      "The number of variants in the region is: 55393\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 11151\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_93428644_94118142.variants\n",
      "06_93428644_94118142.variants\n",
      "The number of variants in the region is: 23965\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 5330\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_94118142_94441175.variants\n",
      "06_94118142_94441175.variants\n",
      "The number of variants in the region is: 10826\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 2153\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_94441175_97093511.variants\n",
      "06_94441175_97093511.variants\n",
      "The number of variants in the region is: 86365\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 18467\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_97093511_97842284.variants\n",
      "06_97093511_97842284.variants\n",
      "The number of variants in the region is: 22948\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 4679\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_97842284_100630146.variants\n",
      "06_97842284_100630146.variants\n",
      "The number of variants in the region is: 86246\n",
      "The number of variants in /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt is: 5751712\n",
      "The number of variants in the maf 0.001 filtered z_file is: 16314\n"
     ]
    }
   ],
   "source": [
    "# Creation of the z file: it has to be space delimited and not tab delimited\n",
    "import glob, os, sys\n",
    "import pandas as pd\n",
    "log_file= open('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/regions1_22.log', \"w\")\n",
    "regions=glob.glob('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/*.variants', recursive=True)\n",
    "mfi_files=glob.glob('/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr*[0-9]_v3.txt', recursive=True)\n",
    "for filename in regions:\n",
    "    print(filename)\n",
    "    basename=os.path.basename(filename)\n",
    "    sp = basename.split(\"_\")[0]\n",
    "    chrom= sp.lstrip(\"0\")\n",
    "    print(basename)\n",
    "    fullregion = pd.read_csv(filename,header=0,sep=\"\\t\", skiprows=1, skipfooter=1, names=['alternate_ids', 'rsid', 'chromosome', 'position', 'number_of_alleles', 'allele1', 'allele2'])\n",
    "    print('The number of variants in the region is:',fullregion.shape[0])\n",
    "    mfi_file = [file for file in mfi_files if \"chr\" + chrom + \"_\" in file][0]\n",
    "    chrom_file = pd.read_csv(mfi_file,header=None,sep=\"\\t\", names=[\"alternate_id\", \"rsid\", \"position\", \"allele1\", \"allele2\", \"maf\", \"minor_allele\", \"info_score\"])\n",
    "    print('The number of variants in',mfi_file,'is:',chrom_file.shape[0])\n",
    "    subset_region=chrom_file.join(fullregion.set_index(['alternate_ids', 'rsid', 'position', 'allele1', 'allele2']), on=['alternate_id', 'rsid', 'position', 'allele1', 'allele2'], how='right')\n",
    "    maf=0.001\n",
    "    subset_maf=subset_region[subset_region['maf'] > maf]\n",
    "    print('The number of variants in the maf',maf,'filtered z_file is:',subset_maf.shape[0])\n",
    "    subset_maf.rename(columns={'rsid': 'rsid_old', 'alternate_id': 'rsid'}, inplace=True)\n",
    "    subset_maf[[\"rsid\",\"chromosome\", \"position\", \"allele1\", \"allele2\", \"rsid_old\"]].to_csv(filename + '_' + str(maf) + '.z', sep=' ', index=False, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9d21489a-983e-4f36-b230-2f2be2800af3",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "cf26a914-9374-4d0e-b2bd-f74f000174db",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "INFO: Checking \u001b[32mbcor\u001b[0m: Create bcor files\n",
      "HINT: /bin/bash SCRIPT\n",
      "~/ldstore_v2.0_x86_64/./ldstore_v2.0_x86_64  \\\n",
      "--in-files /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/masterfile_chr06_31571218_32682664_v0.001_list_unrelated_whiteEur.txt \\\n",
      "--write-bcor \\\n",
      "--read-only-bgen \\\n",
      "--n-threads 10 \n",
      "\n",
      "\n",
      "\n",
      "INFO: \u001b[32mbcor\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: Workflow bcor (ID=wb3b61b1e243f0d82) is tested successfully with 1 completed step.\n"
     ]
    }
   ],
   "source": [
    "sos dryrun /home/dmc2245/project/UKBB_GWAS_dev/workflow/111722_LDstore.ipynb \\\n",
    "     bcor\\\n",
    "     --cwd /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06\\\n",
    "     --masterfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/masterfile_chr06_31571218_32682664_v0.001_list_unrelated_whiteEur.txt\\\n",
    "     --numThreads 10 \\\n",
    "     --mem 100G \\\n",
    "     --job_size 1 "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 78,
   "id": "8ee73528-098e-4fa4-a15d-a1aa948f68d4",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Error : Cannot allocate buffer for storing allelic dosages!. At least 33.8 GB memory are required.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "~/ldstore_v2.0_x86_64/./ldstore_v2.0_x86_64  \\\n",
    "--in-files /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/masterfile_chr06_31571218_32682664_v0.001_list_unrelated_whiteEur.txt \\\n",
    "--write-bcor --read-only-bgen --n-threads 10  > ~/test.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "id": "20220c13-413c-42fc-ab4b-7572f4469828",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Creation of the z file: it has to be space delimited and not tab delimited\n",
    "import glob, os, sys\n",
    "import pandas as pd\n",
    "log_file= open('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/chr06.log', \"w\")\n",
    "regions='/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_31571218_32682664.variants'\n",
    "mfi_files='/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_mfi_chr6_v3.txt'\n",
    "\n",
    "basename=os.path.basename(regions)\n",
    "sp = basename.split(\"_\")[0]\n",
    "chrom= sp.lstrip(\"0\")\n",
    "print(basename)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "id": "359b36aa-386f-4b8a-9094-e2aed5fb5942",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "fullregion = pd.read_csv(regions,header=0,sep=\"\\t\", skiprows=1, skipfooter=1, names=['alternate_ids', 'rsid', 'chromosome', 'position', 'number_of_alleles', 'allele1', 'allele2'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "id": "59f2ec75-d1aa-41fb-abe8-91465a275925",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alternate_ids</th>\n",
       "      <th>rsid</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>number_of_alleles</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6:31571218_C_T</td>\n",
       "      <td>rs145265084</td>\n",
       "      <td>6</td>\n",
       "      <td>31571218</td>\n",
       "      <td>2</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6:31571228_A_C</td>\n",
       "      <td>rs191097828</td>\n",
       "      <td>6</td>\n",
       "      <td>31571228</td>\n",
       "      <td>2</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6:31571296_A_G</td>\n",
       "      <td>rs182311656</td>\n",
       "      <td>6</td>\n",
       "      <td>31571296</td>\n",
       "      <td>2</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6:31571308_C_T</td>\n",
       "      <td>rs147609818</td>\n",
       "      <td>6</td>\n",
       "      <td>31571308</td>\n",
       "      <td>2</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6:31571330_C_A</td>\n",
       "      <td>rs115490198</td>\n",
       "      <td>6</td>\n",
       "      <td>31571330</td>\n",
       "      <td>2</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43048</th>\n",
       "      <td>6:32682566_G_C</td>\n",
       "      <td>rs114913515</td>\n",
       "      <td>6</td>\n",
       "      <td>32682566</td>\n",
       "      <td>2</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43049</th>\n",
       "      <td>rs113721192</td>\n",
       "      <td>rs113721192</td>\n",
       "      <td>6</td>\n",
       "      <td>32682590</td>\n",
       "      <td>2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43050</th>\n",
       "      <td>6:32682623_G_T</td>\n",
       "      <td>rs529848903</td>\n",
       "      <td>6</td>\n",
       "      <td>32682623</td>\n",
       "      <td>2</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43051</th>\n",
       "      <td>6:32682656_G_A</td>\n",
       "      <td>rs374522692</td>\n",
       "      <td>6</td>\n",
       "      <td>32682656</td>\n",
       "      <td>2</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43052</th>\n",
       "      <td>rs9275601</td>\n",
       "      <td>rs9275601</td>\n",
       "      <td>6</td>\n",
       "      <td>32682664</td>\n",
       "      <td>2</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>43053 rows  7 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        alternate_ids         rsid  chromosome  position  number_of_alleles  \\\n",
       "0      6:31571218_C_T  rs145265084           6  31571218                  2   \n",
       "1      6:31571228_A_C  rs191097828           6  31571228                  2   \n",
       "2      6:31571296_A_G  rs182311656           6  31571296                  2   \n",
       "3      6:31571308_C_T  rs147609818           6  31571308                  2   \n",
       "4      6:31571330_C_A  rs115490198           6  31571330                  2   \n",
       "...               ...          ...         ...       ...                ...   \n",
       "43048  6:32682566_G_C  rs114913515           6  32682566                  2   \n",
       "43049     rs113721192  rs113721192           6  32682590                  2   \n",
       "43050  6:32682623_G_T  rs529848903           6  32682623                  2   \n",
       "43051  6:32682656_G_A  rs374522692           6  32682656                  2   \n",
       "43052       rs9275601    rs9275601           6  32682664                  2   \n",
       "\n",
       "      allele1 allele2  \n",
       "0           C       T  \n",
       "1           A       C  \n",
       "2           A       G  \n",
       "3           C       T  \n",
       "4           C       A  \n",
       "...       ...     ...  \n",
       "43048       G       C  \n",
       "43049       G       A  \n",
       "43050       G       T  \n",
       "43051       G       A  \n",
       "43052       C       T  \n",
       "\n",
       "[43053 rows x 7 columns]"
      ]
     },
     "execution_count": 103,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "fullregion"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 106,
   "id": "866293bd-0198-4bd1-9d4d-503c52ed3c8b",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "chrom_file = pd.read_csv(mfi_files,header=None,sep=\"\\t\", names=[\"alternate_id\", \"rsid\", \"position\", \"allele1\", \"allele2\", \"maf\", \"minor_allele\", \"info_score\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "id": "f6dad36c-da84-4622-80ca-8509d4aee17c",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alternate_id</th>\n",
       "      <th>rsid</th>\n",
       "      <th>position</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "      <th>maf</th>\n",
       "      <th>minor_allele</th>\n",
       "      <th>info_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>6:63854_T_G</td>\n",
       "      <td>rs544586840</td>\n",
       "      <td>63854</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>G</td>\n",
       "      <td>0.266002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6:63979_C_T</td>\n",
       "      <td>rs561313667</td>\n",
       "      <td>63979</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.101429</td>\n",
       "      <td>T</td>\n",
       "      <td>0.764002</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>6:63980_A_G</td>\n",
       "      <td>rs530120680</td>\n",
       "      <td>63980</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.101839</td>\n",
       "      <td>G</td>\n",
       "      <td>0.762696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>6:73938_A_G</td>\n",
       "      <td>rs540888038</td>\n",
       "      <td>73938</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000758</td>\n",
       "      <td>G</td>\n",
       "      <td>0.247404</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>6:86820_CT_C</td>\n",
       "      <td>6:86820_CT_C</td>\n",
       "      <td>86820</td>\n",
       "      <td>CT</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000124</td>\n",
       "      <td>C</td>\n",
       "      <td>0.319900</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5751707</th>\n",
       "      <td>6:171051277_T_C</td>\n",
       "      <td>rs570527215</td>\n",
       "      <td>171051277</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000075</td>\n",
       "      <td>C</td>\n",
       "      <td>0.497973</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5751708</th>\n",
       "      <td>6:171051313_G_T</td>\n",
       "      <td>rs534363548</td>\n",
       "      <td>171051313</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000037</td>\n",
       "      <td>T</td>\n",
       "      <td>0.353477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5751709</th>\n",
       "      <td>6:171051321_C_A</td>\n",
       "      <td>rs552883708</td>\n",
       "      <td>171051321</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000003</td>\n",
       "      <td>A</td>\n",
       "      <td>0.335025</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5751710</th>\n",
       "      <td>6:171053361_G_A</td>\n",
       "      <td>rs572669761</td>\n",
       "      <td>171053361</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.008100</td>\n",
       "      <td>A</td>\n",
       "      <td>0.415867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5751711</th>\n",
       "      <td>6:171053406_C_A</td>\n",
       "      <td>rs543306648</td>\n",
       "      <td>171053406</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.057212</td>\n",
       "      <td>A</td>\n",
       "      <td>0.427966</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5751712 rows  8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "            alternate_id          rsid   position allele1 allele2       maf  \\\n",
       "0            6:63854_T_G   rs544586840      63854       T       G  0.000014   \n",
       "1            6:63979_C_T   rs561313667      63979       C       T  0.101429   \n",
       "2            6:63980_A_G   rs530120680      63980       A       G  0.101839   \n",
       "3            6:73938_A_G   rs540888038      73938       A       G  0.000758   \n",
       "4           6:86820_CT_C  6:86820_CT_C      86820      CT       C  0.000124   \n",
       "...                  ...           ...        ...     ...     ...       ...   \n",
       "5751707  6:171051277_T_C   rs570527215  171051277       T       C  0.000075   \n",
       "5751708  6:171051313_G_T   rs534363548  171051313       G       T  0.000037   \n",
       "5751709  6:171051321_C_A   rs552883708  171051321       C       A  0.000003   \n",
       "5751710  6:171053361_G_A   rs572669761  171053361       G       A  0.008100   \n",
       "5751711  6:171053406_C_A   rs543306648  171053406       C       A  0.057212   \n",
       "\n",
       "        minor_allele  info_score  \n",
       "0                  G    0.266002  \n",
       "1                  T    0.764002  \n",
       "2                  G    0.762696  \n",
       "3                  G    0.247404  \n",
       "4                  C    0.319900  \n",
       "...              ...         ...  \n",
       "5751707            C    0.497973  \n",
       "5751708            T    0.353477  \n",
       "5751709            A    0.335025  \n",
       "5751710            A    0.415867  \n",
       "5751711            A    0.427966  \n",
       "\n",
       "[5751712 rows x 8 columns]"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "chrom_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "ff715c2b-8dd3-4596-8f62-2e4f6946b758",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "subset_region=chrom_file.join(fullregion.set_index(['alternate_ids', 'rsid', 'position', 'allele1', 'allele2']), on=['alternate_id', 'rsid', 'position', 'allele1', 'allele2'], how='right')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "id": "2c8346be-07bd-47f9-9dd9-5148e6984843",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alternate_id</th>\n",
       "      <th>rsid</th>\n",
       "      <th>position</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "      <th>maf</th>\n",
       "      <th>minor_allele</th>\n",
       "      <th>info_score</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>number_of_alleles</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1106047</th>\n",
       "      <td>6:31571218_C_T</td>\n",
       "      <td>rs145265084</td>\n",
       "      <td>31571218</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000830</td>\n",
       "      <td>T</td>\n",
       "      <td>0.361975</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106048</th>\n",
       "      <td>6:31571228_A_C</td>\n",
       "      <td>rs191097828</td>\n",
       "      <td>31571228</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000033</td>\n",
       "      <td>C</td>\n",
       "      <td>0.208780</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106049</th>\n",
       "      <td>6:31571296_A_G</td>\n",
       "      <td>rs182311656</td>\n",
       "      <td>31571296</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>0.000072</td>\n",
       "      <td>G</td>\n",
       "      <td>0.547544</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106050</th>\n",
       "      <td>6:31571308_C_T</td>\n",
       "      <td>rs147609818</td>\n",
       "      <td>31571308</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>T</td>\n",
       "      <td>0.596251</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106051</th>\n",
       "      <td>6:31571330_C_A</td>\n",
       "      <td>rs115490198</td>\n",
       "      <td>31571330</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.001874</td>\n",
       "      <td>A</td>\n",
       "      <td>0.855969</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149095</th>\n",
       "      <td>6:32682566_G_C</td>\n",
       "      <td>rs114913515</td>\n",
       "      <td>32682566</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>0.000069</td>\n",
       "      <td>C</td>\n",
       "      <td>0.284171</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149096</th>\n",
       "      <td>rs113721192</td>\n",
       "      <td>rs113721192</td>\n",
       "      <td>32682590</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.009876</td>\n",
       "      <td>A</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149097</th>\n",
       "      <td>6:32682623_G_T</td>\n",
       "      <td>rs529848903</td>\n",
       "      <td>32682623</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>T</td>\n",
       "      <td>0.033795</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149098</th>\n",
       "      <td>6:32682656_G_A</td>\n",
       "      <td>rs374522692</td>\n",
       "      <td>32682656</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.000134</td>\n",
       "      <td>A</td>\n",
       "      <td>0.868186</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149099</th>\n",
       "      <td>rs9275601</td>\n",
       "      <td>rs9275601</td>\n",
       "      <td>32682664</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.441535</td>\n",
       "      <td>T</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>43053 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "           alternate_id         rsid  position allele1 allele2       maf  \\\n",
       "1106047  6:31571218_C_T  rs145265084  31571218       C       T  0.000830   \n",
       "1106048  6:31571228_A_C  rs191097828  31571228       A       C  0.000033   \n",
       "1106049  6:31571296_A_G  rs182311656  31571296       A       G  0.000072   \n",
       "1106050  6:31571308_C_T  rs147609818  31571308       C       T  0.000041   \n",
       "1106051  6:31571330_C_A  rs115490198  31571330       C       A  0.001874   \n",
       "...                 ...          ...       ...     ...     ...       ...   \n",
       "1149095  6:32682566_G_C  rs114913515  32682566       G       C  0.000069   \n",
       "1149096     rs113721192  rs113721192  32682590       G       A  0.009876   \n",
       "1149097  6:32682623_G_T  rs529848903  32682623       G       T  0.000005   \n",
       "1149098  6:32682656_G_A  rs374522692  32682656       G       A  0.000134   \n",
       "1149099       rs9275601    rs9275601  32682664       C       T  0.441535   \n",
       "\n",
       "        minor_allele  info_score  chromosome  number_of_alleles  \n",
       "1106047            T    0.361975           6                  2  \n",
       "1106048            C    0.208780           6                  2  \n",
       "1106049            G    0.547544           6                  2  \n",
       "1106050            T    0.596251           6                  2  \n",
       "1106051            A    0.855969           6                  2  \n",
       "...              ...         ...         ...                ...  \n",
       "1149095            C    0.284171           6                  2  \n",
       "1149096            A    1.000000           6                  2  \n",
       "1149097            T    0.033795           6                  2  \n",
       "1149098            A    0.868186           6                  2  \n",
       "1149099            T    1.000000           6                  2  \n",
       "\n",
       "[43053 rows x 10 columns]"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_region"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "id": "1e33fa37-4ea2-4039-9664-94da59a1395a",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "maf=0.001\n",
    "subset_maf=subset_region[subset_region['maf'] > maf]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "id": "960439de-d930-4b64-ae6b-7809bb0024bb",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alternate_id</th>\n",
       "      <th>rsid</th>\n",
       "      <th>position</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "      <th>maf</th>\n",
       "      <th>minor_allele</th>\n",
       "      <th>info_score</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>number_of_alleles</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1106051</th>\n",
       "      <td>6:31571330_C_A</td>\n",
       "      <td>rs115490198</td>\n",
       "      <td>31571330</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.001874</td>\n",
       "      <td>A</td>\n",
       "      <td>0.855969</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106052</th>\n",
       "      <td>6:31571337_C_T</td>\n",
       "      <td>rs140472229</td>\n",
       "      <td>31571337</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.006666</td>\n",
       "      <td>T</td>\n",
       "      <td>0.833402</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106058</th>\n",
       "      <td>6:31571570_T_C</td>\n",
       "      <td>rs115755747</td>\n",
       "      <td>31571570</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.004427</td>\n",
       "      <td>C</td>\n",
       "      <td>0.748554</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106059</th>\n",
       "      <td>6:31571571_C_T</td>\n",
       "      <td>rs558100393</td>\n",
       "      <td>31571571</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.001318</td>\n",
       "      <td>T</td>\n",
       "      <td>0.259413</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106065</th>\n",
       "      <td>6:31571665_T_C</td>\n",
       "      <td>rs148929231</td>\n",
       "      <td>31571665</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001320</td>\n",
       "      <td>C</td>\n",
       "      <td>0.789462</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149090</th>\n",
       "      <td>6:32682452_G_A</td>\n",
       "      <td>rs3104407</td>\n",
       "      <td>32682452</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.439507</td>\n",
       "      <td>G</td>\n",
       "      <td>0.928569</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149093</th>\n",
       "      <td>6:32682509_GGTT_G</td>\n",
       "      <td>6:32682509_GGTT_G</td>\n",
       "      <td>32682509</td>\n",
       "      <td>GGTT</td>\n",
       "      <td>G</td>\n",
       "      <td>0.442607</td>\n",
       "      <td>G</td>\n",
       "      <td>0.878243</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149094</th>\n",
       "      <td>6:32682540_T_C</td>\n",
       "      <td>rs193258417</td>\n",
       "      <td>32682540</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002254</td>\n",
       "      <td>C</td>\n",
       "      <td>0.885513</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149096</th>\n",
       "      <td>rs113721192</td>\n",
       "      <td>rs113721192</td>\n",
       "      <td>32682590</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.009876</td>\n",
       "      <td>A</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149099</th>\n",
       "      <td>rs9275601</td>\n",
       "      <td>rs9275601</td>\n",
       "      <td>32682664</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.441535</td>\n",
       "      <td>T</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24052 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              alternate_id               rsid  position allele1 allele2  \\\n",
       "1106051     6:31571330_C_A        rs115490198  31571330       C       A   \n",
       "1106052     6:31571337_C_T        rs140472229  31571337       C       T   \n",
       "1106058     6:31571570_T_C        rs115755747  31571570       T       C   \n",
       "1106059     6:31571571_C_T        rs558100393  31571571       C       T   \n",
       "1106065     6:31571665_T_C        rs148929231  31571665       T       C   \n",
       "...                    ...                ...       ...     ...     ...   \n",
       "1149090     6:32682452_G_A          rs3104407  32682452       G       A   \n",
       "1149093  6:32682509_GGTT_G  6:32682509_GGTT_G  32682509    GGTT       G   \n",
       "1149094     6:32682540_T_C        rs193258417  32682540       T       C   \n",
       "1149096        rs113721192        rs113721192  32682590       G       A   \n",
       "1149099          rs9275601          rs9275601  32682664       C       T   \n",
       "\n",
       "              maf minor_allele  info_score  chromosome  number_of_alleles  \n",
       "1106051  0.001874            A    0.855969           6                  2  \n",
       "1106052  0.006666            T    0.833402           6                  2  \n",
       "1106058  0.004427            C    0.748554           6                  2  \n",
       "1106059  0.001318            T    0.259413           6                  2  \n",
       "1106065  0.001320            C    0.789462           6                  2  \n",
       "...           ...          ...         ...         ...                ...  \n",
       "1149090  0.439507            G    0.928569           6                  2  \n",
       "1149093  0.442607            G    0.878243           6                  2  \n",
       "1149094  0.002254            C    0.885513           6                  2  \n",
       "1149096  0.009876            A    1.000000           6                  2  \n",
       "1149099  0.441535            T    1.000000           6                  2  \n",
       "\n",
       "[24052 rows x 10 columns]"
      ]
     },
     "execution_count": 111,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "subset_maf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "id": "f355d44c-519e-470b-9e0a-6cab252289da",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df=subset_maf.copy()\n",
    "df= df.drop('rsid', axis=1)\n",
    "df[\"rsid\"]=\"chr\"+ df[\"chromosome\"].astype(str) + \"_\" + df[\"position\"].astype(str) + \"_\" + df[\"allele1\"] + \"_\" + df[\"allele2\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "bde451b5-d8d4-4bbd-9ffe-ef67f668c742",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df['chromosome'] = df.chromosome.astype(str).str.zfill(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "id": "6da958e4-33de-4f95-b18d-1a71ffe9aec2",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>alternate_id</th>\n",
       "      <th>position</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "      <th>maf</th>\n",
       "      <th>minor_allele</th>\n",
       "      <th>info_score</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>number_of_alleles</th>\n",
       "      <th>rsid</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1106051</th>\n",
       "      <td>6:31571330_C_A</td>\n",
       "      <td>31571330</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>0.001874</td>\n",
       "      <td>A</td>\n",
       "      <td>0.855969</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>chr6_31571330_C_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106052</th>\n",
       "      <td>6:31571337_C_T</td>\n",
       "      <td>31571337</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.006666</td>\n",
       "      <td>T</td>\n",
       "      <td>0.833402</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>chr6_31571337_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106058</th>\n",
       "      <td>6:31571570_T_C</td>\n",
       "      <td>31571570</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.004427</td>\n",
       "      <td>C</td>\n",
       "      <td>0.748554</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>chr6_31571570_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106059</th>\n",
       "      <td>6:31571571_C_T</td>\n",
       "      <td>31571571</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.001318</td>\n",
       "      <td>T</td>\n",
       "      <td>0.259413</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>chr6_31571571_C_T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1106065</th>\n",
       "      <td>6:31571665_T_C</td>\n",
       "      <td>31571665</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.001320</td>\n",
       "      <td>C</td>\n",
       "      <td>0.789462</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>chr6_31571665_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149090</th>\n",
       "      <td>6:32682452_G_A</td>\n",
       "      <td>32682452</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.439507</td>\n",
       "      <td>G</td>\n",
       "      <td>0.928569</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>chr6_32682452_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149093</th>\n",
       "      <td>6:32682509_GGTT_G</td>\n",
       "      <td>32682509</td>\n",
       "      <td>GGTT</td>\n",
       "      <td>G</td>\n",
       "      <td>0.442607</td>\n",
       "      <td>G</td>\n",
       "      <td>0.878243</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>chr6_32682509_GGTT_G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149094</th>\n",
       "      <td>6:32682540_T_C</td>\n",
       "      <td>32682540</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>0.002254</td>\n",
       "      <td>C</td>\n",
       "      <td>0.885513</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>chr6_32682540_T_C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149096</th>\n",
       "      <td>rs113721192</td>\n",
       "      <td>32682590</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>0.009876</td>\n",
       "      <td>A</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>chr6_32682590_G_A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1149099</th>\n",
       "      <td>rs9275601</td>\n",
       "      <td>32682664</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>0.441535</td>\n",
       "      <td>T</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>chr6_32682664_C_T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24052 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "              alternate_id  position allele1 allele2       maf minor_allele  \\\n",
       "1106051     6:31571330_C_A  31571330       C       A  0.001874            A   \n",
       "1106052     6:31571337_C_T  31571337       C       T  0.006666            T   \n",
       "1106058     6:31571570_T_C  31571570       T       C  0.004427            C   \n",
       "1106059     6:31571571_C_T  31571571       C       T  0.001318            T   \n",
       "1106065     6:31571665_T_C  31571665       T       C  0.001320            C   \n",
       "...                    ...       ...     ...     ...       ...          ...   \n",
       "1149090     6:32682452_G_A  32682452       G       A  0.439507            G   \n",
       "1149093  6:32682509_GGTT_G  32682509    GGTT       G  0.442607            G   \n",
       "1149094     6:32682540_T_C  32682540       T       C  0.002254            C   \n",
       "1149096        rs113721192  32682590       G       A  0.009876            A   \n",
       "1149099          rs9275601  32682664       C       T  0.441535            T   \n",
       "\n",
       "         info_score  chromosome  number_of_alleles                  rsid  \n",
       "1106051    0.855969           6                  2     chr6_31571330_C_A  \n",
       "1106052    0.833402           6                  2     chr6_31571337_C_T  \n",
       "1106058    0.748554           6                  2     chr6_31571570_T_C  \n",
       "1106059    0.259413           6                  2     chr6_31571571_C_T  \n",
       "1106065    0.789462           6                  2     chr6_31571665_T_C  \n",
       "...             ...         ...                ...                   ...  \n",
       "1149090    0.928569           6                  2     chr6_32682452_G_A  \n",
       "1149093    0.878243           6                  2  chr6_32682509_GGTT_G  \n",
       "1149094    0.885513           6                  2     chr6_32682540_T_C  \n",
       "1149096    1.000000           6                  2     chr6_32682590_G_A  \n",
       "1149099    1.000000           6                  2     chr6_32682664_C_T  \n",
       "\n",
       "[24052 rows x 10 columns]"
      ]
     },
     "execution_count": 119,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "id": "5b887398-c135-4838-b3b4-f6bc77976e0a",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "df[[\"rsid\",\"chromosome\", \"position\", \"allele1\", \"allele2\", \"alternate_id\"]].to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_06_31571218_32682664' + '_' + str(maf) + '.z', sep=' ', index=False, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "2e1e8e4f-f5be-4bd1-9405-9745e45d9fe6",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "import glob\n",
    "import pandas as pd\n",
    "bgen=\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_31571218_32682664.bgen\"\n",
    "masterfile = pd.DataFrame({\n",
    "    \"bgi\" : \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/*.bgi\",\n",
    "    \"z\" : \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_06_31571218_32682664_0.001.z\",\n",
    "    \"bgen\" : \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_31571218_32682664.bgen\",\n",
    "    \"bgi\" : \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_31571218_32682664.bgen.bgi\",\n",
    "    \"bcor\" : [i.replace('bgen','bcor') for i in bgen],\n",
    "    \"ld\" : [i.replace('bgen','ld') for i in bgen],\n",
    "    \"n_samples\" : 351430,\n",
    "    \"sample\": f\"/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\",\n",
    "    \"incl\" : f\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.white_europeans.filtered.092821_ldprun_unrelated.filtered.incl\"\n",
    "    })\n",
    "masterfile = masterfile[['z', 'bgen', 'bgi', 'bcor', 'ld','n_samples', 'sample', 'incl']]\n",
    "\n",
    "masterfile.to_csv(\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/masterfile_chr06_31571218_32682664_v0.001_list_unrelated_whiteEur.txt\", sep=\";\", header=True, index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 138,
   "id": "fcb4983c-c946-46c4-b8d9-0dce3798151e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div class=\"sos_hint\">Cell content saved to <a href=\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/masterfile_chr06_31571218_32682664_v0.001_list_unrelated_whiteEur.txt\" target=\"_blank\">/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/masterfile_chr06_31571218_32682664_v0.001_list_unrelated_whiteEur.txt</a>, use option -r to also execute the cell.</div>"
      ],
      "text/plain": [
       "Cell content saved to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/masterfile_chr06_31571218_32682664_v0.001_list_unrelated_whiteEur.txt, use option -r to also execute the cell.\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%save /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/masterfile_chr06_31571218_32682664_v0.001_list_unrelated_whiteEur.txt -f\n",
    "z;bgen;bgi;bcor;ld;n_samples;sample;incl\n",
    "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_06_31571218_32682664_0.001.z;/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.bgen;/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.bgen.bgi;/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.bcor;/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.ld;351430;/mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample;/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/UKB_genotypedatadownloaded083019.090221_sample_variant_qc_final_callrate90.filtered.extracted.white_europeans.filtered.092821_ldprun_unrelated.filtered.incl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "eb4d9843-fedd-43c6-a60f-46c145ecdcad",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [],
   "source": [
    "## Set the bash variables \n",
    "cwd=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/\n",
    "ldstore_sbatch=$cwd/ldstore_chr1_chr22_0.001variants_$(date +\"%Y-%m-%d\").sbatch\n",
    "masterfile=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/masterfile_chr1_22_v0.001_list_unrelated_whiteEur.txt\n",
    "jobsize=1\n",
    "ldstore_sos=~/project/UKBB_GWAS_dev/workflow/111722_LDstore.ipynb\n",
    "tpl_file=~/project/bioworkflows/admin/csg2.yml\n",
    "mem='100G'\n",
    "job_size=1\n",
    "numThreads=10\n",
    "\n",
    "ldstore_args=\"\"\"bcor\n",
    " --cwd $cwd\n",
    " --masterfile $masterfile\n",
    " --numThreads $numThreads \n",
    " --mem $mem \n",
    " --job_size $job_size\n",
    "\"\"\"\n",
    "\n",
    "sos run ~/project/UKBB_GWAS_dev/admin/Get_Job_Script.ipynb csg2 \\\n",
    " --template-file $tpl_file \\\n",
    " --workflow-file $ldstore_sos \\\n",
    " --to-script $ldstore_sbatch \\\n",
    " --args \"$ldstore_args\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 140,
   "id": "f4598d9f-7b0b-43a1-a7ed-0f22341f28c1",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div class=\"sos_hint\">Cell content saved to <a href=\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/chr06_31571218_32682664_bcor.sh\" target=\"_blank\">/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/chr06_31571218_32682664_bcor.sh</a>, use option -r to also execute the cell.</div>"
      ],
      "text/plain": [
       "Cell content saved to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/chr06_31571218_32682664_bcor.sh, use option -r to also execute the cell.\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%save /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/chr06_31571218_32682664_bcor.sh -f\n",
    "#!/bin/sh\n",
    "#$ -l h_rt=90:00:00\n",
    "#$ -l h_vmem=16G\n",
    "#$ -N ldstore_blocks_chr06\n",
    "#$ -o /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ldstore_blocks_chr06_$JOB_ID.out\n",
    "#$ -e /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ldstore_blocks_chr06_$JOB_ID.err\n",
    "#$ -j y\n",
    "#$ -q csg.q\n",
    "#$ -S /bin/bash\n",
    "export PATH=$HOME/miniconda3/bin:$PATH\n",
    "module load Singularity/3.5.3\n",
    "sos run /home/dmc2245/project/UKBB_GWAS_dev/workflow/111722_LDstore.ipynb \\\n",
    "    bcor\\\n",
    "    --cwd /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22\\\n",
    "    --masterfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/masterfile_chr06_31571218_32682664_v0.001_list_unrelated_whiteEur.txt\\\n",
    "    --numThreads 10 \\\n",
    "    --mem 100G \\\n",
    "    --job_size 1 \\\n",
    "    -c /home/dmc2245/project/bioworkflows/admin/csg.yml \\\n",
    "    -q csg -s force \\\n",
    "    &> /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ldstore_blocks_testchr22_2023-02-06.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 172,
   "id": "34a47fba-1bae-449c-a783-e80319efa8fd",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "def load_npz_ld(path):\n",
    "        np_ld_loaded = np.load(path,allow_pickle=True)\n",
    "        # sort by start position\n",
    "        snp_id = [x.replace(\":\",\"_\") for x in np_ld_loaded.get(\"arr_1\")]\n",
    "        np_ld_loaded = np_ld_loaded.get(\"arr_0\")\n",
    "        new = np_ld_loaded + np_ld_loaded.T\n",
    "        np.fill_diagonal(new, np.diag(new)/2)\n",
    "        return new,snp_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 177,
   "id": "40a109a0-8307-4b33-8290-e2d1135d7302",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "import numpy as np\n",
    "np_ld_loaded = np.load('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.ld.float16.npz',allow_pickle=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 178,
   "id": "b3f3f996-1b14-44a3-9691-59fe44781cc2",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<numpy.lib.npyio.NpzFile at 0x14f4f3a76040>"
      ]
     },
     "execution_count": 178,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np_ld_loaded"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 181,
   "id": "acf6b028-5416-475c-bc56-427baadb57a9",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "snp_id = np_ld_loaded.get(\"arr_1\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 182,
   "id": "8f13a47a-a0eb-44cb-9808-fd85a0a98004",
   "metadata": {
    "kernel": "Python3",
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['6:31571330:C:A', '6:31571337:C:T', '6:31571570:T:C', ...,\n",
       "       '6:32682540:T:C', 'rs113721192', 'rs9275601'], dtype=object)"
      ]
     },
     "execution_count": 182,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "dba14dff-6772-4b64-942f-5cab616fa7a3",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [],
   "source": [
    "module load Plink/2.00a"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "id": "4a04c082-e321-495e-9ed0-24c6e425beb1",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.log.\n",
      "Options in effect:\n",
      "  --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_31571218_32682664.bgen ref-first\n",
      "  --geno 0.1\n",
      "  --make-bed\n",
      "  --new-id-max-allele-len 100\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664\n",
      "  --sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
      "  --set-all-var-ids chr@:#:$r:$a\n",
      "\n",
      "Start time: Wed Jun  7 16:01:50 2023\n",
      "257482 MiB RAM detected, ~200168 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 40734 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "--bgen: 43053 variants detected, format v1.2.\n",
      "487409 samples imported from .sample file to\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664-temporary.psam\n",
      ".\n",
      "--bgen: 36k variants converted. \n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664-temporary.pgen\n",
      "+\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664-temporary.pvar\n",
      "written.\n",
      "487409 samples (264303 females, 222994 males, 112 ambiguous; 487409 founders)\n",
      "loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664-temporary.psam.\n",
      "43053 variants loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664-temporary.pvar.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... done.\n",
      "--geno: 5210 variants removed due to missing genotype data.\n",
      "37843 variants remaining after main filters.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.fam\n",
      "... done.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.bim\n",
      "... done.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.bed\n",
      "... 4377done.\n",
      "End time: Wed Jun  7 16:02:30 2023\n"
     ]
    }
   ],
   "source": [
    "## filtering for missigness at the genotyope level\n",
    "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_31571218_32682664.bgen ref-first \\\n",
    "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
    "--set-all-var-ids 'chr@:#:$r:$a' \\\n",
    "--new-id-max-allele-len 100 \\\n",
    "--geno 0.1 \\\n",
    "--make-bed \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 230,
   "id": "6fc81aab-543a-4fe5-8025-4ff198981fbf",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664\n",
      "  --export bgen-1.2 bits=8 sample-v2 ref-first\n",
      "  --memory 10000\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664\n",
      "  --threads 8\n",
      "\n",
      "Start time: Wed Jun  7 16:04:47 2023\n",
      "257482 MiB RAM detected, ~200149 available; reserving 10000 MiB for main\n",
      "workspace.\n",
      "Using up to 8 compute threads.\n",
      "487409 samples (264303 females, 222994 males, 112 ambiguous; 487409 founders)\n",
      "loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.fam.\n",
      "37843 variants loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.bim.\n",
      "Note: No phenotype data present.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.bgen\n",
      "... 101316182124272932353740434648515456596264677073757881838689929497done.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.sample\n",
      "... done.\n",
      "End time: Wed Jun  7 16:05:43 2023\n"
     ]
    }
   ],
   "source": [
    "plink2 --threads 8 --memory 10000 \\\n",
    "        --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664 \\\n",
    "        --export bgen-1.2 'bits=8' 'sample-v2' 'ref-first' \\\n",
    "        --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 171,
   "id": "45dd6a4e-8e7a-482c-a879-5648fb6f9acf",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div class=\"sos_hint\">Cell content saved to <a href=\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/chr06_31571218_32682664.susie_rss.sh\" target=\"_blank\">/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/chr06_31571218_32682664.susie_rss.sh</a>, use option -r to also execute the cell.</div>"
      ],
      "text/plain": [
       "Cell content saved to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/chr06_31571218_32682664.susie_rss.sh, use option -r to also execute the cell.\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%save /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/chr06_31571218_32682664.susie_rss.sh\n",
    "\n",
    "#!/bin/sh\n",
    "#$ -l h_rt=36:00:00\n",
    "#$ -l h_vmem=10G\n",
    "#$ -N susie_rss\n",
    "#$ -o /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/chr06_31571218_32682664-$JOB_ID.out\n",
    "#$ -e /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/chr06_31571218_32682664-$JOB_ID.out\n",
    "#$ -j y\n",
    "#$ -q csg.q\n",
    "#$ -S /bin/bash\n",
    "export PATH=$HOME/miniconda3/bin:$PATH\n",
    "module load Singularity/3.9.4\n",
    "sos run ~/project/xqtl-pipeline/code/fine_mapping/SuSiE/SuSiE_RSS.ipynb SuSiE_RSS \\\n",
    "    --LD_list /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS_ld_list.txt\\\n",
    "    --sumstat_list /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS_sumstats_list.txt \\\n",
    "    --container ~/containers/stephenslab.sif --impute --cwd /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06 \\\n",
    "    --mem 180G \\\n",
    "    --job_size 1 \\\n",
    "    -c /home/dmc2245/project/bioworkflows/admin/csg.yml \\\n",
    "    -q csg -s force \\\n",
    "    &> /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/chr06_31571218_32682664.susie_rss.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 183,
   "id": "2682afe7-64eb-4ee2-98a6-3ee7c269a468",
   "metadata": {
    "kernel": "Bash",
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/sos/eval.py:375: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only\n",
      "  exec(\n",
      "INFO: Running \u001b[32mSuSiE_RSS_1\u001b[0m: \n",
      "\u001b[91mERROR\u001b[0m: \u001b[91mSuSiE_RSS_1 (id=c2288a48b3eb3cd7) returns an error.\u001b[0m\n",
      "\u001b[91mERROR\u001b[0m: \u001b[91m[SuSiE_RSS_1]: [0]: Executing script in Singularity returns an error (exitcode=1, stderr=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.stderr).\n",
      "The script has been saved to /home/dmc2245/.sos/fb769d0479f65da0/singularity_run_65482.py. To reproduce the error please run:\n",
      "\u001b[0m\u001b[32msingularity exec  /home/dmc2245/containers/stephenslab.sif python /home/dmc2245/.sos/fb769d0479f65da0/singularity_run_65482.py\u001b[0m\u001b[91m\n",
      "[SuSiE_RSS]: Exits with 3 pending steps (SuSiE_RSS_2, SuSiE_RSS_3, SuSiE_RSS_4)\u001b[0m\n",
      "[1]+  Exit 1                  sos run ~/project/xqtl-pipeline/code/fine_mapping/SuSiE/SuSiE_RSS.ipynb SuSiE_RSS --LD_list /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS_ld_list.txt --sumstat_list /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS_sumstats_list.txt --container ~/containers/stephenslab.sif --impute --cwd /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06\n",
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/sos/eval.py:375: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only\n",
      "  exec(\n",
      "INFO: Checking \u001b[32mSuSiE_RSS_1\u001b[0m: \n",
      "HINT: singularity exec  /home/dmc2245/containers/stephenslab.sif python /mnt/mfs/hgrcgrid/homes/dmc2245/project/UKBB_GWAS_dev/analysis/pleio_AD_ARHI/tmpt1pb7fvh/singularity_run_41215.py\n",
      "import pandas as pd\n",
      "import numpy as np\n",
      "import rpy2\n",
      "from backports import zoneinfo\n",
      "import rpy2.robjects.numpy2ri as numpy2ri\n",
      "from rpy2.robjects import pandas2ri\n",
      "from rpy2.robjects.packages import SignatureTranslatedAnonymousPackage\n",
      "import rpy2.robjects as ro\n",
      "numpy2ri.activate()\n",
      "pandas2ri.activate()\n",
      "def load_npz_ld(path):\n",
      "    np_ld_loaded = np.load(path,allow_pickle=True)\n",
      "    # sort by start position\n",
      "    snp_id = [x.replace(\":\",\"_\") for x in np_ld_loaded.get(\"arr_1\")]\n",
      "    np_ld_loaded = np_ld_loaded.get(\"arr_0\")\n",
      "    new = np_ld_loaded + np_ld_loaded.T\n",
      "    np.fill_diagonal(new, np.diag(new)/2)\n",
      "    return new,snp_id\n",
      "\n",
      "def get_bcor_meta(bcor_obj):\n",
      "    df_ld_snps = bcor_obj.getMeta()\n",
      "    df_ld_snps.rename(columns={'rsid':'SNP', 'position':'BP', 'chromosome':'CHR', 'allele1':'A1', 'allele2':'A2'}, inplace=True, errors='raise')\n",
      "    ###df_ld_snps['CHR'] = df_ld_snps['CHR'].astype(np.int64)\n",
      "    df_ld_snps['BP'] = df_ld_snps['BP'].astype(np.int64)\n",
      "    return df_ld_snps\n",
      "\n",
      "def load_ld_bcor(ld_prefix):\n",
      "    bcor_file = ld_prefix+'.bcor'\n",
      "    import os\n",
      "    import time\n",
      "    from ldstore.bcor import bcor\n",
      "    if not os.path.exists(bcor_file):\n",
      "        raise IOError('%s not found'%(bcor_file))\n",
      "    t0 = time.time()\n",
      "    bcor_obj = bcor(bcor_file)\n",
      "    df_ld_snps = get_bcor_meta(bcor_obj)\n",
      "    ld_arr = bcor_obj.readCorr([])\n",
      "    assert np.all(~np.isnan(ld_arr))\n",
      "    return ld_arr, df_ld_snps\n",
      "\n",
      "def abf(beta, se, W=0.04):\n",
      "    from scipy import special \n",
      "    z = beta / se\n",
      "    V = se ** 2\n",
      "    r = W / (W + V)\n",
      "    lbf = 0.5 * (np.log(1 - r) + (r * z ** 2))\n",
      "    denom = special.logsumexp(lbf)\n",
      "    prob = np.exp(lbf - denom)\n",
      "    return lbf, prob\n",
      "\n",
      "def get_cs(variant, prob, coverage=0.95):\n",
      "    ordering = np.argsort(prob)[::-1]\n",
      "    idx = np.where(np.cumsum(prob[ordering]) > coverage)[0][0]\n",
      "    cs = variant[ordering][: (idx + 1)]\n",
      "    return cs\n",
      "def slalom(df,LD,abf_prior_variance = 0.4 ,nlog10p_dentist_s_threshold = 4, r2_threshold = 0.6  ):\n",
      "    from scipy import stats\n",
      "    lbf, prob = abf(df.beta, df.se, W=abf_prior_variance)\n",
      "    cs = get_cs(df.variant, prob, coverage=0.95)\n",
      "    cs_99 = get_cs(df.variant, prob, coverage=0.99)\n",
      "    df[\"lbf\"] = lbf\n",
      "    df[\"prob\"] = prob\n",
      "    df[\"cs\"] = df.variant.isin(cs)\n",
      "    df[\"cs_99\"] = df.variant.isin(cs_99)\n",
      "\n",
      "    if \"pvalue\" == \"pvalue\":\n",
      "        lead_idx_snp = df.pvalue.idxmin()\n",
      "    else:\n",
      "        lead_idx_snp = df.prob.idxmax()\n",
      "\n",
      "    lead_variant = df.variant[lead_idx_snp]\n",
      "    df[\"lead_variant\"] = False\n",
      "    df[\"lead_variant\"].iloc[lead_idx_snp] = True\n",
      "    # annotate LD     \n",
      "    ## This is to identify the R for each snp vs the lead snp\n",
      "    df[\"r\"] = [LD[np.where(np.in1d(df.variant,lead_variant))][:,np.where(np.in1d(df.variant,x))][0][0][0] for x in df.variant]\n",
      "    lead_z = (df.beta / df.se).iloc[lead_idx_snp]\n",
      "    df[\"t_dentist_s\"] = ((df.beta / df.se) - df.r * lead_z) ** 2 / (1 - df.r ** 2)\n",
      "    df[\"t_dentist_s\"] = np.where(df[\"t_dentist_s\"] < 0, np.inf, df[\"t_dentist_s\"])\n",
      "    df[\"t_dentist_s\"].iloc[lead_idx_snp] = np.nan\n",
      "    df[\"nlog10p_dentist_s\"] = stats.chi2.logsf(df[\"t_dentist_s\"], df=1) / -np.log(10)\n",
      "    df[\"r2\"] = df.r ** 2\n",
      "    df[\"outliers\"] = (df.r2 > r2_threshold) & (df.nlog10p_dentist_s > nlog10p_dentist_s_threshold)\n",
      "    df_output = df\n",
      "    n_r2 = np.sum(df.r2 > r2_threshold)\n",
      "    n_dentist_s_outlier = np.sum(\n",
      "        (df.r2 > r2_threshold) & (df.nlog10p_dentist_s > nlog10p_dentist_s_threshold)\n",
      "    )\n",
      "    max_pip_idx = df.prob.idxmax()\n",
      "    df_summary = pd.DataFrame(\n",
      "        {\n",
      "            \"lead_pip_variant\": [df.variant.iloc[max_pip_idx]],\n",
      "            \"n_total\": [len(df.index)],\n",
      "            \"n_r2\": [n_r2],\n",
      "            \"n_dentist_s_outlier\": [n_dentist_s_outlier],\n",
      "            \"fraction\": [n_dentist_s_outlier / n_r2 if n_r2 > 0 else 0],\n",
      "            \"max_pip\": [np.max(df.prob)]\n",
      "        }\n",
      "        )\n",
      "    return df, df_summary\n",
      "\n",
      "## Load LD\n",
      "if \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.ld.float16.npz\".endswith(\"npz\"):\n",
      "    LD,snp_id = load_npz_ld('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.ld.float16.npz') \n",
      "if \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.ld.float16.npz\".endswith(\"bcor\"):\n",
      "    LD,snp_id = load_ld_bcor('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.ld.float16') \n",
      "\n",
      "sumstat = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS/ADGWAS_GRCh37_Bellenguez_2022.6/ADGWAS2022.GRCh37.chr6.sumstat.tsv', \"\\t\")\n",
      "## Remove nan in LD\n",
      "where = np.where(np.isnan(LD))\n",
      "na_snp = np.array(snp_id)[[x for x in where[0]] + [x for x in where[1]]]\n",
      "### Remove NA from LD\n",
      "LD = LD[np.ix_(~np.in1d(snp_id, na_snp),~np.in1d(snp_id, na_snp))]\n",
      "### Remove NA from snp_list\n",
      "snp_id = np.array(snp_id)[~np.in1d(snp_id, na_snp)]\n",
      "\n",
      "## Get only intersect snp\n",
      "intersct = np.intersect1d(sumstat.variant.to_numpy(),snp_id)\n",
      "sumstat = sumstat.query(\"variant in @intersct\").reset_index()\n",
      "indice = np.where(np.in1d(snp_id, intersct))\n",
      "## slalom\n",
      "ss_qc,ss_qc_sum = slalom(sumstat,LD[np.ix_(indice[0].tolist(), indice[0].tolist())],0.4,4,0.6)\n",
      "print(ss_qc_sum)\n",
      "ss_qc.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.ss_qced.tsv',\"\\t\",index = 0 )\n",
      "## Filter out outlier\n",
      "### Get outliers\n",
      "outliers = ss_qc[ss_qc.outliers].variant\n",
      "### Remove outliers from LD\n",
      "LD = LD[np.ix_(~np.in1d(snp_id, outliers),~np.in1d(snp_id, outliers))]\n",
      "### Remove outliers from snp_list\n",
      "snp_id = np.array(snp_id)[~np.in1d(snp_id, outliers)]\n",
      "ss_qc = ss_qc.assign(z = ss_qc.beta/ss_qc.se)\n",
      "ss_qc = ss_qc[~ss_qc.outliers]\n",
      "\n",
      "if True:\n",
      "    cur_LD = LD\n",
      "    cur_miss = ~np.in1d(snp_id, ss_qc.variant)\n",
      "    print(f'{np.mean(cur_miss)} of snps in the LD panel does not have sumstat. Their sumstat are imputed')\n",
      "    ## Sumstat imputation based on fusion code\n",
      "    cur_wgt = cur_LD[np.ix_(cur_miss, np.logical_not(cur_miss))].dot(np.linalg.inv(cur_LD[np.logical_not(cur_miss), np.logical_not(cur_miss)] + 0.1 * np.eye(np.sum(np.logical_not(cur_miss)))))\n",
      "    cur_impz  = cur_wgt @ ss_qc.z\n",
      "    cur_r2pred = np.diag(cur_wgt.dot(cur_LD[np.ix_(np.logical_not(cur_miss),np.logical_not(cur_miss))]).dot(cur_wgt.T))\n",
      "    imputed_z = cur_impz/np.sqrt(cur_r2pred)\n",
      "    working_ss = pd.concat([ss_qc[[\"variant\",\"z\"]], pd.DataFrame({\"z\" : imputed_z, \"variant\": np.array(snp_id)[cur_miss]})]).set_index(\"variant\").reindex(snp_id).reset_index()\n",
      "    import scipy.stats as st\n",
      "    working_ss[\"beta\"] = working_ss.z\n",
      "    working_ss[\"se\"] = 1\n",
      "    working_ss[\"pvalue\"] = (1 - st.norm.cdf(working_ss.z))*2\n",
      "\n",
      "\n",
      "    working_ss_qc, working_ss_qc_sum = slalom(working_ss,cur_LD,0.4,4,0.6)\n",
      "    print(working_ss_qc_sum)\n",
      "    working_ss_qc.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.imputed_ss_qced.tsv',\"\\t\",index = 0 )\n",
      "    imputed_outliers = working_ss_qc[working_ss_qc.outliers].variant\n",
      "    ### Remove outliers from LD\n",
      "    cur_LD = LD[np.ix_(~np.in1d(snp_id, imputed_outliers),~np.in1d(snp_id, imputed_outliers))]\n",
      "    ### Remove outliers from snp_list\n",
      "    snp_id = np.array(snp_id)[~np.in1d(snp_id, imputed_outliers)]\n",
      "    working_ss_qc = working_ss_qc[~working_ss_qc.outliers]\n",
      "    ss_qc = working_ss_qc\n",
      "    LD = cur_LD \n",
      "### For non-imputed LD, retained only the overlap snps. For imputed LD, this should do nothing as snp_id, ss_qc.variant .\n",
      "indice = np.where(np.in1d(snp_id, ss_qc.variant))\n",
      "LD = LD[np.ix_(indice[0].tolist(), indice[0].tolist())]\n",
      "## SuSiERSS\n",
      "string=\"\"\"\n",
      "susie_rss_analysis=function(ss_df, R, n, var_y, z_ld_weight = 0, estimate_residual_variance = FALSE, \n",
      "prior_variance = 50, check_prior = TRUE, output_path ){\n",
      "res = susieR:::susie_rss(as.matrix(ss_df$z),as.matrix(R),n,var_y, z_ld_weight = 0, estimate_residual_variance = FALSE, \n",
      "prior_variance = 50, check_prior = TRUE, max_iter = 1000,verbose = TRUE )\n",
      "condz = susieR::kriging_rss(as.matrix(ss_df$z),R)\n",
      "ggplot2::ggsave(plot = condz$plot,filename = \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.fit.rds.kriging_rss.pdf\", device = \"pdf\")\n",
      "res$variants = ss_df$variant\n",
      "res$z = ss_df$z\n",
      "res$corr = susieR:::get_cs_correlation(res, X = NULL, Xcorr = R, max = FALSE)\n",
      "rownames(res$corr) <- names(res$cs)\n",
      "colnames(res$corr) <- names(res$cs)\n",
      "if (length(res$sets$cs) > 1) {\n",
      "    index_combos = expand.grid(1:length(res$sets$cs),1:length(res$sets$cs))\n",
      "    in_common = apply(index_combos, 1, function(x) intersect(res$sets$cs[[x[1]]], res$sets$cs[[x[2]]]))\n",
      "    counts = unlist(lapply(in_common, length))\n",
      "    ovlp_mat = matrix(counts, ncol = length(res$sets$cs), byrow = T)\n",
      "    ovlp_mat[lower.tri(ovlp_mat)] = NA\n",
      "    rownames(ovlp_mat) = names(res$sets$cs)\n",
      "    colnames(ovlp_mat) = names(res$sets$cs)\n",
      "    print(ovlp_mat)\n",
      "    res$sets[[\"ovlp_mat\"]] = ovlp_mat\n",
      "}\n",
      "saveRDS(res,output_path)\n",
      "return(res)\n",
      "}\n",
      "\"\"\"\n",
      "susie_analysis = SignatureTranslatedAnonymousPackage(string, \"susie_analysis\")\n",
      "analysis_result = susie_analysis.susie_rss_analysis(ss_df = ss_qc,R = LD,output_path='/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.fit.rds' )\n",
      "\n",
      "\n",
      "INFO: \u001b[32mSuSiE_RSS_1\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mSuSiE_RSS_1\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.fit.rds /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.ss_qced.tsv\u001b[0m\n",
      "INFO: Checking \u001b[32mSuSiE_RSS_2\u001b[0m: \n",
      "HINT: singularity exec  /home/dmc2245/containers/stephenslab.sif Rscript /mnt/mfs/hgrcgrid/homes/dmc2245/project/UKBB_GWAS_dev/analysis/pleio_AD_ARHI/tmp1war81g8/singularity_run_41215.R\n",
      "res = readRDS('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.fit.rds' '/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.ss_qced.tsv')\n",
      "png('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.fit ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.ss_qced.png', width = 14, height=6, unit='in', res=300)\n",
      "par(mfrow=c(1,2))\n",
      "susieR::susie_plot(res, y= \"PIP\", pos=list(attr='pos',start=res$pos[1],end=res$pos[length(res$pos)]), add_legend=T, xlab=\"position\")\n",
      "susieR::susie_plot(res, y= \"z\", pos=list(attr='pos',start=res$pos[1],end=res$pos[length(res$pos)]), add_legend=T, xlab=\"position\", ylab=\"-log10(p)\")\n",
      "dev.off()\n",
      "\n",
      "\n",
      "INFO: \u001b[32mSuSiE_RSS_2\u001b[0m is \u001b[32mcompleted\u001b[0m.\n",
      "INFO: \u001b[32mSuSiE_RSS_2\u001b[0m output:   \u001b[32m/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.fit ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.ss_qced.png\u001b[0m\n",
      "INFO: Checking \u001b[32mSuSiE_RSS_3\u001b[0m: \n",
      "\u001b[91mERROR\u001b[0m: \u001b[91m[SuSiE_RSS_3]: [SuSiE_RSS_3]: Failed to process step output (analysis_summary = f'{cwd}/{sumstats_path:bnn}.analysis_summary.md', variants_csv = f'{cwd}/{sumstats_path:bnn}.variants.csv'): name 'sumstats_path' is not defined\n",
      "[SuSiE_RSS]: Exits with 1 pending step (SuSiE_RSS_4)\u001b[0m\n"
     ]
    },
    {
     "ename": "",
     "evalue": "1",
     "output_type": "error",
     "traceback": []
    }
   ],
   "source": [
    "module load Singularity/3.9.4\n",
    "sos dryrun ~/project/xqtl-pipeline/code/fine_mapping/SuSiE/SuSiE_RSS.ipynb SuSiE_RSS \\\n",
    "    --LD_list /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS_ld_list.txt\\\n",
    "    --sumstat_list /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS_sumstats_list.txt \\\n",
    "    --container ~/containers/stephenslab.sif --impute --cwd /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 207,
   "id": "26e28d5b-c151-4005-ac06-091db9fe7bad",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "from ldstore.bcor import bcor"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "id": "bf2d3bc2-63df-4839-9ab1-08a18c31edb1",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "myBcor = bcor( '/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.bcor' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 209,
   "id": "c6fb91fb-56a5-4a98-83e2-8b9493541c10",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.bcor'"
      ]
     },
     "execution_count": 209,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "myBcor.getFname()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 211,
   "id": "ea7dc6ec-f7d7-47e0-b96a-04031b842c16",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1158014644"
      ]
     },
     "execution_count": 211,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "myBcor.getFsize()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 214,
   "id": "be951a8f-d75b-4cbc-bdb5-f3ddc5e80972",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsid</th>\n",
       "      <th>position</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6:31571330:C:A</td>\n",
       "      <td>31571330</td>\n",
       "      <td>6</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr6:31571337:C:T</td>\n",
       "      <td>31571337</td>\n",
       "      <td>6</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6:31571570:T:C</td>\n",
       "      <td>31571570</td>\n",
       "      <td>6</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr6:31571571:C:T</td>\n",
       "      <td>31571571</td>\n",
       "      <td>6</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr6:31571665:T:C</td>\n",
       "      <td>31571665</td>\n",
       "      <td>6</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>chr6:31622606:C:A</td>\n",
       "      <td>31622606</td>\n",
       "      <td>6</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>396</th>\n",
       "      <td>chr6:31622630:A:C</td>\n",
       "      <td>31622630</td>\n",
       "      <td>6</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>397</th>\n",
       "      <td>chr6:31622816:T:C</td>\n",
       "      <td>31622816</td>\n",
       "      <td>6</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>398</th>\n",
       "      <td>chr6:31622893:T:C</td>\n",
       "      <td>31622893</td>\n",
       "      <td>6</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>399</th>\n",
       "      <td>chr6:31622948:GC:G</td>\n",
       "      <td>31622948</td>\n",
       "      <td>6</td>\n",
       "      <td>GC</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>400 rows  5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                   rsid  position chromosome allele1 allele2\n",
       "0     chr6:31571330:C:A  31571330          6       C       A\n",
       "1     chr6:31571337:C:T  31571337          6       C       T\n",
       "2     chr6:31571570:T:C  31571570          6       T       C\n",
       "3     chr6:31571571:C:T  31571571          6       C       T\n",
       "4     chr6:31571665:T:C  31571665          6       T       C\n",
       "..                  ...       ...        ...     ...     ...\n",
       "395   chr6:31622606:C:A  31622606          6       C       A\n",
       "396   chr6:31622630:A:C  31622630          6       A       C\n",
       "397   chr6:31622816:T:C  31622816          6       T       C\n",
       "398   chr6:31622893:T:C  31622893          6       T       C\n",
       "399  chr6:31622948:GC:G  31622948          6      GC       G\n",
       "\n",
       "[400 rows x 5 columns]"
      ]
     },
     "execution_count": 214,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "myBcor.getMeta().loc[ range( 400 ) ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 215,
   "id": "e2f7fb4b-77d0-42a5-b749-3586aa6615c8",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "24052"
      ]
     },
     "execution_count": 215,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "myBcor.getNumOfSNPs()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 216,
   "id": "9fa212dc-f889-430f-81ff-87cdee874faa",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "351430"
      ]
     },
     "execution_count": 216,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "myBcor.getNumOfSamples()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "fec5da09-c516-46eb-936c-1ab675d3b084",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[1.        , 0.02703174],\n",
       "       [0.02703174, 1.        ]])"
      ]
     },
     "execution_count": 228,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "myBcor.readCorr( [0,100] , [0,100] )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "869da058-3f57-48f9-88bb-499e4c9d999c",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "## Understand the number of variants shared by the z file and the summary stats for the specific chromosome 6 region "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f5fd2033-dac9-44e4-8fce-96a9a1664896",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "# Creation of the z file: it has to be space delimited and not tab delimited\n",
    "import glob, os, sys\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 242,
   "id": "bc151b18-a27c-4af2-9896-d23d1770fdab",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "# Variants in bim file\n",
    "bim = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.bim', sep='\\t', names=['rsid', 'chromosome', 'position', 'allele2', 'allele1'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 243,
   "id": "f82f64b9-b242-4c12-a91f-dab907a90f99",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsid</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>allele2</th>\n",
       "      <th>allele1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:31571218:C:T</td>\n",
       "      <td>0</td>\n",
       "      <td>31571218</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:31571228:A:C</td>\n",
       "      <td>0</td>\n",
       "      <td>31571228</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:31571296:A:G</td>\n",
       "      <td>0</td>\n",
       "      <td>31571296</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:31571308:C:T</td>\n",
       "      <td>0</td>\n",
       "      <td>31571308</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:31571330:C:A</td>\n",
       "      <td>0</td>\n",
       "      <td>31571330</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:32682566:G:C</td>\n",
       "      <td>0</td>\n",
       "      <td>32682566</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:32682590:G:A</td>\n",
       "      <td>0</td>\n",
       "      <td>32682590</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:32682623:G:T</td>\n",
       "      <td>0</td>\n",
       "      <td>32682623</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:32682656:G:A</td>\n",
       "      <td>0</td>\n",
       "      <td>32682656</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:32682664:C:T</td>\n",
       "      <td>0</td>\n",
       "      <td>32682664</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>37843 rows  5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 rsid  chromosome  position allele2 allele1\n",
       "6   chr6:31571218:C:T           0  31571218       T       C\n",
       "6   chr6:31571228:A:C           0  31571228       C       A\n",
       "6   chr6:31571296:A:G           0  31571296       G       A\n",
       "6   chr6:31571308:C:T           0  31571308       T       C\n",
       "6   chr6:31571330:C:A           0  31571330       A       C\n",
       "..                ...         ...       ...     ...     ...\n",
       "6   chr6:32682566:G:C           0  32682566       C       G\n",
       "6   chr6:32682590:G:A           0  32682590       A       G\n",
       "6   chr6:32682623:G:T           0  32682623       T       G\n",
       "6   chr6:32682656:G:A           0  32682656       A       G\n",
       "6   chr6:32682664:C:T           0  32682664       T       C\n",
       "\n",
       "[37843 rows x 5 columns]"
      ]
     },
     "execution_count": 243,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bim"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 247,
   "id": "cf2c5ccc-4021-4777-80f3-8de25de8a005",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "bimfinal= bim.replace({'chromosome': 0}, {'chromosome': 6}, regex=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 248,
   "id": "bea0ad7f-befe-4bb2-a760-7bb9af0b69b7",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsid</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>allele2</th>\n",
       "      <th>allele1</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:31571218:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>31571218</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:31571228:A:C</td>\n",
       "      <td>6</td>\n",
       "      <td>31571228</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:31571296:A:G</td>\n",
       "      <td>6</td>\n",
       "      <td>31571296</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:31571308:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>31571308</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:31571330:C:A</td>\n",
       "      <td>6</td>\n",
       "      <td>31571330</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:32682566:G:C</td>\n",
       "      <td>6</td>\n",
       "      <td>32682566</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:32682590:G:A</td>\n",
       "      <td>6</td>\n",
       "      <td>32682590</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:32682623:G:T</td>\n",
       "      <td>6</td>\n",
       "      <td>32682623</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:32682656:G:A</td>\n",
       "      <td>6</td>\n",
       "      <td>32682656</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:32682664:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>32682664</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>37843 rows  5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 rsid  chromosome  position allele2 allele1\n",
       "6   chr6:31571218:C:T           6  31571218       T       C\n",
       "6   chr6:31571228:A:C           6  31571228       C       A\n",
       "6   chr6:31571296:A:G           6  31571296       G       A\n",
       "6   chr6:31571308:C:T           6  31571308       T       C\n",
       "6   chr6:31571330:C:A           6  31571330       A       C\n",
       "..                ...         ...       ...     ...     ...\n",
       "6   chr6:32682566:G:C           6  32682566       C       G\n",
       "6   chr6:32682590:G:A           6  32682590       A       G\n",
       "6   chr6:32682623:G:T           6  32682623       T       G\n",
       "6   chr6:32682656:G:A           6  32682656       A       G\n",
       "6   chr6:32682664:C:T           6  32682664       T       C\n",
       "\n",
       "[37843 rows x 5 columns]"
      ]
     },
     "execution_count": 248,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bimfinal"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 249,
   "id": "fdf21ab7-102f-4911-8864-b4ff4ff9e030",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "bimfinal[[\"rsid\",\"chromosome\", \"position\", \"allele1\", \"allele2\"]].to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664' + '_' + '0.01' + '.z', sep=' ', index=False, header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "id": "f9022553-17c7-42d0-8b80-d9d4b7e6b346",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "## Variants in z file\n",
    "\n",
    "z_file = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_06_31571218_32682664_0.001.z', sep=' ')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "id": "ca161eca-62b5-429a-89b9-a65c27aa8ce1",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsid</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "      <th>alternate:id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6:31571330:C:A</td>\n",
       "      <td>6</td>\n",
       "      <td>31571330</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>6:31571330:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr6:31571337:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>31571337</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>6:31571337:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6:31571570:T:C</td>\n",
       "      <td>6</td>\n",
       "      <td>31571570</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>6:31571570:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr6:31571571:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>31571571</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>6:31571571:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr6:31571665:T:C</td>\n",
       "      <td>6</td>\n",
       "      <td>31571665</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>6:31571665:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24047</th>\n",
       "      <td>chr6:32682452:G:A</td>\n",
       "      <td>6</td>\n",
       "      <td>32682452</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32682452:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24048</th>\n",
       "      <td>chr6:32682509:GGTT:G</td>\n",
       "      <td>6</td>\n",
       "      <td>32682509</td>\n",
       "      <td>GGTT</td>\n",
       "      <td>G</td>\n",
       "      <td>6:32682509:GGTT:G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24049</th>\n",
       "      <td>chr6:32682540:T:C</td>\n",
       "      <td>6</td>\n",
       "      <td>32682540</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>6:32682540:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24050</th>\n",
       "      <td>chr6:32682590:G:A</td>\n",
       "      <td>6</td>\n",
       "      <td>32682590</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs113721192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24051</th>\n",
       "      <td>chr6:32682664:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>32682664</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs9275601</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>24052 rows  6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       rsid  chromosome  position allele1 allele2  \\\n",
       "0         chr6:31571330:C:A           6  31571330       C       A   \n",
       "1         chr6:31571337:C:T           6  31571337       C       T   \n",
       "2         chr6:31571570:T:C           6  31571570       T       C   \n",
       "3         chr6:31571571:C:T           6  31571571       C       T   \n",
       "4         chr6:31571665:T:C           6  31571665       T       C   \n",
       "...                     ...         ...       ...     ...     ...   \n",
       "24047     chr6:32682452:G:A           6  32682452       G       A   \n",
       "24048  chr6:32682509:GGTT:G           6  32682509    GGTT       G   \n",
       "24049     chr6:32682540:T:C           6  32682540       T       C   \n",
       "24050     chr6:32682590:G:A           6  32682590       G       A   \n",
       "24051     chr6:32682664:C:T           6  32682664       C       T   \n",
       "\n",
       "            alternate:id  \n",
       "0         6:31571330:C:A  \n",
       "1         6:31571337:C:T  \n",
       "2         6:31571570:T:C  \n",
       "3         6:31571571:C:T  \n",
       "4         6:31571665:T:C  \n",
       "...                  ...  \n",
       "24047     6:32682452:G:A  \n",
       "24048  6:32682509:GGTT:G  \n",
       "24049     6:32682540:T:C  \n",
       "24050        rs113721192  \n",
       "24051          rs9275601  \n",
       "\n",
       "[24052 rows x 6 columns]"
      ]
     },
     "execution_count": 239,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "z_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "id": "93d701c3-ff63-4cf1-960e-f9f04fb74552",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsid</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>allele1</th>\n",
       "      <th>allele2</th>\n",
       "      <th>alternate:id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6:31571330:C:A</td>\n",
       "      <td>6</td>\n",
       "      <td>31571330</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>6:31571330:C:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr6:31571337:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>31571337</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>6:31571337:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6:31571570:T:C</td>\n",
       "      <td>6</td>\n",
       "      <td>31571570</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>6:31571570:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr6:31571571:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>31571571</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>6:31571571:C:T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr6:31571665:T:C</td>\n",
       "      <td>6</td>\n",
       "      <td>31571665</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>6:31571665:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24045</th>\n",
       "      <td>chr6:32682429:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>32682429</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs9275599</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24046</th>\n",
       "      <td>chr6:32682443:G:A</td>\n",
       "      <td>6</td>\n",
       "      <td>32682443</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>6:32682443:G:A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24049</th>\n",
       "      <td>chr6:32682540:T:C</td>\n",
       "      <td>6</td>\n",
       "      <td>32682540</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>6:32682540:T:C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24050</th>\n",
       "      <td>chr6:32682590:G:A</td>\n",
       "      <td>6</td>\n",
       "      <td>32682590</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>rs113721192</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24051</th>\n",
       "      <td>chr6:32682664:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>32682664</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>rs9275601</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18842 rows  6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                    rsid  chromosome  position allele1 allele2    alternate:id\n",
       "0      chr6:31571330:C:A           6  31571330       C       A  6:31571330:C:A\n",
       "1      chr6:31571337:C:T           6  31571337       C       T  6:31571337:C:T\n",
       "2      chr6:31571570:T:C           6  31571570       T       C  6:31571570:T:C\n",
       "3      chr6:31571571:C:T           6  31571571       C       T  6:31571571:C:T\n",
       "4      chr6:31571665:T:C           6  31571665       T       C  6:31571665:T:C\n",
       "...                  ...         ...       ...     ...     ...             ...\n",
       "24045  chr6:32682429:C:T           6  32682429       C       T       rs9275599\n",
       "24046  chr6:32682443:G:A           6  32682443       G       A  6:32682443:G:A\n",
       "24049  chr6:32682540:T:C           6  32682540       T       C  6:32682540:T:C\n",
       "24050  chr6:32682590:G:A           6  32682590       G       A     rs113721192\n",
       "24051  chr6:32682664:C:T           6  32682664       C       T       rs9275601\n",
       "\n",
       "[18842 rows x 6 columns]"
      ]
     },
     "execution_count": 240,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "z_file[z_file['rsid'].isin(bim['rsid'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 241,
   "id": "1ed1079a-54fe-441e-9d74-74ecfce2d433",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>rsid</th>\n",
       "      <th>cm</th>\n",
       "      <th>pos</th>\n",
       "      <th>alt</th>\n",
       "      <th>ref</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:31571330:C:A</td>\n",
       "      <td>0</td>\n",
       "      <td>31571330</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:31571337:C:T</td>\n",
       "      <td>0</td>\n",
       "      <td>31571337</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:31571570:T:C</td>\n",
       "      <td>0</td>\n",
       "      <td>31571570</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:31571571:C:T</td>\n",
       "      <td>0</td>\n",
       "      <td>31571571</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:31571665:T:C</td>\n",
       "      <td>0</td>\n",
       "      <td>31571665</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:32682429:C:T</td>\n",
       "      <td>0</td>\n",
       "      <td>32682429</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:32682443:G:A</td>\n",
       "      <td>0</td>\n",
       "      <td>32682443</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:32682540:T:C</td>\n",
       "      <td>0</td>\n",
       "      <td>32682540</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:32682590:G:A</td>\n",
       "      <td>0</td>\n",
       "      <td>32682590</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>chr6:32682664:C:T</td>\n",
       "      <td>0</td>\n",
       "      <td>32682664</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18842 rows  5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 rsid  cm       pos alt ref\n",
       "6   chr6:31571330:C:A   0  31571330   A   C\n",
       "6   chr6:31571337:C:T   0  31571337   T   C\n",
       "6   chr6:31571570:T:C   0  31571570   C   T\n",
       "6   chr6:31571571:C:T   0  31571571   T   C\n",
       "6   chr6:31571665:T:C   0  31571665   C   T\n",
       "..                ...  ..       ...  ..  ..\n",
       "6   chr6:32682429:C:T   0  32682429   T   C\n",
       "6   chr6:32682443:G:A   0  32682443   A   G\n",
       "6   chr6:32682540:T:C   0  32682540   C   T\n",
       "6   chr6:32682590:G:A   0  32682590   A   G\n",
       "6   chr6:32682664:C:T   0  32682664   T   C\n",
       "\n",
       "[18842 rows x 5 columns]"
      ]
     },
     "execution_count": 241,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bim[bim['rsid'].isin(z_file['rsid'])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "6e6aa2b1-c808-4ce2-b97c-da9ea5889e97",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.26 64-bit (2 Apr 2022)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664\n",
      "  --keep /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/unrelated.txt\n",
      "  --make-bed\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated\n",
      "\n",
      "257482 MB RAM detected; reserving 128741 MB for main workspace.\n",
      "Allocated 40734 MB successfully, after larger attempt(s) failed.\n",
      "37843 variants loaded from .bim file.\n",
      "487409 people (222994 males, 264303 females, 112 ambiguous) loaded from .fam.\n",
      "Ambiguous sex IDs written to\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.nosex\n",
      ".\n",
      "--keep: 351430 people remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 351430 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate in remaining samples is 0.99345.\n",
      "37843 variants and 351430 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.bed\n",
      "+\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.bim\n",
      "+\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n"
     ]
    }
   ],
   "source": [
    "module load Plink/1.9.10\n",
    "plink --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664 \\\n",
    "--keep /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/unrelated.txt \\\n",
    "--make-bed \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "6cb88dd1-db9d-4fbd-9241-5fe15c7460d2",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.26 64-bit (2 Apr 2022)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.ld.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.ld\n",
      "  --r square0\n",
      "\n",
      "257482 MB RAM detected; reserving 128741 MB for main workspace.\n",
      "Allocated 40734 MB successfully, after larger attempt(s) failed.\n",
      "37843 variants loaded from .bim file.\n",
      "351430 people (162935 males, 188495 females) loaded from .fam.\n",
      "Using up to 63 threads (change this with --threads).\n",
      "Before main variant filters, 351430 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99345.\n",
      "37843 variants and 351430 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r square0 to\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.ld.ld\n",
      "... 0% [processingwriting]          done.\n"
     ]
    }
   ],
   "source": [
    "module load Plink/1.9.10\n",
    "plink --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated \\\n",
    "--r square0 \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.ld"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "ce99a9a8-d64a-40fd-961f-471dccb36bf2",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/IPython/core/interactiveshell.py:3441: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only\n",
      "  exec(code_obj, self.user_global_ns, self.user_ns)\n"
     ]
    }
   ],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "z_file=pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664_0.01.z', \" \", skiprows=1, header=None)[0].to_numpy()\n",
    "np_ld = np.loadtxt('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.ld.ld', dtype = \"float16\")\n",
    "np.savez_compressed(\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif.06_31571218_32682664.unrelated.ld.float16.npz\", np_ld, z_file, allow_pickel = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "698c8d17-926a-44b6-900f-3739445c3542",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "z_file=pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664_0.01.z', \" \", skiprows=1, header=None)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "6d901fe1-ee41-4814-8808-c364b80e58d4",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>chr6:31571218:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>31571218</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>chr6:31571228:A:C</td>\n",
       "      <td>6</td>\n",
       "      <td>31571228</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>chr6:31571296:A:G</td>\n",
       "      <td>6</td>\n",
       "      <td>31571296</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>chr6:31571308:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>31571308</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>chr6:31571330:C:A</td>\n",
       "      <td>6</td>\n",
       "      <td>31571330</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37838</th>\n",
       "      <td>chr6:32682566:G:C</td>\n",
       "      <td>6</td>\n",
       "      <td>32682566</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37839</th>\n",
       "      <td>chr6:32682590:G:A</td>\n",
       "      <td>6</td>\n",
       "      <td>32682590</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37840</th>\n",
       "      <td>chr6:32682623:G:T</td>\n",
       "      <td>6</td>\n",
       "      <td>32682623</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37841</th>\n",
       "      <td>chr6:32682656:G:A</td>\n",
       "      <td>6</td>\n",
       "      <td>32682656</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37842</th>\n",
       "      <td>chr6:32682664:C:T</td>\n",
       "      <td>6</td>\n",
       "      <td>32682664</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>37843 rows  5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                       0  1         2  3  4\n",
       "0      chr6:31571218:C:T  6  31571218  C  T\n",
       "1      chr6:31571228:A:C  6  31571228  A  C\n",
       "2      chr6:31571296:A:G  6  31571296  A  G\n",
       "3      chr6:31571308:C:T  6  31571308  C  T\n",
       "4      chr6:31571330:C:A  6  31571330  C  A\n",
       "...                  ... ..       ... .. ..\n",
       "37838  chr6:32682566:G:C  6  32682566  G  C\n",
       "37839  chr6:32682590:G:A  6  32682590  G  A\n",
       "37840  chr6:32682623:G:T  6  32682623  G  T\n",
       "37841  chr6:32682656:G:A  6  32682656  G  A\n",
       "37842  chr6:32682664:C:T  6  32682664  C  T\n",
       "\n",
       "[37843 rows x 5 columns]"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "z_file"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "58d3c58f-f48a-4ca2-94f2-f690e78b6d7b",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div class=\"sos_hint\">Cell content saved to <a href=\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/chr06_31571218_32682664.susie_rss.sh\" target=\"_blank\">/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/chr06_31571218_32682664.susie_rss.sh</a>, use option -r to also execute the cell.</div>"
      ],
      "text/plain": [
       "Cell content saved to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/chr06_31571218_32682664.susie_rss.sh, use option -r to also execute the cell.\n"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "%save /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/chr06_31571218_32682664.susie_rss.sh -f\n",
    "\n",
    "#!/bin/sh\n",
    "#$ -l h_rt=36:00:00\n",
    "#$ -l h_vmem=10G\n",
    "#$ -N susie_rss\n",
    "#$ -o /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/chr06_31571218_32682664-$JOB_ID.out\n",
    "#$ -e /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/chr06_31571218_32682664-$JOB_ID.out\n",
    "#$ -j y\n",
    "#$ -q csg.q\n",
    "#$ -S /bin/bash\n",
    "export PATH=$HOME/miniconda3/bin:$PATH\n",
    "module load Singularity/3.9.4\n",
    "sos run ~/project/xqtl-pipeline/code/fine_mapping/SuSiE/SuSiE_RSS.ipynb SuSiE_RSS \\\n",
    "    --LD_list /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS_ld_list.txt\\\n",
    "    --sumstat_list /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS_sumstats_list.txt \\\n",
    "    --container ~/containers/stephenslab.sif --impute --cwd /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06 \\\n",
    "    --mem 180G \\\n",
    "    --job_size 1 \\\n",
    "    -c /home/dmc2245/project/bioworkflows/admin/csg.yml \\\n",
    "    -q csg -s force \\\n",
    "    &> /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/chr06_31571218_32682664.susie_rss.log"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "aa07fd3d-a1b8-4228-812f-6d04b358e45d",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "import pandas as pd\n",
      "import numpy as np\n",
      "import rpy2\n",
      "from backports import zoneinfo\n",
      "import rpy2.robjects.numpy2ri as numpy2ri\n",
      "from rpy2.robjects import pandas2ri\n",
      "from rpy2.robjects.packages import SignatureTranslatedAnonymousPackage\n",
      "import rpy2.robjects as ro\n",
      "numpy2ri.activate()\n",
      "pandas2ri.activate()\n",
      "def load_npz_ld(path):\n",
      "    np_ld_loaded = np.load(path,allow_pickle=True)\n",
      "    # sort by start position\n",
      "    snp_id = [x.replace(\":\",\"_\") for x in np_ld_loaded.get(\"arr_1\")]\n",
      "    np_ld_loaded = np_ld_loaded.get(\"arr_0\")\n",
      "    new = np_ld_loaded + np_ld_loaded.T\n",
      "    np.fill_diagonal(new, np.diag(new)/2)\n",
      "    return new,snp_id\n",
      "\n",
      "def get_bcor_meta(bcor_obj):\n",
      "    df_ld_snps = bcor_obj.getMeta()\n",
      "    df_ld_snps.rename(columns={'rsid':'SNP', 'position':'BP', 'chromosome':'CHR', 'allele1':'A1', 'allele2':'A2'}, inplace=True, errors='raise')\n",
      "    ###df_ld_snps['CHR'] = df_ld_snps['CHR'].astype(np.int64)\n",
      "    df_ld_snps['BP'] = df_ld_snps['BP'].astype(np.int64)\n",
      "    return df_ld_snps\n",
      "\n",
      "def load_ld_bcor(ld_prefix):\n",
      "    bcor_file = ld_prefix+'.bcor'\n",
      "    import os\n",
      "    import time\n",
      "    from ldstore.bcor import bcor\n",
      "    if not os.path.exists(bcor_file):\n",
      "        raise IOError('%s not found'%(bcor_file))\n",
      "    t0 = time.time()\n",
      "    bcor_obj = bcor(bcor_file)\n",
      "    df_ld_snps = get_bcor_meta(bcor_obj)\n",
      "    ld_arr = bcor_obj.readCorr([])\n",
      "    assert np.all(~np.isnan(ld_arr))\n",
      "    return ld_arr, df_ld_snps\n",
      "\n",
      "def abf(beta, se, W=0.04):\n",
      "    from scipy import special \n",
      "    z = beta / se\n",
      "    V = se ** 2\n",
      "    r = W / (W + V)\n",
      "    lbf = 0.5 * (np.log(1 - r) + (r * z ** 2))\n",
      "    denom = special.logsumexp(lbf)\n",
      "    prob = np.exp(lbf - denom)\n",
      "    return lbf, prob\n",
      "\n",
      "def get_cs(variant, prob, coverage=0.95):\n",
      "    ordering = np.argsort(prob)[::-1]\n",
      "    idx = np.where(np.cumsum(prob[ordering]) > coverage)[0][0]\n",
      "    cs = variant[ordering][: (idx + 1)]\n",
      "    return cs\n",
      "def slalom(df,LD,abf_prior_variance = 0.4 ,nlog10p_dentist_s_threshold = 4, r2_threshold = 0.6  ):\n",
      "    from scipy import stats\n",
      "    lbf, prob = abf(df.beta, df.se, W=abf_prior_variance)\n",
      "    cs = get_cs(df.variant, prob, coverage=0.95)\n",
      "    cs_99 = get_cs(df.variant, prob, coverage=0.99)\n",
      "    df[\"lbf\"] = lbf\n",
      "    df[\"prob\"] = prob\n",
      "    df[\"cs\"] = df.variant.isin(cs)\n",
      "    df[\"cs_99\"] = df.variant.isin(cs_99)\n",
      "\n",
      "    if \"pvalue\" == \"pvalue\":\n",
      "        lead_idx_snp = df.pvalue.idxmin()\n",
      "    else:\n",
      "        lead_idx_snp = df.prob.idxmax()\n",
      "\n",
      "    lead_variant = df.variant[lead_idx_snp]\n",
      "    df[\"lead_variant\"] = False\n",
      "    df[\"lead_variant\"].iloc[lead_idx_snp] = True\n",
      "    # annotate LD     \n",
      "    ## This is to identify the R for each snp vs the lead snp\n",
      "    df[\"r\"] = [LD[np.where(np.in1d(df.variant,lead_variant))][:,np.where(np.in1d(df.variant,x))][0][0][0] for x in df.variant]\n",
      "    lead_z = (df.beta / df.se).iloc[lead_idx_snp]\n",
      "    df[\"t_dentist_s\"] = ((df.beta / df.se) - df.r * lead_z) ** 2 / (1 - df.r ** 2)\n",
      "    df[\"t_dentist_s\"] = np.where(df[\"t_dentist_s\"] < 0, np.inf, df[\"t_dentist_s\"])\n",
      "    df[\"t_dentist_s\"].iloc[lead_idx_snp] = np.nan\n",
      "    df[\"nlog10p_dentist_s\"] = stats.chi2.logsf(df[\"t_dentist_s\"], df=1) / -np.log(10)\n",
      "    df[\"r2\"] = df.r ** 2\n",
      "    df[\"outliers\"] = (df.r2 > r2_threshold) & (df.nlog10p_dentist_s > nlog10p_dentist_s_threshold)\n",
      "    df_output = df\n",
      "    n_r2 = np.sum(df.r2 > r2_threshold)\n",
      "    n_dentist_s_outlier = np.sum(\n",
      "        (df.r2 > r2_threshold) & (df.nlog10p_dentist_s > nlog10p_dentist_s_threshold)\n",
      "    )\n",
      "    max_pip_idx = df.prob.idxmax()\n",
      "    df_summary = pd.DataFrame(\n",
      "        {\n",
      "            \"lead_pip_variant\": [df.variant.iloc[max_pip_idx]],\n",
      "            \"n_total\": [len(df.index)],\n",
      "            \"n_r2\": [n_r2],\n",
      "            \"n_dentist_s_outlier\": [n_dentist_s_outlier],\n",
      "            \"fraction\": [n_dentist_s_outlier / n_r2 if n_r2 > 0 else 0],\n",
      "            \"max_pip\": [np.max(df.prob)]\n",
      "        }\n",
      "        )\n",
      "    return df, df_summary\n",
      "\n",
      "## Load LD\n",
      "if \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif.06_31571218_32682664.unrelated.ld.float16.npz\".endswith(\"npz\"):\n",
      "    LD,snp_id = load_npz_ld('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif.06_31571218_32682664.unrelated.ld.float16.npz') \n",
      "if \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif.06_31571218_32682664.unrelated.ld.float16.npz\".endswith(\"bcor\"):\n",
      "    LD,snp_id = load_ld_bcor('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif.06_31571218_32682664.unrelated.ld.float16') \n",
      "\n",
      "sumstat = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS/ADGWAS_GRCh37_Bellenguez_2022.6/ADGWAS2022.GRCh37.chr6.sumstat.tsv', \"\\t\")\n",
      "## Remove nan in LD\n",
      "where = np.where(np.isnan(LD))\n",
      "na_snp = np.array(snp_id)[[x for x in where[0]] + [x for x in where[1]]]\n",
      "### Remove NA from LD\n",
      "LD = LD[np.ix_(~np.in1d(snp_id, na_snp),~np.in1d(snp_id, na_snp))]\n",
      "### Remove NA from snp_list\n",
      "snp_id = np.array(snp_id)[~np.in1d(snp_id, na_snp)]\n",
      "\n",
      "## Get only intersect snp\n",
      "intersct = np.intersect1d(sumstat.variant.to_numpy(),snp_id)\n",
      "sumstat = sumstat.query(\"variant in @intersct\").reset_index()\n",
      "indice = np.where(np.in1d(snp_id, intersct))\n",
      "## slalom\n",
      "ss_qc,ss_qc_sum = slalom(sumstat,LD[np.ix_(indice[0].tolist(), indice[0].tolist())],0.4,4,0.6)\n",
      "print(ss_qc_sum)\n",
      "ss_qc.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.ss_qced.tsv',\"\\t\",index = 0 )\n",
      "## Filter out outlier\n",
      "### Get outliers\n",
      "outliers = ss_qc[ss_qc.outliers].variant\n",
      "### Remove outliers from LD\n",
      "LD = LD[np.ix_(~np.in1d(snp_id, outliers),~np.in1d(snp_id, outliers))]\n",
      "### Remove outliers from snp_list\n",
      "snp_id = np.array(snp_id)[~np.in1d(snp_id, outliers)]\n",
      "ss_qc = ss_qc.assign(z = ss_qc.beta/ss_qc.se)\n",
      "ss_qc = ss_qc[~ss_qc.outliers]\n",
      "\n",
      "if True:\n",
      "    cur_LD = LD\n",
      "    cur_miss = ~np.in1d(snp_id, ss_qc.variant)\n",
      "    print(f'{np.mean(cur_miss)} of snps in the LD panel does not have sumstat. Their sumstat are imputed')\n",
      "    ## Sumstat imputation based on fusion code\n",
      "    cur_wgt = cur_LD[np.ix_(cur_miss, np.logical_not(cur_miss))].dot(np.linalg.inv(cur_LD[np.logical_not(cur_miss), np.logical_not(cur_miss)] + 0.1 * np.eye(np.sum(np.logical_not(cur_miss)))))\n",
      "    cur_impz  = cur_wgt @ ss_qc.z\n",
      "    cur_r2pred = np.diag(cur_wgt.dot(cur_LD[np.ix_(np.logical_not(cur_miss),np.logical_not(cur_miss))]).dot(cur_wgt.T))\n",
      "    imputed_z = cur_impz/np.sqrt(cur_r2pred)\n",
      "    working_ss = pd.concat([ss_qc[[\"variant\",\"z\"]], pd.DataFrame({\"z\" : imputed_z, \"variant\": np.array(snp_id)[cur_miss]})]).set_index(\"variant\").reindex(snp_id).reset_index()\n",
      "    import scipy.stats as st\n",
      "    working_ss[\"beta\"] = working_ss.z\n",
      "    working_ss[\"se\"] = 1\n",
      "    working_ss[\"pvalue\"] = (1 - st.norm.cdf(working_ss.z))*2\n",
      "\n",
      "\n",
      "    working_ss_qc, working_ss_qc_sum = slalom(working_ss,cur_LD,0.4,4,0.6)\n",
      "    print(working_ss_qc_sum)\n",
      "    working_ss_qc.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.imputed_ss_qced.tsv',\"\\t\",index = 0 )\n",
      "    imputed_outliers = working_ss_qc[working_ss_qc.outliers].variant\n",
      "    ### Remove outliers from LD\n",
      "    cur_LD = LD[np.ix_(~np.in1d(snp_id, imputed_outliers),~np.in1d(snp_id, imputed_outliers))]\n",
      "    ### Remove outliers from snp_list\n",
      "    snp_id = np.array(snp_id)[~np.in1d(snp_id, imputed_outliers)]\n",
      "    working_ss_qc = working_ss_qc[~working_ss_qc.outliers]\n",
      "    ss_qc = working_ss_qc\n",
      "    LD = cur_LD \n",
      "### For non-imputed LD, retained only the overlap snps. For imputed LD, this should do nothing as snp_id, ss_qc.variant .\n",
      "indice = np.where(np.in1d(snp_id, ss_qc.variant))\n",
      "LD = LD[np.ix_(indice[0].tolist(), indice[0].tolist())]\n",
      "## SuSiERSS\n",
      "string=\"\"\"\n",
      "susie_rss_analysis=function(ss_df, R, n, var_y, z_ld_weight = 0, estimate_residual_variance = FALSE, \n",
      "prior_variance = 50, check_prior = TRUE, output_path ){\n",
      "res = susieR:::susie_rss(as.matrix(ss_df$z),as.matrix(R),n,var_y, z_ld_weight = 0, estimate_residual_variance = FALSE, \n",
      "prior_variance = 50, check_prior = TRUE, max_iter = 1000,verbose = TRUE )\n",
      "condz = susieR::kriging_rss(as.matrix(ss_df$z),R)\n",
      "ggplot2::ggsave(plot = condz$plot,filename = \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.fit.rds.kriging_rss.pdf\", device = \"pdf\")\n",
      "res$variants = ss_df$variant\n",
      "res$z = ss_df$z\n",
      "res$corr = susieR:::get_cs_correlation(res, X = NULL, Xcorr = R, max = FALSE)\n",
      "rownames(res$corr) <- names(res$cs)\n",
      "colnames(res$corr) <- names(res$cs)\n",
      "if (length(res$sets$cs) > 1) {\n",
      "    index_combos = expand.grid(1:length(res$sets$cs),1:length(res$sets$cs))\n",
      "    in_common = apply(index_combos, 1, function(x) intersect(res$sets$cs[[x[1]]], res$sets$cs[[x[2]]]))\n",
      "    counts = unlist(lapply(in_common, length))\n",
      "    ovlp_mat = matrix(counts, ncol = length(res$sets$cs), byrow = T)\n",
      "    ovlp_mat[lower.tri(ovlp_mat)] = NA\n",
      "    rownames(ovlp_mat) = names(res$sets$cs)\n",
      "    colnames(ovlp_mat) = names(res$sets$cs)\n",
      "    print(ovlp_mat)\n",
      "    res$sets[[\"ovlp_mat\"]] = ovlp_mat\n",
      "}\n",
      "saveRDS(res,output_path)\n",
      "return(res)\n",
      "}\n",
      "\"\"\"\n",
      "susie_analysis = SignatureTranslatedAnonymousPackage(string, \"susie_analysis\")\n",
      "analysis_result = susie_analysis.susie_rss_analysis(ss_df = ss_qc,R = LD,output_path='/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.fit.rds' )\n",
      "\n"
     ]
    }
   ],
   "source": [
    "cat /home/dmc2245/.sos/64b127122595da70/singularity_run_57303.py"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "0efca3df-ed52-4754-830a-0f0b7482b506",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import rpy2\n",
    "\n",
    "def load_npz_ld(path):\n",
    "    np_ld_loaded = np.load(path,allow_pickle=True)\n",
    "    # sort by start position\n",
    "    snp_id = [x.replace(\":\",\"_\") for x in np_ld_loaded.get(\"arr_1\")]\n",
    "    np_ld_loaded = np_ld_loaded.get(\"arr_0\")\n",
    "    new = np_ld_loaded + np_ld_loaded.T\n",
    "    np.fill_diagonal(new, np.diag(new)/2)\n",
    "    return new,snp_id\n",
    "\n",
    "def get_bcor_meta(bcor_obj):\n",
    "    df_ld_snps = bcor_obj.getMeta()\n",
    "    df_ld_snps.rename(columns={'rsid':'SNP', 'position':'BP', 'chromosome':'CHR', 'allele1':'A1', 'allele2':'A2'}, inplace=True, errors='raise')\n",
    "    ###df_ld_snps['CHR'] = df_ld_snps['CHR'].astype(np.int64)\n",
    "    df_ld_snps['BP'] = df_ld_snps['BP'].astype(np.int64)\n",
    "    return df_ld_snps\n",
    "\n",
    "def load_ld_bcor(ld_prefix):\n",
    "    bcor_file = ld_prefix+'.bcor'\n",
    "    import os\n",
    "    import time\n",
    "    from ldstore.bcor import bcor\n",
    "    if not os.path.exists(bcor_file):\n",
    "        raise IOError('%s not found'%(bcor_file))\n",
    "    t0 = time.time()\n",
    "    bcor_obj = bcor(bcor_file)\n",
    "    df_ld_snps = get_bcor_meta(bcor_obj)\n",
    "    ld_arr = bcor_obj.readCorr([])\n",
    "    assert np.all(~np.isnan(ld_arr))\n",
    "    return ld_arr, df_ld_snps\n",
    "\n",
    "def abf(beta, se, W=0.04):\n",
    "    from scipy import special \n",
    "    z = beta / se\n",
    "    V = se ** 2\n",
    "    r = W / (W + V)\n",
    "    lbf = 0.5 * (np.log(1 - r) + (r * z ** 2))\n",
    "    denom = special.logsumexp(lbf)\n",
    "    prob = np.exp(lbf - denom)\n",
    "    return lbf, prob\n",
    "\n",
    "def get_cs(variant, prob, coverage=0.95):\n",
    "    ordering = np.argsort(prob)[::-1]\n",
    "    idx = np.where(np.cumsum(prob[ordering]) > coverage)[0][0]\n",
    "    cs = variant[ordering][: (idx + 1)]\n",
    "    return cs\n",
    "def slalom(df,LD,abf_prior_variance = 0.4 ,nlog10p_dentist_s_threshold = 4, r2_threshold = 0.6  ):\n",
    "    from scipy import stats\n",
    "    lbf, prob = abf(df.beta, df.se, W=abf_prior_variance)\n",
    "    cs = get_cs(df.variant, prob, coverage=0.95)\n",
    "    cs_99 = get_cs(df.variant, prob, coverage=0.99)\n",
    "    df[\"lbf\"] = lbf\n",
    "    df[\"prob\"] = prob\n",
    "    df[\"cs\"] = df.variant.isin(cs)\n",
    "    df[\"cs_99\"] = df.variant.isin(cs_99)\n",
    "\n",
    "    if \"pvalue\" == \"pvalue\":\n",
    "        lead_idx_snp = df.pvalue.idxmin()\n",
    "    else:\n",
    "        lead_idx_snp = df.prob.idxmax()\n",
    "\n",
    "    lead_variant = df.variant[lead_idx_snp]\n",
    "    df[\"lead_variant\"] = False\n",
    "    df[\"lead_variant\"].iloc[lead_idx_snp] = True\n",
    "    # annotate LD     \n",
    "    ## This is to identify the R for each snp vs the lead snp\n",
    "    df[\"r\"] = [LD[np.where(np.in1d(df.variant,lead_variant))][:,np.where(np.in1d(df.variant,x))][0][0][0] for x in df.variant]\n",
    "    lead_z = (df.beta / df.se).iloc[lead_idx_snp]\n",
    "    df[\"t_dentist_s\"] = ((df.beta / df.se) - df.r * lead_z) ** 2 / (1 - df.r ** 2)\n",
    "    df[\"t_dentist_s\"] = np.where(df[\"t_dentist_s\"] < 0, np.inf, df[\"t_dentist_s\"])\n",
    "    df[\"t_dentist_s\"].iloc[lead_idx_snp] = np.nan\n",
    "    df[\"nlog10p_dentist_s\"] = stats.chi2.logsf(df[\"t_dentist_s\"], df=1) / -np.log(10)\n",
    "    df[\"r2\"] = df.r ** 2\n",
    "    df[\"outliers\"] = (df.r2 > r2_threshold) & (df.nlog10p_dentist_s > nlog10p_dentist_s_threshold)\n",
    "    df_output = df\n",
    "    n_r2 = np.sum(df.r2 > r2_threshold)\n",
    "    n_dentist_s_outlier = np.sum(\n",
    "        (df.r2 > r2_threshold) & (df.nlog10p_dentist_s > nlog10p_dentist_s_threshold)\n",
    "    )\n",
    "    max_pip_idx = df.prob.idxmax()\n",
    "    df_summary = pd.DataFrame(\n",
    "        {\n",
    "            \"lead_pip_variant\": [df.variant.iloc[max_pip_idx]],\n",
    "            \"n_total\": [len(df.index)],\n",
    "            \"n_r2\": [n_r2],\n",
    "            \"n_dentist_s_outlier\": [n_dentist_s_outlier],\n",
    "            \"fraction\": [n_dentist_s_outlier / n_r2 if n_r2 > 0 else 0],\n",
    "            \"max_pip\": [np.max(df.prob)]\n",
    "        }\n",
    "        )\n",
    "    return df, df_summary\n",
    "\n",
    "## Load LD\n",
    "if \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif.06_31571218_32682664.unrelated.ld.float16.npz\".endswith(\"npz\"):\n",
    "    LD,snp_id = load_npz_ld('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif.06_31571218_32682664.unrelated.ld.float16.npz') \n",
    "if \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif.06_31571218_32682664.unrelated.ld.float16.npz\".endswith(\"bcor\"):\n",
    "    LD,snp_id = load_ld_bcor('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif.06_31571218_32682664.unrelated.ld.float16') \n",
    "\n",
    "sumstat = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS/ADGWAS_GRCh37_Bellenguez_2022.6/ADGWAS2022.GRCh37.chr6.sumstat.tsv', \"\\t\")\n",
    "## Remove nan in LD\n",
    "where = np.where(np.isnan(LD))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "50ac625d-837e-45be-9284-656b13b31f5d",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array([    0,     0,     0, ..., 37842, 37842, 37842]),\n",
       " array([    1,     3,     7, ..., 37836, 37838, 37840]))"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "where"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "97c62a10-770a-44f4-82e8-c5e75eb1bd60",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "ename": "IndexError",
     "evalue": "tuple index out of range",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m/tmp/5607933.1.csg.q/ipykernel_49455/2493078317.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mwhere\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m2\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mIndexError\u001b[0m: tuple index out of range"
     ]
    }
   ],
   "source": [
    "len(where[2])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "cd8e1196-447b-45cd-ba4b-dddf2382d84e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "552718713"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(where[0])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "e63db57f-b0cd-4807-b13e-2d4b63e3c9a4",
   "metadata": {
    "kernel": "Bash",
    "tags": []
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) Plink/1.9.10 => Plink/2.00a\n",
      "\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.log.\n",
      "Options in effect:\n",
      "  --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_31571218_32682664.bgen ref-first\n",
      "  --keep /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/unrelated.txt\n",
      "  --make-bed\n",
      "  --new-id-max-allele-len 100\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated\n",
      "  --sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
      "  --set-all-var-ids chr@:#:$r:$a\n",
      "\n",
      "Start time: Wed Jul 12 13:36:03 2023\n",
      "257482 MiB RAM detected, ~242581 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Allocated 72416 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "--bgen: 43053 variants detected, format v1.2.\n",
      "487409 samples imported from .sample file to\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated-temporary.psam\n",
      ".\n",
      "--bgen: 32k variants converted.  \n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated-temporary.pgen\n",
      "+\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated-temporary.pvar\n",
      "written.\n",
      "487409 samples (264303 females, 222994 males, 112 ambiguous; 487409 founders)\n",
      "loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated-temporary.psam.\n",
      "43053 variants loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated-temporary.pvar.\n",
      "Note: No phenotype data present.\n",
      "--keep: 351430 samples remaining.\n",
      "351430 samples (188495 females, 162935 males; 351430 founders) remaining after\n",
      "main filters.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.fam\n",
      "... done.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.bim\n",
      "... done.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.bed\n",
      "... done.\n",
      "End time: Wed Jul 12 13:36:43 2023\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_31571218_32682664.bgen ref-first \\\n",
    "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
    "--set-all-var-ids 'chr@:#:$r:$a' \\\n",
    "--new-id-max-allele-len 100 \\\n",
    "--keep /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/unrelated.txt \\\n",
    "--make-bed \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "5448b9cb-7dcb-42fe-bcd3-27a7178ddd0e",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.frequency.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664\n",
      "  --freq\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.frequency\n",
      "\n",
      "Start time: Wed Jun 14 15:30:21 2023\n",
      "515529 MiB RAM detected, ~430107 available; reserving 257764 MiB for main\n",
      "workspace.\n",
      "Allocated 81558 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "351430 samples (188495 females, 162935 males; 351430 founders) loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.fam.\n",
      "28795 variants loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.bim.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... done.\n",
      "--freq: Allele frequencies (founders only) written to12222232324242525262627272828292930303131323233333434353536363737383839404041414242434344444545464647474848494950505151525253535454555556565757585859606061616262636364646565666667676868696970707171727273737474757576767777787879808081818282838384848585868687878888898990909191929293939494959596969797989899%\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.frequency.afreq\n",
      ".\n",
      "End time: Wed Jun 14 15:30:22 2023\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664 \\\n",
    "--freq \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.frequency"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "7cf57c4d-a4cf-4998-b48a-44c1b3962db9",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.26 64-bit (2 Apr 2022)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated\n",
      "  --mac 5\n",
      "  --make-bed\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5\n",
      "\n",
      "257482 MB RAM detected; reserving 128741 MB for main workspace.\n",
      "Allocated 72416 MB successfully, after larger attempt(s) failed.\n",
      "43053 variants loaded from .bim file.\n",
      "351430 people (162935 males, 188495 females) loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 351430 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.96097.\n",
      "12186 variants removed due to minor allele threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "30867 variants and 351430 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.bed\n",
      "+\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.bim\n",
      "+\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/1.9.10\n",
    "plink --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated \\\n",
    "--mac 5 \\\n",
    "--make-bed \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 64,
   "id": "8738a0c8-24cc-4241-8390-ae4b02c2e76a",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.mac5.plink2.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664\n",
      "  --mac 5\n",
      "  --make-bed\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.mac5.plink2\n",
      "\n",
      "Start time: Wed Jun 14 15:56:38 2023\n",
      "515529 MiB RAM detected, ~430096 available; reserving 257764 MiB for main\n",
      "workspace.\n",
      "Allocated 81558 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "351430 samples (188495 females, 162935 males; 351430 founders) loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.fam.\n",
      "28795 variants loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.bim.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... done.\n",
      "3556 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "25239 variants remaining after main filters.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.mac5.plink2.fam\n",
      "... done.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.mac5.plink2.bim\n",
      "... done.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.mac5.plink2.bed\n",
      "... done.\n",
      "End time: Wed Jun 14 15:56:43 2023\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664 \\\n",
    "--mac 5 \\\n",
    "--make-bed \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.mac5.plink2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "422f9d85-285d-4351-9c63-c2f65d2362db",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) Plink/2.00a => Plink/1.9.10\n",
      "\n",
      "PLINK v1.90b6.26 64-bit (2 Apr 2022)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664_ld.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664_ld\n",
      "  --r square0\n",
      "\n",
      "257482 MB RAM detected; reserving 128741 MB for main workspace.\n",
      "Allocated 72416 MB successfully, after larger attempt(s) failed.\n",
      "43053 variants loaded from .bim file.\n",
      "351430 people (162935 males, 188495 females) loaded from .fam.\n",
      "Using up to 63 threads (change this with --threads).\n",
      "Before main variant filters, 351430 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.96097.\n",
      "43053 variants and 351430 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r square0 to\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664_ld.ld\n",
      "... 0% [processingwriting]          done.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/1.9.10\n",
    "plink --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated \\\n",
    "--r square0 \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664_ld"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7e40d83f-f177-4144-bc9d-1531f0fe1701",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "z_file=pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.bim', sep='\\t', header=None)[1].to_numpy()\n",
    "np_ld = np.loadtxt('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.ld', dtype = \"float16\")\n",
    "np.savez_compressed(\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.ld.float16.npz\", np_ld, z_file, allow_pickel = True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "04c269b9-c1d6-4aa1-8e72-d6a7badc299e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import rpy2\n",
    "\n",
    "def load_npz_ld(path):\n",
    "    np_ld_loaded = np.load(path,allow_pickle=True)\n",
    "    # sort by start position\n",
    "    snp_id = [x.replace(\":\",\"_\") for x in np_ld_loaded.get(\"arr_1\")]\n",
    "    np_ld_loaded = np_ld_loaded.get(\"arr_0\")\n",
    "    new = np_ld_loaded + np_ld_loaded.T\n",
    "    np.fill_diagonal(new, np.diag(new)/2)\n",
    "    return new,snp_id\n",
    "\n",
    "def get_bcor_meta(bcor_obj):\n",
    "    df_ld_snps = bcor_obj.getMeta()\n",
    "    df_ld_snps.rename(columns={'rsid':'SNP', 'position':'BP', 'chromosome':'CHR', 'allele1':'A1', 'allele2':'A2'}, inplace=True, errors='raise')\n",
    "    ###df_ld_snps['CHR'] = df_ld_snps['CHR'].astype(np.int64)\n",
    "    df_ld_snps['BP'] = df_ld_snps['BP'].astype(np.int64)\n",
    "    return df_ld_snps\n",
    "\n",
    "def load_ld_bcor(ld_prefix):\n",
    "    bcor_file = ld_prefix+'.bcor'\n",
    "    import os\n",
    "    import time\n",
    "    from ldstore.bcor import bcor\n",
    "    if not os.path.exists(bcor_file):\n",
    "        raise IOError('%s not found'%(bcor_file))\n",
    "    t0 = time.time()\n",
    "    bcor_obj = bcor(bcor_file)\n",
    "    df_ld_snps = get_bcor_meta(bcor_obj)\n",
    "    ld_arr = bcor_obj.readCorr([])\n",
    "    assert np.all(~np.isnan(ld_arr))\n",
    "    return ld_arr, df_ld_snps\n",
    "\n",
    "def abf(beta, se, W=0.04):\n",
    "    from scipy import special \n",
    "    z = beta / se\n",
    "    V = se ** 2\n",
    "    r = W / (W + V)\n",
    "    lbf = 0.5 * (np.log(1 - r) + (r * z ** 2))\n",
    "    denom = special.logsumexp(lbf)\n",
    "    prob = np.exp(lbf - denom)\n",
    "    return lbf, prob\n",
    "\n",
    "def get_cs(variant, prob, coverage=0.95):\n",
    "    ordering = np.argsort(prob)[::-1]\n",
    "    idx = np.where(np.cumsum(prob[ordering]) > coverage)[0][0]\n",
    "    cs = variant[ordering][: (idx + 1)]\n",
    "    return cs\n",
    "def slalom(df,LD,abf_prior_variance = 0.4 ,nlog10p_dentist_s_threshold = 4, r2_threshold = 0.6  ):\n",
    "    from scipy import stats\n",
    "    lbf, prob = abf(df.beta, df.se, W=abf_prior_variance)\n",
    "    cs = get_cs(df.variant, prob, coverage=0.95)\n",
    "    cs_99 = get_cs(df.variant, prob, coverage=0.99)\n",
    "    df[\"lbf\"] = lbf\n",
    "    df[\"prob\"] = prob\n",
    "    df[\"cs\"] = df.variant.isin(cs)\n",
    "    df[\"cs_99\"] = df.variant.isin(cs_99)\n",
    "\n",
    "    if \"pvalue\" == \"pvalue\":\n",
    "        lead_idx_snp = df.pvalue.idxmin()\n",
    "    else:\n",
    "        lead_idx_snp = df.prob.idxmax()\n",
    "\n",
    "    lead_variant = df.variant[lead_idx_snp]\n",
    "    df[\"lead_variant\"] = False\n",
    "    df[\"lead_variant\"].iloc[lead_idx_snp] = True\n",
    "    # annotate LD     \n",
    "    ## This is to identify the R for each snp vs the lead snp\n",
    "    df[\"r\"] = [LD[np.where(np.in1d(df.variant,lead_variant))][:,np.where(np.in1d(df.variant,x))][0][0][0] for x in df.variant]\n",
    "    lead_z = (df.beta / df.se).iloc[lead_idx_snp]\n",
    "    df[\"t_dentist_s\"] = ((df.beta / df.se) - df.r * lead_z) ** 2 / (1 - df.r ** 2)\n",
    "    df[\"t_dentist_s\"] = np.where(df[\"t_dentist_s\"] < 0, np.inf, df[\"t_dentist_s\"])\n",
    "    df[\"t_dentist_s\"].iloc[lead_idx_snp] = np.nan\n",
    "    df[\"nlog10p_dentist_s\"] = stats.chi2.logsf(df[\"t_dentist_s\"], df=1) / -np.log(10)\n",
    "    df[\"r2\"] = df.r ** 2\n",
    "    df[\"outliers\"] = (df.r2 > r2_threshold) & (df.nlog10p_dentist_s > nlog10p_dentist_s_threshold)\n",
    "    df_output = df\n",
    "    n_r2 = np.sum(df.r2 > r2_threshold)\n",
    "    n_dentist_s_outlier = np.sum(\n",
    "        (df.r2 > r2_threshold) & (df.nlog10p_dentist_s > nlog10p_dentist_s_threshold)\n",
    "    )\n",
    "    max_pip_idx = df.prob.idxmax()\n",
    "    df_summary = pd.DataFrame(\n",
    "        {\n",
    "            \"lead_pip_variant\": [df.variant.iloc[max_pip_idx]],\n",
    "            \"n_total\": [len(df.index)],\n",
    "            \"n_r2\": [n_r2],\n",
    "            \"n_dentist_s_outlier\": [n_dentist_s_outlier],\n",
    "            \"fraction\": [n_dentist_s_outlier / n_r2 if n_r2 > 0 else 0],\n",
    "            \"max_pip\": [np.max(df.prob)]\n",
    "        }\n",
    "        )\n",
    "    return df, df_summary\n",
    "\n",
    "## Load LD\n",
    "if \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.ld.float16.npz\".endswith(\"npz\"):\n",
    "    LD,snp_id = load_npz_ld('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.ld.float16.npz') \n",
    "if \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.ld.float16.npz\".endswith(\"bcor\"):\n",
    "    LD,snp_id = load_ld_bcor('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.ld.float16.npz') \n",
    "\n",
    "sumstat = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS/ADGWAS_GRCh37_Bellenguez_2022.6/ADGWAS2022.GRCh37.chr6.sumstat.tsv', \"\\t\")\n",
    "## Remove nan in LD\n",
    "where = np.where(np.isnan(LD))\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "09882a59-4966-4103-ac44-7b4f146b056a",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array([    2,     2,     2, ..., 30862, 30862, 30862]),\n",
       " array([ 4100,  4148,  9245, ..., 11733, 12121, 21306]))"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "where "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0ceb727a-4893-438a-bd13-1f80672545b3",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[ 1.000e+00, -1.482e-04, -6.566e-04, ..., -2.478e-03, -1.165e-04,\n",
       "         6.657e-03],\n",
       "       [-1.482e-04,  1.000e+00, -2.651e-04, ...,  1.335e-02, -4.703e-05,\n",
       "        -3.952e-03],\n",
       "       [-6.566e-04, -2.651e-04,  1.000e+00, ..., -3.139e-03, -2.085e-04,\n",
       "        -9.186e-03],\n",
       "       ...,\n",
       "       [-2.478e-03,  1.335e-02, -3.139e-03, ...,  1.000e+00,  4.669e-02,\n",
       "        -4.160e-02],\n",
       "       [-1.165e-04, -4.703e-05, -2.085e-04, ...,  4.669e-02,  1.000e+00,\n",
       "         2.361e-03],\n",
       "       [ 6.657e-03, -3.952e-03, -9.186e-03, ..., -4.160e-02,  2.361e-03,\n",
       "         1.000e+00]], dtype=float16)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "1f5131b6-31a4-4694-8290-cb4364a91f05",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "345980"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(where[1])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "712ba6af-41e6-4e4a-a2dc-eef865091a5c",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "where[1:]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "685ea307-cb44-4c34-af99-a70c7cb034a7",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[False,  True, False, ...,  True, False, False],\n",
       "       [ True,  True,  True, ...,  True,  True,  True],\n",
       "       [False,  True, False, ...,  True, False, False],\n",
       "       ...,\n",
       "       [ True,  True,  True, ...,  True,  True,  True],\n",
       "       [False,  True, False, ...,  True, False, False],\n",
       "       [False,  True, False, ...,  True, False, False]])"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "np.isnan(LD)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "3fdd9c49-b756-4d52-ac57-b4fd1ba740d9",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/sos/eval.py:375: FutureWarning: In a future version of pandas all arguments of read_csv except for the argument 'filepath_or_buffer' will be keyword-only\n",
      "  exec(\n",
      "INFO: Running \u001b[32mSuSiE_RSS_1\u001b[0m: \n",
      "\u001b[91mERROR\u001b[0m: \u001b[91mSuSiE_RSS_1 (id=c2288a48b3eb3cd7) returns an error.\u001b[0m\n",
      "\u001b[91mERROR\u001b[0m: \u001b[91m[SuSiE_RSS_1]: [0]: Executing script in Singularity returns an error (exitcode=1, stderr=/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.stderr).\n",
      "The script has been saved to /home/dmc2245/.sos/fb769d0479f65da0/singularity_run_18179.py. To reproduce the error please run:\n",
      "\u001b[0m\u001b[32msingularity exec  /home/dmc2245/containers/stephenslab.sif python /home/dmc2245/.sos/fb769d0479f65da0/singularity_run_18179.py\u001b[0m\u001b[91m\n",
      "[SuSiE_RSS]: Exits with 3 pending steps (SuSiE_RSS_2, SuSiE_RSS_3, SuSiE_RSS_4)\u001b[0m\n",
      "[1]+  Exit 1                  sos run ~/project/xqtl-pipeline/code/fine_mapping/SuSiE/SuSiE_RSS.ipynb SuSiE_RSS --LD_list /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS_ld_list.txt --sumstat_list /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS_sumstats_list.txt --container ~/containers/stephenslab.sif --impute --cwd /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06\n",
      "[1] 19428\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Singularity/3.9.4\n",
    "sos dryrun ~/project/xqtl-pipeline/code/fine_mapping/SuSiE/SuSiE_RSS.ipynb SuSiE_RSS \\\n",
    "    --LD_list /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS_ld_list.txt\\\n",
    "    --sumstat_list /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS_sumstats_list.txt \\\n",
    "    --container ~/containers/stephenslab.sif --impute --cwd /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06 &\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "5e48ca63-a221-481a-b3ae-719dd10af529",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.26 64-bit (2 Apr 2022)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664\n",
      "  --freqx\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated\n",
      "\n",
      "515529 MB RAM detected; reserving 257764 MB for main workspace.\n",
      "Allocated 81558 MB successfully, after larger attempt(s) failed.\n",
      "37687 variants loaded from .bim file.\n",
      "351430 people (162935 males, 188495 females) loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 351430 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.99343.\n",
      "--freqx: Allele frequencies (founders only) written to\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.frqx\n",
      ".\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/1.9.10\n",
    "plink --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664 \\\n",
    "--freqx \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "id": "80b5ddf0-1815-4d4b-a69c-3811190c376f",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.26 64-bit (2 Apr 2022)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.maf1e8.mac10.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664\n",
      "  --freqx\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.maf1e8.mac10\n",
      "\n",
      "515529 MB RAM detected; reserving 257764 MB for main workspace.\n",
      "Allocated 81558 MB successfully, after larger attempt(s) failed.\n",
      "28795 variants loaded from .bim file.\n",
      "351430 people (162935 males, 188495 females) loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 351430 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.991403.\n",
      "--freqx: Allele frequencies (founders only) written to\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.maf1e8.mac10.frqx\n",
      ".\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/1.9.10\n",
    "plink --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664 \\\n",
    "--freqx \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_06_31571218_32682664.unrelated.maf1e8.mac10"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "8cbb780a-3f4a-4e8f-9c48-d827606cd134",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.26 64-bit (2 Apr 2022)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5\n",
      "  --r square0\n",
      "\n",
      "257482 MB RAM detected; reserving 128741 MB for main workspace.\n",
      "Allocated 72416 MB successfully, after larger attempt(s) failed.\n",
      "30867 variants loaded from .bim file.\n",
      "351430 people (162935 males, 188495 females) loaded from .fam.\n",
      "Using up to 63 threads (change this with --threads).\n",
      "Before main variant filters, 351430 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.945617.\n",
      "30867 variants and 351430 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--r square0 to\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.ld\n",
      "... 0% [processingwriting]          done.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/1.9.10\n",
    "plink --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5 \\\n",
    "--r square0 \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "c96ac029-1266-40bb-9d30-54487f1bb191",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.26 64-bit (2 Apr 2022)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/toy_example.unrelated.100vars.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated\n",
      "  --extract /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/variants_to_keep.txt\n",
      "  --make-bed\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/toy_example.unrelated.100vars\n",
      "\n",
      "257482 MB RAM detected; reserving 128741 MB for main workspace.\n",
      "Allocated 72416 MB successfully, after larger attempt(s) failed.\n",
      "43053 variants loaded from .bim file.\n",
      "351430 people (162935 males, 188495 females) loaded from .fam.\n",
      "--extract: 100 variants remaining.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 351430 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.995285.\n",
      "100 variants and 351430 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--make-bed to\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/toy_example.unrelated.100vars.bed\n",
      "+\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/toy_example.unrelated.100vars.bim\n",
      "+\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/toy_example.unrelated.100vars.fam\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/1.9.10\n",
    "plink --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated \\\n",
    "--extract  /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/variants_to_keep.txt \\\n",
    "--make-bed \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/toy_example.unrelated.100vars"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "32904983-483c-43ce-b1c9-80f3190db346",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "def load_npz_ld(path):\n",
    "    np_ld_loaded = np.load(path,allow_pickle=True)\n",
    "    # sort by start position\n",
    "    snp_id = [x.replace(\":\",\":\") for x in np_ld_loaded.get(\"arr_1\")]\n",
    "    np_ld_loaded = np_ld_loaded.get(\"arr_0\")\n",
    "    new = np_ld_loaded + np_ld_loaded.T\n",
    "    np.fill_diagonal(new, np.diag(new)/2)\n",
    "    return new,snp_id\n",
    "\n",
    "def get_bcor_meta(bcor_obj):\n",
    "    df_ld_snps = bcor_obj.getMeta()\n",
    "    df_ld_snps.rename(columns={'rsid':'SNP', 'position':'BP', 'chromosome':'CHR', 'allele1':'A1', 'allele2':'A2'}, inplace=True, errors='raise')\n",
    "    ###df_ld_snps['CHR'] = df_ld_snps['CHR'].astype(np.int64)\n",
    "    df_ld_snps['BP'] = df_ld_snps['BP'].astype(np.int64)\n",
    "    return df_ld_snps\n",
    "\n",
    "def load_ld_bcor(ld_prefix):\n",
    "    bcor_file = ld_prefix+'.bcor'\n",
    "    import os\n",
    "    import time\n",
    "    from ldstore.bcor import bcor\n",
    "    if not os.path.exists(bcor_file):\n",
    "        raise IOError('%s not found'%(bcor_file))\n",
    "    t0 = time.time()\n",
    "    bcor_obj = bcor(bcor_file)\n",
    "    df_ld_snps = get_bcor_meta(bcor_obj)\n",
    "    ld_arr = bcor_obj.readCorr([])\n",
    "    assert np.all(~np.isnan(ld_arr))\n",
    "    return ld_arr, df_ld_snps\n",
    "\n",
    "def abf(beta, se, W=0.04):\n",
    "    from scipy import special\n",
    "    z = beta / se\n",
    "    V = se ** 2\n",
    "    r = W / (W + V)\n",
    "    lbf = 0.5 * (np.log(1 - r) + (r * z ** 2))\n",
    "    denom = special.logsumexp(lbf)\n",
    "    prob = np.exp(lbf - denom)\n",
    "    return lbf, prob\n",
    "\n",
    "def get_cs(variant, prob, coverage=0.95):\n",
    "    ordering = np.argsort(prob)[::-1]\n",
    "    idx = np.where(np.cumsum(prob[ordering]) > coverage)[0][0]\n",
    "    cs = variant[ordering][: (idx + 1)]\n",
    "    return cs\n",
    "def slalom(df,LD,abf_prior_variance = 0.4 ,nlog10p_dentist_s_threshold = 4, r2_threshold = 0.6  ):\n",
    "    from scipy import stats\n",
    "    lbf, prob = abf(df.beta, df.se, W=abf_prior_variance)\n",
    "    cs = get_cs(df.variant, prob, coverage=0.95)\n",
    "    cs_99 = get_cs(df.variant, prob, coverage=0.99)\n",
    "    df[\"lbf\"] = lbf\n",
    "    df[\"prob\"] = prob\n",
    "    df[\"cs\"] = df.variant.isin(cs)\n",
    "    df[\"cs_99\"] = df.variant.isin(cs_99)\n",
    "\n",
    "    if \"pvalue\" == \"pvalue\":\n",
    "        lead_idx_snp = df.pvalue.idxmin()\n",
    "    else:\n",
    "        lead_idx_snp = df.prob.idxmax()\n",
    "\n",
    "    lead_variant = df.variant[lead_idx_snp]\n",
    "    df[\"lead_variant\"] = False\n",
    "    df[\"lead_variant\"].iloc[lead_idx_snp] = True\n",
    "    # annotate LD\n",
    "    ## This is to identify the R for each snp vs the lead snp\n",
    "    df[\"r\"] = [LD[np.where(np.in1d(df.variant,lead_variant))][:,np.where(np.in1d(df.variant,x))][0][0][0] for x in df.variant]\n",
    "    lead_z = (df.beta / df.se).iloc[lead_idx_snp]\n",
    "    df[\"t_dentist_s\"] = ((df.beta / df.se) - df.r * lead_z) ** 2 / (1 - df.r ** 2)\n",
    "    df[\"t_dentist_s\"] = np.where(df[\"t_dentist_s\"] < 0, np.inf, df[\"t_dentist_s\"])\n",
    "    df[\"t_dentist_s\"].iloc[lead_idx_snp] = np.nan\n",
    "    df[\"nlog10p_dentist_s\"] = stats.chi2.logsf(df[\"t_dentist_s\"], df=1) / -np.log(10)\n",
    "    df[\"r2\"] = df.r ** 2\n",
    "    df[\"outliers\"] = (df.r2 > r2_threshold) & (df.nlog10p_dentist_s > nlog10p_dentist_s_threshold)\n",
    "    df_output = df\n",
    "    n_r2 = np.sum(df.r2 > r2_threshold)\n",
    "    n_dentist_s_outlier = np.sum(\n",
    "        (df.r2 > r2_threshold) & (df.nlog10p_dentist_s > nlog10p_dentist_s_threshold)\n",
    "    )\n",
    "    max_pip_idx = df.prob.idxmax()\n",
    "    df_summary = pd.DataFrame(\n",
    "        {\n",
    "            \"lead_pip_variant\": [df.variant.iloc[max_pip_idx]],\n",
    "            \"n_total\": [len(df.index)],\n",
    "            \"n_r2\": [n_r2],\n",
    "            \"n_dentist_s_outlier\": [n_dentist_s_outlier],\n",
    "            \"fraction\": [n_dentist_s_outlier / n_r2 if n_r2 > 0 else 0],\n",
    "            \"max_pip\": [np.max(df.prob)]\n",
    "        }\n",
    "        )\n",
    "    return df, df_summary\n",
    "\n",
    "## Load LD\n",
    "if \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.ld.float16.npz\".endswith(\"npz\"):\n",
    "    LD,snp_id = load_npz_ld('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.ld.float16.npz')\n",
    "if \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.ld.float16.npz\".endswith(\"bcor\"):\n",
    "    LD,snp_id = load_ld_bcor('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.ld.float16')\n",
    "\n",
    "sumstat = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS/ADGWAS_GRCh37_Bellenguez_2022.6/ADGWAS2022.GRCh37.chr6.sumstat.tsv', \"\\t\")\n",
    "## Remove nan in LD\n",
    "where = np.where(np.isnan(LD))\n",
    "na_snp = np.array(snp_id)[[x for x in where[0]] + [x for x in where[1]]]\n",
    "### Remove NA from LD\n",
    "LD = LD[np.ix_(~np.in1d(snp_id, na_snp),~np.in1d(snp_id, na_snp))]\n",
    "### Remove NA from snp_list\n",
    "snp_id = np.array(snp_id)[~np.in1d(snp_id, na_snp)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "0540d335-8cca-4181-8d9e-5733158be97e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[ 1.000e+00, -1.482e-04, -1.957e-03, ..., -2.478e-03, -1.165e-04,\n",
       "         6.657e-03],\n",
       "       [-1.482e-04,  1.000e+00, -7.939e-04, ...,  1.335e-02, -4.703e-05,\n",
       "        -3.952e-03],\n",
       "       [-1.957e-03, -7.939e-04,  1.000e+00, ..., -7.568e-03, -6.247e-04,\n",
       "        -3.345e-02],\n",
       "       ...,\n",
       "       [-2.478e-03,  1.335e-02, -7.568e-03, ...,  1.000e+00,  4.669e-02,\n",
       "        -4.160e-02],\n",
       "       [-1.165e-04, -4.703e-05, -6.247e-04, ...,  4.669e-02,  1.000e+00,\n",
       "         2.361e-03],\n",
       "       [ 6.657e-03, -3.952e-03, -3.345e-02, ..., -4.160e-02,  2.361e-03,\n",
       "         1.000e+00]], dtype=float16)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "LD"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "31e67a7c-34a6-451e-bdb1-2dc7395d848e",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['chr6:31571218:C:T', 'chr6:31571296:A:G', 'chr6:31571337:C:T', ...,\n",
       "       'chr6:32682590:G:A', 'chr6:32682656:G:A', 'chr6:32682664:C:T'],\n",
       "      dtype='<U80')"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "70c652a4-cfcb-42ea-b450-45adc1004c13",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "## Get only intersect snp\n",
    "intersct = np.intersect1d(sumstat.variant.to_numpy(),snp_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "4a7485de-2209-4d57-822d-90d22ab131dc",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "8193"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(intersct)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "d72fa3c1-4cdf-4b3f-8021-b810f6fb836e",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "sumstat = sumstat.query(\"variant in @intersct\").reset_index()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "bb1b4b49-87a8-4dad-881f-7ec1b0aaf6c7",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>level_0</th>\n",
       "      <th>index</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>ref</th>\n",
       "      <th>alt</th>\n",
       "      <th>variant</th>\n",
       "      <th>beta</th>\n",
       "      <th>se</th>\n",
       "      <th>pvalue</th>\n",
       "      <th>lbf</th>\n",
       "      <th>prob</th>\n",
       "      <th>cs</th>\n",
       "      <th>cs_99</th>\n",
       "      <th>lead_variant</th>\n",
       "      <th>r</th>\n",
       "      <th>t_dentist_s</th>\n",
       "      <th>nlog10p_dentist_s</th>\n",
       "      <th>r2</th>\n",
       "      <th>outliers</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0</td>\n",
       "      <td>230517</td>\n",
       "      <td>6</td>\n",
       "      <td>31571218</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr6:31571218:C:T</td>\n",
       "      <td>-0.0293</td>\n",
       "      <td>0.0938</td>\n",
       "      <td>7.546000e-01</td>\n",
       "      <td>-1.871587</td>\n",
       "      <td>6.304009e-16</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>-0.008240</td>\n",
       "      <td>0.145977</td>\n",
       "      <td>0.153409</td>\n",
       "      <td>0.000068</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>230518</td>\n",
       "      <td>6</td>\n",
       "      <td>31571296</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr6:31571296:A:G</td>\n",
       "      <td>0.2547</td>\n",
       "      <td>0.3087</td>\n",
       "      <td>4.094000e-01</td>\n",
       "      <td>-0.549201</td>\n",
       "      <td>2.365495e-15</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>-0.001299</td>\n",
       "      <td>0.662737</td>\n",
       "      <td>0.381329</td>\n",
       "      <td>0.000002</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>2</td>\n",
       "      <td>230520</td>\n",
       "      <td>6</td>\n",
       "      <td>31571337</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr6:31571337:C:T</td>\n",
       "      <td>-0.1073</td>\n",
       "      <td>0.0601</td>\n",
       "      <td>7.420000e-02</td>\n",
       "      <td>-0.778607</td>\n",
       "      <td>1.880583e-15</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.084656</td>\n",
       "      <td>1.151298</td>\n",
       "      <td>0.547788</td>\n",
       "      <td>0.007168</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>230521</td>\n",
       "      <td>6</td>\n",
       "      <td>31571570</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr6:31571570:T:C</td>\n",
       "      <td>-0.0898</td>\n",
       "      <td>0.0662</td>\n",
       "      <td>1.749000e-01</td>\n",
       "      <td>-1.352309</td>\n",
       "      <td>1.059587e-15</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.043488</td>\n",
       "      <td>0.979194</td>\n",
       "      <td>0.491608</td>\n",
       "      <td>0.001891</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>4</td>\n",
       "      <td>230522</td>\n",
       "      <td>6</td>\n",
       "      <td>31571601</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>chr6:31571601:A:G</td>\n",
       "      <td>-0.1719</td>\n",
       "      <td>0.2852</td>\n",
       "      <td>5.468000e-01</td>\n",
       "      <td>-0.738023</td>\n",
       "      <td>1.958472e-15</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>-0.003122</td>\n",
       "      <td>0.395827</td>\n",
       "      <td>0.276337</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8188</th>\n",
       "      <td>8188</td>\n",
       "      <td>240066</td>\n",
       "      <td>6</td>\n",
       "      <td>32682402</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr6:32682402:C:T</td>\n",
       "      <td>-0.0065</td>\n",
       "      <td>0.0087</td>\n",
       "      <td>4.510000e-01</td>\n",
       "      <td>-4.007335</td>\n",
       "      <td>7.448577e-17</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.061279</td>\n",
       "      <td>0.052450</td>\n",
       "      <td>0.086794</td>\n",
       "      <td>0.003756</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8189</th>\n",
       "      <td>8189</td>\n",
       "      <td>240067</td>\n",
       "      <td>6</td>\n",
       "      <td>32682429</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr6:32682429:C:T</td>\n",
       "      <td>-0.1075</td>\n",
       "      <td>0.0142</td>\n",
       "      <td>3.767000e-14</td>\n",
       "      <td>24.844591</td>\n",
       "      <td>2.525257e-04</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>0.654297</td>\n",
       "      <td>7.244619</td>\n",
       "      <td>2.148046</td>\n",
       "      <td>0.428223</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8190</th>\n",
       "      <td>8190</td>\n",
       "      <td>240071</td>\n",
       "      <td>6</td>\n",
       "      <td>32682540</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>chr6:32682540:T:C</td>\n",
       "      <td>0.0611</td>\n",
       "      <td>0.0935</td>\n",
       "      <td>5.133000e-01</td>\n",
       "      <td>-1.713510</td>\n",
       "      <td>7.383615e-16</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>-0.021011</td>\n",
       "      <td>0.226441</td>\n",
       "      <td>0.197790</td>\n",
       "      <td>0.000442</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8191</th>\n",
       "      <td>8191</td>\n",
       "      <td>240072</td>\n",
       "      <td>6</td>\n",
       "      <td>32682590</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>chr6:32682590:G:A</td>\n",
       "      <td>0.0034</td>\n",
       "      <td>0.0429</td>\n",
       "      <td>9.368000e-01</td>\n",
       "      <td>-2.689907</td>\n",
       "      <td>2.781155e-16</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>-0.043762</td>\n",
       "      <td>0.084909</td>\n",
       "      <td>0.113086</td>\n",
       "      <td>0.001915</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8192</th>\n",
       "      <td>8192</td>\n",
       "      <td>240073</td>\n",
       "      <td>6</td>\n",
       "      <td>32682664</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>chr6:32682664:C:T</td>\n",
       "      <td>-0.0023</td>\n",
       "      <td>0.0084</td>\n",
       "      <td>7.811000e-01</td>\n",
       "      <td>-4.283987</td>\n",
       "      <td>5.648393e-17</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>-0.131226</td>\n",
       "      <td>1.949214</td>\n",
       "      <td>0.788688</td>\n",
       "      <td>0.017227</td>\n",
       "      <td>False</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>8193 rows  20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      level_0   index  chromosome  position ref alt            variant  \\\n",
       "0           0  230517           6  31571218   C   T  chr6:31571218:C:T   \n",
       "1           1  230518           6  31571296   A   G  chr6:31571296:A:G   \n",
       "2           2  230520           6  31571337   C   T  chr6:31571337:C:T   \n",
       "3           3  230521           6  31571570   T   C  chr6:31571570:T:C   \n",
       "4           4  230522           6  31571601   A   G  chr6:31571601:A:G   \n",
       "...       ...     ...         ...       ...  ..  ..                ...   \n",
       "8188     8188  240066           6  32682402   C   T  chr6:32682402:C:T   \n",
       "8189     8189  240067           6  32682429   C   T  chr6:32682429:C:T   \n",
       "8190     8190  240071           6  32682540   T   C  chr6:32682540:T:C   \n",
       "8191     8191  240072           6  32682590   G   A  chr6:32682590:G:A   \n",
       "8192     8192  240073           6  32682664   C   T  chr6:32682664:C:T   \n",
       "\n",
       "        beta      se        pvalue        lbf          prob     cs  cs_99  \\\n",
       "0    -0.0293  0.0938  7.546000e-01  -1.871587  6.304009e-16  False  False   \n",
       "1     0.2547  0.3087  4.094000e-01  -0.549201  2.365495e-15  False  False   \n",
       "2    -0.1073  0.0601  7.420000e-02  -0.778607  1.880583e-15  False  False   \n",
       "3    -0.0898  0.0662  1.749000e-01  -1.352309  1.059587e-15  False  False   \n",
       "4    -0.1719  0.2852  5.468000e-01  -0.738023  1.958472e-15  False  False   \n",
       "...      ...     ...           ...        ...           ...    ...    ...   \n",
       "8188 -0.0065  0.0087  4.510000e-01  -4.007335  7.448577e-17  False  False   \n",
       "8189 -0.1075  0.0142  3.767000e-14  24.844591  2.525257e-04  False  False   \n",
       "8190  0.0611  0.0935  5.133000e-01  -1.713510  7.383615e-16  False  False   \n",
       "8191  0.0034  0.0429  9.368000e-01  -2.689907  2.781155e-16  False  False   \n",
       "8192 -0.0023  0.0084  7.811000e-01  -4.283987  5.648393e-17  False  False   \n",
       "\n",
       "      lead_variant         r  t_dentist_s  nlog10p_dentist_s        r2  \\\n",
       "0            False -0.008240     0.145977           0.153409  0.000068   \n",
       "1            False -0.001299     0.662737           0.381329  0.000002   \n",
       "2            False  0.084656     1.151298           0.547788  0.007168   \n",
       "3            False  0.043488     0.979194           0.491608  0.001891   \n",
       "4            False -0.003122     0.395827           0.276337  0.000010   \n",
       "...            ...       ...          ...                ...       ...   \n",
       "8188         False  0.061279     0.052450           0.086794  0.003756   \n",
       "8189         False  0.654297     7.244619           2.148046  0.428223   \n",
       "8190         False -0.021011     0.226441           0.197790  0.000442   \n",
       "8191         False -0.043762     0.084909           0.113086  0.001915   \n",
       "8192         False -0.131226     1.949214           0.788688  0.017227   \n",
       "\n",
       "      outliers  \n",
       "0        False  \n",
       "1        False  \n",
       "2        False  \n",
       "3        False  \n",
       "4        False  \n",
       "...        ...  \n",
       "8188     False  \n",
       "8189     False  \n",
       "8190     False  \n",
       "8191     False  \n",
       "8192     False  \n",
       "\n",
       "[8193 rows x 20 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sumstat"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "85636e56-2d7b-43fb-a9b1-ac78534c4186",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "indice = np.where(np.in1d(snp_id, intersct))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "f2c5eaa8-0d74-4964-a380-476df7a5577b",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array([    0,     1,     2, ..., 18467, 18468, 18470]),)"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "indice"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "a2635ce3-8aec-41cf-8cf8-b4b17674c594",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "ename": "SyntaxError",
     "evalue": "invalid syntax (1723443811.py, line 1)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"/tmp/5760179.1.csg.q/ipykernel_12619/1723443811.py\"\u001b[0;36m, line \u001b[0;32m1\u001b[0m\n\u001b[0;31m    ggplot2::ggsave(plot = condz$plot,filename = \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.fit.rds.kriging_rss.pdf\", device = \"pdf\")\u001b[0m\n\u001b[0m            ^\u001b[0m\n\u001b[0;31mSyntaxError\u001b[0m\u001b[0;31m:\u001b[0m invalid syntax\n"
     ]
    }
   ],
   "source": [
    "ggplot2::ggsave(plot = condz$plot,filename = \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.fit.rds.kriging_rss.pdf\", device = \"pdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c1cc8704-1fe1-4b6c-b6dd-c482512389ff",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_block(indexer, value, name)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    lead_pip_variant  n_total  n_r2  n_dentist_s_outlier  fraction   max_pip\n",
      "0  chr6:32561283:C:A     8193    28                    2  0.071429  0.252163\n",
      "0.5565481024308375 of snps in the LD panel does not have sumstat. Their sumstat are imputed\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_block(indexer, value, name)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    lead_pip_variant  n_total  n_r2  n_dentist_s_outlier  fraction   max_pip\n",
      "0  chr6:32561283:C:A    18471     7                    0       0.0  0.054285\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'SignatureTranslatedAnonymousPackage' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/tmp/5760179.1.csg.q/ipykernel_12619/3017530074.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     73\u001b[0m }\n\u001b[1;32m     74\u001b[0m \"\"\"\n\u001b[0;32m---> 75\u001b[0;31m \u001b[0msusie_analysis\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mSignatureTranslatedAnonymousPackage\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstring\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"susie_analysis\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     76\u001b[0m \u001b[0manalysis_result\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msusie_analysis\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msusie_rss_analysis\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mss_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mss_qc\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mR\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mLD\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0moutput_path\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.fit.rds'\u001b[0m \u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'SignatureTranslatedAnonymousPackage' is not defined"
     ]
    }
   ],
   "source": [
    "## slalom\n",
    "ss_qc,ss_qc_sum = slalom(sumstat,LD[np.ix_(indice[0].tolist(), indice[0].tolist())],0.4,4,0.6)\n",
    "print(ss_qc_sum)\n",
    "ss_qc.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.ss_qced.tsv',\"\\t\",index = 0 )\n",
    "## Filter out outlier\n",
    "### Get outliers\n",
    "outliers = ss_qc[ss_qc.outliers].variant\n",
    "### Remove outliers from LD\n",
    "LD = LD[np.ix_(~np.in1d(snp_id, outliers),~np.in1d(snp_id, outliers))]\n",
    "### Remove outliers from snp_list\n",
    "snp_id = np.array(snp_id)[~np.in1d(snp_id, outliers)]\n",
    "ss_qc = ss_qc.assign(z = ss_qc.beta/ss_qc.se)\n",
    "ss_qc = ss_qc[~ss_qc.outliers]\n",
    "\n",
    "if True:\n",
    "    cur_LD = LD\n",
    "    cur_miss = ~np.in1d(snp_id, ss_qc.variant)\n",
    "    print(f'{np.mean(cur_miss)} of snps in the LD panel does not have sumstat. Their sumstat are imputed')\n",
    "    ## Sumstat imputation based on fusion code\n",
    "    cur_wgt = cur_LD[np.ix_(cur_miss, np.logical_not(cur_miss))].dot(np.linalg.inv(cur_LD[np.logical_not(cur_miss), np.logical_not(cur_miss)] + 0.1 * np.eye(np.sum(np.logical_not(cur_miss)))))\n",
    "    cur_impz  = cur_wgt @ ss_qc.z\n",
    "    cur_r2pred = np.diag(cur_wgt.dot(cur_LD[np.ix_(np.logical_not(cur_miss),np.logical_not(cur_miss))]).dot(cur_wgt.T))\n",
    "    imputed_z = cur_impz/np.sqrt(cur_r2pred)\n",
    "    working_ss = pd.concat([ss_qc[[\"variant\",\"z\"]], pd.DataFrame({\"z\" : imputed_z, \"variant\": np.array(snp_id)[cur_miss]})]).set_index(\"variant\").reindex(snp_id).reset_index()\n",
    "    import scipy.stats as st\n",
    "    working_ss[\"beta\"] = working_ss.z\n",
    "    working_ss[\"se\"] = 1\n",
    "    working_ss[\"pvalue\"] = (1 - st.norm.cdf(working_ss.z))*2\n",
    "\n",
    "\n",
    "    working_ss_qc, working_ss_qc_sum = slalom(working_ss,cur_LD,0.4,4,0.6)\n",
    "    print(working_ss_qc_sum)\n",
    "    working_ss_qc.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.imputed_ss_qced.tsv',\"\\t\",index = 0 )\n",
    "    imputed_outliers = working_ss_qc[working_ss_qc.outliers].variant\n",
    "    ### Remove outliers from LD\n",
    "    cur_LD = LD[np.ix_(~np.in1d(snp_id, imputed_outliers),~np.in1d(snp_id, imputed_outliers))]\n",
    "    ### Remove outliers from snp_list\n",
    "    snp_id = np.array(snp_id)[~np.in1d(snp_id, imputed_outliers)]\n",
    "    working_ss_qc = working_ss_qc[~working_ss_qc.outliers]\n",
    "    ss_qc = working_ss_qc\n",
    "    LD = cur_LD\n",
    "### For non-imputed LD, retained only the overlap snps. For imputed LD, this should do nothing as snp_id, ss_qc.variant .\n",
    "indice = np.where(np.in1d(snp_id, ss_qc.variant))\n",
    "LD = LD[np.ix_(indice[0].tolist(), indice[0].tolist())]\n",
    "## SuSiERSS\n",
    "string=\"\"\"\n",
    "susie_rss_analysis=function(ss_df, R, n, var_y, z_ld_weight = 0, estimate_residual_variance = FALSE,\n",
    "prior_variance = 50, check_prior = TRUE, output_path ){\n",
    "res = susieR:::susie_rss(as.matrix(ss_df$z),as.matrix(R),n,var_y, z_ld_weight = 0, estimate_residual_variance = FALSE,\n",
    "prior_variance = 50, check_prior = TRUE, max_iter = 1000,verbose = TRUE )\n",
    "condz = susieR::kriging_rss(as.matrix(ss_df$z),R)\n",
    "ggplot2::ggsave(plot = condz$plot,filename = \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.fit.rds.kriging_rss.pdf\", device = \"pdf\")\n",
    "res$variants = ss_df$variant\n",
    "res$z = ss_df$z\n",
    "res$corr = susieR:::get_cs_correlation(res, X = NULL, Xcorr = R, max = FALSE)\n",
    "rownames(res$corr) <- names(res$cs)\n",
    "colnames(res$corr) <- names(res$cs)\n",
    "if (length(res$sets$cs) > 1) {\n",
    "    index_combos = expand.grid(1:length(res$sets$cs),1:length(res$sets$cs))\n",
    "    in_common = apply(index_combos, 1, function(x) intersect(res$sets$cs[[x[1]]], res$sets$cs[[x[2]]]))\n",
    "    counts = unlist(lapply(in_common, length))\n",
    "    ovlp_mat = matrix(counts, ncol = length(res$sets$cs), byrow = T)\n",
    "    ovlp_mat[lower.tri(ovlp_mat)] = NA\n",
    "    rownames(ovlp_mat) = names(res$sets$cs)\n",
    "    colnames(ovlp_mat) = names(res$sets$cs)\n",
    "    print(ovlp_mat)\n",
    "    res$sets[[\"ovlp_mat\"]] = ovlp_mat\n",
    "}\n",
    "saveRDS(res,output_path)\n",
    "return(res)\n",
    "}\n",
    "\"\"\"\n",
    "susie_analysis = SignatureTranslatedAnonymousPackage(string, \"susie_analysis\")\n",
    "analysis_result = susie_analysis.susie_rss_analysis(ss_df = ss_qc,R = LD,output_path='/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06/ADGWAS2022.GRCh37.chr6.sumstat.ld.unisusie_rss.fit.rds' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "8162e3c3-a170-4255-a36f-675d29aee7b2",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "module load Singularity/3.9.4\n",
    "sos run ~/project/xqtl-pipeline/code/fine_mapping/SuSiE/SuSiE_RSS.ipynb SuSiE_RSS \\\n",
    "    --LD_list /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS_ldrosmap_list.txt\\\n",
    "    --sumstat_list /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS_sumstats_list.txt \\\n",
    "    --n 351430 \\\n",
    "    --container ~/containers/stephenslab.sif --impute --cwd /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06_rosmap &"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "cc2302d2-206f-4e72-a78b-266e704556d6",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "def load_npz_ld(path):\n",
    "    np_ld_loaded = np.load(path,allow_pickle=True)\n",
    "    # sort by start position\n",
    "    snp_id = [x.replace(\"_\",\":\") for x in np_ld_loaded.get(\"arr_1\")]\n",
    "    np_ld_loaded = np_ld_loaded.get(\"arr_0\")\n",
    "    new = np_ld_loaded + np_ld_loaded.T\n",
    "    np.fill_diagonal(new, np.diag(new)/2)\n",
    "    return new,snp_id\n",
    "\n",
    "def get_bcor_meta(bcor_obj):\n",
    "    df_ld_snps = bcor_obj.getMeta()\n",
    "    df_ld_snps.rename(columns={'rsid':'SNP', 'position':'BP', 'chromosome':'CHR', 'allele1':'A1', 'allele2':'A2'}, inplace=True, errors='raise')\n",
    "    ###df_ld_snps['CHR'] = df_ld_snps['CHR'].astype(np.int64)\n",
    "    df_ld_snps['BP'] = df_ld_snps['BP'].astype(np.int64)\n",
    "    return df_ld_snps\n",
    "\n",
    "def load_ld_bcor(ld_prefix):\n",
    "    bcor_file = ld_prefix+'.bcor'\n",
    "    import os\n",
    "    import time\n",
    "    from ldstore.bcor import bcor\n",
    "    if not os.path.exists(bcor_file):\n",
    "        raise IOError('%s not found'%(bcor_file))\n",
    "    t0 = time.time()\n",
    "    bcor_obj = bcor(bcor_file)\n",
    "    df_ld_snps = get_bcor_meta(bcor_obj)\n",
    "    ld_arr = bcor_obj.readCorr([])\n",
    "    assert np.all(~np.isnan(ld_arr))\n",
    "    return ld_arr, df_ld_snps\n",
    "\n",
    "def abf(beta, se, W=0.04):\n",
    "    from scipy import special\n",
    "    z = beta / se\n",
    "    V = se ** 2\n",
    "    r = W / (W + V)\n",
    "    lbf = 0.5 * (np.log(1 - r) + (r * z ** 2))\n",
    "    denom = special.logsumexp(lbf)\n",
    "    prob = np.exp(lbf - denom)\n",
    "    return lbf, prob\n",
    "\n",
    "def get_cs(variant, prob, coverage=0.95):\n",
    "    ordering = np.argsort(prob)[::-1]\n",
    "    idx = np.where(np.cumsum(prob[ordering]) > coverage)[0][0]\n",
    "    cs = variant[ordering][: (idx + 1)]\n",
    "    return cs\n",
    "def slalom(df,LD,abf_prior_variance = 0.4 ,nlog10p_dentist_s_threshold = 4, r2_threshold = 0.6  ):\n",
    "    from scipy import stats\n",
    "    lbf, prob = abf(df.beta, df.se, W=abf_prior_variance)\n",
    "    cs = get_cs(df.variant, prob, coverage=0.95)\n",
    "    cs_99 = get_cs(df.variant, prob, coverage=0.99)\n",
    "    df[\"lbf\"] = lbf\n",
    "    df[\"prob\"] = prob\n",
    "    df[\"cs\"] = df.variant.isin(cs)\n",
    "    df[\"cs_99\"] = df.variant.isin(cs_99)\n",
    "\n",
    "    if \"pvalue\" == \"pvalue\":\n",
    "        lead_idx_snp = df.pvalue.idxmin()\n",
    "    else:\n",
    "        lead_idx_snp = df.prob.idxmax()\n",
    "\n",
    "    lead_variant = df.variant[lead_idx_snp]\n",
    "    df[\"lead_variant\"] = False\n",
    "    df[\"lead_variant\"].iloc[lead_idx_snp] = True\n",
    "    # annotate LD\n",
    "    ## This is to identify the R for each snp vs the lead snp\n",
    "    df[\"r\"] = [LD[np.where(np.in1d(df.variant,lead_variant))][:,np.where(np.in1d(df.variant,x))][0][0][0] for x in df.variant]\n",
    "    lead_z = (df.beta / df.se).iloc[lead_idx_snp]\n",
    "    df[\"t_dentist_s\"] = ((df.beta / df.se) - df.r * lead_z) ** 2 / (1 - df.r ** 2)\n",
    "    df[\"t_dentist_s\"] = np.where(df[\"t_dentist_s\"] < 0, np.inf, df[\"t_dentist_s\"])\n",
    "    df[\"t_dentist_s\"].iloc[lead_idx_snp] = np.nan\n",
    "    df[\"nlog10p_dentist_s\"] = stats.chi2.logsf(df[\"t_dentist_s\"], df=1) / -np.log(10)\n",
    "    df[\"r2\"] = df.r ** 2\n",
    "    df[\"outliers\"] = (df.r2 > r2_threshold) & (df.nlog10p_dentist_s > nlog10p_dentist_s_threshold)\n",
    "    df_output = df\n",
    "    n_r2 = np.sum(df.r2 > r2_threshold)\n",
    "    n_dentist_s_outlier = np.sum(\n",
    "        (df.r2 > r2_threshold) & (df.nlog10p_dentist_s > nlog10p_dentist_s_threshold)\n",
    "    )\n",
    "    max_pip_idx = df.prob.idxmax()\n",
    "    df_summary = pd.DataFrame(\n",
    "        {\n",
    "            \"lead_pip_variant\": [df.variant.iloc[max_pip_idx]],\n",
    "            \"n_total\": [len(df.index)],\n",
    "            \"n_r2\": [n_r2],\n",
    "            \"n_dentist_s_outlier\": [n_dentist_s_outlier],\n",
    "            \"fraction\": [n_dentist_s_outlier / n_r2 if n_r2 > 0 else 0],\n",
    "            \"max_pip\": [np.max(df.prob)]\n",
    "        }\n",
    "        )\n",
    "    return df, df_summary\n",
    "\n",
    "## Load LD\n",
    "if \"/mnt/vast/hpc/csg/molecular_phenotype_calling/LD/output/1300_hg38_EUR_LD_blocks_LD/ROSMAP_NIA_WGS.leftnorm.filtered.filtered.chr6_31604117_32714947.float16.npz\".endswith(\"npz\"):\n",
    "    LD,snp_id = load_npz_ld('/mnt/vast/hpc/csg/molecular_phenotype_calling/LD/output/1300_hg38_EUR_LD_blocks_LD/ROSMAP_NIA_WGS.leftnorm.filtered.filtered.chr6_31604117_32714947.float16.npz')\n",
    "if \"/mnt/vast/hpc/csg/molecular_phenotype_calling/LD/output/1300_hg38_EUR_LD_blocks_LD/ROSMAP_NIA_WGS.leftnorm.filtered.filtered.chr6_31604117_32714947.float16.npz\".endswith(\"bcor\"):\n",
    "    LD,snp_id = load_ld_bcor('/mnt/vast/hpc/csg/molecular_phenotype_calling/LD/output/1300_hg38_EUR_LD_blocks_LD/ROSMAP_NIA_WGS.leftnorm.filtered.filtered.chr6_31604117_32714947.float16')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "c968ce8c-d6d6-432e-8506-aff393690a40",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "sumstat = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/ADGWAS/ADGWAS_GRCh37_Bellenguez_2022.6/ADGWAS2022.GRCh37.chr6.sumstat.tsv', sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "5a186a57-04da-450d-bf75-0a22d7ffbeb3",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "## Remove nan in LD\n",
    "where = np.where(np.isnan(LD))\n",
    "na_snp = np.array(snp_id)[[x for x in where[0]] + [x for x in where[1]]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "8629079c-36a5-4d69-9c59-e74085dd98f0",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "### Remove NA from LD\n",
    "LD = LD[np.ix_(~np.in1d(snp_id, na_snp),~np.in1d(snp_id, na_snp))]\n",
    "### Remove NA from snp_list\n",
    "snp_id = np.array(snp_id)[~np.in1d(snp_id, na_snp)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "4d1e193e-a4cc-4557-b3d2-7272dd59facc",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['chr6:31604117:G:A', 'chr6:31604194:T:C', 'chr6:31604232:G:A', ...,\n",
       "       'chr6:32714813:G:A', 'chr6:32714887:C:T', 'chr6:32714947:C:T'],\n",
       "      dtype='<U567')"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "snp_id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "f8742fcd-864c-42a8-bc8c-f95149fba786",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "## Get only intersect snp\n",
    "intersct = np.intersect1d(sumstat.variant.to_numpy(),snp_id)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "be374851-1891-49ee-ada6-bbd9cdf68ed8",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "21"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(intersct)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "e0bd7bbc-6174-4310-8cc4-fbc4b3e33886",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_block(indexer, value, name)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    lead_pip_variant  n_total  n_r2  n_dentist_s_outlier  fraction   max_pip\n",
      "0  chr6:32342087:C:T       21     1                    0       0.0  0.999049\n",
      "0.9988184324537219 of snps in the LD panel does not have sumstat. Their sumstat are imputed\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/pandas/core/indexing.py:1732: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self._setitem_single_block(indexer, value, name)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "    lead_pip_variant  n_total  n_r2  n_dentist_s_outlier  fraction   max_pip\n",
      "0  chr6:31920285:G:A    17773     1                    0       0.0  0.014863\n"
     ]
    },
    {
     "ename": "NameError",
     "evalue": "name 'SignatureTranslatedAnonymousPackage' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m/tmp/5760179.1.csg.q/ipykernel_61024/2853763035.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     74\u001b[0m }\n\u001b[1;32m     75\u001b[0m \"\"\"\n\u001b[0;32m---> 76\u001b[0;31m \u001b[0msusie_analysis\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mSignatureTranslatedAnonymousPackage\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mstring\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"susie_analysis\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     77\u001b[0m \u001b[0manalysis_result\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msusie_analysis\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msusie_rss_analysis\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mss_df\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mss_qc\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mR\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mLD\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0moutput_path\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m'/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06_rosmap/ADGWAS2022.GRCh37.chr6.sumstat.chr6_31604117_32714947.unisusie_rss.fit.rds'\u001b[0m \u001b[0;34m,\u001b[0m \u001b[0mn\u001b[0m \u001b[0;34m=\u001b[0m\u001b[0;36m351430\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mNameError\u001b[0m: name 'SignatureTranslatedAnonymousPackage' is not defined"
     ]
    }
   ],
   "source": [
    "sumstat = sumstat.query(\"variant in @intersct\").reset_index()\n",
    "indice = np.where(np.in1d(snp_id, intersct))\n",
    "## slalom\n",
    "ss_qc,ss_qc_sum = slalom(sumstat,LD[np.ix_(indice[0].tolist(), indice[0].tolist())],0.4,4,0.6)\n",
    "print(ss_qc_sum)\n",
    "ss_qc.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06_rosmap/ADGWAS2022.GRCh37.chr6.sumstat.chr6_31604117_32714947.unisusie_rss.ss_qced.tsv',\"\\t\",index = 0 )\n",
    "## Filter out outlier\n",
    "### Get outliers\n",
    "outliers = ss_qc[ss_qc.outliers].variant\n",
    "### Remove outliers from LD\n",
    "LD = LD[np.ix_(~np.in1d(snp_id, outliers),~np.in1d(snp_id, outliers))]\n",
    "### Remove outliers from snp_list\n",
    "snp_id = np.array(snp_id)[~np.in1d(snp_id, outliers)]\n",
    "ss_qc = ss_qc.assign(z = ss_qc.beta/ss_qc.se)\n",
    "ss_qc = ss_qc[~ss_qc.outliers]\n",
    "\n",
    "if True:\n",
    "    cur_LD = LD\n",
    "    cur_miss = ~np.in1d(snp_id, ss_qc.variant)\n",
    "    print(f'{np.mean(cur_miss)} of snps in the LD panel does not have sumstat. Their sumstat are imputed')\n",
    "    ## Sumstat imputation based on fusion code\n",
    "    cur_wgt = cur_LD[np.ix_(cur_miss, np.logical_not(cur_miss))].dot(np.linalg.inv(cur_LD[np.logical_not(cur_miss), np.logical_not(cur_miss)] + 0.1 * np.eye(np.sum(np.logical_not(cur_miss)))))\n",
    "    cur_impz  = cur_wgt @ ss_qc.z\n",
    "    cur_r2pred = np.diag(cur_wgt.dot(cur_LD[np.ix_(np.logical_not(cur_miss),np.logical_not(cur_miss))]).dot(cur_wgt.T))\n",
    "    imputed_z = cur_impz/np.sqrt(cur_r2pred)\n",
    "    working_ss = pd.concat([ss_qc[[\"variant\",\"z\"]], pd.DataFrame({\"z\" : imputed_z, \"variant\": np.array(snp_id)[cur_miss]})]).set_index(\"variant\").reindex(snp_id).reset_index()\n",
    "    import scipy.stats as st\n",
    "    working_ss[\"beta\"] = working_ss.z\n",
    "    working_ss[\"se\"] = 1\n",
    "    working_ss[\"pvalue\"] = (1 - st.norm.cdf(working_ss.z))*2\n",
    "\n",
    "\n",
    "    working_ss_qc, working_ss_qc_sum = slalom(working_ss,cur_LD,0.4,4,0.6)\n",
    "    print(working_ss_qc_sum)\n",
    "    working_ss_qc.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06_rosmap/ADGWAS2022.GRCh37.chr6.sumstat.chr6_31604117_32714947.unisusie_rss.imputed_ss_qced.tsv',\"\\t\",index = 0 )\n",
    "    imputed_outliers = working_ss_qc[working_ss_qc.outliers].variant\n",
    "    ### Remove outliers from LD\n",
    "    cur_LD = LD[np.ix_(~np.in1d(snp_id, imputed_outliers),~np.in1d(snp_id, imputed_outliers))]\n",
    "    ### Remove outliers from snp_list\n",
    "    snp_id = np.array(snp_id)[~np.in1d(snp_id, imputed_outliers)]\n",
    "    working_ss_qc = working_ss_qc[~working_ss_qc.outliers]\n",
    "    ss_qc = working_ss_qc\n",
    "    LD = cur_LD\n",
    "### For non-imputed LD, retained only the overlap snps. For imputed LD, this should do nothing as snp_id, ss_qc.variant .\n",
    "indice = np.where(np.in1d(snp_id, ss_qc.variant))\n",
    "LD = LD[np.ix_(indice[0].tolist(), indice[0].tolist())]\n",
    "## SuSiERSS\n",
    "string=\"\"\"\n",
    "susie_rss_analysis=function(ss_df, R, n, var_y, z_ld_weight = 0, estimate_residual_variance = FALSE,\n",
    "prior_variance = 50, check_prior = TRUE, output_path,L = 10 , max_iter  = 1000 ){\n",
    "condz = susieR::kriging_rss(as.matrix(ss_df$z),R ,n = 351430)\n",
    "ggplot2::ggsave(plot = condz$plot,filename = \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06_rosmap/ADGWAS2022.GRCh37.chr6.sumstat.chr6_31604117_32714947.unisusie_rss.fit.rds.kriging_rss.pdf\", device = \"pdf\")\n",
    "res = susieR:::susie_rss(as.matrix(ss_df$z),as.matrix(R), n = 351430,var_y, z_ld_weight = 0, estimate_residual_variance = FALSE,\n",
    "prior_variance = 50, check_prior = TRUE, max_iter = 1000,verbose = TRUE,L = 10   )\n",
    "res$variants = ss_df$variant\n",
    "res$z = ss_df$z\n",
    "res$corr = susieR:::get_cs_correlation(res, X = NULL, Xcorr = R, max = FALSE)\n",
    "res$conditional_dist = condz$conditional_dist\n",
    "rownames(res$corr) <- names(res$cs)\n",
    "colnames(res$corr) <- names(res$cs)\n",
    "if (length(res$sets$cs) > 1) {\n",
    "    index_combos = expand.grid(1:length(res$sets$cs),1:length(res$sets$cs))\n",
    "    in_common = apply(index_combos, 1, function(x) intersect(res$sets$cs[[x[1]]], res$sets$cs[[x[2]]]))\n",
    "    counts = unlist(lapply(in_common, length))\n",
    "    ovlp_mat = matrix(counts, ncol = length(res$sets$cs), byrow = T)\n",
    "    ovlp_mat[lower.tri(ovlp_mat)] = NA\n",
    "    rownames(ovlp_mat) = names(res$sets$cs)\n",
    "    colnames(ovlp_mat) = names(res$sets$cs)\n",
    "    print(ovlp_mat)\n",
    "    res$sets[[\"ovlp_mat\"]] = ovlp_mat\n",
    "}\n",
    "saveRDS(res,output_path)\n",
    "return(res)\n",
    "}\n",
    "\"\"\"\n",
    "susie_analysis = SignatureTranslatedAnonymousPackage(string, \"susie_analysis\")\n",
    "analysis_result = susie_analysis.susie_rss_analysis(ss_df = ss_qc,R = LD,output_path='/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/ADx/output_chr06_rosmap/ADGWAS2022.GRCh37.chr6.sumstat.chr6_31604117_32714947.unisusie_rss.fit.rds' , n =351430)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "7c19e55d-bfb7-4215-beae-1709527920b2",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "\n",
    "def load_npz_ld(path):\n",
    "    np_ld_loaded = np.load(path,allow_pickle=True)\n",
    "    # sort by start position\n",
    "    snp_id = [x.replace(\":\",\":\") for x in np_ld_loaded.get(\"arr_1\")]\n",
    "    np_ld_loaded = np_ld_loaded.get(\"arr_0\")\n",
    "    new = np_ld_loaded + np_ld_loaded.T\n",
    "    np.fill_diagonal(new, np.diag(new)/2)\n",
    "    return new,snp_id\n",
    "\n",
    "def get_bcor_meta(bcor_obj):\n",
    "    df_ld_snps = bcor_obj.getMeta()\n",
    "    df_ld_snps.rename(columns={'rsid':'SNP', 'position':'BP', 'chromosome':'CHR', 'allele1':'A1', 'allele2':'A2'}, inplace=True, errors='raise')\n",
    "    ###df_ld_snps['CHR'] = df_ld_snps['CHR'].astype(np.int64)\n",
    "    df_ld_snps['BP'] = df_ld_snps['BP'].astype(np.int64)\n",
    "    return df_ld_snps\n",
    "\n",
    "def load_ld_bcor(ld_prefix):\n",
    "    bcor_file = ld_prefix+'.bcor'\n",
    "    import os\n",
    "    import time\n",
    "    from ldstore.bcor import bcor\n",
    "    if not os.path.exists(bcor_file):\n",
    "        raise IOError('%s not found'%(bcor_file))\n",
    "    t0 = time.time()\n",
    "    bcor_obj = bcor(bcor_file)\n",
    "    df_ld_snps = get_bcor_meta(bcor_obj)\n",
    "    ld_arr = bcor_obj.readCorr([])\n",
    "    assert np.all(~np.isnan(ld_arr))\n",
    "    return ld_arr, df_ld_snps\n",
    "\n",
    "def abf(beta, se, W=0.04):\n",
    "    from scipy import special\n",
    "    z = beta / se\n",
    "    V = se ** 2\n",
    "    r = W / (W + V)\n",
    "    lbf = 0.5 * (np.log(1 - r) + (r * z ** 2))\n",
    "    denom = special.logsumexp(lbf)\n",
    "    prob = np.exp(lbf - denom)\n",
    "    return lbf, prob\n",
    "\n",
    "def get_cs(variant, prob, coverage=0.95):\n",
    "    ordering = np.argsort(prob)[::-1]\n",
    "    idx = np.where(np.cumsum(prob[ordering]) > coverage)[0][0]\n",
    "    cs = variant[ordering][: (idx + 1)]\n",
    "    return cs\n",
    "def slalom(df,LD,abf_prior_variance = 0.4 ,nlog10p_dentist_s_threshold = 4, r2_threshold = 0.6  ):\n",
    "    from scipy import stats\n",
    "    lbf, prob = abf(df.beta, df.se, W=abf_prior_variance)\n",
    "    cs = get_cs(df.variant, prob, coverage=0.95)\n",
    "    cs_99 = get_cs(df.variant, prob, coverage=0.99)\n",
    "    df[\"lbf\"] = lbf\n",
    "    df[\"prob\"] = prob\n",
    "    df[\"cs\"] = df.variant.isin(cs)\n",
    "    df[\"cs_99\"] = df.variant.isin(cs_99)\n",
    "\n",
    "    if \"pvalue\" == \"pvalue\":\n",
    "        lead_idx_snp = df.pvalue.idxmin()\n",
    "    else:\n",
    "        lead_idx_snp = df.prob.idxmax()\n",
    "\n",
    "    lead_variant = df.variant[lead_idx_snp]\n",
    "    df[\"lead_variant\"] = False\n",
    "    df[\"lead_variant\"].iloc[lead_idx_snp] = True\n",
    "    # annotate LD\n",
    "    ## This is to identify the R for each snp vs the lead snp\n",
    "    df[\"r\"] = [LD[np.where(np.in1d(df.variant,lead_variant))][:,np.where(np.in1d(df.variant,x))][0][0][0] for x in df.variant]\n",
    "    lead_z = (df.beta / df.se).iloc[lead_idx_snp]\n",
    "    df[\"t_dentist_s\"] = ((df.beta / df.se) - df.r * lead_z) ** 2 / (1 - df.r ** 2)\n",
    "    df[\"t_dentist_s\"] = np.where(df[\"t_dentist_s\"] < 0, np.inf, df[\"t_dentist_s\"])\n",
    "    df[\"t_dentist_s\"].iloc[lead_idx_snp] = np.nan\n",
    "    df[\"nlog10p_dentist_s\"] = stats.chi2.logsf(df[\"t_dentist_s\"], df=1) / -np.log(10)\n",
    "    df[\"r2\"] = df.r ** 2\n",
    "    df[\"outliers\"] = (df.r2 > r2_threshold) & (df.nlog10p_dentist_s > nlog10p_dentist_s_threshold)\n",
    "    df_output = df\n",
    "    n_r2 = np.sum(df.r2 > r2_threshold)\n",
    "    n_dentist_s_outlier = np.sum(\n",
    "        (df.r2 > r2_threshold) & (df.nlog10p_dentist_s > nlog10p_dentist_s_threshold)\n",
    "    )\n",
    "    max_pip_idx = df.prob.idxmax()\n",
    "    df_summary = pd.DataFrame(\n",
    "        {\n",
    "            \"lead_pip_variant\": [df.variant.iloc[max_pip_idx]],\n",
    "            \"n_total\": [len(df.index)],\n",
    "            \"n_r2\": [n_r2],\n",
    "            \"n_dentist_s_outlier\": [n_dentist_s_outlier],\n",
    "            \"fraction\": [n_dentist_s_outlier / n_r2 if n_r2 > 0 else 0],\n",
    "            \"max_pip\": [np.max(df.prob)]\n",
    "        }\n",
    "        )\n",
    "    return df, df_summary\n",
    "\n",
    "## Load LD\n",
    "if \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.ld.float16.npz\".endswith(\"npz\"):\n",
    "    LD,snp_id = load_npz_ld('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.ld.float16.npz')\n",
    "if \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.ld.float16.npz\".endswith(\"bcor\"):\n",
    "    LD,snp_id = load_ld_bcor('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.mac5.ld.float16')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "fa116511-4d6b-40c5-8c11-2ef11758510a",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "sumstat = pd.read_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/All_Hdiff/UKBB_Hdiff_F2247_4PCs_All_f2247.chr6.regenie.snp_stats.tsv', sep = \"\\t\")\n",
    "## Remove nan in LD\n",
    "where = np.where(np.isnan(LD))\n",
    "na_snp = np.array(snp_id)[[x for x in where[0]] + [x for x in where[1]]]\n",
    "### Remove NA from LD\n",
    "LD = LD[np.ix_(~np.in1d(snp_id, na_snp),~np.in1d(snp_id, na_snp))]\n",
    "### Remove NA from snp_list\n",
    "snp_id = np.array(snp_id)[~np.in1d(snp_id, na_snp)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3b4afb06-16bb-498b-9ca6-6b4cd78a8ea1",
   "metadata": {
    "kernel": "Python3"
   },
   "outputs": [],
   "source": [
    "## Get only intersect snp\n",
    "intersct = np.intersect1d(sumstat.variant.to_numpy(),snp_id)\n",
    "sumstat = sumstat.query(\"variant in @intersct\").reset_index()\n",
    "indice = np.where(np.in1d(snp_id, intersct))\n",
    "## slalom\n",
    "ss_qc,ss_qc_sum = slalom(sumstat,LD[np.ix_(indice[0].tolist(), indice[0].tolist())],0.4,4,0.6)\n",
    "print(ss_qc_sum)\n",
    "ss_qc.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/All_Hdiff/output_finemap/UKBB_Hdiff_F2247_4PCs_All_f2247.chr6.regenie.snp_stats.ld.unisusie_rss.ss_qced.tsv',\"\\t\",index = 0 )\n",
    "## Filter out outlier\n",
    "### Get outliers\n",
    "outliers = ss_qc[ss_qc.outliers].variant\n",
    "### Remove outliers from LD\n",
    "LD = LD[np.ix_(~np.in1d(snp_id, outliers),~np.in1d(snp_id, outliers))]\n",
    "### Remove outliers from snp_list\n",
    "snp_id = np.array(snp_id)[~np.in1d(snp_id, outliers)]\n",
    "ss_qc = ss_qc.assign(z = ss_qc.beta/ss_qc.se)\n",
    "ss_qc = ss_qc[~ss_qc.outliers]\n",
    "\n",
    "if True:\n",
    "    cur_LD = LD\n",
    "    cur_miss = ~np.in1d(snp_id, ss_qc.variant)\n",
    "    print(f'{np.mean(cur_miss)} of snps in the LD panel does not have sumstat. Their sumstat are imputed')\n",
    "    ## Sumstat imputation based on fusion code\n",
    "    cur_wgt = cur_LD[np.ix_(cur_miss, np.logical_not(cur_miss))].dot(np.linalg.inv(cur_LD[np.logical_not(cur_miss), np.logical_not(cur_miss)] + 0.1 * np.eye(np.sum(np.logical_not(cur_miss)))))\n",
    "    cur_impz  = cur_wgt @ ss_qc.z\n",
    "    cur_r2pred = np.diag(cur_wgt.dot(cur_LD[np.ix_(np.logical_not(cur_miss),np.logical_not(cur_miss))]).dot(cur_wgt.T))\n",
    "    imputed_z = cur_impz/np.sqrt(cur_r2pred)\n",
    "    working_ss = pd.concat([ss_qc[[\"variant\",\"z\"]], pd.DataFrame({\"z\" : imputed_z, \"variant\": np.array(snp_id)[cur_miss]})]).set_index(\"variant\").reindex(snp_id).reset_index()\n",
    "    import scipy.stats as st\n",
    "    working_ss[\"beta\"] = working_ss.z\n",
    "    working_ss[\"se\"] = 1\n",
    "    working_ss[\"pvalue\"] = (1 - st.norm.cdf(working_ss.z))*2\n",
    "\n",
    "\n",
    "    working_ss_qc, working_ss_qc_sum = slalom(working_ss,cur_LD,0.4,4,0.6)\n",
    "    print(working_ss_qc_sum)\n",
    "    working_ss_qc.to_csv('/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/All_Hdiff/output_finemap/UKBB_Hdiff_F2247_4PCs_All_f2247.chr6.regenie.snp_stats.ld.unisusie_rss.imputed_ss_qced.tsv',\"\\t\",index = 0 )\n",
    "    imputed_outliers = working_ss_qc[working_ss_qc.outliers].variant\n",
    "    ### Remove outliers from LD\n",
    "    cur_LD = LD[np.ix_(~np.in1d(snp_id, imputed_outliers),~np.in1d(snp_id, imputed_outliers))]\n",
    "    ### Remove outliers from snp_list\n",
    "    snp_id = np.array(snp_id)[~np.in1d(snp_id, imputed_outliers)]\n",
    "    working_ss_qc = working_ss_qc[~working_ss_qc.outliers]\n",
    "    ss_qc = working_ss_qc\n",
    "    LD = cur_LD\n",
    "### For non-imputed LD, retained only the overlap snps. For imputed LD, this should do nothing as snp_id, ss_qc.variant .\n",
    "indice = np.where(np.in1d(snp_id, ss_qc.variant))\n",
    "LD = LD[np.ix_(indice[0].tolist(), indice[0].tolist())]\n",
    "## SuSiERSS\n",
    "string=\"\"\"\n",
    "susie_rss_analysis=function(ss_df, R, n, var_y, z_ld_weight = 0, estimate_residual_variance = FALSE,\n",
    "prior_variance = 50, check_prior = TRUE, output_path,L = 10 , max_iter  = 1000 ){\n",
    "condz = susieR::kriging_rss(as.matrix(ss_df$z),R ,n = 0)\n",
    "ggplot2::ggsave(plot = condz$plot,filename = \"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/All_Hdiff/output_finemap/UKBB_Hdiff_F2247_4PCs_All_f2247.chr6.regenie.snp_stats.ld.unisusie_rss.fit.rds.kriging_rss.pdf\", device = \"pdf\")\n",
    "res = susieR:::susie_rss(as.matrix(ss_df$z),as.matrix(R), n = 0,var_y, z_ld_weight = 0, estimate_residual_variance = FALSE,\n",
    "prior_variance = 50, check_prior = TRUE, max_iter = 1000,verbose = TRUE,L = 10   )\n",
    "res$variants = ss_df$variant\n",
    "res$z = ss_df$z\n",
    "res$corr = susieR:::get_cs_correlation(res, X = NULL, Xcorr = R, max = FALSE)\n",
    "res$conditional_dist = condz$conditional_dist\n",
    "rownames(res$corr) <- names(res$cs)\n",
    "colnames(res$corr) <- names(res$cs)\n",
    "if (length(res$sets$cs) > 1) {\n",
    "    index_combos = expand.grid(1:length(res$sets$cs),1:length(res$sets$cs))\n",
    "    in_common = apply(index_combos, 1, function(x) intersect(res$sets$cs[[x[1]]], res$sets$cs[[x[2]]]))\n",
    "    counts = unlist(lapply(in_common, length))\n",
    "    ovlp_mat = matrix(counts, ncol = length(res$sets$cs), byrow = T)\n",
    "    ovlp_mat[lower.tri(ovlp_mat)] = NA\n",
    "    rownames(ovlp_mat) = names(res$sets$cs)\n",
    "    colnames(ovlp_mat) = names(res$sets$cs)\n",
    "    print(ovlp_mat)\n",
    "    res$sets[[\"ovlp_mat\"]] = ovlp_mat\n",
    "}\n",
    "saveRDS(res,output_path)\n",
    "return(res)\n",
    "}\n",
    "\"\"\"\n",
    "susie_analysis = SignatureTranslatedAnonymousPackage(string, \"susie_analysis\")\n",
    "analysis_result = susie_analysis.susie_rss_analysis(ss_df = ss_qc,R = LD,output_path='/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/All_Hdiff/output_finemap/UKBB_Hdiff_F2247_4PCs_All_f2247.chr6.regenie.snp_stats.ld.unisusie_rss.fit.rds' )"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "c17e8c55-dbb8-4598-bc58-7e376bb2a76e",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "library(\"dplyr\")\n",
    "library(\"tibble\")\n",
    "library(\"purrr\")\n",
    "library(\"tidyr\")\n",
    "library(\"readr\")\n",
    "library(\"stringr\")\n",
    "library(\"susieR\")\n",
    "extract_lbf = function(susie_obj){\n",
    "\n",
    "if(\"variants\" %in% names(susie_obj) ){\n",
    "ss_bf = tibble(snps = susie_obj$variants, cs_index = ifelse(is.null(susie_obj$sets$cs_index), 0, paste0(susie_obj$sets$cs_index,collapse =\",\")),names = \"ld\")\n",
    "}\n",
    "  else\n",
    "  {\n",
    "ss_bf = tibble(snps = susie_obj$variable_name, cs_index = ifelse(is.null(susie_obj$sets$cs_index), 0, paste0(susie_obj$sets$cs_index,collapse =\",\")),names = \"UKBB_Hdiff_F2247_4PCs_All_f2247.chr6.regenie.snp_stats.ld\")\n",
    " }\n",
    "\n",
    "ss_bf = ss_bf%>%cbind(susie_obj$lbf_variable%>%t)%>%as_tibble()\n",
    "\n",
    "return(ss_bf)\n",
    "}\n",
    "\n",
    "extract_variants_pip = function(susie_obj,region_list){\n",
    "susie_tb = tibble( variants =  names(susie_obj$pip)[which( susie_obj$pip >= 0)],\n",
    "                       snps_index = which(( susie_obj$pip >= 0))) %>%\n",
    "    mutate(chromosome = map_chr(variants, ~read.table(text = .x, sep = \":\")$V1%>%str_replace(\"chr\",\"\") ),\n",
    "            position  = map_chr(variants, ~read.table(text = .x, sep = \":\")$V2  ),\n",
    "            ref = map_chr(position , ~read.table(text = .x, sep = \"_\",colClasses = \"character\")$V2  ),\n",
    "            alt = map_chr(position , ~read.table(text = .x, sep = \"_\",colClasses = \"character\")$V3  ),\n",
    "            position  = map_dbl(position , ~read.table(text = .x, sep = \"_\",as.is = T)$V1  )\n",
    "                         )\n",
    "  susie_tb = susie_tb%>%mutate(cs_order =(map(susie_tb$snps_index , ~tryCatch(which(pmap(list( a= susie_obj$sets$cs) , function(a) .x %in% a )%>%unlist()), error = function(e) return(0) )  ))%>%as.character%>%str_replace(\"integer\\\\(0\\\\)\",\"0\"),\n",
    "                     cs_id = map_chr(cs_order,~ifelse(.x ==\"0\", \"None\" ,names(susie_obj$sets$cs)[.x%>%str_split(\":\")%>%unlist%>%as.numeric] ) ),\n",
    "                     log10_base_factor = map_chr(snps_index,~paste0( susie_obj$lbf_variable[,.x],  collapse = \";\")),\n",
    "                     pip = susie_obj$pip,\n",
    "                     posterior_mean = coef.susie(susie_obj)[-1],\n",
    "                     posterior_sd = susie_get_posterior_sd(susie_obj),\n",
    "                     z = posterior_mean/posterior_sd)\n",
    "\n",
    "      susie_tb =  susie_tb%>%mutate(  molecular_trait_id = region_list$molecular_trait_id,\n",
    "                         finemapped_region_start = region_list$finemapped_region_start,\n",
    "                         finemapped_region_end = region_list$finemapped_region_end)\n",
    "      return(susie_tb)    }\n",
    "\n",
    "\n",
    "\n",
    " extract_effect_pip = function(susie_obj,region_list,susie_tb){\n",
    "  result_tb =  tibble(phenotype = susie_obj$name,\n",
    "    V = susie_obj$V,effect_id = paste0(\"L\",1:length(V) ) ,\n",
    "    cs_log10bf = susie_obj$lbf)\n",
    "    if(is.null(susie_obj$sets$cs)){\n",
    "        cs_min_r2 = cs_avg_r2 =  coverage =  0\n",
    "        cs = \"None\"} else {         cs = map_chr(susie_obj$sets$cs[result_tb$effect_id],~susie_tb$variants[.x]%>%paste0(collapse = \";\"))\n",
    "    coverage = map(result_tb$effect_id, ~susie_obj$sets$coverage[which(names(susie_obj$sets$cs) == .x )])%>%as.numeric%>%replace_na(0)\n",
    "    cs_min_r2  = (susie_obj$sets$purity[result_tb$effect_id,1])%>%as.numeric%>%replace_na(0)\n",
    "    cs_avg_r2  = (susie_obj$sets$purity[result_tb$effect_id,2])%>%as.numeric%>%replace_na(0) }\n",
    "    result_tb = result_tb%>%mutate(cs_min_r2 = cs_min_r2,cs_avg_r2 = cs_avg_r2 ,coverage = coverage%>%unlist,cs = cs )\n",
    "  return(result_tb)\n",
    "  }\n",
    "\n",
    "\n",
    "susie_obj = readRDS(\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/All_Hdiff/output_finemap/UKBB_Hdiff_F2247_4PCs_All_f2247.chr6.regenie.snp_stats.ld.unisusie_rss.fit.rds\")\n",
    "if(\"variants\" %in% names(susie_obj) ){susie_obj$variants = susie_obj$variants%>%str_replace(\"_\",\":\")}\n",
    "if(is.null(names(susie_obj$pip ))){names(susie_obj$pip) = susie_obj$variants}\n",
    "lbf = extract_lbf(susie_obj)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1fc041bb-718b-45b5-8fa7-796c5ad76259",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "\n",
      "\u001b[36m\u001b[39m \u001b[1mColumn specification\u001b[22m \u001b[36m\u001b[39m\n",
      "cols(\n",
      "  `#chr` = \u001b[31mcol_character()\u001b[39m,\n",
      "  start = \u001b[32mcol_double()\u001b[39m,\n",
      "  end = \u001b[32mcol_double()\u001b[39m\n",
      ")\n",
      "\n"
     ]
    }
   ],
   "source": [
    "region_list = read_delim(\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/All_Hdiff/region_list\",\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "8b91ce25-6549-4de2-99fb-6f4da9034ee9",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "if(ncol(region_list) == 3 ){   region_list =  region_list%>%mutate(`#chr` = `#chr`%>%str_remove_all(\" \") , ID = paste0(`#chr`,\"_\",start,\"_\",end) ) } # LD_list\n",
    "if(region_list$start[1] - region_list$end[1]  == -1 ){\n",
    "    region_list = region_list%>%mutate( start = start - 1000000 ,end = start +1000000) # region_list for fix cis windows\n",
    "      }\n",
    "  if(\"gene_id\" %in% colnames(region_list)){region_list = region_list%>%mutate(ID = gene_id)  } # region_list for gene\n",
    "                                                                   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "0df4e796-4082-4a45-8bca-2dd19f01e80c",
   "metadata": {
    "kernel": "R"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A spec_tbl_df: 1 x 4</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>#chr</th><th scope=col>start</th><th scope=col>end</th><th scope=col>ID</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;chr&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>chr6</td><td>31571218</td><td>32682664</td><td>chr6_31571218_32682664</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A spec\\_tbl\\_df: 1 x 4\n",
       "\\begin{tabular}{llll}\n",
       " \\#chr & start & end & ID\\\\\n",
       " <chr> & <dbl> & <dbl> & <chr>\\\\\n",
       "\\hline\n",
       "\t chr6 & 31571218 & 32682664 & chr6\\_31571218\\_32682664\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A spec_tbl_df: 1 x 4\n",
       "\n",
       "| #chr &lt;chr&gt; | start &lt;dbl&gt; | end &lt;dbl&gt; | ID &lt;chr&gt; |\n",
       "|---|---|---|---|\n",
       "| chr6 | 31571218 | 32682664 | chr6_31571218_32682664 |\n",
       "\n"
      ],
      "text/plain": [
       "  #chr start    end      ID                    \n",
       "1 chr6 31571218 32682664 chr6_31571218_32682664"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "region_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "801dfc0b-c3cc-4fe1-af88-70bd5531f7bd",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [
    {
     "ename": "ERROR",
     "evalue": "\u001b[1m\u001b[33mError\u001b[39m in \u001b[1m\u001b[1m`chr_as_locations()`:\u001b[22m\n\u001b[33m!\u001b[39m Can't subset columns that don't exist.\n\u001b[31m\u001b[39m Column `ID` doesn't exist.\n",
     "output_type": "error",
     "traceback": [
      "\u001b[1m\u001b[33mError\u001b[39m in \u001b[1m\u001b[1m`chr_as_locations()`:\u001b[22m\n\u001b[33m!\u001b[39m Can't subset columns that don't exist.\n\u001b[31m\u001b[39m Column `ID` doesn't exist.\nTraceback:\n",
      "1. region_list %>% select(molecular_trait_id = ID, chromosome = `#chr`, \n .     finemapped_region_start = start, finemapped_region_end = end)",
      "2. select(., molecular_trait_id = ID, chromosome = `#chr`, finemapped_region_start = start, \n .     finemapped_region_end = end)",
      "3. select.data.frame(., molecular_trait_id = ID, chromosome = `#chr`, \n .     finemapped_region_start = start, finemapped_region_end = end)",
      "4. tidyselect::eval_select(expr(c(...)), .data)",
      "5. eval_select_impl(data, names(data), as_quosure(expr, env), include = include, \n .     exclude = exclude, strict = strict, name_spec = name_spec, \n .     allow_rename = allow_rename)",
      "6. with_subscript_errors(vars_select_eval(vars, expr, strict, data = x, \n .     name_spec = name_spec, uniquely_named = uniquely_named, allow_rename = allow_rename, \n .     type = type), type = type)",
      "7. tryCatch(instrument_base_errors(expr), vctrs_error_subscript = function(cnd) {\n .     cnd$subscript_action <- subscript_action(type)\n .     cnd$subscript_elt <- \"column\"\n .     cnd_signal(cnd)\n . })",
      "8. tryCatchList(expr, classes, parentenv, handlers)",
      "9. tryCatchOne(expr, names, parentenv, handlers[[1L]])",
      "10. value[[3L]](cond)",
      "11. cnd_signal(cnd)",
      "12. signal_abort(cnd)"
     ]
    }
   ],
   "source": [
    "region_list = region_list%>%select(molecular_trait_id = ID, chromosome  = `#chr`,finemapped_region_start = start ,finemapped_region_end = end)  # Formatting\n",
    "                                   "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "369d53b2-097c-4a6e-88b2-dbb66365066f",
   "metadata": {
    "kernel": "R",
    "tags": []
   },
   "outputs": [],
   "source": [
    "region_list = region_list%>%filter(molecular_trait_id == \"ld\")\n",
    "variants_pip = extract_variants_pip( susie_obj , region_list)\n",
    "effect_pip = extract_effect_pip( susie_obj , region_list,variants_pip)\n",
    "lbf%>%write_delim(\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/All_Hdiff/output_finemap/UKBB_Hdiff_F2247_4PCs_All_f2247.chr6.regenie.snp_stats.ld.unisusie_rss.fit.lbf.tsv\",\"\\t\")\n",
    "variants_pip%>%write_delim(\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/All_Hdiff/output_finemap/UKBB_Hdiff_F2247_4PCs_All_f2247.chr6.regenie.snp_stats.ld.unisusie_rss.fit.variant.tsv\",\"\\t\")\n",
    "effect_pip%>%write_delim(\"/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/All_Hdiff/output_finemap/UKBB_Hdiff_F2247_4PCs_All_f2247.chr6.regenie.snp_stats.ld.unisusie_rss.fit.effect.tsv\",\"\\t\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "32c67d0c-a88b-4371-a4a7-acbae97e548b",
   "metadata": {
    "kernel": "R"
   },
   "source": [
    "## For Rui"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "bc59533c-b998-4e28-a93c-ee9ef24f257c",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.log.\n",
      "Options in effect:\n",
      "  --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_31571218_32682664.bgen ref-first\n",
      "  --keep /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/unrelated.txt\n",
      "  --maf 0.01\n",
      "  --make-pgen\n",
      "  --new-id-max-allele-len 100\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01\n",
      "  --sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample\n",
      "  --set-all-var-ids chr@:#:$r:$a\n",
      "\n",
      "Start time: Mon Jul 24 11:23:14 2023\n",
      "257482 MiB RAM detected, ~246185 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "--bgen: 43053 variants detected, format v1.2.\n",
      "487409 samples imported from .sample file to\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01-temporary.psam\n",
      ".\n",
      "--bgen: 21k variants converted.  \n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01-temporary.pgen\n",
      "+\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01-temporary.pvar\n",
      "written.\n",
      "487409 samples (264303 females, 222994 males, 112 ambiguous; 487409 founders)\n",
      "loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01-temporary.psam.\n",
      "43053 variants loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01-temporary.pvar.\n",
      "Note: No phenotype data present.\n",
      "--keep: 351430 samples remaining.\n",
      "351430 samples (188495 females, 162935 males; 351430 founders) remaining after\n",
      "main filters.\n",
      "Calculating allele frequencies... done.\n",
      "23535 variants removed due to allele frequency threshold(s)\n",
      "(--maf/--max-maf/--mac/--max-mac).\n",
      "19518 variants remaining after main filters.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.psam\n",
      "... done.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.pvar\n",
      "... 10101111121213131414151516161717181819192020212122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899done.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.pgen\n",
      "... done.\n",
      "End time: Mon Jul 24 11:24:16 2023\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --bgen /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/06_31571218_32682664.bgen ref-first \\\n",
    "--sample /mnt/vast/hpc/csg/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample \\\n",
    "--set-all-var-ids 'chr@:#:$r:$a' \\\n",
    "--new-id-max-allele-len 100 \\\n",
    "--keep /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/unrelated.txt \\\n",
    "--maf 0.01 \\\n",
    "--make-pgen \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "bf4f0a85-2000-4a25-b4c1-b154ca26b22f",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.info0.9.log.\n",
      "Options in effect:\n",
      "  --extract-if-info R2<0.9\n",
      "  --make-pgen\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.info0.9\n",
      "  --pfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01\n",
      "\n",
      "Start time: Mon Jul 24 12:17:17 2023\n",
      "257482 MiB RAM detected, ~246186 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "351430 samples (188495 females, 162935 males; 351430 founders) loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.psam.\n",
      "Error: --extract-if-info used on a variant file with no INFO column.\n",
      "End time: Mon Jul 24 12:17:17 2023\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --pfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01 \\\n",
    "--extract-if-info \"R2<0.9\" \\\n",
    "--make-pgen --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.info0.9"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "30019fe5-a2ed-47e2-bdc3-e767a06f59a8",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to plink2.log.\n",
      "Options in effect:\n",
      "  --pfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01\n",
      "  --pgen-info\n",
      "\n",
      "Start time: Mon Jul 24 12:29:59 2023\n",
      "257482 MiB RAM detected, ~246185 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "351430 samples (188495 females, 162935 males; 351430 founders) loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.psam.\n",
      "19518 variants loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.pvar.\n",
      "--pgen-info on\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.pgen:\n",
      "  Variants: 19518\n",
      "  Samples: 351430\n",
      "  REF alleles are all known\n",
      "  Maximum allele count for a single variant: 2\n",
      "  No hardcalls are explicitly phased\n",
      "  Dosage present, none explicitly phased\n",
      "End time: Mon Jul 24 12:30:00 2023\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --pfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01 \\\n",
    "--pgen-info"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "38c87e27-af42-4a20-87cb-5203980b7665",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.R.log.\n",
      "Options in effect:\n",
      "  --export A\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.R\n",
      "  --pfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01\n",
      "\n",
      "Start time: Mon Jul 24 12:39:26 2023\n",
      "257482 MiB RAM detected, ~246185 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "351430 samples (188495 females, 162935 males; 351430 founders) loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.psam.\n",
      "19518 variants loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.pvar.\n",
      "Note: No phenotype data present.\n",
      "--export A pass 1/1: loading... 0writing... 101011111212131314141515161617171818192020212122222323242425252626272728282930303131323233333434353536363737383839404041414242434344444545464647474848495050515152525353545455555656575758585960606161626263636464656566666767686869707071717272737374747575767677777878798080818182828383848485858686878788888990909191929293939494959596969797989899done.\n",
      "--export A:\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.R.raw\n",
      "written.\n",
      "End time: Mon Jul 24 12:40:49 2023\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --pfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01 \\\n",
    "--export A \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "2eb43b2b-f8a7-4523-88ce-4c2420108302",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.AD.R.log.\n",
      "Options in effect:\n",
      "  --export AD\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.AD.R\n",
      "  --pfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01\n",
      "\n",
      "Start time: Mon Jul 24 13:08:15 2023\n",
      "257482 MiB RAM detected, ~246147 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "351430 samples (188495 females, 162935 males; 351430 founders) loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.psam.\n",
      "19518 variants loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.pvar.\n",
      "Note: No phenotype data present.\n",
      "--export AD pass 1/1: loading... 0writing... 101011111212131314141515161617171818192020212122222323242425252626272728282930303131323233333434353536363737383839404041414242434344444545464647474848495050515152525353545455555656575758585960606161626263636464656566666767686869707071717272737374747575767677777878798080818182828383848485858686878788888990909191929293939494959596969797989899done.\n",
      "--export AD:\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.AD.R.raw\n",
      "written.\n",
      "End time: Mon Jul 24 13:09:59 2023\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --pfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01 \\\n",
    "--export \"AD\" \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.AD.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "bb37322f-7798-4342-844a-6175b73cdb8a",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.log.\n",
      "Options in effect:\n",
      "  --make-bed\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01\n",
      "  --pfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01\n",
      "\n",
      "Start time: Mon Jul 24 13:10:20 2023\n",
      "257482 MiB RAM detected, ~246117 available; reserving 128741 MiB for main\n",
      "workspace.\n",
      "Using up to 64 threads (change this with --threads).\n",
      "351430 samples (188495 females, 162935 males; 351430 founders) loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.psam.\n",
      "19518 variants loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.pvar.\n",
      "Note: No phenotype data present.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.fam\n",
      "... done.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.bim\n",
      "... done.\n",
      "Writing\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.bed\n",
      "... done.\n",
      "End time: Mon Jul 24 13:10:24 2023\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --pfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01 \\\n",
    "--make-bed  \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "dca68e67-b3a0-4fc9-a2a0-4d2528501c9d",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.26 64-bit (2 Apr 2022)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.AD.R.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.AD.R\n",
      "  --recode AD\n",
      "\n",
      "257482 MB RAM detected; reserving 128741 MB for main workspace.\n",
      "19518 variants loaded from .bim file.\n",
      "351430 people (162935 males, 188495 females) loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 351430 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.915182.\n",
      "19518 variants and 351430 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode AD to\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.AD.R.raw\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/1.9.10\n",
    "\n",
    "plink --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01 \\\n",
    "--recode 'AD' \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.AD.R\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "id": "dc99b18d-bc54-4699-84b2-aef0747165b6",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.26 64-bit (2 Apr 2022)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.A.R.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.A.R\n",
      "  --recode A\n",
      "\n",
      "257482 MB RAM detected; reserving 128741 MB for main workspace.\n",
      "19518 variants loaded from .bim file.\n",
      "351430 people (162935 males, 188495 females) loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 351430 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.915182.\n",
      "19518 variants and 351430 people pass filters and QC.\n",
      "Note: No phenotypes present.\n",
      "--recode A to\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.A.R.raw\n",
      "... 101112131415161718192021222324252627282930313233343536373839404142434445464748495051525354555657585960616263646566676869707172737475767778798081828384858687888990919293949596979899done.\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/1.9.10\n",
    "\n",
    "plink --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01 \\\n",
    "--recode 'A' \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.A.R\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "3835d4fd-cd80-4159-9d2f-ba0eb54ae41d",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "Traceback (most recent call last):\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/runpy.py\", line 194, in _run_module_as_main\n",
      "    return _run_code(code, main_globals, None,\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/runpy.py\", line 87, in _run_code\n",
      "    exec(code, run_globals)\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/site-packages/calysto_bash/__main__.py\", line 2, in <module>\n",
      "    from .kernel import BashKernel\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/site-packages/calysto_bash/kernel.py\", line 6, in <module>\n",
      "    from metakernel import MetaKernel\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/site-packages/metakernel/__init__.py\", line 1, in <module>\n",
      "    from ._metakernel import (\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/site-packages/metakernel/_metakernel.py\", line 21, in <module>\n",
      "    from IPython.paths import get_ipython_dir\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/site-packages/IPython/__init__.py\", line 56, in <module>\n",
      "    from .terminal.embed import embed\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/site-packages/IPython/terminal/embed.py\", line 17, in <module>\n",
      "    from IPython.terminal.ipapp import load_default_config\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/site-packages/IPython/terminal/ipapp.py\", line 28, in <module>\n",
      "    from IPython.core.magics import (\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/site-packages/IPython/core/magics/__init__.py\", line 21, in <module>\n",
      "    from .execution import ExecutionMagics\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/site-packages/IPython/core/magics/execution.py\", line 24, in <module>\n",
      "    import cProfile as profile\n",
      "  File \"/home/dmc2245/miniconda3/lib/python3.8/cProfile.py\", line 10, in <module>\n",
      "    import profile as _pyprofile\n",
      "  File \"<frozen importlib._bootstrap>\", line 991, in _find_and_load\n",
      "  File \"<frozen importlib._bootstrap>\", line 975, in _find_and_load_unlocked\n",
      "  File \"<frozen importlib._bootstrap>\", line 671, in _load_unlocked\n",
      "  File \"<frozen importlib._bootstrap_external>\", line 779, in exec_module\n",
      "  File \"<frozen importlib._bootstrap_external>\", line 874, in get_code\n",
      "  File \"<frozen importlib._bootstrap_external>\", line 973, in get_data\n",
      "KeyboardInterrupt\n",
      "/home/dmc2245/miniconda3/lib/python3.8/site-packages/sos_notebook/kernel.py:1334: DeprecationWarning: Kernel._parent_header is deprecated in ipykernel 6. Use .get_parent()\n",
      "  msg['msg_id'] = self._parent_header['header']['msg_id']\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PLINK v1.90b6.26 64-bit (2 Apr 2022)           www.cog-genomics.org/plink/1.9/\n",
      "(C) 2005-2022 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.A.R.log.\n",
      "Options in effect:\n",
      "  --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01\n",
      "  --freq\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.A.R\n",
      "\n",
      "515665 MB RAM detected; reserving 257832 MB for main workspace.\n",
      "Allocated 193374 MB successfully, after larger attempt(s) failed.\n",
      "19518 variants loaded from .bim file.\n",
      "351430 people (162935 males, 188495 females) loaded from .fam.\n",
      "Using 1 thread (no multithreaded calculations invoked).\n",
      "Before main variant filters, 351430 founders and 0 nonfounders present.\n",
      "Calculating allele frequencies... 10111213141516171819202122232425262728293031323334353637383940414243444546474849505152535455565758596061626364656667686970717273747576777879808182838485868788899091929394959697989 done.\n",
      "Total genotyping rate is 0.915182.\n",
      "--freq: Allele frequencies (founders only) written to\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.A.R.frq\n",
      ".\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/1.9.10\n",
    "\n",
    "plink --bfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01 \\\n",
    "--freq  \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.A.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "7117ad2a-0b76-47f7-a9d8-7bd1d1e45d26",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "The following have been reloaded with a version change:\n",
      "  1) Plink/1.9.10 => Plink/2.00a\n",
      "\n",
      "PLINK v2.00a4LM 64-bit Intel (11 Apr 2023)     www.cog-genomics.org/plink/2.0/\n",
      "(C) 2005-2023 Shaun Purcell, Christopher Chang   GNU General Public License v3\n",
      "Logging to /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.R.log.\n",
      "Options in effect:\n",
      "  --freq\n",
      "  --out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.R\n",
      "  --pfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01\n",
      "\n",
      "Start time: Tue Jul 25 15:58:39 2023\n",
      "515665 MiB RAM detected, ~472586 available; reserving 257832 MiB for main\n",
      "workspace.\n",
      "Allocated 193374 MiB successfully, after larger attempt(s) failed.\n",
      "Using up to 48 threads (change this with --threads).\n",
      "351430 samples (188495 females, 162935 males; 351430 founders) loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.psam.\n",
      "19518 variants loaded from\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.pvar.\n",
      "Note: No phenotype data present.\n",
      "Calculating allele frequencies... done.\n",
      "--freq: Allele frequencies (founders only) written to12122222323242425252626272728282929303031313232333334343535363637373838393940404141424243434444454546464747484849505051515252535354545555565657575858595960606161626263636464656566666767686869697070717172727373747475757676777778787979808081818282838384848585868687878888898990909191929293939494959596969797989899%\n",
      "/mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.R.afreq\n",
      ".\n",
      "End time: Tue Jul 25 16:20:35 2023\n",
      "\n"
     ]
    }
   ],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --pfile /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01 \\\n",
    "--freq \\\n",
    "--out /mnt/vast/hpc/csg/UKBiobank/results/pleiotropy_AD_ARHI/111822_LDstore_files/regions_chr1_22/06/ukb_modif_keep_06_31571218_32682664.unrelated.0.01.R"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9036320c-443c-410f-b894-df8542fc94ff",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": [
    "module load Plink/2.00a\n",
    "plink2 --bgen ~/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb_hap_chr1_v2.bgen --sample ~/UKBiobank_Yale_transfer/ukb39554_imputeddataset/ukb32285_imputedindiv.sample --oxford-single-chr --make-bed ~/test/test1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "e1e0fb01-959b-4fd6-93b7-f54df7dc8835",
   "metadata": {
    "kernel": "Bash"
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "SoS",
   "language": "sos",
   "name": "sos"
  },
  "language_info": {
   "codemirror_mode": "sos",
   "file_extension": ".sos",
   "mimetype": "text/x-sos",
   "name": "sos",
   "nbconvert_exporter": "sos_notebook.converter.SoS_Exporter",
   "pygments_lexer": "sos"
  },
  "sos": {
   "kernels": [
    [
     "Bash",
     "calysto_bash",
     "Bash",
     "#E6EEFF",
     "shell"
    ],
    [
     "Python3",
     "python3",
     "Python3",
     "#FFD91A",
     {
      "name": "ipython",
      "version": 3
     }
    ],
    [
     "R",
     "ir",
     "R",
     "#DCDCDA",
     "r"
    ]
   ],
   "version": "0.22.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
