{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "3f8c075b-ca06-445a-96db-857a95bdc334",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "48430860-c719-4408-8d48-539f7e7972b2",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "a39f36f9-1e60-4d90-83cb-ab364d66d279",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "file='~/data_public/cadd/gnomad.genomes.r3.0.indel.tsv.gz'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "dccebff0-d7df-4f42-85ab-163b1877deb1",
   "metadata": {
    "kernel": "Python3",
    "tags": []
   },
   "outputs": [],
   "source": [
    "df = pd.read_csv(file, sep=\"\\t\", compression='gzip', skiprows=2, names=[\"Chrom\",\"Pos\",\"Ref\",\"Alt\",\"RawScore\",\"PHRED\"], dtype=\"string\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "8fedffc2-538c-49eb-83e2-85feaf51a4a1",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<StringArray>\n",
       "[ '1',  '2',  '3',  '4',  '5',  '6',  '7',  '8',  '9', '10', '11', '12', '13',\n",
       " '14', '15', '16', '17', '18', '19', '20', '21', '22',  'X',  'Y']\n",
       "Length: 24, dtype: string"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[\"Chrom\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "f65ef49a-be24-45a7-815b-e1961d4949f2",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "df2=df[(df['Chrom'] != 'X') & (df['Chrom'] != 'Y')]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "3f3cc5bb-6a52-48bd-aae5-8da2bd5192f0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>10061</td>\n",
       "      <td>T</td>\n",
       "      <td>TAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCC</td>\n",
       "      <td>0.199964</td>\n",
       "      <td>3.123</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>1</td>\n",
       "      <td>10067</td>\n",
       "      <td>T</td>\n",
       "      <td>TAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCC</td>\n",
       "      <td>0.225348</td>\n",
       "      <td>3.416</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>10108</td>\n",
       "      <td>C</td>\n",
       "      <td>CA</td>\n",
       "      <td>0.390514</td>\n",
       "      <td>5.321</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>10108</td>\n",
       "      <td>C</td>\n",
       "      <td>CAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT</td>\n",
       "      <td>0.215572</td>\n",
       "      <td>3.303</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>10108</td>\n",
       "      <td>CAACCCT</td>\n",
       "      <td>C</td>\n",
       "      <td>0.317701</td>\n",
       "      <td>4.494</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96498716</th>\n",
       "      <td>22</td>\n",
       "      <td>50808323</td>\n",
       "      <td>G</td>\n",
       "      <td>GT</td>\n",
       "      <td>0.419674</td>\n",
       "      <td>5.644</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96498717</th>\n",
       "      <td>22</td>\n",
       "      <td>50808323</td>\n",
       "      <td>GTTAGGCTTAGGGTTAGGGTAGGCTTAGGC</td>\n",
       "      <td>G</td>\n",
       "      <td>0.246563</td>\n",
       "      <td>3.664</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96498718</th>\n",
       "      <td>22</td>\n",
       "      <td>50808327</td>\n",
       "      <td>GGCTTAGGGTTAGGGTAGGCTTAGGCTTAGGGTTAGGGTTAGGGTTA</td>\n",
       "      <td>G</td>\n",
       "      <td>0.169190</td>\n",
       "      <td>2.780</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96498719</th>\n",
       "      <td>22</td>\n",
       "      <td>50808329</td>\n",
       "      <td>CTTAGGGTTAGGGTAGGCTTAGGCTTAGGGTTAGGGTTAGGGTTAG...</td>\n",
       "      <td>C</td>\n",
       "      <td>0.156602</td>\n",
       "      <td>2.645</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>96498720</th>\n",
       "      <td>22</td>\n",
       "      <td>50808368</td>\n",
       "      <td>GGGTTA</td>\n",
       "      <td>G</td>\n",
       "      <td>0.340612</td>\n",
       "      <td>4.757</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>96498721 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "         Chrom       Pos                                                Ref  \\\n",
       "0            1     10061                                                  T   \n",
       "1            1     10067                                                  T   \n",
       "2            1     10108                                                  C   \n",
       "3            1     10108                                                  C   \n",
       "4            1     10108                                            CAACCCT   \n",
       "...        ...       ...                                                ...   \n",
       "96498716    22  50808323                                                  G   \n",
       "96498717    22  50808323                     GTTAGGCTTAGGGTTAGGGTAGGCTTAGGC   \n",
       "96498718    22  50808327    GGCTTAGGGTTAGGGTAGGCTTAGGCTTAGGGTTAGGGTTAGGGTTA   \n",
       "96498719    22  50808329  CTTAGGGTTAGGGTAGGCTTAGGCTTAGGGTTAGGGTTAGGGTTAG...   \n",
       "96498720    22  50808368                                             GGGTTA   \n",
       "\n",
       "                                                       Alt  RawScore  PHRED  \n",
       "0         TAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCC  0.199964  3.123  \n",
       "1               TAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCC  0.225348  3.416  \n",
       "2                                                       CA  0.390514  5.321  \n",
       "3              CAACCCTAACCCTAACCCTAACCCTAACCCTAACCCTAACCCT  0.215572  3.303  \n",
       "4                                                        C  0.317701  4.494  \n",
       "...                                                    ...       ...    ...  \n",
       "96498716                                                GT  0.419674  5.644  \n",
       "96498717                                                 G  0.246563  3.664  \n",
       "96498718                                                 G  0.169190  2.780  \n",
       "96498719                                                 C  0.156602  2.645  \n",
       "96498720                                                 G  0.340612  4.757  \n",
       "\n",
       "[96498721 rows x 6 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "a51bb024-4d4c-493d-9f47-53427bc70c68",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<StringArray>\n",
       "[ '1',  '2',  '3',  '4',  '5',  '6',  '7',  '8',  '9', '10', '11', '12', '13',\n",
       " '14', '15', '16', '17', '18', '19', '20', '21', '22']\n",
       "Length: 22, dtype: string"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df2[\"Chrom\"].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "c278138d-e086-419a-887a-735ea62290bf",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/6586085.1.csg.q/ipykernel_39415/882595014.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df2['Chrom']=df2['Chrom'].astype(int)\n"
     ]
    }
   ],
   "source": [
    "df2['Chrom']=df2['Chrom'].astype(int)\n",
    "df3=df2[df2['Chrom']==3]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 68,
   "id": "47db3963-665d-4b42-8c23-a941e1d37cc0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/tmp/6586085.1.csg.q/ipykernel_39415/2214076356.py:1: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df3['Pos']=df3['Pos'].astype(int)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Chrom</th>\n",
       "      <th>Pos</th>\n",
       "      <th>Ref</th>\n",
       "      <th>Alt</th>\n",
       "      <th>RawScore</th>\n",
       "      <th>PHRED</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>17616955</th>\n",
       "      <td>3</td>\n",
       "      <td>44659107</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>3.986693</td>\n",
       "      <td>26.9</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17616956</th>\n",
       "      <td>3</td>\n",
       "      <td>44659227</td>\n",
       "      <td>T</td>\n",
       "      <td>TCA</td>\n",
       "      <td>3.690562</td>\n",
       "      <td>25.4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17616957</th>\n",
       "      <td>3</td>\n",
       "      <td>44659252</td>\n",
       "      <td>C</td>\n",
       "      <td>CA</td>\n",
       "      <td>3.952939</td>\n",
       "      <td>26.7</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17616958</th>\n",
       "      <td>3</td>\n",
       "      <td>44659277</td>\n",
       "      <td>CTT</td>\n",
       "      <td>C</td>\n",
       "      <td>3.850260</td>\n",
       "      <td>26.1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17616959</th>\n",
       "      <td>3</td>\n",
       "      <td>44659354</td>\n",
       "      <td>G</td>\n",
       "      <td>GA</td>\n",
       "      <td>4.668134</td>\n",
       "      <td>32</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22716654</th>\n",
       "      <td>3</td>\n",
       "      <td>198232718</td>\n",
       "      <td>TA</td>\n",
       "      <td>T</td>\n",
       "      <td>0.108415</td>\n",
       "      <td>2.167</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22716655</th>\n",
       "      <td>3</td>\n",
       "      <td>198232734</td>\n",
       "      <td>T</td>\n",
       "      <td>TA</td>\n",
       "      <td>0.133540</td>\n",
       "      <td>2.408</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22716656</th>\n",
       "      <td>3</td>\n",
       "      <td>198232739</td>\n",
       "      <td>AG</td>\n",
       "      <td>A</td>\n",
       "      <td>0.040894</td>\n",
       "      <td>1.612</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22716657</th>\n",
       "      <td>3</td>\n",
       "      <td>198232740</td>\n",
       "      <td>GAC</td>\n",
       "      <td>G</td>\n",
       "      <td>0.020106</td>\n",
       "      <td>1.466</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22716658</th>\n",
       "      <td>3</td>\n",
       "      <td>198232918</td>\n",
       "      <td>AT</td>\n",
       "      <td>A</td>\n",
       "      <td>0.658152</td>\n",
       "      <td>8.067</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5099704 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          Chrom        Pos  Ref  Alt  RawScore  PHRED\n",
       "17616955      3   44659107   AC    A  3.986693   26.9\n",
       "17616956      3   44659227    T  TCA  3.690562   25.4\n",
       "17616957      3   44659252    C   CA  3.952939   26.7\n",
       "17616958      3   44659277  CTT    C  3.850260   26.1\n",
       "17616959      3   44659354    G   GA  4.668134     32\n",
       "...         ...        ...  ...  ...       ...    ...\n",
       "22716654      3  198232718   TA    T  0.108415  2.167\n",
       "22716655      3  198232734    T   TA  0.133540  2.408\n",
       "22716656      3  198232739   AG    A  0.040894  1.612\n",
       "22716657      3  198232740  GAC    G  0.020106  1.466\n",
       "22716658      3  198232918   AT    A  0.658152  8.067\n",
       "\n",
       "[5099704 rows x 6 columns]"
      ]
     },
     "execution_count": 68,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df3['Pos']=df3['Pos'].astype(int)\n",
    "df3[df3['Pos']>=44659074]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "bed192d8-a537-42d7-8578-ae350bb8fbff",
   "metadata": {},
   "source": [
    "## Post-processing of vep - code "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "7af1c615-5c76-47b4-8346-4b4ed273e5ee",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "vep = pd.read_csv(\"~/test/bcftools/snps_indels_vep_test_sorted.rare.VEP.CADD_gnomAD.vcf.gz\", compression=\"gzip\", comment='#', sep='\\t', names=[\"CHROM\",\"POS\",\"ID\",\"REF\",\"ALT\",\"QUAL\",\"FILTER\",\"INFO\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "106b450b-bed0-4d10-8e16-4c0c920a6ab8",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "vep"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "id": "c345fddc-d7ab-4173-a1bb-7ed29407e305",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import gzip\n",
    "vcf_file = \"/home/dmc2245/test/bcftools/snps_indels_vep_test_sorted.rare.VEP.CADD_gnomAD.vcf.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "id": "6780c07a-c893-4f4f-9012-454446fbd477",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "df = pd.read_csv(vcf_file, compression=\"gzip\", comment='#', sep='\\t', names=[\"CHROM\",\"POS\",\"ID\",\"REF\",\"ALT\",\"QUAL\",\"FILTER\",\"INFO\"])\n",
    "# Open the gzip-compressed VCF file and read it as text\n",
    "\n",
    "with gzip.open(vcf_file, 'rt') as file:\n",
    "    # Read the VCF file into a DataFrame\n",
    "    for line in file:\n",
    "        if line.startswith(\"##INFO=<ID=CSQ\"):\n",
    "            info_description = line.strip()\n",
    "            break"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "id": "4038b845-4a3c-4825-bc8a-0a790bc31584",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>ID</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>QUAL</th>\n",
       "      <th>FILTER</th>\n",
       "      <th>INFO</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>43343506</td>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>G</td>\n",
       "      <td>GT</td>\n",
       "      <td>31</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AF=5e-06;AQ=31;AC=0;AN=4;CSQ=T|intron_variant|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>44244604</td>\n",
       "      <td>3:44244604:AAAG:A</td>\n",
       "      <td>AAAG</td>\n",
       "      <td>A</td>\n",
       "      <td>54</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AF=7e-06;AQ=54;AC=0;AN=6;CSQ=-|inframe_deletio...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>44359012</td>\n",
       "      <td>3:44359012:GTT:G</td>\n",
       "      <td>GTT</td>\n",
       "      <td>G</td>\n",
       "      <td>57</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AF=2e-06;AQ=57;AC=0;AN=6;CSQ=-|intron_variant|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>44568370</td>\n",
       "      <td>3:44568370:CTCAG:C</td>\n",
       "      <td>CTCAG</td>\n",
       "      <td>C</td>\n",
       "      <td>58</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AF=0.000214;AQ=58;AC=0;AN=6;CSQ=-|frameshift_v...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>44570113</td>\n",
       "      <td>3:44570113:GA:G</td>\n",
       "      <td>GA</td>\n",
       "      <td>G</td>\n",
       "      <td>48</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AF=1.5e-05;AQ=48;AC=0;AN=6;CSQ=-|frameshift_va...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>3</td>\n",
       "      <td>44631109</td>\n",
       "      <td>3:44631109:TG:T</td>\n",
       "      <td>TG</td>\n",
       "      <td>T</td>\n",
       "      <td>41</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AF=5e-06;AQ=41;AC=0;AN=6;CSQ=-|3_prime_UTR_var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>3</td>\n",
       "      <td>44642021</td>\n",
       "      <td>3:44642021:TGAA:T</td>\n",
       "      <td>TGAA</td>\n",
       "      <td>T</td>\n",
       "      <td>51</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AF=2e-06;AQ=51;AC=0;AN=6;CSQ=-|upstream_gene_v...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>3</td>\n",
       "      <td>44754210</td>\n",
       "      <td>3:44754210:AC:A</td>\n",
       "      <td>AC</td>\n",
       "      <td>A</td>\n",
       "      <td>48</td>\n",
       "      <td>PASS</td>\n",
       "      <td>AF=7e-06;AQ=48;AC=0;AN=6;CSQ=-|intron_variant|...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>3</td>\n",
       "      <td>45225871</td>\n",
       "      <td>3:45225871:A:T</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>48</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=5e-06;AQ=48;AC=0;AN=20;CSQ=T|missense_varia...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>3</td>\n",
       "      <td>45417390</td>\n",
       "      <td>3:45417390:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>47</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=7e-06;AQ=47;AC=0;AN=20;CSQ=G|intron_variant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>3</td>\n",
       "      <td>45419610</td>\n",
       "      <td>3:45419610:G:GT</td>\n",
       "      <td>G</td>\n",
       "      <td>GT</td>\n",
       "      <td>56</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.535414;AQ=56;AC=17;AN=20;CSQ=T|intron_var...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>3</td>\n",
       "      <td>45419757</td>\n",
       "      <td>3:45419757:C:G</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>54</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=1.2e-05;AQ=48;AC=0;AN=20;CSQ=G|intron_varia...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>3</td>\n",
       "      <td>45419757</td>\n",
       "      <td>3:45419757:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>54</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=0.000122;AQ=54;AC=0;AN=20;CSQ=T|intron_vari...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>3</td>\n",
       "      <td>45419779</td>\n",
       "      <td>3:45419779:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>44</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=44;AC=0;AN=20;CSQ=G|intron_variant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>3</td>\n",
       "      <td>45458900</td>\n",
       "      <td>3:45458900:T:C</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>50</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2.7e-05;AQ=50;AC=0;AN=20;CSQ=C|intron_varia...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>3</td>\n",
       "      <td>45488901</td>\n",
       "      <td>3:45488901:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>39</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=39;AC=0;AN=20;CSQ=T|intron_variant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>3</td>\n",
       "      <td>45496262</td>\n",
       "      <td>3:45496262:A:G</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>44</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=44;AC=0;AN=20;CSQ=G|splice_polypyr...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>41</td>\n",
       "      <td>.</td>\n",
       "      <td>AF=2e-06;AQ=41;AC=0;AN=20;CSQ=T|synonymous_var...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    CHROM       POS                  ID    REF ALT  QUAL FILTER  \\\n",
       "0       3  43343506     3:43343506:G:GT      G  GT    31   PASS   \n",
       "1       3  44244604   3:44244604:AAAG:A   AAAG   A    54   PASS   \n",
       "2       3  44359012    3:44359012:GTT:G    GTT   G    57   PASS   \n",
       "3       3  44568370  3:44568370:CTCAG:C  CTCAG   C    58   PASS   \n",
       "4       3  44570113     3:44570113:GA:G     GA   G    48   PASS   \n",
       "5       3  44631109     3:44631109:TG:T     TG   T    41   PASS   \n",
       "6       3  44642021   3:44642021:TGAA:T   TGAA   T    51   PASS   \n",
       "7       3  44754210     3:44754210:AC:A     AC   A    48   PASS   \n",
       "8       3  45225871      3:45225871:A:T      A   T    48      .   \n",
       "9       3  45417390      3:45417390:A:G      A   G    47      .   \n",
       "10      3  45419610     3:45419610:G:GT      G  GT    56      .   \n",
       "11      3  45419757      3:45419757:C:G      C   G    54      .   \n",
       "12      3  45419757      3:45419757:C:T      C   T    54      .   \n",
       "13      3  45419779      3:45419779:A:G      A   G    44      .   \n",
       "14      3  45458900      3:45458900:T:C      T   C    50      .   \n",
       "15      3  45488901      3:45488901:C:T      C   T    39      .   \n",
       "16      3  45496262      3:45496262:A:G      A   G    44      .   \n",
       "17      3  45496305      3:45496305:C:T      C   T    41      .   \n",
       "\n",
       "                                                 INFO  \n",
       "0   AF=5e-06;AQ=31;AC=0;AN=4;CSQ=T|intron_variant|...  \n",
       "1   AF=7e-06;AQ=54;AC=0;AN=6;CSQ=-|inframe_deletio...  \n",
       "2   AF=2e-06;AQ=57;AC=0;AN=6;CSQ=-|intron_variant|...  \n",
       "3   AF=0.000214;AQ=58;AC=0;AN=6;CSQ=-|frameshift_v...  \n",
       "4   AF=1.5e-05;AQ=48;AC=0;AN=6;CSQ=-|frameshift_va...  \n",
       "5   AF=5e-06;AQ=41;AC=0;AN=6;CSQ=-|3_prime_UTR_var...  \n",
       "6   AF=2e-06;AQ=51;AC=0;AN=6;CSQ=-|upstream_gene_v...  \n",
       "7   AF=7e-06;AQ=48;AC=0;AN=6;CSQ=-|intron_variant|...  \n",
       "8   AF=5e-06;AQ=48;AC=0;AN=20;CSQ=T|missense_varia...  \n",
       "9   AF=7e-06;AQ=47;AC=0;AN=20;CSQ=G|intron_variant...  \n",
       "10  AF=0.535414;AQ=56;AC=17;AN=20;CSQ=T|intron_var...  \n",
       "11  AF=1.2e-05;AQ=48;AC=0;AN=20;CSQ=G|intron_varia...  \n",
       "12  AF=0.000122;AQ=54;AC=0;AN=20;CSQ=T|intron_vari...  \n",
       "13  AF=2e-06;AQ=44;AC=0;AN=20;CSQ=G|intron_variant...  \n",
       "14  AF=2.7e-05;AQ=50;AC=0;AN=20;CSQ=C|intron_varia...  \n",
       "15  AF=2e-06;AQ=39;AC=0;AN=20;CSQ=T|intron_variant...  \n",
       "16  AF=2e-06;AQ=44;AC=0;AN=20;CSQ=G|splice_polypyr...  \n",
       "17  AF=2e-06;AQ=41;AC=0;AN=20;CSQ=T|synonymous_var...  "
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "id": "b533860c-a690-4cb2-a61b-ed2d071b7ea4",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Extract the INFO field header from the description\n",
    "info_header = info_description.split(\": \")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "id": "a7a87174-555c-430c-80f1-c5080af80e13",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "info_columns = df['INFO'].str.split(',', expand=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 133,
   "id": "7ee8844b-690f-4f7b-a49a-4544b4364228",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>10</th>\n",
       "      <th>11</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AF=5e-06;AQ=31;AC=0;AN=4;CSQ=T|intron_variant|...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|SNRK-AS1|EN...</td>\n",
       "      <td>T|intron_variant|MODIFIER|SNRK|ENSG00000163788...</td>\n",
       "      <td>T|intron_variant|MODIFIER|SNRK|ENSG00000163788...</td>\n",
       "      <td>T|intron_variant|MODIFIER|SNRK|ENSG00000163788...</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|SNRK|ENSG00...</td>\n",
       "      <td>T|downstream_gene_variant|MODIFIER|SNRK-AS1|EN...</td>\n",
       "      <td>T|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AF=7e-06;AQ=54;AC=0;AN=6;CSQ=-|inframe_deletio...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AF=2e-06;AQ=57;AC=0;AN=6;CSQ=-|intron_variant|...</td>\n",
       "      <td>-|3_prime_UTR_variant|MODIFIER|TCAIM|ENSG00000...</td>\n",
       "      <td>-|3_prime_UTR_variant|MODIFIER|TCAIM|ENSG00000...</td>\n",
       "      <td>-|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>-|intron_variant|MODIFIER|TCAIM|ENSG0000017915...</td>\n",
       "      <td>-|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>-|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>-|intron_variant|MODIFIER|TCAIM|ENSG0000017915...</td>\n",
       "      <td>-|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>-|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>-|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AF=0.000214;AQ=58;AC=0;AN=6;CSQ=-|frameshift_v...</td>\n",
       "      <td>-|frameshift_variant|HIGH|ZKSCAN7|ENSG00000196...</td>\n",
       "      <td>-|upstream_gene_variant|MODIFIER|ZNF660|ENSG00...</td>\n",
       "      <td>-|3_prime_UTR_variant&amp;NMD_transcript_variant|M...</td>\n",
       "      <td>-|frameshift_variant|HIGH|ZKSCAN7|ENSG00000196...</td>\n",
       "      <td>-|frameshift_variant|HIGH|ZKSCAN7|ENSG00000196...</td>\n",
       "      <td>-|frameshift_variant|HIGH|ZKSCAN7|ENSG00000196...</td>\n",
       "      <td>-|frameshift_variant|HIGH|ZKSCAN7|ENSG00000196...</td>\n",
       "      <td>-|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>-|non_coding_transcript_exon_variant|MODIFIER|...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AF=1.5e-05;AQ=48;AC=0;AN=6;CSQ=-|frameshift_va...</td>\n",
       "      <td>-|intron_variant|MODIFIER|ZKSCAN7|ENSG00000196...</td>\n",
       "      <td>-|upstream_gene_variant|MODIFIER|ZNF660|ENSG00...</td>\n",
       "      <td>-|3_prime_UTR_variant&amp;NMD_transcript_variant|M...</td>\n",
       "      <td>-|intron_variant|MODIFIER|ZKSCAN7|ENSG00000196...</td>\n",
       "      <td>-|frameshift_variant|HIGH|ZKSCAN7|ENSG00000196...</td>\n",
       "      <td>-|intron_variant|MODIFIER|ZKSCAN7|ENSG00000196...</td>\n",
       "      <td>-|frameshift_variant|HIGH|ZKSCAN7|ENSG00000196...</td>\n",
       "      <td>-|downstream_gene_variant|MODIFIER|MPRIPP1|ENS...</td>\n",
       "      <td>-|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>-|downstream_gene_variant|MODIFIER|ZKSCAN7|ENS...</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AF=5e-06;AQ=41;AC=0;AN=6;CSQ=-|3_prime_UTR_var...</td>\n",
       "      <td>-|frameshift_variant|HIGH|ZNF197|ENSG000001864...</td>\n",
       "      <td>-|frameshift_variant|HIGH|ZNF197|ENSG000001864...</td>\n",
       "      <td>-|frameshift_variant|HIGH|ZNF197|ENSG000001864...</td>\n",
       "      <td>-|frameshift_variant|HIGH|ZNF197|ENSG000001864...</td>\n",
       "      <td>-|downstream_gene_variant|MODIFIER|ZNF197|ENSG...</td>\n",
       "      <td>-|upstream_gene_variant|MODIFIER|ZNF197-AS1|EN...</td>\n",
       "      <td>-|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AF=2e-06;AQ=51;AC=0;AN=6;CSQ=-|upstream_gene_v...</td>\n",
       "      <td>-|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>-|inframe_deletion|MODERATE|ZNF197|ENSG0000018...</td>\n",
       "      <td>-|intron_variant|MODIFIER|ZNF197|ENSG000001864...</td>\n",
       "      <td>-|intron_variant|MODIFIER|ZNF197|ENSG000001864...</td>\n",
       "      <td>-|upstream_gene_variant|MODIFIER|ZNF35|ENSG000...</td>\n",
       "      <td>-|inframe_deletion|MODERATE|ZNF197|ENSG0000018...</td>\n",
       "      <td>-|upstream_gene_variant|MODIFIER|ZNF35|ENSG000...</td>\n",
       "      <td>-|upstream_gene_variant|MODIFIER|ZNF35|ENSG000...</td>\n",
       "      <td>-|upstream_gene_variant|MODIFIER|ZNF35|ENSG000...</td>\n",
       "      <td>-|upstream_gene_variant|MODIFIER|ZNF35|ENSG000...</td>\n",
       "      <td>-|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AF=7e-06;AQ=48;AC=0;AN=6;CSQ=-|intron_variant|...</td>\n",
       "      <td>-|upstream_gene_variant|MODIFIER|KIF15|ENSG000...</td>\n",
       "      <td>-|upstream_gene_variant|MODIFIER|KIF15|ENSG000...</td>\n",
       "      <td>-|upstream_gene_variant|MODIFIER|KIF15|ENSG000...</td>\n",
       "      <td>-|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AF=5e-06;AQ=48;AC=0;AN=20;CSQ=T|missense_varia...</td>\n",
       "      <td>T|regulatory_region_variant|MODIFIER|||Regulat...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AF=7e-06;AQ=47;AC=0;AN=20;CSQ=G|intron_variant...</td>\n",
       "      <td>G|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>G|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>G|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>G|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>G|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>G|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>G|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>AF=0.535414;AQ=56;AC=17;AN=20;CSQ=T|intron_var...</td>\n",
       "      <td>T|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>T|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>T|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>T|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>T|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>AF=1.2e-05;AQ=48;AC=0;AN=20;CSQ=G|intron_varia...</td>\n",
       "      <td>G|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>G|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>G|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>G|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>G|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>G|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>G|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>AF=0.000122;AQ=54;AC=0;AN=20;CSQ=T|intron_vari...</td>\n",
       "      <td>T|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>T|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>T|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>T|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>T|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>AF=2e-06;AQ=44;AC=0;AN=20;CSQ=G|intron_variant...</td>\n",
       "      <td>G|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>G|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>G|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>G|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>G|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>G|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>G|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>AF=2.7e-05;AQ=50;AC=0;AN=20;CSQ=C|intron_varia...</td>\n",
       "      <td>C|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>C|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>C|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>C|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>C|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>C|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>AF=2e-06;AQ=39;AC=0;AN=20;CSQ=T|intron_variant...</td>\n",
       "      <td>T|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>T|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>T|intron_variant|MODIFIER|LARS2|ENSG0000001137...</td>\n",
       "      <td>T|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>T|intron_variant&amp;NMD_transcript_variant|MODIFI...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>AF=2e-06;AQ=44;AC=0;AN=20;CSQ=G|splice_polypyr...</td>\n",
       "      <td>G|splice_polypyrimidine_tract_variant&amp;intron_v...</td>\n",
       "      <td>G|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>G|upstream_gene_variant|MODIFIER|LARS2|ENSG000...</td>\n",
       "      <td>G|splice_polypyrimidine_tract_variant&amp;intron_v...</td>\n",
       "      <td>G|splice_polypyrimidine_tract_variant&amp;intron_v...</td>\n",
       "      <td>G|splice_polypyrimidine_tract_variant&amp;intron_v...</td>\n",
       "      <td>G|splice_polypyrimidine_tract_variant&amp;intron_v...</td>\n",
       "      <td>G|splice_polypyrimidine_tract_variant&amp;intron_v...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>AF=2e-06;AQ=41;AC=0;AN=20;CSQ=T|synonymous_var...</td>\n",
       "      <td>T|synonymous_variant|LOW|LARS2|ENSG00000011376...</td>\n",
       "      <td>T|intron_variant&amp;non_coding_transcript_variant...</td>\n",
       "      <td>T|upstream_gene_variant|MODIFIER|LARS2|ENSG000...</td>\n",
       "      <td>T|synonymous_variant|LOW|LARS2|ENSG00000011376...</td>\n",
       "      <td>T|synonymous_variant|LOW|LARS2|ENSG00000011376...</td>\n",
       "      <td>T|synonymous_variant|LOW|LARS2|ENSG00000011376...</td>\n",
       "      <td>T|synonymous_variant&amp;NMD_transcript_variant|LO...</td>\n",
       "      <td>T|3_prime_UTR_variant&amp;NMD_transcript_variant|M...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   0   \\\n",
       "0   AF=5e-06;AQ=31;AC=0;AN=4;CSQ=T|intron_variant|...   \n",
       "1   AF=7e-06;AQ=54;AC=0;AN=6;CSQ=-|inframe_deletio...   \n",
       "2   AF=2e-06;AQ=57;AC=0;AN=6;CSQ=-|intron_variant|...   \n",
       "3   AF=0.000214;AQ=58;AC=0;AN=6;CSQ=-|frameshift_v...   \n",
       "4   AF=1.5e-05;AQ=48;AC=0;AN=6;CSQ=-|frameshift_va...   \n",
       "5   AF=5e-06;AQ=41;AC=0;AN=6;CSQ=-|3_prime_UTR_var...   \n",
       "6   AF=2e-06;AQ=51;AC=0;AN=6;CSQ=-|upstream_gene_v...   \n",
       "7   AF=7e-06;AQ=48;AC=0;AN=6;CSQ=-|intron_variant|...   \n",
       "8   AF=5e-06;AQ=48;AC=0;AN=20;CSQ=T|missense_varia...   \n",
       "9   AF=7e-06;AQ=47;AC=0;AN=20;CSQ=G|intron_variant...   \n",
       "10  AF=0.535414;AQ=56;AC=17;AN=20;CSQ=T|intron_var...   \n",
       "11  AF=1.2e-05;AQ=48;AC=0;AN=20;CSQ=G|intron_varia...   \n",
       "12  AF=0.000122;AQ=54;AC=0;AN=20;CSQ=T|intron_vari...   \n",
       "13  AF=2e-06;AQ=44;AC=0;AN=20;CSQ=G|intron_variant...   \n",
       "14  AF=2.7e-05;AQ=50;AC=0;AN=20;CSQ=C|intron_varia...   \n",
       "15  AF=2e-06;AQ=39;AC=0;AN=20;CSQ=T|intron_variant...   \n",
       "16  AF=2e-06;AQ=44;AC=0;AN=20;CSQ=G|splice_polypyr...   \n",
       "17  AF=2e-06;AQ=41;AC=0;AN=20;CSQ=T|synonymous_var...   \n",
       "\n",
       "                                                   1   \\\n",
       "0   T|downstream_gene_variant|MODIFIER|SNRK-AS1|EN...   \n",
       "1                                                None   \n",
       "2   -|3_prime_UTR_variant|MODIFIER|TCAIM|ENSG00000...   \n",
       "3   -|frameshift_variant|HIGH|ZKSCAN7|ENSG00000196...   \n",
       "4   -|intron_variant|MODIFIER|ZKSCAN7|ENSG00000196...   \n",
       "5   -|frameshift_variant|HIGH|ZNF197|ENSG000001864...   \n",
       "6   -|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "7   -|upstream_gene_variant|MODIFIER|KIF15|ENSG000...   \n",
       "8   T|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "9   G|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "10  T|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "11  G|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "12  T|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "13  G|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "14  C|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "15  T|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "16  G|splice_polypyrimidine_tract_variant&intron_v...   \n",
       "17  T|synonymous_variant|LOW|LARS2|ENSG00000011376...   \n",
       "\n",
       "                                                   2   \\\n",
       "0   T|intron_variant|MODIFIER|SNRK|ENSG00000163788...   \n",
       "1                                                None   \n",
       "2   -|3_prime_UTR_variant|MODIFIER|TCAIM|ENSG00000...   \n",
       "3   -|upstream_gene_variant|MODIFIER|ZNF660|ENSG00...   \n",
       "4   -|upstream_gene_variant|MODIFIER|ZNF660|ENSG00...   \n",
       "5   -|frameshift_variant|HIGH|ZNF197|ENSG000001864...   \n",
       "6   -|inframe_deletion|MODERATE|ZNF197|ENSG0000018...   \n",
       "7   -|upstream_gene_variant|MODIFIER|KIF15|ENSG000...   \n",
       "8                                                None   \n",
       "9   G|intron_variant&non_coding_transcript_variant...   \n",
       "10  T|intron_variant&non_coding_transcript_variant...   \n",
       "11  G|intron_variant&non_coding_transcript_variant...   \n",
       "12  T|intron_variant&non_coding_transcript_variant...   \n",
       "13  G|intron_variant&non_coding_transcript_variant...   \n",
       "14  C|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "15  T|intron_variant&non_coding_transcript_variant...   \n",
       "16  G|intron_variant&non_coding_transcript_variant...   \n",
       "17  T|intron_variant&non_coding_transcript_variant...   \n",
       "\n",
       "                                                   3   \\\n",
       "0   T|intron_variant|MODIFIER|SNRK|ENSG00000163788...   \n",
       "1                                                None   \n",
       "2   -|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "3   -|3_prime_UTR_variant&NMD_transcript_variant|M...   \n",
       "4   -|3_prime_UTR_variant&NMD_transcript_variant|M...   \n",
       "5   -|frameshift_variant|HIGH|ZNF197|ENSG000001864...   \n",
       "6   -|intron_variant|MODIFIER|ZNF197|ENSG000001864...   \n",
       "7   -|upstream_gene_variant|MODIFIER|KIF15|ENSG000...   \n",
       "8                                                None   \n",
       "9   G|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "10  T|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "11  G|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "12  T|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "13  G|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "14  C|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "15  T|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "16  G|upstream_gene_variant|MODIFIER|LARS2|ENSG000...   \n",
       "17  T|upstream_gene_variant|MODIFIER|LARS2|ENSG000...   \n",
       "\n",
       "                                                   4   \\\n",
       "0   T|intron_variant|MODIFIER|SNRK|ENSG00000163788...   \n",
       "1                                                None   \n",
       "2   -|intron_variant|MODIFIER|TCAIM|ENSG0000017915...   \n",
       "3   -|frameshift_variant|HIGH|ZKSCAN7|ENSG00000196...   \n",
       "4   -|intron_variant|MODIFIER|ZKSCAN7|ENSG00000196...   \n",
       "5   -|frameshift_variant|HIGH|ZNF197|ENSG000001864...   \n",
       "6   -|intron_variant|MODIFIER|ZNF197|ENSG000001864...   \n",
       "7   -|intron_variant&non_coding_transcript_variant...   \n",
       "8                                                None   \n",
       "9   G|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "10  T|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "11  G|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "12  T|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "13  G|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "14  C|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "15  T|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "16  G|splice_polypyrimidine_tract_variant&intron_v...   \n",
       "17  T|synonymous_variant|LOW|LARS2|ENSG00000011376...   \n",
       "\n",
       "                                                   5   \\\n",
       "0   T|intron_variant&non_coding_transcript_variant...   \n",
       "1                                                None   \n",
       "2   -|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "3   -|frameshift_variant|HIGH|ZKSCAN7|ENSG00000196...   \n",
       "4   -|frameshift_variant|HIGH|ZKSCAN7|ENSG00000196...   \n",
       "5   -|downstream_gene_variant|MODIFIER|ZNF197|ENSG...   \n",
       "6   -|upstream_gene_variant|MODIFIER|ZNF35|ENSG000...   \n",
       "7                                                None   \n",
       "8                                                None   \n",
       "9   G|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "10  T|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "11  G|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "12  T|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "13  G|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "14  C|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "15  T|intron_variant|MODIFIER|LARS2|ENSG0000001137...   \n",
       "16  G|splice_polypyrimidine_tract_variant&intron_v...   \n",
       "17  T|synonymous_variant|LOW|LARS2|ENSG00000011376...   \n",
       "\n",
       "                                                   6   \\\n",
       "0   T|intron_variant&non_coding_transcript_variant...   \n",
       "1                                                None   \n",
       "2   -|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "3   -|frameshift_variant|HIGH|ZKSCAN7|ENSG00000196...   \n",
       "4   -|intron_variant|MODIFIER|ZKSCAN7|ENSG00000196...   \n",
       "5   -|upstream_gene_variant|MODIFIER|ZNF197-AS1|EN...   \n",
       "6   -|inframe_deletion|MODERATE|ZNF197|ENSG0000018...   \n",
       "7                                                None   \n",
       "8                                                None   \n",
       "9   G|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "10  T|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "11  G|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "12  T|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "13  G|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "14  C|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "15  T|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "16  G|splice_polypyrimidine_tract_variant&intron_v...   \n",
       "17  T|synonymous_variant|LOW|LARS2|ENSG00000011376...   \n",
       "\n",
       "                                                   7   \\\n",
       "0   T|downstream_gene_variant|MODIFIER|SNRK|ENSG00...   \n",
       "1                                                None   \n",
       "2   -|intron_variant|MODIFIER|TCAIM|ENSG0000017915...   \n",
       "3   -|frameshift_variant|HIGH|ZKSCAN7|ENSG00000196...   \n",
       "4   -|frameshift_variant|HIGH|ZKSCAN7|ENSG00000196...   \n",
       "5   -|intron_variant&non_coding_transcript_variant...   \n",
       "6   -|upstream_gene_variant|MODIFIER|ZNF35|ENSG000...   \n",
       "7                                                None   \n",
       "8                                                None   \n",
       "9   G|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "10  T|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "11  G|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "12  T|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "13  G|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "14                                               None   \n",
       "15  T|intron_variant&NMD_transcript_variant|MODIFI...   \n",
       "16  G|splice_polypyrimidine_tract_variant&intron_v...   \n",
       "17  T|synonymous_variant&NMD_transcript_variant|LO...   \n",
       "\n",
       "                                                   8   \\\n",
       "0   T|downstream_gene_variant|MODIFIER|SNRK-AS1|EN...   \n",
       "1                                                None   \n",
       "2   -|intron_variant&non_coding_transcript_variant...   \n",
       "3   -|intron_variant&non_coding_transcript_variant...   \n",
       "4   -|downstream_gene_variant|MODIFIER|MPRIPP1|ENS...   \n",
       "5                                                None   \n",
       "6   -|upstream_gene_variant|MODIFIER|ZNF35|ENSG000...   \n",
       "7                                                None   \n",
       "8                                                None   \n",
       "9                                                None   \n",
       "10                                               None   \n",
       "11                                               None   \n",
       "12                                               None   \n",
       "13                                               None   \n",
       "14                                               None   \n",
       "15                                               None   \n",
       "16  G|splice_polypyrimidine_tract_variant&intron_v...   \n",
       "17  T|3_prime_UTR_variant&NMD_transcript_variant|M...   \n",
       "\n",
       "                                                   9   \\\n",
       "0   T|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "1                                                None   \n",
       "2   -|intron_variant&non_coding_transcript_variant...   \n",
       "3   -|non_coding_transcript_exon_variant|MODIFIER|...   \n",
       "4   -|intron_variant&non_coding_transcript_variant...   \n",
       "5                                                None   \n",
       "6   -|upstream_gene_variant|MODIFIER|ZNF35|ENSG000...   \n",
       "7                                                None   \n",
       "8                                                None   \n",
       "9                                                None   \n",
       "10                                               None   \n",
       "11                                               None   \n",
       "12                                               None   \n",
       "13                                               None   \n",
       "14                                               None   \n",
       "15                                               None   \n",
       "16                                               None   \n",
       "17                                               None   \n",
       "\n",
       "                                                   10  \\\n",
       "0                                                None   \n",
       "1                                                None   \n",
       "2   -|regulatory_region_variant|MODIFIER|||Regulat...   \n",
       "3                                                None   \n",
       "4   -|downstream_gene_variant|MODIFIER|ZKSCAN7|ENS...   \n",
       "5                                                None   \n",
       "6   -|upstream_gene_variant|MODIFIER|ZNF35|ENSG000...   \n",
       "7                                                None   \n",
       "8                                                None   \n",
       "9                                                None   \n",
       "10                                               None   \n",
       "11                                               None   \n",
       "12                                               None   \n",
       "13                                               None   \n",
       "14                                               None   \n",
       "15                                               None   \n",
       "16                                               None   \n",
       "17                                               None   \n",
       "\n",
       "                                                   11  \n",
       "0                                                None  \n",
       "1                                                None  \n",
       "2                                                None  \n",
       "3                                                None  \n",
       "4                                                None  \n",
       "5                                                None  \n",
       "6   -|intron_variant&non_coding_transcript_variant...  \n",
       "7                                                None  \n",
       "8                                                None  \n",
       "9                                                None  \n",
       "10                                               None  \n",
       "11                                               None  \n",
       "12                                               None  \n",
       "13                                               None  \n",
       "14                                               None  \n",
       "15                                               None  \n",
       "16                                               None  \n",
       "17                                               None  "
      ]
     },
     "execution_count": 133,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info_columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "id": "7ac9c877-97c8-4eea-8096-3f9e32094db0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>73</th>\n",
       "      <th>74</th>\n",
       "      <th>75</th>\n",
       "      <th>76</th>\n",
       "      <th>77</th>\n",
       "      <th>78</th>\n",
       "      <th>79</th>\n",
       "      <th>80</th>\n",
       "      <th>81</th>\n",
       "      <th>82</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>T</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SNRK-AS1</td>\n",
       "      <td>ENSG00000234617</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000422681</td>\n",
       "      <td>lncRNA</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>19.04</td>\n",
       "      <td>1.970863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>-</td>\n",
       "      <td>3_prime_UTR_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>TCAIM</td>\n",
       "      <td>ENSG00000179152</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000383746</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>4/4</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>ZKSCAN7</td>\n",
       "      <td>ENSG00000196345</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000341840</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>5/6</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>HC</td>\n",
       "      <td></td>\n",
       "      <td>PHYLOCSF_WEAK</td>\n",
       "      <td>PERCENTILE:0.904933814681107&amp;GERP_DIST:-1.1370...</td>\n",
       "      <td>17.44</td>\n",
       "      <td>1.759156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>-</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>ZKSCAN7</td>\n",
       "      <td>ENSG00000196345</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000341840</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>5/5</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>ZNF197</td>\n",
       "      <td>ENSG00000186448</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000344387</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>3/6</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>HC</td>\n",
       "      <td></td>\n",
       "      <td>PHYLOCSF_WEAK</td>\n",
       "      <td>PERCENTILE:0.142071197411003&amp;GERP_DIST:-760.75...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>-</td>\n",
       "      <td>intron_variant&amp;NMD_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>ZNF197</td>\n",
       "      <td>ENSG00000186448</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000334075</td>\n",
       "      <td>nonsense_mediated_decay</td>\n",
       "      <td></td>\n",
       "      <td>6/6</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>-</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>KIF15</td>\n",
       "      <td>ENSG00000163808</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000326047</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>6.635</td>\n",
       "      <td>0.512506</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>T</td>\n",
       "      <td>regulatory_region_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>RegulatoryFeature</td>\n",
       "      <td>ENSR00000151685</td>\n",
       "      <td>promoter</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>20.8</td>\n",
       "      <td>2.185077</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>G</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414984</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>2/19</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>7.541</td>\n",
       "      <td>0.602889</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>T</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414984</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>3/19</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>3.950</td>\n",
       "      <td>0.270969</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>G</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414984</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>4/19</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.641</td>\n",
       "      <td>-0.153895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>T</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414984</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>4/19</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.785</td>\n",
       "      <td>-0.111623</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>G</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414984</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>4/19</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.039</td>\n",
       "      <td>-0.720353</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>C</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414984</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>6/19</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.612</td>\n",
       "      <td>-0.163293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>T</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414984</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>10/19</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>0.144</td>\n",
       "      <td>-0.459438</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>G</td>\n",
       "      <td>splice_polypyrimidine_tract_variant&amp;intron_var...</td>\n",
       "      <td>LOW</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414984</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>11/19</td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>8.892</td>\n",
       "      <td>0.747722</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>T</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000414984</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>12/20</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>7.467</td>\n",
       "      <td>0.595288</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18 rows × 83 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      0                                                  1         2   \\\n",
       "0      T                            downstream_gene_variant  MODIFIER   \n",
       "1   None                                               None      None   \n",
       "2      -                                3_prime_UTR_variant  MODIFIER   \n",
       "3      -                                 frameshift_variant      HIGH   \n",
       "4      -                                     intron_variant  MODIFIER   \n",
       "5      -                                 frameshift_variant      HIGH   \n",
       "6      -              intron_variant&NMD_transcript_variant  MODIFIER   \n",
       "7      -                              upstream_gene_variant  MODIFIER   \n",
       "8      T                          regulatory_region_variant  MODIFIER   \n",
       "9      G                                     intron_variant  MODIFIER   \n",
       "10     T                                     intron_variant  MODIFIER   \n",
       "11     G                                     intron_variant  MODIFIER   \n",
       "12     T                                     intron_variant  MODIFIER   \n",
       "13     G                                     intron_variant  MODIFIER   \n",
       "14     C                                     intron_variant  MODIFIER   \n",
       "15     T                                     intron_variant  MODIFIER   \n",
       "16     G  splice_polypyrimidine_tract_variant&intron_var...       LOW   \n",
       "17     T                                 synonymous_variant       LOW   \n",
       "\n",
       "          3                4                  5                6   \\\n",
       "0   SNRK-AS1  ENSG00000234617         Transcript  ENST00000422681   \n",
       "1       None             None               None             None   \n",
       "2      TCAIM  ENSG00000179152         Transcript  ENST00000383746   \n",
       "3    ZKSCAN7  ENSG00000196345         Transcript  ENST00000341840   \n",
       "4    ZKSCAN7  ENSG00000196345         Transcript  ENST00000341840   \n",
       "5     ZNF197  ENSG00000186448         Transcript  ENST00000344387   \n",
       "6     ZNF197  ENSG00000186448         Transcript  ENST00000334075   \n",
       "7      KIF15  ENSG00000163808         Transcript  ENST00000326047   \n",
       "8                              RegulatoryFeature  ENSR00000151685   \n",
       "9      LARS2  ENSG00000011376         Transcript  ENST00000414984   \n",
       "10     LARS2  ENSG00000011376         Transcript  ENST00000414984   \n",
       "11     LARS2  ENSG00000011376         Transcript  ENST00000414984   \n",
       "12     LARS2  ENSG00000011376         Transcript  ENST00000414984   \n",
       "13     LARS2  ENSG00000011376         Transcript  ENST00000414984   \n",
       "14     LARS2  ENSG00000011376         Transcript  ENST00000414984   \n",
       "15     LARS2  ENSG00000011376         Transcript  ENST00000414984   \n",
       "16     LARS2  ENSG00000011376         Transcript  ENST00000414984   \n",
       "17     LARS2  ENSG00000011376         Transcript  ENST00000414984   \n",
       "\n",
       "                         7      8      9   ...    73    74    75    76    77  \\\n",
       "0                    lncRNA                ...                                 \n",
       "1                      None   None   None  ...  None  None  None  None  None   \n",
       "2            protein_coding    4/4         ...                                 \n",
       "3            protein_coding    5/6         ...                            HC   \n",
       "4            protein_coding           5/5  ...                                 \n",
       "5            protein_coding    3/6         ...                            HC   \n",
       "6   nonsense_mediated_decay           6/6  ...                                 \n",
       "7            protein_coding                ...                                 \n",
       "8                  promoter                ...                                 \n",
       "9            protein_coding          2/19  ...                                 \n",
       "10           protein_coding          3/19  ...                                 \n",
       "11           protein_coding          4/19  ...                                 \n",
       "12           protein_coding          4/19  ...                                 \n",
       "13           protein_coding          4/19  ...                                 \n",
       "14           protein_coding          6/19  ...                                 \n",
       "15           protein_coding         10/19  ...                                 \n",
       "16           protein_coding         11/19  ...                                 \n",
       "17           protein_coding  12/20         ...                                 \n",
       "\n",
       "      78             79                                                 80  \\\n",
       "0                                                                            \n",
       "1   None           None                                               None   \n",
       "2                                                                            \n",
       "3         PHYLOCSF_WEAK  PERCENTILE:0.904933814681107&GERP_DIST:-1.1370...   \n",
       "4                                                                            \n",
       "5         PHYLOCSF_WEAK  PERCENTILE:0.142071197411003&GERP_DIST:-760.75...   \n",
       "6                                                                            \n",
       "7                                                                            \n",
       "8                                                                            \n",
       "9                                                                            \n",
       "10                                                                           \n",
       "11                                                                           \n",
       "12                                                                           \n",
       "13                                                                           \n",
       "14                                                                           \n",
       "15                                                                           \n",
       "16                                                                           \n",
       "17                                                                           \n",
       "\n",
       "       81         82  \n",
       "0   19.04   1.970863  \n",
       "1    None       None  \n",
       "2                     \n",
       "3   17.44   1.759156  \n",
       "4                     \n",
       "5                     \n",
       "6                     \n",
       "7   6.635   0.512506  \n",
       "8    20.8   2.185077  \n",
       "9   7.541   0.602889  \n",
       "10  3.950   0.270969  \n",
       "11  0.641  -0.153895  \n",
       "12  0.785  -0.111623  \n",
       "13  0.039  -0.720353  \n",
       "14  0.612  -0.163293  \n",
       "15  0.144  -0.459438  \n",
       "16  8.892   0.747722  \n",
       "17  7.467   0.595288  \n",
       "\n",
       "[18 rows x 83 columns]"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info_columns[1].str.split('|', expand=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 114,
   "id": "b7c5b9b5-a1dc-4c65-b4cb-6574cb2e9ab2",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Allele',\n",
       " 'Consequence',\n",
       " 'IMPACT',\n",
       " 'SYMBOL',\n",
       " 'Gene',\n",
       " 'Feature_type',\n",
       " 'Feature',\n",
       " 'BIOTYPE',\n",
       " 'EXON',\n",
       " 'INTRON',\n",
       " 'HGVSc',\n",
       " 'HGVSp',\n",
       " 'cDNA_position',\n",
       " 'CDS_position',\n",
       " 'Protein_position',\n",
       " 'Amino_acids',\n",
       " 'Codons',\n",
       " 'Existing_variation',\n",
       " 'DISTANCE',\n",
       " 'STRAND',\n",
       " 'FLAGS',\n",
       " 'VARIANT_CLASS',\n",
       " 'SYMBOL_SOURCE',\n",
       " 'HGNC_ID',\n",
       " 'CANONICAL',\n",
       " 'MANE_SELECT',\n",
       " 'MANE_PLUS_CLINICAL',\n",
       " 'TSL',\n",
       " 'APPRIS',\n",
       " 'CCDS',\n",
       " 'ENSP',\n",
       " 'SWISSPROT',\n",
       " 'TREMBL',\n",
       " 'UNIPARC',\n",
       " 'UNIPROT_ISOFORM',\n",
       " 'GENE_PHENO',\n",
       " 'SIFT',\n",
       " 'PolyPhen',\n",
       " 'DOMAINS',\n",
       " 'miRNA',\n",
       " 'AF',\n",
       " 'AFR_AF',\n",
       " 'AMR_AF',\n",
       " 'EAS_AF',\n",
       " 'EUR_AF',\n",
       " 'SAS_AF',\n",
       " 'gnomADe_AF',\n",
       " 'gnomADe_AFR_AF',\n",
       " 'gnomADe_AMR_AF',\n",
       " 'gnomADe_ASJ_AF',\n",
       " 'gnomADe_EAS_AF',\n",
       " 'gnomADe_FIN_AF',\n",
       " 'gnomADe_NFE_AF',\n",
       " 'gnomADe_OTH_AF',\n",
       " 'gnomADe_SAS_AF',\n",
       " 'gnomADg_AF',\n",
       " 'gnomADg_AFR_AF',\n",
       " 'gnomADg_AMI_AF',\n",
       " 'gnomADg_AMR_AF',\n",
       " 'gnomADg_ASJ_AF',\n",
       " 'gnomADg_EAS_AF',\n",
       " 'gnomADg_FIN_AF',\n",
       " 'gnomADg_MID_AF',\n",
       " 'gnomADg_NFE_AF',\n",
       " 'gnomADg_OTH_AF',\n",
       " 'gnomADg_SAS_AF',\n",
       " 'MAX_AF',\n",
       " 'MAX_AF_POPS',\n",
       " 'CLIN_SIG',\n",
       " 'SOMATIC',\n",
       " 'PHENO',\n",
       " 'PUBMED',\n",
       " 'MOTIF_NAME',\n",
       " 'MOTIF_POS',\n",
       " 'HIGH_INF_POS',\n",
       " 'MOTIF_SCORE_CHANGE',\n",
       " 'TRANSCRIPTION_FACTORS',\n",
       " 'LoF',\n",
       " 'LoF_filter',\n",
       " 'LoF_flags',\n",
       " 'LoF_info',\n",
       " 'CADD_PHRED',\n",
       " 'CADD_RAW\">']"
      ]
     },
     "execution_count": 114,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info_header[1].split(\"|\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 115,
   "id": "cafbe910-f877-4631-b763-6cbb308b5eb3",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Split the INFO column on the pipe ('|') delimiter and expand into new columns\n",
    "info_columns = df['INFO'].str.split('|', expand=True)\n",
    "#info_columns.columns = info_header.split('|')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "id": "607d9274-4051-4f2a-9a45-256d96ba80a0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>975</th>\n",
       "      <th>976</th>\n",
       "      <th>977</th>\n",
       "      <th>978</th>\n",
       "      <th>979</th>\n",
       "      <th>980</th>\n",
       "      <th>981</th>\n",
       "      <th>982</th>\n",
       "      <th>983</th>\n",
       "      <th>984</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>AF=5e-06;AQ=31;AC=0;AN=4;CSQ=T</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>SNRK</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000296088</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>6/6</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AF=7e-06;AQ=54;AC=0;AN=6;CSQ=-</td>\n",
       "      <td>inframe_deletion</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TOPAZ1</td>\n",
       "      <td>ENSG00000173769</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000309765</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>2/20</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AF=2e-06;AQ=57;AC=0;AN=6;CSQ=-</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>TCAIM</td>\n",
       "      <td>ENSG00000179152</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000342649</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>3/10</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AF=0.000214;AQ=58;AC=0;AN=6;CSQ=-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>ZKSCAN7</td>\n",
       "      <td>ENSG00000196345</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000273320</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>5/6</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AF=1.5e-05;AQ=48;AC=0;AN=6;CSQ=-</td>\n",
       "      <td>frameshift_variant</td>\n",
       "      <td>HIGH</td>\n",
       "      <td>ZKSCAN7</td>\n",
       "      <td>ENSG00000196345</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000273320</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>6/6</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AF=5e-06;AQ=41;AC=0;AN=6;CSQ=-</td>\n",
       "      <td>3_prime_UTR_variant&amp;NMD_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>ZNF197</td>\n",
       "      <td>ENSG00000186448</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000334075</td>\n",
       "      <td>nonsense_mediated_decay</td>\n",
       "      <td>4/7</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>AF=2e-06;AQ=51;AC=0;AN=6;CSQ=-</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>ZNF35</td>\n",
       "      <td>ENSG00000169981</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000296092</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "      <td></td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>AF=7e-06;AQ=48;AC=0;AN=6;CSQ=-</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>KIAA1143</td>\n",
       "      <td>ENSG00000163807</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000296121</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td></td>\n",
       "      <td>2/2</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>AF=5e-06;AQ=48;AC=0;AN=20;CSQ=T</td>\n",
       "      <td>missense_variant</td>\n",
       "      <td>MODERATE</td>\n",
       "      <td>TMEM158</td>\n",
       "      <td>ENSG00000249992</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000503771</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>1/1</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>AF=7e-06;AQ=47;AC=0;AN=20;CSQ=G</td>\n",
       "      <td>intron_variant&amp;NMD_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000265537</td>\n",
       "      <td>nonsense_mediated_decay</td>\n",
       "      <td></td>\n",
       "      <td>4/22</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>AF=0.535414;AQ=56;AC=17;AN=20;CSQ=T</td>\n",
       "      <td>intron_variant&amp;NMD_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000265537</td>\n",
       "      <td>nonsense_mediated_decay</td>\n",
       "      <td></td>\n",
       "      <td>5/22</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>AF=1.2e-05;AQ=48;AC=0;AN=20;CSQ=G</td>\n",
       "      <td>intron_variant&amp;NMD_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000265537</td>\n",
       "      <td>nonsense_mediated_decay</td>\n",
       "      <td></td>\n",
       "      <td>6/22</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>AF=0.000122;AQ=54;AC=0;AN=20;CSQ=T</td>\n",
       "      <td>intron_variant&amp;NMD_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000265537</td>\n",
       "      <td>nonsense_mediated_decay</td>\n",
       "      <td></td>\n",
       "      <td>6/22</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>AF=2e-06;AQ=44;AC=0;AN=20;CSQ=G</td>\n",
       "      <td>intron_variant&amp;NMD_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000265537</td>\n",
       "      <td>nonsense_mediated_decay</td>\n",
       "      <td></td>\n",
       "      <td>6/22</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>AF=2.7e-05;AQ=50;AC=0;AN=20;CSQ=C</td>\n",
       "      <td>intron_variant&amp;NMD_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000265537</td>\n",
       "      <td>nonsense_mediated_decay</td>\n",
       "      <td></td>\n",
       "      <td>8/22</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>AF=2e-06;AQ=39;AC=0;AN=20;CSQ=T</td>\n",
       "      <td>intron_variant&amp;NMD_transcript_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000265537</td>\n",
       "      <td>nonsense_mediated_decay</td>\n",
       "      <td></td>\n",
       "      <td>12/22</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>AF=2e-06;AQ=44;AC=0;AN=20;CSQ=G</td>\n",
       "      <td>splice_polypyrimidine_tract_variant&amp;intron_var...</td>\n",
       "      <td>LOW</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000265537</td>\n",
       "      <td>nonsense_mediated_decay</td>\n",
       "      <td></td>\n",
       "      <td>13/22</td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>AF=2e-06;AQ=41;AC=0;AN=20;CSQ=T</td>\n",
       "      <td>synonymous_variant&amp;NMD_transcript_variant</td>\n",
       "      <td>LOW</td>\n",
       "      <td>LARS2</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>ENST00000265537</td>\n",
       "      <td>nonsense_mediated_decay</td>\n",
       "      <td>14/23</td>\n",
       "      <td></td>\n",
       "      <td>...</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>18 rows × 985 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                    0    \\\n",
       "0        AF=5e-06;AQ=31;AC=0;AN=4;CSQ=T   \n",
       "1        AF=7e-06;AQ=54;AC=0;AN=6;CSQ=-   \n",
       "2        AF=2e-06;AQ=57;AC=0;AN=6;CSQ=-   \n",
       "3     AF=0.000214;AQ=58;AC=0;AN=6;CSQ=-   \n",
       "4      AF=1.5e-05;AQ=48;AC=0;AN=6;CSQ=-   \n",
       "5        AF=5e-06;AQ=41;AC=0;AN=6;CSQ=-   \n",
       "6        AF=2e-06;AQ=51;AC=0;AN=6;CSQ=-   \n",
       "7        AF=7e-06;AQ=48;AC=0;AN=6;CSQ=-   \n",
       "8       AF=5e-06;AQ=48;AC=0;AN=20;CSQ=T   \n",
       "9       AF=7e-06;AQ=47;AC=0;AN=20;CSQ=G   \n",
       "10  AF=0.535414;AQ=56;AC=17;AN=20;CSQ=T   \n",
       "11    AF=1.2e-05;AQ=48;AC=0;AN=20;CSQ=G   \n",
       "12   AF=0.000122;AQ=54;AC=0;AN=20;CSQ=T   \n",
       "13      AF=2e-06;AQ=44;AC=0;AN=20;CSQ=G   \n",
       "14    AF=2.7e-05;AQ=50;AC=0;AN=20;CSQ=C   \n",
       "15      AF=2e-06;AQ=39;AC=0;AN=20;CSQ=T   \n",
       "16      AF=2e-06;AQ=44;AC=0;AN=20;CSQ=G   \n",
       "17      AF=2e-06;AQ=41;AC=0;AN=20;CSQ=T   \n",
       "\n",
       "                                                  1         2         3    \\\n",
       "0                                      intron_variant  MODIFIER      SNRK   \n",
       "1                                    inframe_deletion  MODERATE    TOPAZ1   \n",
       "2                                      intron_variant  MODIFIER     TCAIM   \n",
       "3                                  frameshift_variant      HIGH   ZKSCAN7   \n",
       "4                                  frameshift_variant      HIGH   ZKSCAN7   \n",
       "5          3_prime_UTR_variant&NMD_transcript_variant  MODIFIER    ZNF197   \n",
       "6                               upstream_gene_variant  MODIFIER     ZNF35   \n",
       "7                                      intron_variant  MODIFIER  KIAA1143   \n",
       "8                                    missense_variant  MODERATE   TMEM158   \n",
       "9               intron_variant&NMD_transcript_variant  MODIFIER     LARS2   \n",
       "10              intron_variant&NMD_transcript_variant  MODIFIER     LARS2   \n",
       "11              intron_variant&NMD_transcript_variant  MODIFIER     LARS2   \n",
       "12              intron_variant&NMD_transcript_variant  MODIFIER     LARS2   \n",
       "13              intron_variant&NMD_transcript_variant  MODIFIER     LARS2   \n",
       "14              intron_variant&NMD_transcript_variant  MODIFIER     LARS2   \n",
       "15              intron_variant&NMD_transcript_variant  MODIFIER     LARS2   \n",
       "16  splice_polypyrimidine_tract_variant&intron_var...       LOW     LARS2   \n",
       "17          synonymous_variant&NMD_transcript_variant       LOW     LARS2   \n",
       "\n",
       "                4           5                6                        7    \\\n",
       "0   ENSG00000163788  Transcript  ENST00000296088           protein_coding   \n",
       "1   ENSG00000173769  Transcript  ENST00000309765           protein_coding   \n",
       "2   ENSG00000179152  Transcript  ENST00000342649           protein_coding   \n",
       "3   ENSG00000196345  Transcript  ENST00000273320           protein_coding   \n",
       "4   ENSG00000196345  Transcript  ENST00000273320           protein_coding   \n",
       "5   ENSG00000186448  Transcript  ENST00000334075  nonsense_mediated_decay   \n",
       "6   ENSG00000169981  Transcript  ENST00000296092           protein_coding   \n",
       "7   ENSG00000163807  Transcript  ENST00000296121           protein_coding   \n",
       "8   ENSG00000249992  Transcript  ENST00000503771           protein_coding   \n",
       "9   ENSG00000011376  Transcript  ENST00000265537  nonsense_mediated_decay   \n",
       "10  ENSG00000011376  Transcript  ENST00000265537  nonsense_mediated_decay   \n",
       "11  ENSG00000011376  Transcript  ENST00000265537  nonsense_mediated_decay   \n",
       "12  ENSG00000011376  Transcript  ENST00000265537  nonsense_mediated_decay   \n",
       "13  ENSG00000011376  Transcript  ENST00000265537  nonsense_mediated_decay   \n",
       "14  ENSG00000011376  Transcript  ENST00000265537  nonsense_mediated_decay   \n",
       "15  ENSG00000011376  Transcript  ENST00000265537  nonsense_mediated_decay   \n",
       "16  ENSG00000011376  Transcript  ENST00000265537  nonsense_mediated_decay   \n",
       "17  ENSG00000011376  Transcript  ENST00000265537  nonsense_mediated_decay   \n",
       "\n",
       "      8      9    ...   975   976   977   978   979   980   981   982   983  \\\n",
       "0            6/6  ...  None  None  None  None  None  None  None  None  None   \n",
       "1    2/20         ...  None  None  None  None  None  None  None  None  None   \n",
       "2           3/10  ...  None  None  None  None  None  None  None  None  None   \n",
       "3     5/6         ...  None  None  None  None  None  None  None  None  None   \n",
       "4     6/6         ...  None  None  None  None  None  None  None  None  None   \n",
       "5     4/7         ...  None  None  None  None  None  None  None  None  None   \n",
       "6                 ...                                                         \n",
       "7            2/2  ...  None  None  None  None  None  None  None  None  None   \n",
       "8     1/1         ...  None  None  None  None  None  None  None  None  None   \n",
       "9           4/22  ...  None  None  None  None  None  None  None  None  None   \n",
       "10          5/22  ...  None  None  None  None  None  None  None  None  None   \n",
       "11          6/22  ...  None  None  None  None  None  None  None  None  None   \n",
       "12          6/22  ...  None  None  None  None  None  None  None  None  None   \n",
       "13          6/22  ...  None  None  None  None  None  None  None  None  None   \n",
       "14          8/22  ...  None  None  None  None  None  None  None  None  None   \n",
       "15         12/22  ...  None  None  None  None  None  None  None  None  None   \n",
       "16         13/22  ...  None  None  None  None  None  None  None  None  None   \n",
       "17  14/23         ...  None  None  None  None  None  None  None  None  None   \n",
       "\n",
       "     984  \n",
       "0   None  \n",
       "1   None  \n",
       "2   None  \n",
       "3   None  \n",
       "4   None  \n",
       "5   None  \n",
       "6         \n",
       "7   None  \n",
       "8   None  \n",
       "9   None  \n",
       "10  None  \n",
       "11  None  \n",
       "12  None  \n",
       "13  None  \n",
       "14  None  \n",
       "15  None  \n",
       "16  None  \n",
       "17  None  \n",
       "\n",
       "[18 rows x 985 columns]"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "info_columns"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "63997f8e-7542-4937-bb15-e406f13aa5c7",
   "metadata": {},
   "source": [
    "## Using the tab option for VEP output"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "cb67fe8a-1ee2-4675-8198-e98e9a65761f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import gzip\n",
    "vcf_file = \"/home/dmc2245/test/bcftools/snps_indels_vep_test_sorted.VEP.CADD_gnomAD.tsv.gz\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "ec1dc0a3-a85a-4d97-a1b0-a9f8bc0d4a34",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "df = pd.read_csv(vcf_file, comment='#', sep='\\t', header=None)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "6114f930-a61c-4081-8140-af9d30a05d5d",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "      <th>1</th>\n",
       "      <th>2</th>\n",
       "      <th>3</th>\n",
       "      <th>4</th>\n",
       "      <th>5</th>\n",
       "      <th>6</th>\n",
       "      <th>7</th>\n",
       "      <th>8</th>\n",
       "      <th>9</th>\n",
       "      <th>...</th>\n",
       "      <th>86</th>\n",
       "      <th>87</th>\n",
       "      <th>88</th>\n",
       "      <th>89</th>\n",
       "      <th>90</th>\n",
       "      <th>91</th>\n",
       "      <th>92</th>\n",
       "      <th>93</th>\n",
       "      <th>94</th>\n",
       "      <th>95</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000296088</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000234617</td>\n",
       "      <td>ENST00000422681</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000429705</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000437827</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000454177</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000642274</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1912</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000645846</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1746</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000650792</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1695</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000651549</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant,NMD_transcript_variant</td>\n",
       "      <td>1680</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000652135</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>3_prime_UTR_variant,NMD_transcript_variant</td>\n",
       "      <td>1845</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>143 rows × 96 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  0                    1  2                3   \\\n",
       "0    3:43343506:G:GT  3:43343506-43343507  T  ENSG00000163788   \n",
       "1    3:43343506:G:GT  3:43343506-43343507  T  ENSG00000234617   \n",
       "2    3:43343506:G:GT  3:43343506-43343507  T  ENSG00000163788   \n",
       "3    3:43343506:G:GT  3:43343506-43343507  T  ENSG00000163788   \n",
       "4    3:43343506:G:GT  3:43343506-43343507  T  ENSG00000163788   \n",
       "..               ...                  ... ..              ...   \n",
       "138   3:45496305:C:T           3:45496305  T  ENSG00000011376   \n",
       "139   3:45496305:C:T           3:45496305  T  ENSG00000011376   \n",
       "140   3:45496305:C:T           3:45496305  T  ENSG00000011376   \n",
       "141   3:45496305:C:T           3:45496305  T  ENSG00000011376   \n",
       "142   3:45496305:C:T           3:45496305  T  ENSG00000011376   \n",
       "\n",
       "                  4           5                                           6   \\\n",
       "0    ENST00000296088  Transcript                              intron_variant   \n",
       "1    ENST00000422681  Transcript                     downstream_gene_variant   \n",
       "2    ENST00000429705  Transcript                              intron_variant   \n",
       "3    ENST00000437827  Transcript                              intron_variant   \n",
       "4    ENST00000454177  Transcript                              intron_variant   \n",
       "..               ...         ...                                         ...   \n",
       "138  ENST00000642274  Transcript                          synonymous_variant   \n",
       "139  ENST00000645846  Transcript                          synonymous_variant   \n",
       "140  ENST00000650792  Transcript                          synonymous_variant   \n",
       "141  ENST00000651549  Transcript   synonymous_variant,NMD_transcript_variant   \n",
       "142  ENST00000652135  Transcript  3_prime_UTR_variant,NMD_transcript_variant   \n",
       "\n",
       "       7     8    9   ... 86 87           88 89 90 91           92  \\\n",
       "0       -     -    -  ...  -  -  rs781680268  0  0  0  4.58211e-05   \n",
       "1       -     -    -  ...  -  -  rs781680268  0  0  0  4.58211e-05   \n",
       "2       -     -    -  ...  -  -  rs781680268  0  0  0  4.58211e-05   \n",
       "3       -     -    -  ...  -  -  rs781680268  0  0  0  4.58211e-05   \n",
       "4       -     -    -  ...  -  -  rs781680268  0  0  0  4.58211e-05   \n",
       "..    ...   ...  ...  ... .. ..          ... .. .. ..          ...   \n",
       "138  1912  1554  518  ...  -  -            -  -  -  -            -   \n",
       "139  1746  1554  518  ...  -  -            -  -  -  -            -   \n",
       "140  1695  1554  518  ...  -  -            -  -  -  -            -   \n",
       "141  1680  1554  518  ...  -  -            -  -  -  -            -   \n",
       "142  1845     -    -  ...  -  -            -  -  -  -            -   \n",
       "\n",
       "              93 94 95  \n",
       "0    0.000113296  0  0  \n",
       "1    0.000113296  0  0  \n",
       "2    0.000113296  0  0  \n",
       "3    0.000113296  0  0  \n",
       "4    0.000113296  0  0  \n",
       "..           ... .. ..  \n",
       "138            -  -  -  \n",
       "139            -  -  -  \n",
       "140            -  -  -  \n",
       "141            -  -  -  \n",
       "142            -  -  -  \n",
       "\n",
       "[143 rows x 96 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "a9150cc9-8658-43b0-af94-c1a846ec22e6",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Initialize an empty list to store the comments\n",
    "comments = []\n",
    "\n",
    "# Open the gzip-compressed VCF file and read it line by line\n",
    "with gzip.open(vcf_file, 'rt') as file:\n",
    "    for line in file:\n",
    "        if line.startswith('#'):\n",
    "            comments.append(line.strip())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "2c6fda45-785c-4d33-87d1-cbd6b866884f",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# The last item in the 'comments' list is the header comment\n",
    "header_comment = comments[-1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "d8bd7ad8-7fe4-426a-9995-555766305aa0",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'#Uploaded_variation\\tLocation\\tAllele\\tGene\\tFeature\\tFeature_type\\tConsequence\\tcDNA_position\\tCDS_position\\tProtein_position\\tAmino_acids\\tCodons\\tExisting_variation\\tIMPACT\\tDISTANCE\\tSTRAND\\tFLAGS\\tVARIANT_CLASS\\tSYMBOL\\tSYMBOL_SOURCE\\tHGNC_ID\\tBIOTYPE\\tCANONICAL\\tMANE_SELECT\\tMANE_PLUS_CLINICAL\\tTSL\\tAPPRIS\\tCCDS\\tENSP\\tSWISSPROT\\tTREMBL\\tUNIPARC\\tUNIPROT_ISOFORM\\tSOURCE\\tGENE_PHENO\\tSIFT\\tPolyPhen\\tEXON\\tINTRON\\tDOMAINS\\tmiRNA\\tAF\\tAFR_AF\\tAMR_AF\\tEAS_AF\\tEUR_AF\\tSAS_AF\\tgnomADe_AF\\tgnomADe_AFR_AF\\tgnomADe_AMR_AF\\tgnomADe_ASJ_AF\\tgnomADe_EAS_AF\\tgnomADe_FIN_AF\\tgnomADe_NFE_AF\\tgnomADe_OTH_AF\\tgnomADe_SAS_AF\\tgnomADg_AF\\tgnomADg_AFR_AF\\tgnomADg_AMI_AF\\tgnomADg_AMR_AF\\tgnomADg_ASJ_AF\\tgnomADg_EAS_AF\\tgnomADg_FIN_AF\\tgnomADg_MID_AF\\tgnomADg_NFE_AF\\tgnomADg_OTH_AF\\tgnomADg_SAS_AF\\tMAX_AF\\tMAX_AF_POPS\\tCLIN_SIG\\tSOMATIC\\tPHENO\\tPUBMED\\tMOTIF_NAME\\tMOTIF_POS\\tHIGH_INF_POS\\tMOTIF_SCORE_CHANGE\\tTRANSCRIPTION_FACTORS\\tLoF\\tLoF_filter\\tLoF_flags\\tLoF_info\\tCADD_PHRED\\tCADD_RAW\\tClinVar\\tClinVar_CLNSIG\\tClinVar_CLNREVSTAT\\tClinVar_CLNDN\\tgnomADg\\tgnomADg_AF_joint_afr\\tgnomADg_AF_joint_amr\\tgnomADg_AF_joint_asj\\tgnomADg_AF_joint_eas\\tgnomADg_AF_joint_sas\\tgnomADg_AF_joint_fin\\tgnomADg_AF_joint_nfe'"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "header_comment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "d6261b77-e028-45f2-97eb-fd7cb6956e2d",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Extract the header from the comment\n",
    "header = header_comment.split('#')[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "030a78d6-64f4-4191-b22a-f762a55c81cf",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'Uploaded_variation\\tLocation\\tAllele\\tGene\\tFeature\\tFeature_type\\tConsequence\\tcDNA_position\\tCDS_position\\tProtein_position\\tAmino_acids\\tCodons\\tExisting_variation\\tIMPACT\\tDISTANCE\\tSTRAND\\tFLAGS\\tVARIANT_CLASS\\tSYMBOL\\tSYMBOL_SOURCE\\tHGNC_ID\\tBIOTYPE\\tCANONICAL\\tMANE_SELECT\\tMANE_PLUS_CLINICAL\\tTSL\\tAPPRIS\\tCCDS\\tENSP\\tSWISSPROT\\tTREMBL\\tUNIPARC\\tUNIPROT_ISOFORM\\tSOURCE\\tGENE_PHENO\\tSIFT\\tPolyPhen\\tEXON\\tINTRON\\tDOMAINS\\tmiRNA\\tAF\\tAFR_AF\\tAMR_AF\\tEAS_AF\\tEUR_AF\\tSAS_AF\\tgnomADe_AF\\tgnomADe_AFR_AF\\tgnomADe_AMR_AF\\tgnomADe_ASJ_AF\\tgnomADe_EAS_AF\\tgnomADe_FIN_AF\\tgnomADe_NFE_AF\\tgnomADe_OTH_AF\\tgnomADe_SAS_AF\\tgnomADg_AF\\tgnomADg_AFR_AF\\tgnomADg_AMI_AF\\tgnomADg_AMR_AF\\tgnomADg_ASJ_AF\\tgnomADg_EAS_AF\\tgnomADg_FIN_AF\\tgnomADg_MID_AF\\tgnomADg_NFE_AF\\tgnomADg_OTH_AF\\tgnomADg_SAS_AF\\tMAX_AF\\tMAX_AF_POPS\\tCLIN_SIG\\tSOMATIC\\tPHENO\\tPUBMED\\tMOTIF_NAME\\tMOTIF_POS\\tHIGH_INF_POS\\tMOTIF_SCORE_CHANGE\\tTRANSCRIPTION_FACTORS\\tLoF\\tLoF_filter\\tLoF_flags\\tLoF_info\\tCADD_PHRED\\tCADD_RAW\\tClinVar\\tClinVar_CLNSIG\\tClinVar_CLNREVSTAT\\tClinVar_CLNDN\\tgnomADg\\tgnomADg_AF_joint_afr\\tgnomADg_AF_joint_amr\\tgnomADg_AF_joint_asj\\tgnomADg_AF_joint_eas\\tgnomADg_AF_joint_sas\\tgnomADg_AF_joint_fin\\tgnomADg_AF_joint_nfe'"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "header"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "b24455e1-fc47-4056-9237-04a22a68fdba",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Split the header into column names\n",
    "columns = header.split('\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "92a9d498-df12-4020-8cc6-442ca2a6770b",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Uploaded_variation',\n",
       " 'Location',\n",
       " 'Allele',\n",
       " 'Gene',\n",
       " 'Feature',\n",
       " 'Feature_type',\n",
       " 'Consequence',\n",
       " 'cDNA_position',\n",
       " 'CDS_position',\n",
       " 'Protein_position',\n",
       " 'Amino_acids',\n",
       " 'Codons',\n",
       " 'Existing_variation',\n",
       " 'IMPACT',\n",
       " 'DISTANCE',\n",
       " 'STRAND',\n",
       " 'FLAGS',\n",
       " 'VARIANT_CLASS',\n",
       " 'SYMBOL',\n",
       " 'SYMBOL_SOURCE',\n",
       " 'HGNC_ID',\n",
       " 'BIOTYPE',\n",
       " 'CANONICAL',\n",
       " 'MANE_SELECT',\n",
       " 'MANE_PLUS_CLINICAL',\n",
       " 'TSL',\n",
       " 'APPRIS',\n",
       " 'CCDS',\n",
       " 'ENSP',\n",
       " 'SWISSPROT',\n",
       " 'TREMBL',\n",
       " 'UNIPARC',\n",
       " 'UNIPROT_ISOFORM',\n",
       " 'SOURCE',\n",
       " 'GENE_PHENO',\n",
       " 'SIFT',\n",
       " 'PolyPhen',\n",
       " 'EXON',\n",
       " 'INTRON',\n",
       " 'DOMAINS',\n",
       " 'miRNA',\n",
       " 'AF',\n",
       " 'AFR_AF',\n",
       " 'AMR_AF',\n",
       " 'EAS_AF',\n",
       " 'EUR_AF',\n",
       " 'SAS_AF',\n",
       " 'gnomADe_AF',\n",
       " 'gnomADe_AFR_AF',\n",
       " 'gnomADe_AMR_AF',\n",
       " 'gnomADe_ASJ_AF',\n",
       " 'gnomADe_EAS_AF',\n",
       " 'gnomADe_FIN_AF',\n",
       " 'gnomADe_NFE_AF',\n",
       " 'gnomADe_OTH_AF',\n",
       " 'gnomADe_SAS_AF',\n",
       " 'gnomADg_AF',\n",
       " 'gnomADg_AFR_AF',\n",
       " 'gnomADg_AMI_AF',\n",
       " 'gnomADg_AMR_AF',\n",
       " 'gnomADg_ASJ_AF',\n",
       " 'gnomADg_EAS_AF',\n",
       " 'gnomADg_FIN_AF',\n",
       " 'gnomADg_MID_AF',\n",
       " 'gnomADg_NFE_AF',\n",
       " 'gnomADg_OTH_AF',\n",
       " 'gnomADg_SAS_AF',\n",
       " 'MAX_AF',\n",
       " 'MAX_AF_POPS',\n",
       " 'CLIN_SIG',\n",
       " 'SOMATIC',\n",
       " 'PHENO',\n",
       " 'PUBMED',\n",
       " 'MOTIF_NAME',\n",
       " 'MOTIF_POS',\n",
       " 'HIGH_INF_POS',\n",
       " 'MOTIF_SCORE_CHANGE',\n",
       " 'TRANSCRIPTION_FACTORS',\n",
       " 'LoF',\n",
       " 'LoF_filter',\n",
       " 'LoF_flags',\n",
       " 'LoF_info',\n",
       " 'CADD_PHRED',\n",
       " 'CADD_RAW',\n",
       " 'ClinVar',\n",
       " 'ClinVar_CLNSIG',\n",
       " 'ClinVar_CLNREVSTAT',\n",
       " 'ClinVar_CLNDN',\n",
       " 'gnomADg',\n",
       " 'gnomADg_AF_joint_afr',\n",
       " 'gnomADg_AF_joint_amr',\n",
       " 'gnomADg_AF_joint_asj',\n",
       " 'gnomADg_AF_joint_eas',\n",
       " 'gnomADg_AF_joint_sas',\n",
       " 'gnomADg_AF_joint_fin',\n",
       " 'gnomADg_AF_joint_nfe']"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "3c4e38fd-9947-4b64-ade3-8a37f00ea334",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "df = pd.read_csv(vcf_file, comment='#', sep='\\t', names=columns, compression='gzip')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "945e8c45-f437-4284-8e27-3dc0cb27367e",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>...</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>gnomADg</th>\n",
       "      <th>gnomADg_AF_joint_afr</th>\n",
       "      <th>gnomADg_AF_joint_amr</th>\n",
       "      <th>gnomADg_AF_joint_asj</th>\n",
       "      <th>gnomADg_AF_joint_eas</th>\n",
       "      <th>gnomADg_AF_joint_sas</th>\n",
       "      <th>gnomADg_AF_joint_fin</th>\n",
       "      <th>gnomADg_AF_joint_nfe</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000296088</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000234617</td>\n",
       "      <td>ENST00000422681</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000429705</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000437827</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000454177</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000642274</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1912</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000645846</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1746</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000650792</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1695</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000651549</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant,NMD_transcript_variant</td>\n",
       "      <td>1680</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000652135</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>3_prime_UTR_variant,NMD_transcript_variant</td>\n",
       "      <td>1845</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>143 rows × 96 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Uploaded_variation             Location Allele             Gene  \\\n",
       "0      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "1      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000234617   \n",
       "2      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "3      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "4      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "..                 ...                  ...    ...              ...   \n",
       "138     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "139     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "140     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "141     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "142     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "\n",
       "             Feature Feature_type                                 Consequence  \\\n",
       "0    ENST00000296088   Transcript                              intron_variant   \n",
       "1    ENST00000422681   Transcript                     downstream_gene_variant   \n",
       "2    ENST00000429705   Transcript                              intron_variant   \n",
       "3    ENST00000437827   Transcript                              intron_variant   \n",
       "4    ENST00000454177   Transcript                              intron_variant   \n",
       "..               ...          ...                                         ...   \n",
       "138  ENST00000642274   Transcript                          synonymous_variant   \n",
       "139  ENST00000645846   Transcript                          synonymous_variant   \n",
       "140  ENST00000650792   Transcript                          synonymous_variant   \n",
       "141  ENST00000651549   Transcript   synonymous_variant,NMD_transcript_variant   \n",
       "142  ENST00000652135   Transcript  3_prime_UTR_variant,NMD_transcript_variant   \n",
       "\n",
       "    cDNA_position CDS_position Protein_position  ... ClinVar_CLNREVSTAT  \\\n",
       "0               -            -                -  ...                  -   \n",
       "1               -            -                -  ...                  -   \n",
       "2               -            -                -  ...                  -   \n",
       "3               -            -                -  ...                  -   \n",
       "4               -            -                -  ...                  -   \n",
       "..            ...          ...              ...  ...                ...   \n",
       "138          1912         1554              518  ...                  -   \n",
       "139          1746         1554              518  ...                  -   \n",
       "140          1695         1554              518  ...                  -   \n",
       "141          1680         1554              518  ...                  -   \n",
       "142          1845            -                -  ...                  -   \n",
       "\n",
       "    ClinVar_CLNDN      gnomADg gnomADg_AF_joint_afr gnomADg_AF_joint_amr  \\\n",
       "0               -  rs781680268                    0                    0   \n",
       "1               -  rs781680268                    0                    0   \n",
       "2               -  rs781680268                    0                    0   \n",
       "3               -  rs781680268                    0                    0   \n",
       "4               -  rs781680268                    0                    0   \n",
       "..            ...          ...                  ...                  ...   \n",
       "138             -            -                    -                    -   \n",
       "139             -            -                    -                    -   \n",
       "140             -            -                    -                    -   \n",
       "141             -            -                    -                    -   \n",
       "142             -            -                    -                    -   \n",
       "\n",
       "    gnomADg_AF_joint_asj gnomADg_AF_joint_eas gnomADg_AF_joint_sas  \\\n",
       "0                      0          4.58211e-05          0.000113296   \n",
       "1                      0          4.58211e-05          0.000113296   \n",
       "2                      0          4.58211e-05          0.000113296   \n",
       "3                      0          4.58211e-05          0.000113296   \n",
       "4                      0          4.58211e-05          0.000113296   \n",
       "..                   ...                  ...                  ...   \n",
       "138                    -                    -                    -   \n",
       "139                    -                    -                    -   \n",
       "140                    -                    -                    -   \n",
       "141                    -                    -                    -   \n",
       "142                    -                    -                    -   \n",
       "\n",
       "    gnomADg_AF_joint_fin gnomADg_AF_joint_nfe  \n",
       "0                      0                    0  \n",
       "1                      0                    0  \n",
       "2                      0                    0  \n",
       "3                      0                    0  \n",
       "4                      0                    0  \n",
       "..                   ...                  ...  \n",
       "138                    -                    -  \n",
       "139                    -                    -  \n",
       "140                    -                    -  \n",
       "141                    -                    -  \n",
       "142                    -                    -  \n",
       "\n",
       "[143 rows x 96 columns]"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "53b1fb9e-ac65-4760-8e53-894a18a692c4",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Existing_variation</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>rs781680268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>rs781680268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>rs781680268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>rs781680268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>rs781680268</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>143 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Uploaded_variation Existing_variation\n",
       "0      3:43343506:G:GT        rs781680268\n",
       "1      3:43343506:G:GT        rs781680268\n",
       "2      3:43343506:G:GT        rs781680268\n",
       "3      3:43343506:G:GT        rs781680268\n",
       "4      3:43343506:G:GT        rs781680268\n",
       "..                 ...                ...\n",
       "138     3:45496305:C:T                  -\n",
       "139     3:45496305:C:T                  -\n",
       "140     3:45496305:C:T                  -\n",
       "141     3:45496305:C:T                  -\n",
       "142     3:45496305:C:T                  -\n",
       "\n",
       "[143 rows x 2 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[['Uploaded_variation','Existing_variation']]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "f707749a-3feb-483f-91b6-bfac125641ae",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Split the values in 'your_column' and create new columns\n",
    "df[['chromosome', 'position']] = df['Location'].str.split(':', expand=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "afa43f5c-0724-4dc6-862f-2822d1829e0c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomADg</th>\n",
       "      <th>gnomADg_AF_joint_afr</th>\n",
       "      <th>gnomADg_AF_joint_amr</th>\n",
       "      <th>gnomADg_AF_joint_asj</th>\n",
       "      <th>gnomADg_AF_joint_eas</th>\n",
       "      <th>gnomADg_AF_joint_sas</th>\n",
       "      <th>gnomADg_AF_joint_fin</th>\n",
       "      <th>gnomADg_AF_joint_nfe</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000296088</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>43343506-43343507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000234617</td>\n",
       "      <td>ENST00000422681</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>43343506-43343507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000429705</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>43343506-43343507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000437827</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>43343506-43343507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000454177</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>43343506-43343507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000642274</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1912</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000645846</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1746</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000650792</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1695</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000651549</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant,NMD_transcript_variant</td>\n",
       "      <td>1680</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000652135</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>3_prime_UTR_variant,NMD_transcript_variant</td>\n",
       "      <td>1845</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>143 rows × 98 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Uploaded_variation             Location Allele             Gene  \\\n",
       "0      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "1      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000234617   \n",
       "2      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "3      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "4      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "..                 ...                  ...    ...              ...   \n",
       "138     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "139     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "140     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "141     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "142     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "\n",
       "             Feature Feature_type                                 Consequence  \\\n",
       "0    ENST00000296088   Transcript                              intron_variant   \n",
       "1    ENST00000422681   Transcript                     downstream_gene_variant   \n",
       "2    ENST00000429705   Transcript                              intron_variant   \n",
       "3    ENST00000437827   Transcript                              intron_variant   \n",
       "4    ENST00000454177   Transcript                              intron_variant   \n",
       "..               ...          ...                                         ...   \n",
       "138  ENST00000642274   Transcript                          synonymous_variant   \n",
       "139  ENST00000645846   Transcript                          synonymous_variant   \n",
       "140  ENST00000650792   Transcript                          synonymous_variant   \n",
       "141  ENST00000651549   Transcript   synonymous_variant,NMD_transcript_variant   \n",
       "142  ENST00000652135   Transcript  3_prime_UTR_variant,NMD_transcript_variant   \n",
       "\n",
       "    cDNA_position CDS_position Protein_position  ...      gnomADg  \\\n",
       "0               -            -                -  ...  rs781680268   \n",
       "1               -            -                -  ...  rs781680268   \n",
       "2               -            -                -  ...  rs781680268   \n",
       "3               -            -                -  ...  rs781680268   \n",
       "4               -            -                -  ...  rs781680268   \n",
       "..            ...          ...              ...  ...          ...   \n",
       "138          1912         1554              518  ...            -   \n",
       "139          1746         1554              518  ...            -   \n",
       "140          1695         1554              518  ...            -   \n",
       "141          1680         1554              518  ...            -   \n",
       "142          1845            -                -  ...            -   \n",
       "\n",
       "    gnomADg_AF_joint_afr gnomADg_AF_joint_amr gnomADg_AF_joint_asj  \\\n",
       "0                      0                    0                    0   \n",
       "1                      0                    0                    0   \n",
       "2                      0                    0                    0   \n",
       "3                      0                    0                    0   \n",
       "4                      0                    0                    0   \n",
       "..                   ...                  ...                  ...   \n",
       "138                    -                    -                    -   \n",
       "139                    -                    -                    -   \n",
       "140                    -                    -                    -   \n",
       "141                    -                    -                    -   \n",
       "142                    -                    -                    -   \n",
       "\n",
       "    gnomADg_AF_joint_eas gnomADg_AF_joint_sas gnomADg_AF_joint_fin  \\\n",
       "0            4.58211e-05          0.000113296                    0   \n",
       "1            4.58211e-05          0.000113296                    0   \n",
       "2            4.58211e-05          0.000113296                    0   \n",
       "3            4.58211e-05          0.000113296                    0   \n",
       "4            4.58211e-05          0.000113296                    0   \n",
       "..                   ...                  ...                  ...   \n",
       "138                    -                    -                    -   \n",
       "139                    -                    -                    -   \n",
       "140                    -                    -                    -   \n",
       "141                    -                    -                    -   \n",
       "142                    -                    -                    -   \n",
       "\n",
       "    gnomADg_AF_joint_nfe chromosome           position  \n",
       "0                      0          3  43343506-43343507  \n",
       "1                      0          3  43343506-43343507  \n",
       "2                      0          3  43343506-43343507  \n",
       "3                      0          3  43343506-43343507  \n",
       "4                      0          3  43343506-43343507  \n",
       "..                   ...        ...                ...  \n",
       "138                    -          3           45496305  \n",
       "139                    -          3           45496305  \n",
       "140                    -          3           45496305  \n",
       "141                    -          3           45496305  \n",
       "142                    -          3           45496305  \n",
       "\n",
       "[143 rows x 98 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "97539d6d-863b-49f2-9b41-6559efe4d049",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "df[['start', 'end']] = df['position'].str.split('-', expand=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "e704777b-0c71-4b9f-8066-3cb13b7bb91c",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomADg_AF_joint_amr</th>\n",
       "      <th>gnomADg_AF_joint_asj</th>\n",
       "      <th>gnomADg_AF_joint_eas</th>\n",
       "      <th>gnomADg_AF_joint_sas</th>\n",
       "      <th>gnomADg_AF_joint_fin</th>\n",
       "      <th>gnomADg_AF_joint_nfe</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000296088</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>43343506-43343507</td>\n",
       "      <td>43343506</td>\n",
       "      <td>43343507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000234617</td>\n",
       "      <td>ENST00000422681</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>43343506-43343507</td>\n",
       "      <td>43343506</td>\n",
       "      <td>43343507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000429705</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>43343506-43343507</td>\n",
       "      <td>43343506</td>\n",
       "      <td>43343507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000437827</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>43343506-43343507</td>\n",
       "      <td>43343506</td>\n",
       "      <td>43343507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000454177</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>43343506-43343507</td>\n",
       "      <td>43343506</td>\n",
       "      <td>43343507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000642274</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1912</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000645846</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1746</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000650792</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1695</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000651549</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant,NMD_transcript_variant</td>\n",
       "      <td>1680</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000652135</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>3_prime_UTR_variant,NMD_transcript_variant</td>\n",
       "      <td>1845</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>143 rows × 100 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Uploaded_variation             Location Allele             Gene  \\\n",
       "0      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "1      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000234617   \n",
       "2      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "3      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "4      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "..                 ...                  ...    ...              ...   \n",
       "138     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "139     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "140     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "141     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "142     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "\n",
       "             Feature Feature_type                                 Consequence  \\\n",
       "0    ENST00000296088   Transcript                              intron_variant   \n",
       "1    ENST00000422681   Transcript                     downstream_gene_variant   \n",
       "2    ENST00000429705   Transcript                              intron_variant   \n",
       "3    ENST00000437827   Transcript                              intron_variant   \n",
       "4    ENST00000454177   Transcript                              intron_variant   \n",
       "..               ...          ...                                         ...   \n",
       "138  ENST00000642274   Transcript                          synonymous_variant   \n",
       "139  ENST00000645846   Transcript                          synonymous_variant   \n",
       "140  ENST00000650792   Transcript                          synonymous_variant   \n",
       "141  ENST00000651549   Transcript   synonymous_variant,NMD_transcript_variant   \n",
       "142  ENST00000652135   Transcript  3_prime_UTR_variant,NMD_transcript_variant   \n",
       "\n",
       "    cDNA_position CDS_position Protein_position  ... gnomADg_AF_joint_amr  \\\n",
       "0               -            -                -  ...                    0   \n",
       "1               -            -                -  ...                    0   \n",
       "2               -            -                -  ...                    0   \n",
       "3               -            -                -  ...                    0   \n",
       "4               -            -                -  ...                    0   \n",
       "..            ...          ...              ...  ...                  ...   \n",
       "138          1912         1554              518  ...                    -   \n",
       "139          1746         1554              518  ...                    -   \n",
       "140          1695         1554              518  ...                    -   \n",
       "141          1680         1554              518  ...                    -   \n",
       "142          1845            -                -  ...                    -   \n",
       "\n",
       "    gnomADg_AF_joint_asj gnomADg_AF_joint_eas gnomADg_AF_joint_sas  \\\n",
       "0                      0          4.58211e-05          0.000113296   \n",
       "1                      0          4.58211e-05          0.000113296   \n",
       "2                      0          4.58211e-05          0.000113296   \n",
       "3                      0          4.58211e-05          0.000113296   \n",
       "4                      0          4.58211e-05          0.000113296   \n",
       "..                   ...                  ...                  ...   \n",
       "138                    -                    -                    -   \n",
       "139                    -                    -                    -   \n",
       "140                    -                    -                    -   \n",
       "141                    -                    -                    -   \n",
       "142                    -                    -                    -   \n",
       "\n",
       "    gnomADg_AF_joint_fin gnomADg_AF_joint_nfe chromosome           position  \\\n",
       "0                      0                    0          3  43343506-43343507   \n",
       "1                      0                    0          3  43343506-43343507   \n",
       "2                      0                    0          3  43343506-43343507   \n",
       "3                      0                    0          3  43343506-43343507   \n",
       "4                      0                    0          3  43343506-43343507   \n",
       "..                   ...                  ...        ...                ...   \n",
       "138                    -                    -          3           45496305   \n",
       "139                    -                    -          3           45496305   \n",
       "140                    -                    -          3           45496305   \n",
       "141                    -                    -          3           45496305   \n",
       "142                    -                    -          3           45496305   \n",
       "\n",
       "        start       end  \n",
       "0    43343506  43343507  \n",
       "1    43343506  43343507  \n",
       "2    43343506  43343507  \n",
       "3    43343506  43343507  \n",
       "4    43343506  43343507  \n",
       "..        ...       ...  \n",
       "138  45496305      None  \n",
       "139  45496305      None  \n",
       "140  45496305      None  \n",
       "141  45496305      None  \n",
       "142  45496305      None  \n",
       "\n",
       "[143 rows x 100 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "d0141bc4-662d-4288-8687-0837371272c2",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# If there are only two values, set the end column equal to the start column\n",
    "df['end'].fillna(df['start'], inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "cd7bcc6a-56da-4577-a912-277cde638843",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>cDNA_position</th>\n",
       "      <th>CDS_position</th>\n",
       "      <th>Protein_position</th>\n",
       "      <th>...</th>\n",
       "      <th>gnomADg_AF_joint_amr</th>\n",
       "      <th>gnomADg_AF_joint_asj</th>\n",
       "      <th>gnomADg_AF_joint_eas</th>\n",
       "      <th>gnomADg_AF_joint_sas</th>\n",
       "      <th>gnomADg_AF_joint_fin</th>\n",
       "      <th>gnomADg_AF_joint_nfe</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>position</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000296088</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>43343506-43343507</td>\n",
       "      <td>43343506</td>\n",
       "      <td>43343507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000234617</td>\n",
       "      <td>ENST00000422681</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>43343506-43343507</td>\n",
       "      <td>43343506</td>\n",
       "      <td>43343507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000429705</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>43343506-43343507</td>\n",
       "      <td>43343506</td>\n",
       "      <td>43343507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000437827</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>43343506-43343507</td>\n",
       "      <td>43343506</td>\n",
       "      <td>43343507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000454177</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>3</td>\n",
       "      <td>43343506-43343507</td>\n",
       "      <td>43343506</td>\n",
       "      <td>43343507</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000642274</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1912</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000645846</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1746</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000650792</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>1695</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000651549</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant,NMD_transcript_variant</td>\n",
       "      <td>1680</td>\n",
       "      <td>1554</td>\n",
       "      <td>518</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000652135</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>3_prime_UTR_variant,NMD_transcript_variant</td>\n",
       "      <td>1845</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>143 rows × 100 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    Uploaded_variation             Location Allele             Gene  \\\n",
       "0      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "1      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000234617   \n",
       "2      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "3      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "4      3:43343506:G:GT  3:43343506-43343507      T  ENSG00000163788   \n",
       "..                 ...                  ...    ...              ...   \n",
       "138     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "139     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "140     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "141     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "142     3:45496305:C:T           3:45496305      T  ENSG00000011376   \n",
       "\n",
       "             Feature Feature_type                                 Consequence  \\\n",
       "0    ENST00000296088   Transcript                              intron_variant   \n",
       "1    ENST00000422681   Transcript                     downstream_gene_variant   \n",
       "2    ENST00000429705   Transcript                              intron_variant   \n",
       "3    ENST00000437827   Transcript                              intron_variant   \n",
       "4    ENST00000454177   Transcript                              intron_variant   \n",
       "..               ...          ...                                         ...   \n",
       "138  ENST00000642274   Transcript                          synonymous_variant   \n",
       "139  ENST00000645846   Transcript                          synonymous_variant   \n",
       "140  ENST00000650792   Transcript                          synonymous_variant   \n",
       "141  ENST00000651549   Transcript   synonymous_variant,NMD_transcript_variant   \n",
       "142  ENST00000652135   Transcript  3_prime_UTR_variant,NMD_transcript_variant   \n",
       "\n",
       "    cDNA_position CDS_position Protein_position  ... gnomADg_AF_joint_amr  \\\n",
       "0               -            -                -  ...                    0   \n",
       "1               -            -                -  ...                    0   \n",
       "2               -            -                -  ...                    0   \n",
       "3               -            -                -  ...                    0   \n",
       "4               -            -                -  ...                    0   \n",
       "..            ...          ...              ...  ...                  ...   \n",
       "138          1912         1554              518  ...                    -   \n",
       "139          1746         1554              518  ...                    -   \n",
       "140          1695         1554              518  ...                    -   \n",
       "141          1680         1554              518  ...                    -   \n",
       "142          1845            -                -  ...                    -   \n",
       "\n",
       "    gnomADg_AF_joint_asj gnomADg_AF_joint_eas gnomADg_AF_joint_sas  \\\n",
       "0                      0          4.58211e-05          0.000113296   \n",
       "1                      0          4.58211e-05          0.000113296   \n",
       "2                      0          4.58211e-05          0.000113296   \n",
       "3                      0          4.58211e-05          0.000113296   \n",
       "4                      0          4.58211e-05          0.000113296   \n",
       "..                   ...                  ...                  ...   \n",
       "138                    -                    -                    -   \n",
       "139                    -                    -                    -   \n",
       "140                    -                    -                    -   \n",
       "141                    -                    -                    -   \n",
       "142                    -                    -                    -   \n",
       "\n",
       "    gnomADg_AF_joint_fin gnomADg_AF_joint_nfe chromosome           position  \\\n",
       "0                      0                    0          3  43343506-43343507   \n",
       "1                      0                    0          3  43343506-43343507   \n",
       "2                      0                    0          3  43343506-43343507   \n",
       "3                      0                    0          3  43343506-43343507   \n",
       "4                      0                    0          3  43343506-43343507   \n",
       "..                   ...                  ...        ...                ...   \n",
       "138                    -                    -          3           45496305   \n",
       "139                    -                    -          3           45496305   \n",
       "140                    -                    -          3           45496305   \n",
       "141                    -                    -          3           45496305   \n",
       "142                    -                    -          3           45496305   \n",
       "\n",
       "        start       end  \n",
       "0    43343506  43343507  \n",
       "1    43343506  43343507  \n",
       "2    43343506  43343507  \n",
       "3    43343506  43343507  \n",
       "4    43343506  43343507  \n",
       "..        ...       ...  \n",
       "138  45496305  45496305  \n",
       "139  45496305  45496305  \n",
       "140  45496305  45496305  \n",
       "141  45496305  45496305  \n",
       "142  45496305  45496305  \n",
       "\n",
       "[143 rows x 100 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "9c06eefa-ac10-42f2-b9e0-75dcb4f2ed3c",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "# Drop the intermediate columns\n",
    "df.drop(['position'], axis=1, inplace=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "02dfbdfa-d447-4309-a939-f34a1d8af1f2",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "column_order = ['chromosome', 'start', 'end'] + [col for col in df.columns if col not in ['chromosome', 'start', 'end']]\n",
    "df = df[column_order]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "0efa55b7-336d-475b-87f4-723225422708",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chromosome</th>\n",
       "      <th>start</th>\n",
       "      <th>end</th>\n",
       "      <th>Uploaded_variation</th>\n",
       "      <th>Location</th>\n",
       "      <th>Allele</th>\n",
       "      <th>Gene</th>\n",
       "      <th>Feature</th>\n",
       "      <th>Feature_type</th>\n",
       "      <th>Consequence</th>\n",
       "      <th>...</th>\n",
       "      <th>ClinVar_CLNREVSTAT</th>\n",
       "      <th>ClinVar_CLNDN</th>\n",
       "      <th>gnomADg</th>\n",
       "      <th>gnomADg_AF_joint_afr</th>\n",
       "      <th>gnomADg_AF_joint_amr</th>\n",
       "      <th>gnomADg_AF_joint_asj</th>\n",
       "      <th>gnomADg_AF_joint_eas</th>\n",
       "      <th>gnomADg_AF_joint_sas</th>\n",
       "      <th>gnomADg_AF_joint_fin</th>\n",
       "      <th>gnomADg_AF_joint_nfe</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>3</td>\n",
       "      <td>43343506</td>\n",
       "      <td>43343507</td>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000296088</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>3</td>\n",
       "      <td>43343506</td>\n",
       "      <td>43343507</td>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000234617</td>\n",
       "      <td>ENST00000422681</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>3</td>\n",
       "      <td>43343506</td>\n",
       "      <td>43343507</td>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000429705</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>3</td>\n",
       "      <td>43343506</td>\n",
       "      <td>43343507</td>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000437827</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>3</td>\n",
       "      <td>43343506</td>\n",
       "      <td>43343507</td>\n",
       "      <td>3:43343506:G:GT</td>\n",
       "      <td>3:43343506-43343507</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000163788</td>\n",
       "      <td>ENST00000454177</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>rs781680268</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>4.58211e-05</td>\n",
       "      <td>0.000113296</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>138</th>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000642274</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>139</th>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000645846</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>140</th>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000650792</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>141</th>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000651549</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>synonymous_variant,NMD_transcript_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>3</td>\n",
       "      <td>45496305</td>\n",
       "      <td>45496305</td>\n",
       "      <td>3:45496305:C:T</td>\n",
       "      <td>3:45496305</td>\n",
       "      <td>T</td>\n",
       "      <td>ENSG00000011376</td>\n",
       "      <td>ENST00000652135</td>\n",
       "      <td>Transcript</td>\n",
       "      <td>3_prime_UTR_variant,NMD_transcript_variant</td>\n",
       "      <td>...</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>143 rows × 99 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    chromosome     start       end Uploaded_variation             Location  \\\n",
       "0            3  43343506  43343507    3:43343506:G:GT  3:43343506-43343507   \n",
       "1            3  43343506  43343507    3:43343506:G:GT  3:43343506-43343507   \n",
       "2            3  43343506  43343507    3:43343506:G:GT  3:43343506-43343507   \n",
       "3            3  43343506  43343507    3:43343506:G:GT  3:43343506-43343507   \n",
       "4            3  43343506  43343507    3:43343506:G:GT  3:43343506-43343507   \n",
       "..         ...       ...       ...                ...                  ...   \n",
       "138          3  45496305  45496305     3:45496305:C:T           3:45496305   \n",
       "139          3  45496305  45496305     3:45496305:C:T           3:45496305   \n",
       "140          3  45496305  45496305     3:45496305:C:T           3:45496305   \n",
       "141          3  45496305  45496305     3:45496305:C:T           3:45496305   \n",
       "142          3  45496305  45496305     3:45496305:C:T           3:45496305   \n",
       "\n",
       "    Allele             Gene          Feature Feature_type  \\\n",
       "0        T  ENSG00000163788  ENST00000296088   Transcript   \n",
       "1        T  ENSG00000234617  ENST00000422681   Transcript   \n",
       "2        T  ENSG00000163788  ENST00000429705   Transcript   \n",
       "3        T  ENSG00000163788  ENST00000437827   Transcript   \n",
       "4        T  ENSG00000163788  ENST00000454177   Transcript   \n",
       "..     ...              ...              ...          ...   \n",
       "138      T  ENSG00000011376  ENST00000642274   Transcript   \n",
       "139      T  ENSG00000011376  ENST00000645846   Transcript   \n",
       "140      T  ENSG00000011376  ENST00000650792   Transcript   \n",
       "141      T  ENSG00000011376  ENST00000651549   Transcript   \n",
       "142      T  ENSG00000011376  ENST00000652135   Transcript   \n",
       "\n",
       "                                    Consequence  ... ClinVar_CLNREVSTAT  \\\n",
       "0                                intron_variant  ...                  -   \n",
       "1                       downstream_gene_variant  ...                  -   \n",
       "2                                intron_variant  ...                  -   \n",
       "3                                intron_variant  ...                  -   \n",
       "4                                intron_variant  ...                  -   \n",
       "..                                          ...  ...                ...   \n",
       "138                          synonymous_variant  ...                  -   \n",
       "139                          synonymous_variant  ...                  -   \n",
       "140                          synonymous_variant  ...                  -   \n",
       "141   synonymous_variant,NMD_transcript_variant  ...                  -   \n",
       "142  3_prime_UTR_variant,NMD_transcript_variant  ...                  -   \n",
       "\n",
       "    ClinVar_CLNDN      gnomADg gnomADg_AF_joint_afr gnomADg_AF_joint_amr  \\\n",
       "0               -  rs781680268                    0                    0   \n",
       "1               -  rs781680268                    0                    0   \n",
       "2               -  rs781680268                    0                    0   \n",
       "3               -  rs781680268                    0                    0   \n",
       "4               -  rs781680268                    0                    0   \n",
       "..            ...          ...                  ...                  ...   \n",
       "138             -            -                    -                    -   \n",
       "139             -            -                    -                    -   \n",
       "140             -            -                    -                    -   \n",
       "141             -            -                    -                    -   \n",
       "142             -            -                    -                    -   \n",
       "\n",
       "    gnomADg_AF_joint_asj gnomADg_AF_joint_eas gnomADg_AF_joint_sas  \\\n",
       "0                      0          4.58211e-05          0.000113296   \n",
       "1                      0          4.58211e-05          0.000113296   \n",
       "2                      0          4.58211e-05          0.000113296   \n",
       "3                      0          4.58211e-05          0.000113296   \n",
       "4                      0          4.58211e-05          0.000113296   \n",
       "..                   ...                  ...                  ...   \n",
       "138                    -                    -                    -   \n",
       "139                    -                    -                    -   \n",
       "140                    -                    -                    -   \n",
       "141                    -                    -                    -   \n",
       "142                    -                    -                    -   \n",
       "\n",
       "    gnomADg_AF_joint_fin gnomADg_AF_joint_nfe  \n",
       "0                      0                    0  \n",
       "1                      0                    0  \n",
       "2                      0                    0  \n",
       "3                      0                    0  \n",
       "4                      0                    0  \n",
       "..                   ...                  ...  \n",
       "138                    -                    -  \n",
       "139                    -                    -  \n",
       "140                    -                    -  \n",
       "141                    -                    -  \n",
       "142                    -                    -  \n",
       "\n",
       "[143 rows x 99 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0c1b5e04-95ad-4e33-a823-9f10629ac023",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Print the result\n",
    "print(df)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 167,
   "id": "21a9397a-3787-4b4f-9df4-9bfa3bd01e07",
   "metadata": {
    "tags": []
   },
   "outputs": [],
   "source": [
    "df.to_csv('/home/dmc2245/test/bcftools/test_out.tsv', index=False, sep='\\t', header=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 232,
   "id": "bebb0409-a28e-41d4-9797-c8afe0ed053d",
   "metadata": {
    "scrolled": true,
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>MAX_AF</th>\n",
       "      <th>MAX_AF_POPS</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0004168</td>\n",
       "      <td>gnomADg_SAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0004168</td>\n",
       "      <td>gnomADg_SAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0004168</td>\n",
       "      <td>gnomADg_SAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0004168</td>\n",
       "      <td>gnomADg_SAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0004168</td>\n",
       "      <td>gnomADg_SAS</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>-</td>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>147 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        MAX_AF  MAX_AF_POPS\n",
       "0    0.0004168  gnomADg_SAS\n",
       "1    0.0004168  gnomADg_SAS\n",
       "2    0.0004168  gnomADg_SAS\n",
       "3    0.0004168  gnomADg_SAS\n",
       "4    0.0004168  gnomADg_SAS\n",
       "..         ...          ...\n",
       "142          -            -\n",
       "143          -            -\n",
       "144          -            -\n",
       "145          -            -\n",
       "146          -            -\n",
       "\n",
       "[147 rows x 2 columns]"
      ]
     },
     "execution_count": 232,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.columns[pd.Series(df.columns).str.startswith('MAX_AF')]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 220,
   "id": "b69719bd-5563-4cf7-b74c-d80638e5bbcb",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>CADD_PHRED</th>\n",
       "      <th>CADD_RAW</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>19.04</td>\n",
       "      <td>1.970863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>19.04</td>\n",
       "      <td>1.970863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>19.04</td>\n",
       "      <td>1.970863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>19.04</td>\n",
       "      <td>1.970863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>19.04</td>\n",
       "      <td>1.970863</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>7.467</td>\n",
       "      <td>0.595288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>7.467</td>\n",
       "      <td>0.595288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>7.467</td>\n",
       "      <td>0.595288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>7.467</td>\n",
       "      <td>0.595288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>7.467</td>\n",
       "      <td>0.595288</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>147 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    CADD_PHRED  CADD_RAW\n",
       "0        19.04  1.970863\n",
       "1        19.04  1.970863\n",
       "2        19.04  1.970863\n",
       "3        19.04  1.970863\n",
       "4        19.04  1.970863\n",
       "..         ...       ...\n",
       "142      7.467  0.595288\n",
       "143      7.467  0.595288\n",
       "144      7.467  0.595288\n",
       "145      7.467  0.595288\n",
       "146      7.467  0.595288\n",
       "\n",
       "[147 rows x 2 columns]"
      ]
     },
     "execution_count": 220,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.columns[pd.Series(df.columns).str.startswith('CADD')]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 228,
   "id": "c1705c63-4b47-40eb-8d18-c492098fca73",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['1.970863', '1.492370', '-', '1.759156', '0.512506', '2.185077',\n",
       "       '0.602889', '0.270969', '-0.153895', '-0.111623', '-0.720353',\n",
       "       '-0.163293', '-0.459438', '0.747722', '0.595288'], dtype=object)"
      ]
     },
     "execution_count": 228,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['CADD_RAW'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 229,
   "id": "e750882f-6d19-45a2-9074-6d743124c045",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['19.04', '15.69', '-', '17.44', '6.635', '20.8', '7.541', '3.950',\n",
       "       '0.641', '0.785', '0.039', '0.612', '0.144', '8.892', '7.467'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 229,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['CADD_PHRED'].unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 231,
   "id": "549c7db0-03ea-4c0c-af71-e428d1aa9d61",
   "metadata": {
    "tags": []
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gnomADg_SAS_AF</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0004168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0004168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0004168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0004168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0004168</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>142</th>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>143</th>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>144</th>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>145</th>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>-</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>147 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gnomADg_SAS_AF\n",
       "0        0.0004168\n",
       "1        0.0004168\n",
       "2        0.0004168\n",
       "3        0.0004168\n",
       "4        0.0004168\n",
       "..             ...\n",
       "142              -\n",
       "143              -\n",
       "144              -\n",
       "145              -\n",
       "146              -\n",
       "\n",
       "[147 rows x 1 columns]"
      ]
     },
     "execution_count": 231,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.filter(regex='gnomADg_SAS')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "f2f42ef0-ee5b-4ce8-b001-aba5854d5cef",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.13"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
