<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="ipynb_website:version" content="0.9.7" />
<meta name="viewport" content="width=device-width, initial-scale=1" />

<link rel="stylesheet" type="text/css" href="../css/jt.css">

<link rel="stylesheet" type="text/css" href="../css/toc2.css">

<link href="../site_libs/jqueryui-1.11.4/jquery-ui.css">
<link rel="stylesheet" href="../site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<link rel="stylesheet" href="../site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
<script type="text/javascript" src="https://ajax.googleapis.com/ajax/libs/jqueryui/1.9.1/jquery-ui.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="../site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>

<link rel="stylesheet"
      href="../site_libs/highlightjs/null.min.css"
      type="text/css" />

<script src="../site_libs/highlightjs/highlight.pack.js"></script>
<script>hljs.initHighlightingOnLoad();</script>
<script type="text/javascript">
if (window.hljs && document.readyState && document.readyState === "complete") {
   window.setTimeout(function() {
      hljs.initHighlighting();
   }, 0);
}
</script>

<script src="../js/doc_toc.js"></script>
<script src="../js/docs.js"></script>

<script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.2/MathJax.js?config=TeX-MML-AM_CHTML"></script>
<script>
    MathJax.Hub.Config({
        extensions: ["tex2jax.js"],
        jax: ["input/TeX", "output/HTML-CSS"],
        tex2jax: {
        inlineMath: [ ['$','$'], ["\\(","\\)"] ],
        displayMath: [ ['$$','$$'], ["\\[","\\]"] ],
        processEscapes: true
        },
        "HTML-CSS": {
            preferredFont: "TeX",
            availableFonts: ["TeX"],
            styles: {
                scale: 110,
                ".MathJax_Display": {
                    "font-size": "110%",
                }
            }
        }
    });
</script>
<script>
function filterDataFrame(id) {
    var input = document.getElementById("search_" + id);
    var filter = input.value.toUpperCase();
    var table = document.getElementById("dataframe_" + id);
    var tr = table.getElementsByTagName("tr");
    // Loop through all table rows, and hide those who don't match the search query
    for (var i = 1; i < tr.length; i++) {
        for (var j = 0; j < tr[i].cells.length; ++j) {
            var matched = false;
            if (tr[i].cells[j].innerHTML.toUpperCase().indexOf(filter) != -1) {
                tr[i].style.display = "";
                matched = true
                break;
            }
            if (!matched)
                tr[i].style.display = "none";
        }
    }
}
function sortDataFrame(id, n, dtype) {
    var table = document.getElementById("dataframe_" + id);
    var tb = table.tBodies[0]; // use `<tbody>` to ignore `<thead>` and `<tfoot>` rows
    var tr = Array.prototype.slice.call(tb.rows, 0); // put rows into array
    if (dtype === 'numeric') {
        var fn = function(a, b) { 
            return parseFloat(a.cells[n].textContent) <= parseFloat(b.cells[n].textContent) ? -1 : 1;
        }
    } else {
        var fn = function(a, b) {
            var c = a.cells[n].textContent.trim().localeCompare(b.cells[n].textContent.trim()); 
            return c > 0 ? 1 : (c < 0 ? -1 : 0) }
    }
    var isSorted = function(array, fn) {
        if (array.length < 2)
            return 1;
        var direction = fn(array[0], array[1]); 
        for (var i = 1; i < array.length - 1; ++i) {
            var d = fn(array[i], array[i+1]);
            if (d == 0)
                continue;
            else if (direction == 0)
                direction = d;
            else if (direction != d)
                return 0;
            }
        return direction;
    }
    var sorted = isSorted(tr, fn);
    if (sorted == 1 || sorted == -1) {
        // if sorted already, reverse it
        for(var i = tr.length - 1; i >= 0; --i)
            tb.appendChild(tr[i]); // append each row in order
    } else {
        tr = tr.sort(fn);
        for(var i = 0; i < tr.length; ++i)
            tb.appendChild(tr[i]); // append each row in order
    }
}
</script>

<script>
$( document ).ready(function(){
            var cfg={'threshold':3,     // depth of toc (number of levels)
             'number_sections': false,
             'toc_cell': false,          // useless here
             'toc_window_display': true, // display the toc window
             "toc_section_display": "block", // display toc contents in the window
             'sideBar':true,       // sidebar or floating window
             'navigate_menu':false       // navigation menu (only in liveNotebook -- do not change)
            }
            var st={};                  // some variables used in the script
            st.rendering_toc_cell = false;
            st.config_loaded = false;
            st.extension_initialized=false;
            st.nbcontainer_marginleft = $('#notebook-container').css('margin-left')
            st.nbcontainer_marginright = $('#notebook-container').css('margin-right')
            st.nbcontainer_width = $('#notebook-container').css('width')
            st.oldTocHeight = undefined
            st.cell_toc = undefined;
            st.toc_index=0;
            // fire the main function with these parameters
            table_of_contents(cfg, st);
            var file=workflowDict[$("h1:first").attr("id")];
            $("#toc-level0 a").css("color","#126dce");
            $('a[href="#'+$("h1:first").attr("id")+'"]').hide()
            var docs=workflowArray;
            var docs_map=workflowArrayMap;
            var pos=workflowArray.indexOf(file);
            for (var a=pos;a>=0;a--){
                  $('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>').insertBefore("#toc-level0 li:eq(0)");
            }
            $('a[href="'+file+'.html'+'"]').css("color","#126dce");
            for (var a=pos+1;a<docs.length;a++){
                  $(".toc #toc-level0").append('<li><a href="'+docs[a]+'.html"><font color="#073642"><b>'+docs_map[docs[a]].replace(/_/g," ")+'</b></font></a></li>');
            }
            // $("#toc-header").hide(); // comment out because it prevents search bar from displaying
    });
</script>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');
  // mark it active
  menuAnchor.parent().addClass('active');
  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>
<div class="container-fluid main-container">
<!-- tabsets -->
<script src="../site_libs/navigation-1.1/tabsets.js"></script>
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>



<title>Pleiotropy analysis in UKBB</title>

<style type = "text/css">
body {
  font-family: "Droid Sans";
  padding-top: 66px;
  padding-bottom: 40px;
}
</style>
</head>

<body>
<div tabindex="-1" id="notebook" class="border-box-sizing">
<div class="container" id="notebook-container">

<!-- code folding -->

<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="../index.html">Pleiotropy analysis in UKBB</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        
<li>
  <a href="../index.html">Overview</a>
</li>
        
<li>
  <a href="../workflow.html">Workflow</a>
</li>
        
<li>
  <a href="../analysis.html">Analysis</a>
</li>
        
      </ul>
        
<ul class="nav navbar-nav navbar-right">
<li>
   <a href="http://github.com/dianacornejo/pleiotropy_UKB"> <span class="fa fa-github"></span> </a>
</li>
</ul>
        
      </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h1 id="Extracting-data-for-genomic-regions-of-interest">Extracting data for genomic regions of interest<a class="anchor-link" href="#Extracting-data-for-genomic-regions-of-interest">&#182;</a></h1>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Aim">Aim<a class="anchor-link" href="#Aim">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>To extract the summary statistics and genotype on specific genomic regions and calculate their LD matrix.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Input-and-Output">Input and Output<a class="anchor-link" href="#Input-and-Output">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Input">Input<a class="anchor-link" href="#Input">&#182;</a></h3><ul>
<li>list of regions e.g. SNP rs849335 is at chr 7:28223990; and we extract 1Mb radius around it.<ul>
<li>The locus will be represented by one line in the region list with 3 columns <code>7 27723990 28723990</code></li>
</ul>
</li>
<li>list of all genotypes, in <code>bgen</code> format and in <code>plink</code> format.<ul>
<li>The list is a file with 2 columns: <code>chr genotype_file_chr.ext</code>.</li>
<li>The first column is chromosome ID, the 2nd file is genotype for that chromosome.</li>
<li>When chromosome ID is 0, it implies that the genotype file contains all the genotypes.</li>
</ul>
</li>
<li>all summary statistics (eg, $\hat{\beta}$, $SE(\hat{\beta})$ and p-values)<ul>
<li>These summary statistics should contain at least these columns: </li>
<li><code>chrom, pos, a0, a1, snp_id, bhat, sbhat, z, p</code></li>
<li>If SNP id is not available we make it up as <code>chrom_pos_a1_a2</code></li>
</ul>
</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<ul>
<li>list of unrelated individuals (from which LD matrices will be computed)</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p><strong>FIXME: examplain format of summary stats file and format of unrelated individual list</strong></p>
<h3 id="Output">Output<a class="anchor-link" href="#Output">&#182;</a></h3><ul>
<li>regional genotype</li>
<li>regional LD accounting for related individuals</li>
<li>reginonal summary stats</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Workflow-usage">Workflow usage<a class="anchor-link" href="#Workflow-usage">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>Using our minimal working example data-set,</p>

<pre><code>sos run Region_Extraction.ipynb \
    --cwd output \
    --region-file data/regions.txt \
    --pheno-path data/phenotypes.txt \
    --geno-path data/genotype_inventory.txt \
    --sumstats-path output/ukb_imp_v3.UKB_caucasians_BMIwaisthip_AsthmaAndT2D_INT-BMI_withagesex_041720.BoltLMM.snp_stats.all_chr.gz \
    --unrelated-samples data/UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620 \
    --chrom-col 2 --pos-col 3 --a0-col 6 --a1-col 5 --bhat-col 11 --sbhat-col 12 --p-col 16 --snp-col 1</code></pre>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Workflow-codes">Workflow codes<a class="anchor-link" href="#Workflow-codes">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="p">[</span><span class="k">global</span><span class="p">]</span>
<span class="c1"># Work directory where output will be saved to</span>
<span class="kn">parameter:</span> <span class="n">cwd</span> <span class="o">=</span> <span class="n">path</span>
<span class="c1"># Region specifications</span>
<span class="kn">parameter:</span> <span class="n">region_file</span> <span class="o">=</span> <span class="n">path</span>
<span class="c1"># Genotype file inventory</span>
<span class="kn">parameter:</span> <span class="n">geno_path</span> <span class="o">=</span> <span class="n">path</span>
<span class="c1"># Phenotype path</span>
<span class="kn">parameter:</span> <span class="n">pheno_path</span> <span class="o">=</span> <span class="n">path</span>
<span class="c1"># Path to summary stats file</span>
<span class="kn">parameter:</span> <span class="n">sumstats_path</span> <span class="o">=</span> <span class="n">path</span>
<span class="c1"># Path to samples of unrelated individuals</span>
<span class="kn">parameter:</span> <span class="n">unrelated_samples</span> <span class="o">=</span> <span class="n">path</span>
<span class="c1"># Column ID for summary stats, 1-based index</span>
<span class="kn">parameter:</span> <span class="n">chrom_col</span> <span class="o">=</span> <span class="nb">int</span>
<span class="kn">parameter:</span> <span class="n">pos_col</span> <span class="o">=</span> <span class="nb">int</span>
<span class="kn">parameter:</span> <span class="n">a0_col</span> <span class="o">=</span> <span class="nb">int</span>
<span class="kn">parameter:</span> <span class="n">a1_col</span> <span class="o">=</span> <span class="nb">int</span>
<span class="kn">parameter:</span> <span class="n">bhat_col</span> <span class="o">=</span> <span class="nb">int</span>
<span class="kn">parameter:</span> <span class="n">sbhat_col</span> <span class="o">=</span> <span class="nb">int</span>
<span class="kn">parameter:</span> <span class="n">snp_col</span> <span class="o">=</span> <span class="nb">int</span>
<span class="kn">parameter:</span> <span class="n">p_col</span> <span class="o">=</span> <span class="nb">int</span>
<span class="n">fail_if</span><span class="p">(</span><span class="ow">not</span> <span class="n">region_file</span><span class="o">.</span><span class="n">is_file</span><span class="p">(),</span> <span class="n">msg</span> <span class="o">=</span> <span class="s1">&#39;Cannot find regions to extract. Please specify them using ``--region-file`` option.&#39;</span><span class="p">)</span>
<span class="c1"># Load all regions of interest. Each item in the list will be a region: (chr, start, end)</span>
<span class="n">regions</span> <span class="o">=</span> <span class="p">[</span><span class="n">x</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span><span class="o">.</span><span class="n">split</span><span class="p">()</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">open</span><span class="p">(</span><span class="n">region_file</span><span class="p">)</span><span class="o">.</span><span class="n">readlines</span><span class="p">()</span> <span class="k">if</span> <span class="n">x</span><span class="o">.</span><span class="n">strip</span><span class="p">()]</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Some-utility-functions">Some utility functions<a class="anchor-link" href="#Some-utility-functions">&#182;</a></h2><ul>
<li><code>plink_slice</code>: The function of extracting regional genotype<ul>
<li>p: list of bim, fam, bed</li>
<li>pb: index of bim</li>
<li>pf: index of fam</li>
</ul>
</li>
</ul>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="p">[</span><span class="n">utils</span><span class="p">]</span>
<span class="kn">output:</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">cwd</span><span class="si">:</span><span class="s1">a</span><span class="si">}</span><span class="s1">/utils.py&#39;</span>
<span class="kn">report:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s1">&#39;${ }&#39;</span><span class="p">,</span> <span class="n">output</span><span class="o">=</span><span class="n">_output</span>
    <span class="kn">import</span> <span class="nn">numpy</span> <span class="k">as</span> <span class="nn">np</span>
    <span class="kn">import</span> <span class="nn">pandas</span> <span class="k">as</span> <span class="nn">pd</span>
    <span class="k">def</span> <span class="nf">region_index</span><span class="p">(</span><span class="n">bim</span><span class="p">,</span><span class="n">chrom</span><span class="p">,</span><span class="n">start</span><span class="p">,</span><span class="n">end</span><span class="p">,</span> <span class="n">chrom_col</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">chrom_col</span> <span class="o">-</span> <span class="mi">1</span><span class="p">},</span> <span class="n">pos_col</span><span class="o">=</span><span class="err">$</span><span class="p">{</span><span class="n">pos_col</span> <span class="o">-</span> <span class="mi">1</span><span class="p">}):</span>
        <span class="n">chr_bool</span> <span class="o">=</span> <span class="n">bim</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span><span class="n">chrom_col</span><span class="p">]</span> <span class="o">==</span> <span class="n">chrom</span>
        <span class="n">chr_ind</span> <span class="o">=</span> <span class="n">chr_bool</span><span class="p">[</span><span class="n">chr_bool</span><span class="p">]</span><span class="o">.</span><span class="n">index</span>
        <span class="n">reg_bool</span> <span class="o">=</span> <span class="p">(</span><span class="n">bim</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span><span class="n">pos_col</span><span class="p">][</span><span class="n">chr_ind</span><span class="p">]</span><span class="o">&gt;</span><span class="n">start</span><span class="p">)</span> <span class="o">&amp;</span> <span class="p">(</span><span class="n">bim</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span><span class="n">pos_col</span><span class="p">][</span><span class="n">chr_ind</span><span class="p">]</span><span class="o">&lt;</span><span class="n">end</span><span class="p">)</span>
        <span class="k">return</span> <span class="n">chr_ind</span><span class="p">[</span><span class="n">reg_bool</span><span class="p">]</span>

    <span class="kn">from</span> <span class="nn">scipy.stats</span> <span class="kn">import</span> <span class="n">norm</span>

    <span class="k">def</span> <span class="nf">p2z</span><span class="p">(</span><span class="n">pval</span><span class="p">,</span><span class="n">beta</span><span class="p">):</span>
        <span class="n">z</span> <span class="o">=</span> <span class="p">[</span><span class="n">norm</span><span class="o">.</span><span class="n">pdf</span><span class="p">(</span><span class="n">p</span><span class="p">)</span> <span class="k">if</span> <span class="n">b</span><span class="o">&gt;</span><span class="mi">0</span> <span class="k">else</span> <span class="o">-</span><span class="n">norm</span><span class="o">.</span><span class="n">pdf</span><span class="p">(</span><span class="n">p</span><span class="p">)</span> <span class="k">for</span> <span class="n">p</span><span class="p">,</span><span class="n">b</span> <span class="ow">in</span> <span class="nb">zip</span><span class="p">(</span><span class="n">pval</span><span class="p">,</span><span class="n">beta</span><span class="p">)]</span>
        <span class="k">return</span> <span class="n">z</span>

    <span class="k">def</span> <span class="nf">regional_stat</span><span class="p">(</span><span class="n">ss</span><span class="p">,</span><span class="n">ind</span><span class="p">):</span>
        <span class="n">rg_ss</span> <span class="o">=</span> <span class="n">ss</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">ind</span><span class="p">,:]</span>
        <span class="c1"># FIXME: please take into consideration here different columns</span>
        <span class="c1"># ultimately, the output format should be what&#39;s outline in this post</span>
        <span class="c1"># https://github.com/dianacornejo/pleiotropy_UKB/issues/1#issuecomment-640680333</span>
        <span class="c1"># you need to reconstruct rg_ss using all those `*_col` parameters, see the example below</span>
        <span class="c1"># This way we dont have to worry about whatever summary statistics format as long as the file contain</span>
        <span class="c1"># these relevant information</span>
        <span class="c1"># here I&#39;m just using pvalues as one example</span>
        <span class="c1"># If a column is zero then we have to use its default value</span>
        <span class="c1"># For example, if ${snp_col} == 0 then we create a SNP ID using `chr_pos_a0_a1`; otherwise we use provided SNPs</span>
        <span class="n">rg_ss</span><span class="p">[</span><span class="s1">&#39;z&#39;</span><span class="p">]</span><span class="o">=</span><span class="n">p2z</span><span class="p">(</span><span class="n">rg_ss</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="err">$</span><span class="p">{</span><span class="n">p_col</span> <span class="o">-</span> <span class="mi">1</span><span class="p">}],</span> <span class="n">rg_ss</span><span class="o">.</span><span class="n">iloc</span><span class="p">[:,</span> <span class="err">$</span><span class="p">{</span><span class="n">bhat_col</span> <span class="o">-</span> <span class="mi">1</span><span class="p">}])</span>
        <span class="k">return</span> <span class="n">ss</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">ind</span><span class="p">,:]</span>

    <span class="k">def</span> <span class="nf">plink_slice</span><span class="p">(</span><span class="n">p</span><span class="p">,</span><span class="n">pb</span><span class="o">=</span><span class="kc">None</span><span class="p">,</span><span class="n">pf</span><span class="o">=</span><span class="kc">None</span><span class="p">):</span>
        <span class="p">(</span><span class="n">bim</span><span class="p">,</span><span class="n">fam</span><span class="p">,</span><span class="n">bed</span><span class="p">)</span><span class="o">=</span><span class="n">p</span>
        <span class="k">if</span> <span class="n">pb</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="n">bim</span> <span class="o">=</span> <span class="n">bim</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">pb</span><span class="p">]</span>
            <span class="n">bed</span> <span class="o">=</span> <span class="n">bed</span><span class="p">[</span><span class="n">pb</span><span class="p">,:]</span>
        <span class="k">if</span> <span class="n">pf</span> <span class="ow">is</span> <span class="ow">not</span> <span class="kc">None</span><span class="p">:</span>
            <span class="n">fam</span> <span class="o">=</span> <span class="n">fam</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="n">pf</span><span class="p">]</span>
            <span class="n">bed</span> <span class="o">=</span> <span class="n">bed</span><span class="p">[:,</span><span class="n">pf</span><span class="p">]</span>
        <span class="k">return</span><span class="p">(</span><span class="n">bim</span><span class="p">,</span><span class="n">fam</span><span class="p">,</span><span class="n">bed</span><span class="p">)</span>

    <span class="k">def</span> <span class="nf">LD_matrix</span><span class="p">(</span><span class="n">bed</span><span class="p">):</span>
        <span class="n">snps</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">DataFrame</span><span class="p">(</span><span class="n">bed</span><span class="o">.</span><span class="n">transpose</span><span class="p">())</span>
        <span class="n">ld</span> <span class="o">=</span> <span class="n">snps</span><span class="o">.</span><span class="n">corr</span><span class="p">()</span>
        <span class="k">return</span> <span class="n">ld</span>

    <span class="k">def</span> <span class="nf">extract_region</span><span class="p">(</span><span class="n">region</span><span class="p">,</span><span class="n">gwas</span><span class="p">,</span><span class="n">geno</span><span class="p">,</span><span class="n">iid</span><span class="p">):</span>
        <span class="c1"># FIXME: sample &amp; variants matching are still problematic</span>
        <span class="c1"># genoteyp data should be the ones used for a specific phenotype, not that of all the genotype samples</span>
        <span class="c1"># you need to load phenotype files too, matching samples in phenotypes</span>
        <span class="c1"># then get genotypes only for these phenotypes</span>
        <span class="c1"># and LD matrix based on unrelated individuals in the given phenotypes</span>
        <span class="c1"># please let me know if you need additional clarifications</span>
        <span class="n">rg_ind</span> <span class="o">=</span> <span class="n">region_index</span><span class="p">(</span><span class="n">geno</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span><span class="n">region</span><span class="p">[</span><span class="mi">0</span><span class="p">],</span><span class="n">region</span><span class="p">[</span><span class="mi">1</span><span class="p">],</span><span class="n">region</span><span class="p">[</span><span class="mi">2</span><span class="p">])</span>
        <span class="n">rg_stat</span> <span class="o">=</span> <span class="n">regional_stat</span><span class="p">(</span><span class="n">gwas</span><span class="p">,</span><span class="n">rg_ind</span><span class="p">)</span>
        <span class="n">rg_geno</span> <span class="o">=</span> <span class="n">plink_slice</span><span class="p">(</span><span class="n">geno</span><span class="p">,</span><span class="n">pb</span><span class="o">=</span><span class="nb">list</span><span class="p">(</span><span class="n">rg_ind</span><span class="p">))</span>
        <span class="n">rg_bed</span> <span class="o">=</span> <span class="n">rg_geno</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="o">.</span><span class="n">compute</span><span class="p">()</span>
        <span class="n">rg_ld</span> <span class="o">=</span> <span class="n">LD_matrix</span><span class="p">(</span><span class="n">rg_bed</span><span class="p">[:,</span><span class="n">iid</span><span class="p">])</span>
        <span class="k">return</span> <span class="nb">dict</span><span class="p">(</span><span class="n">stats</span><span class="o">=</span><span class="n">rg_stat</span><span class="p">,</span><span class="n">geno</span><span class="o">=</span><span class="n">rg_bed</span><span class="p">,</span><span class="n">ld</span><span class="o">=</span><span class="n">rg_ld</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Extract-data">Extract data<a class="anchor-link" href="#Extract-data">&#182;</a></h2><p>This step runs in parallel for all loci listed in the region file (via <code>for_each</code>).</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[1]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="p">[</span><span class="n">default</span> <span class="p">(</span><span class="n">extract</span> <span class="n">genotypes</span><span class="p">)]</span>
<span class="kn">depends:</span> <span class="n">Py_Module</span><span class="p">(</span><span class="s1">&#39;pandas_plink&#39;</span><span class="p">)</span>
<span class="kn">input:</span> <span class="n">geno_path</span><span class="p">,</span> <span class="n">pheno_path</span><span class="p">,</span> <span class="n">sumstats_path</span><span class="p">,</span> <span class="n">unrelated_samples</span><span class="p">,</span> <span class="n">output_from</span><span class="p">(</span><span class="s1">&#39;utils&#39;</span><span class="p">),</span> <span class="n">for_each</span> <span class="o">=</span> <span class="s1">&#39;regions&#39;</span>
<span class="kn">output:</span> <span class="n">sumstats</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">cwd</span><span class="si">:</span><span class="s1">a</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">sumstats_path</span><span class="si">:</span><span class="s1">bn</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">_regions</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">_regions</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">_regions</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="si">}</span><span class="s1">.sumstats.gz&#39;</span><span class="p">,</span>
        <span class="n">genotype</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">cwd</span><span class="si">:</span><span class="s1">a</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">geno_path</span><span class="si">:</span><span class="s1">bn</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">pheno_path</span><span class="si">:</span><span class="s1">bn</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">_regions</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">_regions</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">_regions</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="si">}</span><span class="s1">.genotype.gz&#39;</span><span class="p">,</span>
        <span class="n">ld</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">cwd</span><span class="si">:</span><span class="s1">a</span><span class="si">}</span><span class="s1">/</span><span class="si">{</span><span class="n">geno_path</span><span class="si">:</span><span class="s1">bn</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">pheno_path</span><span class="si">:</span><span class="s1">bn</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">_regions</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">_regions</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="si">}</span><span class="s1">_</span><span class="si">{</span><span class="n">_regions</span><span class="p">[</span><span class="mi">2</span><span class="p">]</span><span class="si">}</span><span class="s1">.ld.gz&#39;</span>
<span class="kn">python:</span> <span class="n">expand</span> <span class="o">=</span> <span class="s1">&#39;${ }&#39;</span><span class="p">,</span> <span class="nb">input</span> <span class="o">=</span> <span class="n">_input</span><span class="p">[</span><span class="mi">4</span><span class="p">],</span> <span class="n">stderr</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">:</span><span class="s1">n</span><span class="si">}</span><span class="s1">.stderr&#39;</span><span class="p">,</span> <span class="n">stdout</span> <span class="o">=</span> <span class="sa">f</span><span class="s1">&#39;</span><span class="si">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span><span class="si">:</span><span class="s1">n</span><span class="si">}</span><span class="s1">.stdout&#39;</span>
    <span class="c1"># Load genotype file for the region of interest</span>
    <span class="n">geno_inventory</span> <span class="o">=</span> <span class="nb">dict</span><span class="p">([</span><span class="n">x</span><span class="o">.</span><span class="n">strip</span><span class="p">()</span><span class="o">.</span><span class="n">split</span><span class="p">()</span> <span class="k">for</span> <span class="n">x</span> <span class="ow">in</span> <span class="nb">open</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">})</span><span class="o">.</span><span class="n">readlines</span><span class="p">()</span> <span class="k">if</span> <span class="n">x</span><span class="o">.</span><span class="n">strip</span><span class="p">()])</span>
    <span class="k">if</span> <span class="s2">&quot;$</span><span class="si">{_regions[0]}</span><span class="s2">&quot;</span> <span class="ow">not</span> <span class="ow">in</span> <span class="n">geno_inventory</span><span class="p">:</span>
        <span class="n">geno_file</span> <span class="o">=</span> <span class="n">geno_inventory</span><span class="p">[</span><span class="s1">&#39;0&#39;</span><span class="p">]</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="n">geno_file</span> <span class="o">=</span> <span class="n">geno_inventory</span><span class="p">[</span><span class="s2">&quot;$</span><span class="si">{_regions[0]}</span><span class="s2">&quot;</span><span class="p">]</span>
    <span class="kn">import</span> <span class="nn">os</span>
    <span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="n">geno_file</span><span class="p">):</span>
        <span class="c1"># relative path</span>
        <span class="k">if</span> <span class="ow">not</span> <span class="n">os</span><span class="o">.</span><span class="n">path</span><span class="o">.</span><span class="n">exists</span><span class="p">(</span><span class="s1">&#39;${_input[0]:ad}/&#39;</span> <span class="o">+</span> <span class="n">geno_file</span><span class="p">):</span>
            <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="sa">f</span><span class="s2">&quot;Cannot find genotype file </span><span class="si">{</span><span class="n">geno_file</span><span class="si">}</span><span class="s2">&quot;</span><span class="p">)</span>
        <span class="k">else</span><span class="p">:</span>
            <span class="n">geno_file</span> <span class="o">=</span> <span class="s1">&#39;${_input[0]:ad}/&#39;</span> <span class="o">+</span> <span class="n">geno_file</span>
    <span class="k">if</span> <span class="n">geno_file</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s1">&#39;.bed&#39;</span><span class="p">):</span>
        <span class="kn">from</span> <span class="nn">pandas_plink</span> <span class="kn">import</span> <span class="n">read_plink</span>
        <span class="n">geno</span> <span class="o">=</span> <span class="n">read_plink</span><span class="p">(</span><span class="n">geno_file</span><span class="p">)</span>
    <span class="k">else</span><span class="p">:</span>
        <span class="c1"># FIXME: load from bgen file</span>
        <span class="c1"># https://bgen-reader.readthedocs.io/en/latest/read.html</span>
        <span class="k">raise</span> <span class="ne">ValueError</span><span class="p">(</span><span class="s1">&#39;FIXME&#39;</span><span class="p">)</span>
    <span class="c1"># Load other files</span>
    <span class="n">sumstats</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">2</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">compression</span><span class="o">=</span><span class="s1">&#39;gzip&#39;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">quotechar</span><span class="o">=</span><span class="s1">&#39;&quot;&#39;</span><span class="p">)</span>
    <span class="n">unr</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_input</span><span class="p">[</span><span class="mi">3</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39; &#39;</span><span class="p">,</span> <span class="n">quotechar</span><span class="o">=</span><span class="s1">&#39;&quot;&#39;</span><span class="p">)</span>
    <span class="n">fid</span> <span class="o">=</span> <span class="n">geno</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">fid</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">int</span><span class="p">)</span>
    <span class="n">fid_bool</span> <span class="o">=</span> <span class="n">fid</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">unr</span><span class="o">.</span><span class="n">FID</span><span class="p">)</span>
    <span class="n">fid_ind</span> <span class="o">=</span> <span class="n">fid_bool</span><span class="p">[</span><span class="n">fid_bool</span><span class="p">]</span><span class="o">.</span><span class="n">index</span>
    <span class="c1"># FIXME: fid should be the ones used for a specific phenotype, not that of all the genotype samples</span>
    <span class="c1"># you need to load phenotype files too, matching samples in phenotypes</span>
    <span class="c1"># then get genotypes only for these phenotypes</span>
    <span class="c1"># and LD matrix based on unrelated individuals in the given phenotypes</span>
    <span class="n">rg_info</span> <span class="o">=</span> <span class="n">extract_region</span><span class="p">((</span><span class="s1">&#39;$</span><span class="si">{_regions[0]}</span><span class="s1">&#39;</span><span class="p">,</span> <span class="err">$</span><span class="p">{</span><span class="n">_regions</span><span class="p">[</span><span class="mi">1</span><span class="p">]},</span> <span class="err">$</span><span class="p">{</span><span class="n">_regions</span><span class="p">[</span><span class="mi">2</span><span class="p">]}),</span> <span class="n">sumstats</span><span class="p">,</span> <span class="n">geno</span><span class="p">,</span> <span class="n">fid_ind</span><span class="p">)</span>
    <span class="n">rg_info</span><span class="p">[</span><span class="s1">&#39;stats&#39;</span><span class="p">]</span><span class="o">.</span><span class="n">to_csv</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">0</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">sep</span> <span class="o">=</span> <span class="s2">&quot;</span><span class="se">\t</span><span class="s2">&quot;</span><span class="p">,</span> <span class="n">header</span> <span class="o">=</span> <span class="kc">True</span><span class="p">,</span> <span class="n">index</span> <span class="o">=</span> <span class="kc">False</span><span class="p">)</span>
    <span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">1</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">rg_info</span><span class="p">[</span><span class="s1">&#39;geno&#39;</span><span class="p">])</span>
    <span class="n">np</span><span class="o">.</span><span class="n">savetxt</span><span class="p">(</span><span class="err">$</span><span class="p">{</span><span class="n">_output</span><span class="p">[</span><span class="mi">2</span><span class="p">]:</span><span class="n">r</span><span class="p">},</span> <span class="n">rg_info</span><span class="p">[</span><span class="s1">&#39;ld&#39;</span><span class="p">])</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="FIXME:-codes-below-are-all-what-you-have-used-for-prototyping.-The-can-be-removed-after-the-workflow-is-developed.">FIXME: codes below are all what you have used for prototyping. The can be removed after the workflow is developed.<a class="anchor-link" href="#FIXME:-codes-below-are-all-what-you-have-used-for-prototyping.-The-can-be-removed-after-the-workflow-is-developed.">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[105]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">rg_infos</span><span class="o">=</span> <span class="n">extract_regions</span><span class="p">([[</span><span class="mi">16</span><span class="p">,</span><span class="mi">53800754</span><span class="p">,</span><span class="mi">53845487</span><span class="p">],[</span><span class="mi">7</span><span class="p">,</span><span class="mi">28223990</span><span class="o">-</span><span class="mi">500000</span><span class="p">,</span><span class="mi">28223990</span><span class="o">+</span><span class="mi">500000</span><span class="p">]],</span>
                            <span class="n">sumstat_path</span><span class="p">,</span><span class="n">ukb_geno_path</span><span class="p">,</span><span class="n">unrelated_samples</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stderr output_text">
<pre>Mapping files: 100%|##########| 3/3 [00:42&lt;00:00, 14.28s/it]
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>If you want to make comments on input and output files, you can always load and show part of them. See my <code>%preview</code> example above.</p>

</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Summary-Stat">Summary Stat<a class="anchor-link" href="#Summary-Stat">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Read-GWAS-stat">Read GWAS stat<a class="anchor-link" href="#Read-GWAS-stat">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[21]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">bmi_bolt_path</span> <span class="o">=</span> <span class="n">ukb_gwas_path</span><span class="o">/</span><span class="s1">&#39;INT-BMI/ukb_imp_v3.UKB_caucasians_BMIwaisthip_AsthmaAndT2D_INT-BMI_withagesex_041720.BoltLMM.stats.all_chr.gz&#39;</span>
<span class="n">bmi_bolt</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">bmi_bolt_path</span><span class="p">,</span> <span class="n">compression</span><span class="o">=</span><span class="s1">&#39;gzip&#39;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">quotechar</span><span class="o">=</span><span class="s1">&#39;&quot;&#39;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[37]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">bmi_bolt</span><span class="o">.</span><span class="n">iloc</span><span class="p">[</span><span class="mi">3</span><span class="p">:</span><span class="mi">7</span><span class="p">,]</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt output_prompt">Out[37]:</div>



<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>SNP</th>
      <th>CHR</th>
      <th>BP</th>
      <th>GENPOS</th>
      <th>ALLELE1</th>
      <th>ALLELE0</th>
      <th>A1FREQ</th>
      <th>F_MISS</th>
      <th>CHISQ_LINREG</th>
      <th>P_LINREG</th>
      <th>BETA</th>
      <th>SE</th>
      <th>CHISQ_BOLT_LMM_INF</th>
      <th>P_BOLT_LMM_INF</th>
      <th>CHISQ_BOLT_LMM</th>
      <th>P_BOLT_LMM</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>3</th>
      <td>rs79373928</td>
      <td>1</td>
      <td>801536</td>
      <td>0.005872</td>
      <td>G</td>
      <td>T</td>
      <td>0.014871</td>
      <td>0.000596</td>
      <td>0.009184</td>
      <td>0.9200</td>
      <td>-0.005915</td>
      <td>0.008938</td>
      <td>0.43801</td>
      <td>0.510</td>
      <td>0.358727</td>
      <td>0.550</td>
    </tr>
    <tr>
      <th>4</th>
      <td>rs11240779</td>
      <td>1</td>
      <td>808631</td>
      <td>0.006208</td>
      <td>G</td>
      <td>A</td>
      <td>0.224724</td>
      <td>0.006986</td>
      <td>1.305990</td>
      <td>0.2500</td>
      <td>0.003471</td>
      <td>0.002604</td>
      <td>1.77719</td>
      <td>0.180</td>
      <td>1.598230</td>
      <td>0.210</td>
    </tr>
    <tr>
      <th>5</th>
      <td>rs57181708</td>
      <td>1</td>
      <td>809876</td>
      <td>0.006208</td>
      <td>G</td>
      <td>A</td>
      <td>0.100445</td>
      <td>0.000874</td>
      <td>6.646970</td>
      <td>0.0099</td>
      <td>0.008757</td>
      <td>0.003594</td>
      <td>5.93807</td>
      <td>0.015</td>
      <td>5.179680</td>
      <td>0.023</td>
    </tr>
    <tr>
      <th>6</th>
      <td>rs4970383</td>
      <td>1</td>
      <td>838555</td>
      <td>0.006208</td>
      <td>A</td>
      <td>C</td>
      <td>0.245613</td>
      <td>0.001787</td>
      <td>0.433580</td>
      <td>0.5100</td>
      <td>0.000175</td>
      <td>0.002512</td>
      <td>0.00487</td>
      <td>0.940</td>
      <td>0.003883</td>
      <td>0.950</td>
    </tr>
  </tbody>
</table>
</div>
</div>

</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[22]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">bmi_bolt</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt output_prompt">Out[22]:</div>



<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>SNP</th>
      <th>CHR</th>
      <th>BP</th>
      <th>GENPOS</th>
      <th>ALLELE1</th>
      <th>ALLELE0</th>
      <th>A1FREQ</th>
      <th>F_MISS</th>
      <th>CHISQ_LINREG</th>
      <th>P_LINREG</th>
      <th>BETA</th>
      <th>SE</th>
      <th>CHISQ_BOLT_LMM_INF</th>
      <th>P_BOLT_LMM_INF</th>
      <th>CHISQ_BOLT_LMM</th>
      <th>P_BOLT_LMM</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>rs3131962</td>
      <td>1</td>
      <td>756604</td>
      <td>0.004907</td>
      <td>A</td>
      <td>G</td>
      <td>0.129619</td>
      <td>0.001543</td>
      <td>0.143797</td>
      <td>0.700</td>
      <td>-0.000596</td>
      <td>0.003224</td>
      <td>0.034136</td>
      <td>0.850</td>
      <td>0.008625</td>
      <td>0.930</td>
    </tr>
    <tr>
      <th>1</th>
      <td>rs12562034</td>
      <td>1</td>
      <td>768448</td>
      <td>0.004957</td>
      <td>A</td>
      <td>G</td>
      <td>0.105098</td>
      <td>0.001601</td>
      <td>3.206650</td>
      <td>0.073</td>
      <td>0.006521</td>
      <td>0.003525</td>
      <td>3.423000</td>
      <td>0.064</td>
      <td>2.742840</td>
      <td>0.098</td>
    </tr>
    <tr>
      <th>2</th>
      <td>rs4040617</td>
      <td>1</td>
      <td>779322</td>
      <td>0.005007</td>
      <td>G</td>
      <td>A</td>
      <td>0.127412</td>
      <td>0.002416</td>
      <td>0.681178</td>
      <td>0.410</td>
      <td>-0.001951</td>
      <td>0.003249</td>
      <td>0.360661</td>
      <td>0.550</td>
      <td>0.294739</td>
      <td>0.590</td>
    </tr>
    <tr>
      <th>3</th>
      <td>rs79373928</td>
      <td>1</td>
      <td>801536</td>
      <td>0.005872</td>
      <td>G</td>
      <td>T</td>
      <td>0.014871</td>
      <td>0.000596</td>
      <td>0.009184</td>
      <td>0.920</td>
      <td>-0.005915</td>
      <td>0.008938</td>
      <td>0.438010</td>
      <td>0.510</td>
      <td>0.358727</td>
      <td>0.550</td>
    </tr>
    <tr>
      <th>4</th>
      <td>rs11240779</td>
      <td>1</td>
      <td>808631</td>
      <td>0.006208</td>
      <td>G</td>
      <td>A</td>
      <td>0.224724</td>
      <td>0.006986</td>
      <td>1.305990</td>
      <td>0.250</td>
      <td>0.003471</td>
      <td>0.002604</td>
      <td>1.777190</td>
      <td>0.180</td>
      <td>1.598230</td>
      <td>0.210</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>529019</th>
      <td>rs4824147</td>
      <td>22</td>
      <td>51173542</td>
      <td>0.750810</td>
      <td>T</td>
      <td>C</td>
      <td>0.061359</td>
      <td>0.000558</td>
      <td>0.011819</td>
      <td>0.910</td>
      <td>-0.001627</td>
      <td>0.004553</td>
      <td>0.127749</td>
      <td>0.720</td>
      <td>0.211562</td>
      <td>0.650</td>
    </tr>
    <tr>
      <th>529020</th>
      <td>rs73174435</td>
      <td>22</td>
      <td>51174939</td>
      <td>0.750825</td>
      <td>T</td>
      <td>C</td>
      <td>0.053825</td>
      <td>0.001157</td>
      <td>3.009070</td>
      <td>0.083</td>
      <td>0.009323</td>
      <td>0.004843</td>
      <td>3.705620</td>
      <td>0.054</td>
      <td>3.406890</td>
      <td>0.065</td>
    </tr>
    <tr>
      <th>529021</th>
      <td>rs3810648</td>
      <td>22</td>
      <td>51175626</td>
      <td>0.750832</td>
      <td>G</td>
      <td>A</td>
      <td>0.061006</td>
      <td>0.002949</td>
      <td>0.137834</td>
      <td>0.710</td>
      <td>-0.000627</td>
      <td>0.004571</td>
      <td>0.018814</td>
      <td>0.890</td>
      <td>0.000217</td>
      <td>0.990</td>
    </tr>
    <tr>
      <th>529022</th>
      <td>rs3865764</td>
      <td>22</td>
      <td>51185848</td>
      <td>0.750911</td>
      <td>G</td>
      <td>A</td>
      <td>0.050675</td>
      <td>0.001809</td>
      <td>1.385920</td>
      <td>0.240</td>
      <td>0.011019</td>
      <td>0.004979</td>
      <td>4.896610</td>
      <td>0.027</td>
      <td>4.455700</td>
      <td>0.035</td>
    </tr>
    <tr>
      <th>529023</th>
      <td>rs142680588</td>
      <td>22</td>
      <td>51193629</td>
      <td>0.750970</td>
      <td>G</td>
      <td>A</td>
      <td>0.075823</td>
      <td>0.001268</td>
      <td>5.535070</td>
      <td>0.019</td>
      <td>0.008501</td>
      <td>0.004133</td>
      <td>4.229520</td>
      <td>0.040</td>
      <td>4.460670</td>
      <td>0.035</td>
    </tr>
  </tbody>
</table>
<p>529024 rows  16 columns</p>
</div>
</div>

</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[6]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">asthma_fastgwa_path</span> <span class="o">=</span> <span class="n">ukb_gwas_path</span><span class="o">/</span><span class="s1">&#39;asthma/UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv.Asthma_casesbyICD10codesANDselfreport_controlsbyselfreportandicd10_noautoimmuneincontrols.fastGWA.stats.all_chr&#39;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[7]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">asthma_fastgwa</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">asthma_fastgwa_path</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">quotechar</span><span class="o">=</span><span class="s1">&#39;&quot;&#39;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[8]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">asthma_fastgwa</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt output_prompt">Out[8]:</div>



<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>CHR</th>
      <th>SNP</th>
      <th>POS</th>
      <th>A1</th>
      <th>A2</th>
      <th>N</th>
      <th>AF1</th>
      <th>BETA</th>
      <th>SE</th>
      <th>P</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1</td>
      <td>rs3131962</td>
      <td>756604</td>
      <td>A</td>
      <td>G</td>
      <td>338837</td>
      <td>0.129612</td>
      <td>-0.001497</td>
      <td>0.001272</td>
      <td>0.239246</td>
    </tr>
    <tr>
      <th>1</th>
      <td>1</td>
      <td>rs12562034</td>
      <td>768448</td>
      <td>A</td>
      <td>G</td>
      <td>338815</td>
      <td>0.105013</td>
      <td>-0.000502</td>
      <td>0.001391</td>
      <td>0.718023</td>
    </tr>
    <tr>
      <th>2</th>
      <td>1</td>
      <td>rs4040617</td>
      <td>779322</td>
      <td>G</td>
      <td>A</td>
      <td>338535</td>
      <td>0.127406</td>
      <td>-0.001131</td>
      <td>0.001281</td>
      <td>0.377623</td>
    </tr>
    <tr>
      <th>3</th>
      <td>1</td>
      <td>rs79373928</td>
      <td>801536</td>
      <td>G</td>
      <td>T</td>
      <td>339152</td>
      <td>0.014847</td>
      <td>-0.002883</td>
      <td>0.003529</td>
      <td>0.413952</td>
    </tr>
    <tr>
      <th>4</th>
      <td>1</td>
      <td>rs11240779</td>
      <td>808631</td>
      <td>G</td>
      <td>A</td>
      <td>336983</td>
      <td>0.224707</td>
      <td>-0.001378</td>
      <td>0.001027</td>
      <td>0.179785</td>
    </tr>
    <tr>
      <th>...</th>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
      <td>...</td>
    </tr>
    <tr>
      <th>529019</th>
      <td>22</td>
      <td>rs4824147</td>
      <td>51173542</td>
      <td>T</td>
      <td>C</td>
      <td>339168</td>
      <td>0.061450</td>
      <td>-0.000804</td>
      <td>0.001775</td>
      <td>0.650797</td>
    </tr>
    <tr>
      <th>529020</th>
      <td>22</td>
      <td>rs73174435</td>
      <td>51174939</td>
      <td>T</td>
      <td>C</td>
      <td>338963</td>
      <td>0.053767</td>
      <td>-0.000395</td>
      <td>0.001891</td>
      <td>0.834663</td>
    </tr>
    <tr>
      <th>529021</th>
      <td>22</td>
      <td>rs3810648</td>
      <td>51175626</td>
      <td>G</td>
      <td>A</td>
      <td>338362</td>
      <td>0.060905</td>
      <td>0.000996</td>
      <td>0.001785</td>
      <td>0.576889</td>
    </tr>
    <tr>
      <th>529022</th>
      <td>22</td>
      <td>rs3865764</td>
      <td>51185848</td>
      <td>G</td>
      <td>A</td>
      <td>338737</td>
      <td>0.050654</td>
      <td>-0.001565</td>
      <td>0.001944</td>
      <td>0.420748</td>
    </tr>
    <tr>
      <th>529023</th>
      <td>22</td>
      <td>rs142680588</td>
      <td>51193629</td>
      <td>G</td>
      <td>A</td>
      <td>338919</td>
      <td>0.075940</td>
      <td>0.000452</td>
      <td>0.001613</td>
      <td>0.779308</td>
    </tr>
  </tbody>
</table>
<p>529024 rows  10 columns</p>
</div>
</div>

</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h3 id="Read-genotype-data">Read genotype data<a class="anchor-link" href="#Read-genotype-data">&#182;</a></h3>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[81]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="c1">#[global]</span>
<span class="n">ukb_geno_path1</span><span class="o">=</span><span class="s1">&#39;/SAY/dbgapstg/scratch/UKBiobank/genotype_files/pleiotropy_geneticfiles/UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv&#39;</span>
<span class="n">ukb_geno</span> <span class="o">=</span> <span class="n">read_plink</span><span class="p">(</span><span class="nb">str</span><span class="p">(</span><span class="n">ukb_geno_path1</span><span class="p">))</span>
<span class="n">ukb_unrelatedcauc_path</span><span class="o">=</span><span class="s1">&#39;/SAY/dbgapstg/scratch/UKBiobank/genotype_files/pleiotropy_geneticfiles/unrelated_n307259/UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620&#39;</span>
<span class="n">ukb_unrelatedcauc</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">ukb_unrelatedcauc_path</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39; &#39;</span><span class="p">,</span> <span class="n">quotechar</span><span class="o">=</span><span class="s1">&#39;&quot;&#39;</span><span class="p">)</span>

<span class="c1">#the index of unrelated individuals</span>
<span class="n">fid</span> <span class="o">=</span> <span class="n">ukb_geno</span><span class="p">[</span><span class="mi">1</span><span class="p">]</span><span class="o">.</span><span class="n">fid</span><span class="o">.</span><span class="n">astype</span><span class="p">(</span><span class="nb">int</span><span class="p">)</span>
<span class="n">fid_bool</span> <span class="o">=</span> <span class="n">fid</span><span class="o">.</span><span class="n">isin</span><span class="p">(</span><span class="n">ukb_unrelatedcauc</span><span class="o">.</span><span class="n">FID</span><span class="p">)</span>
<span class="n">fid_ind</span> <span class="o">=</span> <span class="n">fid_bool</span><span class="p">[</span><span class="n">fid_bool</span><span class="p">]</span><span class="o">.</span><span class="n">index</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stderr output_text">
<pre>Mapping files: 100%|##########| 3/3 [00:43&lt;00:00, 14.66s/it]
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>regional LD accounting for unrelated individuals</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[20]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tmp1</span> <span class="o">=</span> <span class="n">LD_matrix</span><span class="p">(</span><span class="n">asthma_fastgwa</span><span class="p">,</span><span class="n">ukb_geno</span><span class="p">,[</span><span class="mi">7</span><span class="p">,</span><span class="mi">28223990</span><span class="o">-</span><span class="mi">500000</span><span class="p">,</span><span class="mi">28223990</span><span class="o">+</span><span class="mi">500000</span><span class="p">],</span><span class="n">fid_ind</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>There is no BP column,trying POS
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[23]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">tmp2</span> <span class="o">=</span> <span class="n">extract_regions</span><span class="p">(</span><span class="n">bmi_bolt</span><span class="p">,</span><span class="n">ukb_geno</span><span class="p">,[[</span><span class="mi">16</span><span class="p">,</span><span class="mi">53800754</span><span class="p">,</span><span class="mi">53845487</span><span class="p">],[</span><span class="mi">7</span><span class="p">,</span><span class="mi">28223990</span><span class="o">-</span><span class="mi">500000</span><span class="p">,</span><span class="mi">28223990</span><span class="o">+</span><span class="mi">500000</span><span class="p">]],</span><span class="n">fid_ind</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Other">Other<a class="anchor-link" href="#Other">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[2]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">bmi_sr_path</span> <span class="o">=</span> <span class="s2">&quot;/SAY/dbgapstg/scratch/UKBiobank/results/BOLTLMM_results/results_imputed_data/INT-BMI/ukb_imp_v3.UKB_caucasians_BMIwaisthip_AsthmaAndT2D_INT-BMI_withagesex_041720.BoltLMM.snp_stats.all_chr.gz&quot;</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">df</span> <span class="o">=</span> <span class="n">pd</span><span class="o">.</span><span class="n">read_csv</span><span class="p">(</span><span class="n">bmi_sr_path</span><span class="p">,</span> <span class="n">compression</span><span class="o">=</span><span class="s1">&#39;gzip&#39;</span><span class="p">,</span> <span class="n">header</span><span class="o">=</span><span class="mi">0</span><span class="p">,</span> <span class="n">sep</span><span class="o">=</span><span class="s1">&#39;</span><span class="se">\t</span><span class="s1">&#39;</span><span class="p">,</span> <span class="n">quotechar</span><span class="o">=</span><span class="s1">&#39;&quot;&#39;</span><span class="p">)</span>
</pre></div>

    </div>
</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[4]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">df</span><span class="o">.</span><span class="n">head</span><span class="p">()</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt output_prompt">Out[4]:</div>



<div class="output_html rendered_html output_subarea output_execute_result">
<div>
<style scoped>
    .dataframe tbody tr th:only-of-type {
        vertical-align: middle;
    }

    .dataframe tbody tr th {
        vertical-align: top;
    }

    .dataframe thead th {
        text-align: right;
    }
</style>
<table border="1" class="dataframe">
  <thead>
    <tr style="text-align: right;">
      <th></th>
      <th>SNP</th>
      <th>CHR</th>
      <th>BP</th>
      <th>GENPOS</th>
      <th>ALLELE1</th>
      <th>ALLELE0</th>
      <th>A1FREQ</th>
      <th>INFO</th>
      <th>CHISQ_LINREG</th>
      <th>P_LINREG</th>
      <th>BETA</th>
      <th>SE</th>
      <th>CHISQ_BOLT_LMM_INF</th>
      <th>P_BOLT_LMM_INF</th>
      <th>CHISQ_BOLT_LMM</th>
      <th>P_BOLT_LMM</th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>0</th>
      <td>1:692794_CA_C</td>
      <td>1</td>
      <td>692794</td>
      <td>0.003338</td>
      <td>CA</td>
      <td>C</td>
      <td>0.889360</td>
      <td>0.824483</td>
      <td>1.869460</td>
      <td>0.17</td>
      <td>0.003181</td>
      <td>0.003795</td>
      <td>0.702462</td>
      <td>0.40</td>
      <td>0.525906</td>
      <td>0.47</td>
    </tr>
    <tr>
      <th>1</th>
      <td>rs12238997</td>
      <td>1</td>
      <td>693731</td>
      <td>0.003363</td>
      <td>A</td>
      <td>G</td>
      <td>0.884251</td>
      <td>0.875969</td>
      <td>0.314486</td>
      <td>0.57</td>
      <td>0.001202</td>
      <td>0.003587</td>
      <td>0.112243</td>
      <td>0.74</td>
      <td>0.070030</td>
      <td>0.79</td>
    </tr>
    <tr>
      <th>2</th>
      <td>rs371890604</td>
      <td>1</td>
      <td>707522</td>
      <td>0.003733</td>
      <td>G</td>
      <td>C</td>
      <td>0.902904</td>
      <td>0.803693</td>
      <td>0.067607</td>
      <td>0.79</td>
      <td>-0.000160</td>
      <td>0.004037</td>
      <td>0.001580</td>
      <td>0.97</td>
      <td>0.018918</td>
      <td>0.89</td>
    </tr>
    <tr>
      <th>3</th>
      <td>rs149887893</td>
      <td>1</td>
      <td>714596</td>
      <td>0.003923</td>
      <td>T</td>
      <td>C</td>
      <td>0.968095</td>
      <td>0.844433</td>
      <td>0.050703</td>
      <td>0.82</td>
      <td>-0.002333</td>
      <td>0.006792</td>
      <td>0.117972</td>
      <td>0.73</td>
      <td>0.145692</td>
      <td>0.70</td>
    </tr>
    <tr>
      <th>4</th>
      <td>rs12184267</td>
      <td>1</td>
      <td>715265</td>
      <td>0.003941</td>
      <td>C</td>
      <td>T</td>
      <td>0.964573</td>
      <td>0.926915</td>
      <td>0.019258</td>
      <td>0.89</td>
      <td>-0.001602</td>
      <td>0.006167</td>
      <td>0.067493</td>
      <td>0.80</td>
      <td>0.105143</td>
      <td>0.75</td>
    </tr>
  </tbody>
</table>
</div>
</div>

</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Check-files">Check files<a class="anchor-link" href="#Check-files">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="GWAS-results">GWAS results<a class="anchor-link" href="#GWAS-results">&#182;</a></h4>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[1]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">ls</span> <span class="o">/</span><span class="n">SAY</span><span class="o">/</span><span class="n">dbgapstg</span><span class="o">/</span><span class="n">scratch</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">BOLTLMM_results</span><span class="o">/</span><span class="n">results_imputed_data</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre><span class="ansi-blue-intense-fg ansi-bold">asthma</span>/  <span class="ansi-blue-intense-fg ansi-bold">INT-BMI</span>/  <span class="ansi-blue-intense-fg ansi-bold">INT-WAIST</span>/  <span class="ansi-blue-intense-fg ansi-bold">INT-WHR</span>/
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[180]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">ls</span> <span class="o">/</span><span class="n">SAY</span><span class="o">/</span><span class="n">dbgapstg</span><span class="o">/</span><span class="n">scratch</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">results</span><span class="o">/</span><span class="n">BOLTLMM_results</span><span class="o">/</span><span class="n">results_imputed_data</span><span class="o">/</span><span class="n">asthma</span><span class="o">/</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv.Asthma_casesbyICD10codesANDselfreport_controlsbyselfreportandicd10_noautoimmuneincontrols.fastGWA.stats.all_chr
UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv.Asthma_casesbyICD10codesANDselfreport_controlsbyselfreportandicd10_noautoimmuneincontrols.fastGWA.stats.all_chr.numgenosnps.txt
ukb_imp_v3.Asthma_casesbyICD10codesANDselfreport_controlsbyselfreportandicd10_noautoimmuneincontrols.fastGWA.log.all_chr
ukb_imp_v3.Asthma_casesbyICD10codesANDselfreport_controlsbyselfreportandicd10_noautoimmuneincontrols.fastGWA.snp_stats.all_chr.gz
ukb_imp_v3.Asthma_casesbyICD10codesANDselfreport_controlsbyselfreportandicd10_noautoimmuneincontrols.fastGWA.snp_stats.all_chr.numimpsnps.txt
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Genotype-data-and-imputed-data">Genotype data and imputed data<a class="anchor-link" href="#Genotype-data-and-imputed-data">&#182;</a></h4>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[11]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">ls</span> <span class="o">/</span><span class="n">SAY</span><span class="o">/</span><span class="n">dbgapstg</span><span class="o">/</span><span class="n">scratch</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv</span><span class="o">.*</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>/SAY/dbgapstg/scratch/UKBiobank/genotype_files/pleiotropy_geneticfiles/UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv.bed
/SAY/dbgapstg/scratch/UKBiobank/genotype_files/pleiotropy_geneticfiles/UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv.bim
/SAY/dbgapstg/scratch/UKBiobank/genotype_files/pleiotropy_geneticfiles/UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv.fam
/SAY/dbgapstg/scratch/UKBiobank/genotype_files/pleiotropy_geneticfiles/UKB_Caucasians_phenotypeindepqc120319_updated020720removedwithdrawnindiv.log
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[129]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">ls</span> <span class="o">/</span><span class="n">SAY</span><span class="o">/</span><span class="n">dbgapstg</span><span class="o">/</span><span class="n">scratch</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">pleiotropy_geneticfiles</span><span class="o">/</span><span class="n">unrelated_n307259</span><span class="o">/</span><span class="n">UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>/SAY/dbgapstg/scratch/UKBiobank/genotype_files/pleiotropy_geneticfiles/unrelated_n307259/UKB_unrelatedcauc_phenotypes_asthmat2dbmiwaisthip_agesex_waisthipratio_040620
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[3]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">ls</span> <span class="o">/</span><span class="n">SAY</span><span class="o">/</span><span class="n">dbgapstg</span><span class="o">/</span><span class="n">scratch</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">genotype_files</span><span class="o">/</span><span class="n">ukb39554_imputeddataset</span><span class="o">/</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre><span class="ansi-blue-intense-fg ansi-bold">downloadscripts</span>/                            <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr18_v3.bgen</span>*
<span class="ansi-green-intense-fg ansi-bold">encoding.ukb</span>*                               <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr18_v3.bgen.bgi</span>*
<span class="ansi-green-intense-fg ansi-bold">fields.ukb</span>*                                 <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr19_v3.bgen</span>*
README                                      <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr19_v3.bgen.bgi</span>*
samples.txt                                 <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr1_v3.bgen</span>*
subset_sample_variant_ukb_imp_chr1_v3.bgen  <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr1_v3.bgen.bgi</span>*
subset_ukb_imp_chr1_v3.bgen                 <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr20_v3.bgen</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb1_1581369705_4814.tmp_bulk</span>*              <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr20_v3.bgen.bgi</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb1_1581570817_24750.tmp_bulk</span>*             <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr21_v3.bgen</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb1_1581572048_25854.tmp_bulk</span>*             <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr21_v3.bgen.bgi</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb32285_imp_chr22_v3_s487297.sample</span>*       <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr22_v3.bgen</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb32285_imputedindiv.sample</span>*               <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr22_v3.bgen.bgi</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb39554.enc_ukb</span>*                           <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr2_v3.bgen</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb39554.log</span>*                               <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr2_v3.bgen.bgi</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb39554.r</span>*                                 <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr3_v3.bgen</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb39554.tab</span>*                               <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr3_v3.bgen.bgi</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_bgi.md5</span>*                                <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr4_v3.bgen</span>*
<span class="ansi-green-intense-fg ansi-bold">ukbgene</span>*                                    <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr4_v3.bgen.bgi</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_bgi.md5</span>*                            <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr5_v3.bgen</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_bgi.tgz</span>*                            <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr5_v3.bgen.bgi</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr10_v2.bgen</span>*                      <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr6_v3.bgen</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr10_v2.bgen.bgi</span>*                  <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr6_v3.bgen.bgi</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr11_v2.bgen</span>*                      ukb_imp_chr7_v3.bgen
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr11_v2.bgen.bgi</span>*                  <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr7_v3.bgen.bgi</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr12_v2.bgen</span>*                      ukb_imp_chr8_v3.bgen
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr12_v2.bgen.bgi</span>*                  <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr8_v3.bgen.bgi</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr13_v2.bgen</span>*                      ukb_imp_chr9_v3.bgen
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr13_v2.bgen.bgi</span>*                  <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr9_v3.bgen.bgi</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr14_v2.bgen</span>*                      <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chrX_v3.bgen</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr14_v2.bgen.bgi</span>*                  <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chrX_v3.bgen.bgi</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr15_v2.bgen</span>*                      <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chrXY_v3.bgen</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr15_v2.bgen.bgi</span>*                  <span class="ansi-green-intense-fg ansi-bold">ukb_imp_chrXY_v3.bgen.bgi</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr16_v2.bgen</span>*                      <span class="ansi-green-intense-fg ansi-bold">ukb_imp_mfi.tgz</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr16_v2.bgen.bgi</span>*                  <span class="ansi-green-intense-fg ansi-bold">ukbmd5</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr17_v2.bgen</span>*                      <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr10_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr17_v2.bgen.bgi</span>*                  ukb_mfi_chr10_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr18_v2.bgen</span>*                      <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr11_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr18_v2.bgen.bgi</span>*                  ukb_mfi_chr11_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr19_v2.bgen</span>*                      <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr12_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr19_v2.bgen.bgi</span>*                  ukb_mfi_chr12_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr1_v2.bgen</span>*                       <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr13_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr1_v2.bgen.bgi</span>*                   ukb_mfi_chr13_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr20_v2.bgen</span>*                      <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr14_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr20_v2.bgen.bgi</span>*                  ukb_mfi_chr14_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr21_v2.bgen</span>*                      <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr15_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr21_v2.bgen.bgi</span>*                  ukb_mfi_chr15_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr22_v2.bgen</span>*                      <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr16_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr22_v2.bgen.bgi</span>*                  ukb_mfi_chr16_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr2_v2.bgen</span>*                       <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr17_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr2_v2.bgen.bgi</span>*                   ukb_mfi_chr17_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr3_v2.bgen</span>*                       <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr18_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr3_v2.bgen.bgi</span>*                   ukb_mfi_chr18_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr4_v2.bgen</span>*                       <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr19_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr4_v2.bgen.bgi</span>*                   ukb_mfi_chr19_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr5_v2.bgen</span>*                       <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr1_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr5_v2.bgen.bgi</span>*                   ukb_mfi_chr1_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr6_v2.bgen</span>*                       <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr20_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr6_v2.bgen.bgi</span>*                   ukb_mfi_chr20_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr7_v2.bgen</span>*                       <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr21_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr7_v2.bgen.bgi</span>*                   ukb_mfi_chr21_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr8_v2.bgen</span>*                       <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr22_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr8_v2.bgen.bgi</span>*                   ukb_mfi_chr22_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr9_v2.bgen</span>*                       <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr2_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_hap_chr9_v2.bgen.bgi</span>*                   ukb_mfi_chr2_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_bgi.tgz</span>*                            <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr3_v3.txt</span>*
ukb_imp_chr10_v3.bgen                       ukb_mfi_chr3_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr10_v3.bgen.bgi</span>*                  <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr4_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr11_v3.bgen</span>*                      ukb_mfi_chr4_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr11_v3.bgen.bgi</span>*                  <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr5_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr12_v3.bgen</span>*                      ukb_mfi_chr5_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr12_v3.bgen.bgi</span>*                  <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr6_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr13_v3.bgen</span>*                      ukb_mfi_chr6_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr13_v3.bgen.bgi</span>*                  <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr7_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr14_v3.bgen</span>*                      ukb_mfi_chr7_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr14_v3.bgen.bgi</span>*                  <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr8_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr15_v3.bgen</span>*                      ukb_mfi_chr8_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr15_v3.bgen.bgi</span>*                  <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chr9_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr16_v3.bgen</span>*                      ukb_mfi_chr9_v3.txt_infoabove0.8
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr16_v3.bgen.bgi</span>*                  <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chrX_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr17_v3.bgen</span>*                      <span class="ansi-green-intense-fg ansi-bold">ukb_mfi_chrXY_v3.txt</span>*
<span class="ansi-green-intense-fg ansi-bold">ukb_imp_chr17_v3.bgen.bgi</span>*                  <span class="ansi-green-intense-fg ansi-bold">ukb_mfi.md5</span>*
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h4 id="Phenotype-data">Phenotype data<a class="anchor-link" href="#Phenotype-data">&#182;</a></h4>
</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[5]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">ls</span> <span class="o">/</span><span class="n">SAY</span><span class="o">/</span><span class="n">dbgapstg</span><span class="o">/</span><span class="n">scratch</span><span class="o">/</span><span class="n">UKBiobank</span><span class="o">/</span><span class="n">phenotype_files</span><span class="o">/</span><span class="n">pleiotropy_R01</span><span class="o">/</span><span class="n">phenotypesforanalysis</span>
</pre></div>

    </div>
</div>
</div>

<div class="output_wrapper">
<div class="output">


<div class="output_area">

    <div class="prompt"></div>


<div class="output_subarea output_stream output_stdout output_text">
<pre>Asthma_casesbyICD10codesANDselfreport_controlsbyselfreportandicd10_forbolt030720
Asthma_casesbyICD10codesANDselfreport_controlsbyselfreportandicd10_noautoimmuneincontrols_forbolt030720
diabetes_casesbyICD10andselfreport_controls_030620
diabetes_casesbyICD10andselfreport_controlswithoutautoiummune_030720
UKB_caucasians_BMI_nopreg_adjagesex_residuals_andstandardized_022720
UKB_caucasians_BMIwaisthip_AsthmaAndT2D_withagesex_033120
UKB_caucasians_waistcircumference_adjbmiagesex_nopreg_residuals_022720
UKB_caucasians_waisthipratio_adjbmiagesex_nopreg_residuals_022720
UKBrelated_AsthmaT2D_casesbyfieldoricd10_noautoincontrols_dataforbothphenotypes_040820
</pre>
</div>
</div>

</div>
</div>

</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<h2 id="Install-pandas_plink">Install pandas_plink<a class="anchor-link" href="#Install-pandas_plink">&#182;</a></h2>
</div>
</div>
</div>
<div class="cell border-box-sizing text_cell rendered"><div class="prompt input_prompt">
</div><div class="inner_cell">
<div class="text_cell_render border-box-sizing rendered_html">
<p>a Python package for reading PLINK binary file format and PLINK and GCTA realized relationship matrices. The file reading is taken place via lazy loading, meaning that it saves up memory by actually reading only the genotypes that are actually accessed by the user.(<a href="https://pypi.org/project/pandas-plink/">https://pypi.org/project/pandas-plink/</a>)</p>

</div>
</div>
</div>
<div class="cell border-box-sizing code_cell rendered">
<div class="input">
<div class="prompt input_prompt">In&nbsp;[&nbsp;]:</div>
<div class="inner_cell">
    <div class="input_area">
<div class=" highlight hl-sos"><pre><span></span><span class="n">pip</span> <span class="n">install</span> <span class="n">pandas</span><span class="o">-</span><span class="n">plink</span>
</pre></div>

    </div>
</div>
</div>

</div>
<hr>
&copy 2020 Diana Cornejo at Center for Statistical Genetics, Sergievsky Center, Columbia University VP&S
<p><small>Exported from <a href="http://github.com/dianacornejo/pleiotropy_UKB/blob/68ee90230ca76cfddc44c21e1da4a7bcb3b59f08/workflow/Region_Extraction.ipynb"><code>workflow/Region_Extraction.ipynb</code></a> committed by Gao Wang on Wed Jun 10 13:04:22 2020 <a href="http://github.com/dianacornejo/pleiotropy_UKB/commit/68ee90230ca76cfddc44c21e1da4a7bcb3b59f08">revision 7, 68ee902</a> <a href="https://stephenslab.github.io/ipynb-website/notes.html#Note-about-commit-ids"><span class="fa fa-question-circle"></span></a></small></p>
</div>
</div>
</body>
</html>
